---
title: "Discussion Draft"
date: ''
output:
  html_document:
    df_print: paged
    toc: yes
    toc_depth: '5'
    toc_float: yes
  pdf_document: default
  word_document:
    toc: yes
    toc_depth: '5'
bibliography: refs_Main_Report.bib
---

```{r format.Notes, include=FALSE}
#table, wite by all of the columns and rows

#over-view of the synthesis of hypothesis testing
#(this is how we're synthsizing / testing / the framework were using to compare the results to the literature)

#constrast on model types
#-evolution
#-models accounting for chromosome based traits

#HetC traits
# 1. typical landscape
# 2. female variance
# 3. chromatin organization

#faster male evolution traits
# 1. Reversed Direction of Heterochiasmy
# 2. correlated gwRR increase and interference strength

#go through each traits and point out which models support each of the traits
```


```{r Model.testing, include=FALSE}

#imortant of testable hypotheses

#model not included in this comparison


```

Advice from Bret:  

- Don't know where to fit in DSB evolution results  

- Keeping interference section breif (background, explaningillustrating spindle based selection model)

To synthesize our results, we decded to systematically compare models which have been proposed for explaining the evolution of heterochiasmy or meiotic machinery. There are two main types of models: modifers (indirect) modifers or functional (cell physiology based models). We attempted to test the hypotheses (mental test for if the model) (Table X). 

**the models**  We examined 3 modifer models; haploid selection, two locus model, and SACE. We also examined and propose cell physiology based models: the Chromosome Ocsilation (COM) model and a model focused on the spindle differences between males and females.

<we left out Beam film model for XX reasons>

<heterochiasmy refers to different numbers of chiasmata - and recently has been extended to number of crossovers -- this view / charaecterization is slightly incomplete, since it considers just one cell stage and 1 value (a count))
We argue that the sex differences (broaden the definition of heterochiasmy or use a different word) -- which includes more aspects that show conserved sexes differences>

<Our results indicate, sexes have distinct evolutionary trajectories for genome wide recombinaiotn rate (and landscape) in males nd females).

Conserved features of the landscape may play a role in influencing/keeping the sex specific trajectories. In our examination of the panel of mouse strains, we note several features of the recobmination landscape with conserved sexual dirmophism pattern. 
That is the sex within sex these features show no evolution and consistant pattern of variation.>

We focus first on the results for sexually dimorphic traits.


#reversed hetC direction

Reversed heterochaiasmy (briefly noted in the first part)
The common direction of heterochiasmy is female biased, however male biased heterochiasmy has (significant number of observations.
-- has been seen before (cattle, marsupial)-- Our results are novel, because the pattern is indicative of rapid evolution (male specific) -- polymorphism of low and high group. 

models
the prediction from two of the modifier models, haploid selection and SACE are that males will generally evolve lower recombination rates overall compared to females. (cases with exceptions.? we don't have evidence or reason to predict these exception cases are held)
The two locus modifier model  does make predictions / parameterize cases for (reversal of the common pattern of female biased heterochiasmy). Which are dependant on the stage of action for the driver and the linkage status (see table 1 Brandvain and Coop).
There are no prediction for reversal of heterochiasmy direction  under the COM (pairing model, but it should be noted that this model wasn't designed to incorporate this type of data.

Under the spindle based model faster male specific evolution of the genome wide recombination rate is primarily driven by differences in the strength of the SAC.
 leading to different evolutionary trajectories for the genome wide recombination rate.

The SAC 'read's that all tetrads have tension -- (each set of centromeres have sinlge and opposite pole connection (synthellic?) and that the sister cohesion connecting the homologs -- sets up the tension -- opposes the pull from each of the poles.
 
Efficient normal SAC -- works by pausing entry into anaphase or triggers apoptosis -- if a unpaired / tension-lacking tetrad is detected
(sperm SAC is finer tuned (can detect 1-2 tetrads lacking tension -- ), while the female SAC  (has a higher treshold for being triggered  (4-5?)
(Land Kauppi, 

(relaxed selection - / inefficient SAC -- in females -- fits with higher rate of achiasmata bivalents (there might be alternative mechanisms facilitating achiasmate segregation)

  Mechanisms for sex differences in the strength of the SAC are the cell volumne (diffusion of signal molecules) and the centrosome being present at the spindle have been proposed in the literature.
(testing / proving the mechanism for the new optima is for the high recombining group in males is out of the scope of this paper, but we propose some hypothesis are suggested below.
 but the evolutionary pattern fits that of directional.


# 'Typical landscape'


```{r typical.landscape.outline, include=FALSE}

#-long observed
#measured from a variety of approaches,
#larger variety of species

#-Exception to landscape (marisupials, birds?)

#which models make clear predctions for typical landscape?
#SACE
#drivers
#spindle differences

```


Our results on the broad scale pattern of crossover placement are generally supported by the literature. As reviewed in (Sardel Kirkpatrick), sexual dirmorohism in the broad scale landscape is a highly conserved trait.  However there are exceptions, (marsupial, bird?)
the popgen models
The two locus modifier model and SACE model make predictions for (sexual dimorphism ) in the broad scale recombination landscape, for diminishing the effect of drive systems and maintaining larger chromosome blocks respectively.  
functional models
The two mechanical models also both predict sexual dimorphism in the recombination landscape for distinct reasons. The chromosome pairing based model (COM), predicts the sex differences in the positioning of crossovers is due to a combination of the length of the axis and intensity of the RPMs.  The spindle based model proposed that sexual dimorphism in the recombination landscape is driven by difference strength in the spindle assembly checkpoint (Lane Kauppi). The relaxed selection imposed by an ineffective SAC results in a uniform distribution of positions (while maintaining positive interference). Whereas,  in males, the directional selection could cause decreased variance in the recombination landscape via telomeric positioning of crossovers. Under the telomeric bias  recombination landscape, entry into anaphase may be more synchoriznes (and faster) given that the reductional separation of homologs is dependent on a minimal amount of sister cohesion connecting the homologs (Figure X).



[@sardell_sex_2020].

 
# Greater between cell variation


```{r within.mouse.var, include=FALSE}

#our results, female more (greater than tech error) male no significant difference
#between cell variation

#between cell variance (Veller Kleckner), mechnism
#evolution / popgen model from (veller)

#models that predict more within animal variance
#none of the models expliciltly predict fr traits at the between cell level
#so any between cell variation -- patterns is due to DIFFERENCES IN THE STRENGTH OF SELECTION


## connection on evolution of transcriptional noise
# transcriptional noise and complex traits -- both show the same repsonses to different modes of selection

```


**the within animal variation is equvillent to between cell variation**
[@lynn2002] 

Our results of greater between cell variation . these results are supported by human and mouse (Gruhn (Kong)?  (Zwel -- Cole).  These data are also scant, between cell variance isn't often measured especially for both sexes. (given they require single cell observations for both sexes).

While the modifier models parametrize sexual dimorphism in the strength of selection which can shape the variance in a trait these types of models make predictions  for variance between individuals but not within individuals where the genetics are constant. (selection on penetrance).
In the spindle based selection model,  the source of between cell variance is attributed to difference strength of checkpoints. (Sexual dimorphism in the stringency of the spindle assembly checkpoint is a well established result in the literature. Most empirical results come from mammals (is this true?) yet, some of the hypotheses for the mechanism of inefficient SAC in oocytes are attributed to acentrosomal spindles (cite) and cell volume (cite), which are conserved sexual dirmohpic features of gametogenesis (cite).



# Chromatin Structure

```{r chromatin.organization, include=FALSE}
#1. basics and expectations for chromatin organization
#seen in humans, mice (signal comes through even with cell to cell variation in chromatin compaction / organization)

#2. how the model treat this limiting factor

#3. consequence for interference (general patterns) is that interference measured as (mb) will be stronger if DNA loops are stronger
#prediction for general -- pattern
```


0.  Our results from MSM and PWD show that the sc length compaction is independant (of the direction of heterochiasmy. Two strains have have more crossovers in males, but females still have longer sc axis. _This indecates that the sexual dimorphism in chromosome organization / DNA compaction is a conserved part of recombination landscape / heterochiasmy_


**chromatin organization**

The results of sexual dimorphism in chromatin structure is supported by the literature, especially for mammals (Lynn, Gruhn, ). 
models
Very few models make predictions for how chromatin organization relates to variation in recombination rates / the recombination landscape.  (the chromatin organization is how chromosomes with the same number of base pairs) are transformed into bottle brush structures with a central axis and most of the genetic material arranged as loops emanating outwards (figure).
Only the COM model states that chromatin structure will be longer in oocytes due to greater cell volume.  This prediction / model might fit broader pattern (such as in At where pollen is the larger cell and has longer axis length) (cahoon libuda).  
Predictions for this trait are  inconclusive (due to lack of cytology measures from both sexes. We argue that the chromatin structure (and axis length) should be considered as an integral parameter when studying recombination landscape.  The sex differences in chromatin strucutre are closely connected to sex differences  in the interference strength and our results mirror those in Petkov et al.  (prediction, shorter-loop will have stronger interference).


According to the tethered loop model of DSB formation (cite) -- the 'real estate' for COs are the DNA loops - the density of whch is conserved in the axis and loop formation (later the SC str), so the number of loops is equvilent to the length of the axis. **The chromosomes are converted to a 2D space (the axis length of a given chromosome is the limited factor for total CO number)**


2. Only one of the models we examined aligns with our observed results for sexual dimorphism for axis length.

The modifier models do not parameterize this since they do not models / are not built to consider single meiotic events.

**COM** predicts that axis length will be larger in oocytes due to the larger volume -- this would help in the bouquet stage -- may be connected to the 'the search' during homolog pairing. 

the **spindle based selection** model does not specific predictions about this aspect -- assumes that the 2D areas for chromosomes are proportional across stages (during the decondensing stage between pachtyene and dipl? / metaphase) -- the chromosomes condenses uniformly across the length.



# Evolved Traits

```{r Evolve.traits.outline, include=FALSE}

#tricker to fit the models to this, 
#paradox -- how the broad/general patterns from the sex differences

#models weren't designed to explain within species variation in recombination rates 
#or evolution

```

The faster evolution in males with the high recombining rates. BUT applying these above models to the traits connected to the faster male evolution is a bit trickier since none of these models were designed to explain within species variation in recombination rates. (most were designed to explain the general pattern of sexual dimorphism in meiotic recombination rates.)



# Evolution of Interference

[@ottoPaysuer2019].

```{r evo.interference, include=FALSE}
#review results,
#why unexpected, unexpected pattern is great for identifying most parimonious model / explanation
#(we're gonna spend alot of space on this section)

# more convincing that the pattern is real

#lil background, for interferences 
#assuptions for predictions (most modifer models, )

#models that predict this pattern (CONTROL FOR CO NUMBER -- PRIORITIZE THE WEIGHT /PROB of CO ASSURANCE)

#the pattern is driven in part by short IFD (when short IFD removed in reduced data set -- the low strain mean goes up, but they are still shrter)

#**The rapid evolution of the genome wide rate in our high rec strains is PARTLY due to more enrichment of 2CO which have distinct rec landscape from the 2COs in low rec strains.**


```

[@ma2015_cattle]. In *Peromyscus leucopus*, the wild mice had higher gwRR and stronger (gamma nu estimates) compared to lab raised mice of the same speicies [@peterson2019].


Some basics of the constraints / bounds of interference are known, yet there are still many unknowns regarding the relationship of genome wide recombination rate and interference strength. 
 The non-random spacing of crossovers along the 2D length of chromosomes is the first level of proof /meeting the rejection of the (null model of random placement (cite gamma COI papers).
The physical limit of how far apart crossovers can be placed on a chromosome is a function of the length of the meiotic chromosome axis.
Logically a negative correlation is expected; increasing the number of crossovers across chromosomes would most logically be done by  more densely spacing crossovers along chromosomes and decreasing interference strength. This pattern has empirical support -- and describes  -- the fundamental relationship / of SC area of crossover number which is observed in much of the literature (which describes large/broad scale patterns).

Our results support an opposite pattern, a positive correlation between interference strength and genome wide recombination rates; the strains which show rapid evolution of the gwRR also show stronger interference strength.  We've found a number of examples of positive correlation between genome wide recombination rate and interference. The within sex comparison of two breeds of cattle (Ma et al), the Peromyscus species in (Peterson et al ), and preliminary data (HVR un published).

Theoretical models haven't really considered the evolution of interference strength -- While the COM model predicts that interference (and recombination landscape) arises from known oscillatory movements during prophase it lacks an evolutionary based prediction.
While the SACE modifier model does not explicitly model evolution of interference strength -- we note that a logical outcome of the main prediction of maintaining larger chromosome blocks in males, would be a landscape with stronger interference strength.
Veller et al rbar -- and predictions / for this - found a similar pattern (positive correlation in their simulations of male human data.

Models from Goldstein et al (review in Otto and Payseur) suggest that if this pattern is widespread --- interference evolves in X manner.



# REVIEW

We propose that the spindle based selection model is the most parsimounious with our results and known recombination patterns from the literautre.

  - It matches fine scale differences in how DSB hotspots had different sex specific strengths with varied due to sex, epigenetc landscape and the gene architecture (Brick et al)

 - incorporates know cellular / molecular mechanisms with a very common pattern sexual dimorphic recombination rate landscape.
 
- extends modifer model to explain a chromosome based / level pattern
 _weaknesses_, and ways the model diverges from our results

 - the simulation present in the paper are too weak to support the rapid spread of such a modifer system (selection coefficients are too weak to support the evolution -- spread of such a modifer.)
- Other ways this model departs from our results, 
-Always predicting male will be lower recombination rates (or risk breaking up the epistatsis blocks) -- which doesn't fit with our results of faster male specific evolution in gwRR.
However / while not explicitaly arameterized or predicted in this model,
 -- this model predicts larger chuncks of chromosomes will segregate together
 (as outlined in Veller et al)
our results of stronger male interference and stronger interference in high rec strains have the same consequences as predited from this modle


```{r final.review.outline, echo=FALSE}
# comes down to explaning the interference correlation


# spindel selection -- 
#good for sex differences because it frames heterochiasmy within the context of meiosis

#can fit the interference evolution due to the (pressumed) effect on bivalent shape

#caveats and unanswered parts

# SACE is runner up


```



#Future Steps

Be aware of sex-specific and sex average biases. Some cytlogy models (Celegans) mostly focus on oogenesis  and genetic maps are often from hermaphrodites. (lack of data from one sex). Recognize the value of Chromosome level measures and incorporating chromosome behavior into models.

Borader communication across sub-fields of cell physiology and biophysics for understanding recombination rate evolution. Utilize mitosis systems given the conservation of the molecular pathway. THink of meiosis as a program.

Dark matter of the genome. We can't really explude that some of the recombination rate variation is due to strucutreal variation (more of less genome). (since everything is mapped to B6). (also maybe there are crossovers in the centromeres and telomeres -- which might escape detected in linkage maps)

what are the rates of sister-sister crossovers?

How does elevated rates of precocious sister centromere seperation effect oogenesis?

Are selfish elements affecting the evolution of meiosis -- and thus recombination rates (from a functional standpoint).

Keep generating those testable hypotheses (cite).



#References
