---
title: "Discussion Draft"
date: ""
output:
  html_document:
    df_print: paged
    toc: yes
    toc_depth: '5'
    toc_float: yes
  pdf_document: default
  word_document:
    toc: yes
    toc_depth: '5'
bibliography: refs_Main_Report.bib
---

```{r format.Notes, include=FALSE}
#table, wite by all of the columns and rows

#over-view of the synthesis of hypothesis testing
#(this is how we're synthsizing / testing / the framework were using to compare the results to the literature)

#constrast on model types
#-evolution
#-models accounting for chromosome based traits

#HetC traits
# 1. typical landscape
# 2. female variance
# 3. chromatin organization

#faster male evolution traits
# 1. Reversed Direction of Heterochiasmy
# 2. correlated gwRR increase and interference strength

#go through each traits and point out which models support each of the traits
```


```{r Model.testing, include=FALSE}

#imortant of testable hypotheses

#model not included in this comparison


```

Advice from Bret:  

- Don't know where to fit in DSB evolution results  

- Keeping interference section breif (background, explaningillustrating spindle based selection model)

To synthesize our results, we decded to systematically compare models which have been proposed for explaining the evolution of heterochiasmy or meiotic machinery. There are two main types of models: modifers (indirect) modifers or functional (cell physiology based models). We attempted to test the hypotheses (mental test for if the model) (Table X). 

**the models**  We examined 3 modifer models; haploid selection, two locus model, and SACE. We also examined and propose cell physiology based models: the Chromosome Ocsilation (COM) model and a model focused on the spindle differences between males and females.

<we left out Beam film model for XX reasons>

<heterochiasmy refers to different numbers of chiasmata - and recently has been extended to number of crossovers -- this view / charaecterization is slightly incomplete, since it considers just one cell stage and 1 value (a count))
We argue that the sex differences (broaden the definition of heterochiasmy or use a different word) -- which includes more aspects that show conserved sexes differences>

<Our results indicate, sexes have distinct evolutionary trajectories for genome wide recombinaiotn rate (and landscape) in males nd females).

Conserved features of the landscape may play a role in influencing/keeping the sex specific trajectories. In our examination of the panel of mouse strains, we note several features of the recobmination landscape with conserved sexual dirmophism pattern. 
That is the sex within sex these features show no evolution and consistant pattern of variation.>

We focus first on the results for sexually dimorphic traits.

# 'Typical landscape'


```{r typical.landscape.outline, include=FALSE}

#-long observed
#measured from a variety of approaches,
#larger variety of species

#-Exception to landscape (marisupials, birds?)

#which models make clear predctions for typical landscape?
#SACE
#drivers
#spindle differences

```


The 'typical landscapes' refers to the sex differences in the board scale pattern or general placement of crossovers along the length of chromosomes. The primary pattern which has been observed across mny species is that crossovers are shifted to telomereic positions in males while a more unifrom landscape in observed in females [@sardell_sex_2020].

Our results show that the typical landscape is consered across all strains of house mice in this study (for all single chromosome daata and when controlled for crossover number), the broad scale position of crossovers was shifted closer to the telomere end (of the long arm).

<Detangling centromere suppression from telomere bias,  The Sardel guppies paper was able to deduce (dientangle) that this typical landscape pattern was due to // disentangle the specific components of centromere supression and telomere bias between the male and feamel landscapes. we are not able to do that in these cases (because of inability to estimate the chromosome identity across our sample of single bivalent
s. )>

The conclusions from the models -- basically all of the models could account for differences in the general recombination landscape patterns.

 
# Greater between cell variation


```{r within.mouse.var, include=FALSE}

#our results, female more (greater than tech error) male no significant difference
#between cell variation

#between cell variance (Veller Kleckner), mechnism
#evolution / popgen model from (veller)

#models that predict more within animal variance
#none of the models expliciltly predict fr traits at the between cell level
#so any between cell variation -- patterns is due to DIFFERENCES IN THE STRENGTH OF SELECTION


## connection on evolution of transcriptional noise
# transcriptional noise and complex traits -- both show the same repsonses to different modes of selection

```


**the within animal variation is equvillent to between cell variation**

Our results confirm previous findings from mice [@lynn2002] and humans of females having greater between cell variation compared to males.

Transcriptional noise and complex traits both show the same repsonses to different modes of selection, lower variation under stabilizing selection and greater variaion with relaxed selection. The difference in between 

so all of the models could reasonably predict the above results ... examining some of the traits related to the faster evolution in the male specific data indecate different support across the models.

None of the models make predictions for between cell variation -- so we draw conclusions from their preciations based on the strength of interference.

# Chromatin Structure and interference strength

```{r chromatin.organization, include=FALSE}
#1. basics and expectations for chromatin organization
#seen in humans, mice (signal comes through even with cell to cell variation in chromatin compaction / organization)

#2. how the model treat this limiting factor

#3. consequence for interference (general patterns) is that interference measured as (mb) will be stronger if DNA loops are stronger
#prediction for general -- pattern
```


0.  Our results from MSM and PWD show that the sc length compaction is independant (of the direction of heterochiasmy. Two strains have have more crossovers in males, but females still have longer sc axis. _This indecates that the sexual dimorphism in chromosome organization / DNA compaction is a conserved part of recombination landscape / heterochiasmy_


**chromatin organization**

According to the tethered loop model of DSB formation (cite) -- the 'real estate' for COs are the DNA loops - the density of whch is conserved in the axis and loop formation (later the SC str), so the number of loops is equvilent to the length of the axis. **The chromosomes are converted to a 2D space (the axis length of a given chromosome is the limited factor for total CO number)**


2. Only one of the models we examined aligns with our observed results for sexual dimorphism for axis length.

The modifier models do not parameterize this since they do not models / are not built to consider single meiotic events.

**COM** predicts that axis length will be larger in oocytes due to the larger volume -- this would help in the bouquet stage -- may be connected to the 'the search' during homolog pairing. 

the **spindle based selection** model does not specific predictions about this aspect -- assumes that the 2D areas for chromosomes are proportional across stages (during the decondensing stage between pachtyene and dipl? / metaphase) -- the chromosomes condenses uniformly across the length.


Another thing to point out / regarding this model -- is the connection to sex difference in interference strength. Our results of conserved within strain raw IFD lengths - yet when the IFDs are controlled for total SC length, we observed sexual dimorphism with males having longer IFD^normalized^ (stronger interference), mirror those observed in [@petkov2007].
The model / explanation proposed in that paper is that The chromatin structure which has longer loops (and shorter axis) will have more DNA bps folded / sequestered into the DNA loops. Thus the strength of interference will be stronger because with each step (count) along the 2D axis, more bases pairs are being skipped above the axis. These findings point out two things, This part of heterochiasmy generates testable prediction: First, the interference strength measured in chromosome axis units (or SC) is generally conserved between sexes. Second the sex with longer axis and shorter loops will have weaker interference compared to the other sex. Or conversely, the sex with stronger linkage map based interference would have chromatin organization with shorter axis compared to the other sex. There is a plethora of data of sexual dimorphism in linkage map based interference strength indicating sexual dimorphism, but comparative cytological data is lacking (difference in perparness of some sexes and complications due to identification of specific chromosomes between cells.)

# Evolved Traits

```{r Evolve.traits.outline, include=FALSE}

#tricker to fit the models to this, 
#paradox -- how the broad/general patterns from the sex differences

#models weren't designed to explain within species variation in recombination rates 
#or evolution

```

The faster evolution in males with the high recombining rates. BUT applying these above models to the traits connected to the faster male evolution is a bit trickier since none of these models were designed to explain within species variation in recombination rates. (most were designed to explain the general pattern of sexual dimorphism in meiotic recombination rates.)


# Revered heterochiasmy

```{r reverse.HetC, echo=FALSE}
# transition connection to the hetC models

#our results - total gwRR isn't the most conserved, (we observed 3 other traits more)

#what does a reversal of HetC imply?  gwRR
#driven by faster evolution in males

#our model had faster evolution in males resulting in a reversal of the canonical pattern of heterichiasmy ( F gwRR > M gwRR)
#other examples, cattle, marsupials

#what do models say about faster evolution in one sex? (yes -- but almost always for canonical pattern of heterochiasmy)

```

Our results demonstrate that gwRR is decoupled from other sexually dimorphic meotic traits, (we observed 3 other traits more conserved sexual dimorphism examined above). We the general pattern 2 strains attribute this pattern of reversed heterochiasmy to faster evolution of gwRR in two strains males.  

Which models predict different rates of evolution between the sexes. Some of the models could explain a reversal of the direction of sexual dimrohpsim, however most all of the predictions for faster evolution result in males having reduced recombination rates. Under the haploid selection model a  **reversal of the direction of heterochiasmy** would be accompanied by vast changes in haploid stages or degree of imprinting. Following the meiotic drive based model, females should be predicted to have larger linkage maps/gwRR since they are more exposed / weaker to true meiotic drivers. A reversal in heteroschiasmy is not clearly predicted by the SACE model (unless with the -- except under the conditions described below). Under the spindle selection model, the male (centrosome/effect SAC) metaphase could drive directional selection (and faster evolution) in males compared to female. However the mechanism/aspect of different fitness is not clear. Under the COM model, evolutionary predictions are not clear (explicit), but following the logic would come from changes in axis length (to generate another wave point) -- which might be due to meiocyte volume.


# Evolution of Interference

```{r evo.interference, include=FALSE}
#review results,
#why unexpected, unexpected pattern is great for identifying most parimonious model / explanation
#(we're gonna spend alot of space on this section)

# more convincing that the pattern is real

#lil background, for interferences 
#assuptions for predictions (most modifer models, )

#models that predict this pattern (CONTROL FOR CO NUMBER -- PRIORITIZE THE WEIGHT /PROB of CO ASSURANCE)

#the pattern is driven in part by short IFD (when short IFD removed in reduced data set -- the low strain mean goes up, but they are still shrter)

#**The rapid evolution of the genome wide rate in our high rec strains is PARTLY due to more enrichment of 2CO which have distinct rec landscape from the 2COs in low rec strains.**


```

Our results indicated that interference strength has evolved with the genome wide recombination rates in the three strains with faster male evolution.

This is unexpected because most literature and logical models are built / assume that weaker dense spacing of crossovers along chromosomes (ie. weaker interference) is how. The history, models and empirical examples are extensively reviewed in [@ottoPaysuer2019]. Broad scale patterns for the correlation of gwRR and interference strength (gamma parameter) do fit the expected negative pattern [@ottoPaysuer2019].

There's no model that fits evolution of heterochiasmy and interference together. Under our undertsanding of the models, the relatioship of genome wide recombiantion rate and interference strength is assumed to be negative. We now review how some of the assumptions of typical modifer models might not fit realities of the meiotic pathway.

First, the **typical modifer models do not model single meiotic events**. We conclude that under the haploid selection and meiotic driver model any prediction for evolution of interference strength follows from the 'logical' expectation of a negative correlation.

This model predicts the evolution of heterchiasmy -through models (that build on selection to reduce recombiantion for the sex with geater varatince in fitness effect)

**SACE**
While the SACE model doesn't explecitly make predictions aof the out come for interference strength (or averge spacing of crossovers)
We note that the main predicted pattern under this model of maintaining larger blocks of chromosomes in linkage can asrise from stronger interference / greater spacing of the distance between two foci (also maximize rbar). 

Under the **COM** model -- the placement of crossovers (the recombination landscape) is generated by 'waves' / movement from the bouquet stage.  It is conceivable that the pattern of stronger interference could be generated from the waves/movements of chrmosomes in the early prophase stage this would come from a change in the movement or extension of the axis length.


The results from golstein reviewed in [@ottoPaysuer2019], predict a modeifier model (which placed more weight on crossover assurance (holds the number of crossovers constant) and simulates a scnario where a modifer acts on modulating the probabilty of flanking regions having 0 or 2 crossovers. Which has the result of increasing the distance between two foci. 

Additionally there are a handful of empircal results which also show the unexpected possitive correlation between gwRR and interference strength. When comparisons of genetic map length and interference strength are made across strengths within sex, the strain with the higher genome wide recombinaiton rate also has stronger interference (measured as nu gamma parameter) [@ma2015_cattle]. In *Peromyscus leucopus*, the wild mice had higher gwRR and stronger (gamma nu estimates) compared to lab raised mice of the same speicies [@peterson2019].

Additionaly -- preliminary data from a previous cross using MLH1 cytology in spermatocytes in house mouse replicated these results with IFD measures from  M. m. musculus^PWD^  being longer compared to M. m. castenus^Cast^ (HVR unpublished data) (cast has low gwRR).



# REVIEW

We propose that the spindle based selection model is the most parsimounious with our results and known recombination patterns from the literautre.

  - It matches fine scale differences in how DSB hotspots had different sex specific strengths with varied due to sex, epigenetc landscape and the gene architecture (Brick et al)

 - incorporates know cellular / molecular mechanisms with a very common pattern sexual dimorphic recombination rate landscape.
 
- extends modifer model to explain a chromosome based / level pattern
 _weaknesses_, and ways the model diverges from our results

 - the simulation present in the paper are too weak to support the rapid spread of such a modifer system (selection coefficients are too weak to support the evolution -- spread of such a modifer.)
- Other ways this model departs from our results, 
-Always predicting male will be lower recombination rates (or risk breaking up the epistatsis blocks) -- which doesn't fit with our results of faster male specific evolution in gwRR.
However / while not explicitaly arameterized or predicted in this model,
 -- this model predicts larger chuncks of chromosomes will segregate together
 (as outlined in Veller et al)
our results of stronger male interference and stronger interference in high rec strains have the same consequences as predited from this modle


```{r final.review.outline, echo=FALSE}
# comes down to explaning the interference correlation


# spindel selection -- 
#good for sex differences because it frames heterochiasmy within the context of meiosis

#can fit the interference evolution due to the (pressumed) effect on bivalent shape

#caveats and unanswered parts

# SACE is runner up


```


## The CONTEXT of MEIOSIS

the study confirmed preivously identified sexual dimorhisc patterns in regard to the recombiantion landscape (in mammals) -- in the goal of unifying all of these results, we sough to investigate physiologycal aspects of gametogenesis which have conserved sexual dimorphic features. A fundamental difference in meiosis (MI) for males and females is the presence of the centrosome which serves as a nucleation site for MT at each pole (and changes the shape of the spindle). (Our hypothesis is that the tension in a centrosome spindle is stronger (or more uniform) since all MT-KT are anchored at single points across cells (the two centrosomes). This will also make prediction for the spindle assembly checkpoint (SAC), which is stronger/stricter in smaller gametes with centrosomes.  (the SAC is more sensitive to achiasmate bivalents on the spindle / better at detecting a lack of tension).
Whereas in the acentrosome spindle is more diffuse (across a larger area) and the tension across bivalents are anchored by multiple MTOCs, resulting in weaker tension force or less uniform strengths across all the bivalents within the cells. The SAC in eggs seems to be weaker/leakier, that is more achiasmate bivalents are required to trigger the SAC (stop the division).

**main sex differences in spindle** are the anchoring of MT at opposite poles

in males there are centromsomes for Mi and MII divisions
where are in oocytes, a (universal) conserved feature of oogenesis in many species is that they loose centrosomes -- instead the MTs are anchored at mulitple aMTOCs.

There is potential for these aMTOC to be unevenly distributd across the cortical and central side of the spindles (Wu et al, Schuh and ellenberg) --- (could be an origin of different tension / pulling )

The survalence of SAC in oocytes might be decreased due to large volumne difusing the signals.

#CO placement translates to bivalent shape

**still an open question of if and how the CO number and placement of crossovers translates to different fitness values for segregation rates**

the metric / amout of sister cohesion might be an good place to start

the ultimate goal is proper disjunction of chromosomes, the main checkpoint is spindle assembly checkpoint (SAC) which responds to the tension signal across chiasmata

we like this model because it can be applied to **both our results** for i) sexual dimorphism variation and ii) the fast male evolution in gwRR and recombination landscape.

Uses / centers around one of the best known selection factors on meiotic division, anneuploidy. Several other facors related to meiotic recombiniation center on **segregation** (obligate chiasmata, lower bound of the genome wide rate) it's not unbelievable that the recombination 

Consequences of evolution in the number and placement of crossovers / evolution of interference strength:
Spindle / SAC model: The number and placement of crossovers changes the resulting amount of sister cohesin involved the tension signal, which affects the tetrad/chiasmata structure at MI when the homologs are pulled apart (Hollis et al 2020, Lee et al 2019).  This model predicts a difference in the meiotic spindles or division mechanism between high and low rec strains. 
ii. SACE: For the 2CO in the high rec males, larger blocks of chromosomes are kept together in the next generation.
This model would predict greater reproductive variance for males.

#Future Steps

Be aware of sex-specific and sex average biases. Some cytlogy models (Celegans) mostly focus on oogenesis  and genetic maps are often from hermaphrodites. (lack of data from one sex). Recognize the value of Chromosome level measures and incorporating chromosome behavior into models.

Borader communication across sub-fields of cell physiology and biophysics for understanding recombination rate evolution. Utilize mitosis systems given the conservation of the molecular pathway. THink of meiosis as a program.

Dark matter of the genome. We can't really explude that some of the recombination rate variation is due to strucutreal variation (more of less genome). (since everything is mapped to B6). (also maybe there are crossovers in the centromeres and telomeres -- which might escape detected in linkage maps)

what are the rates of sister-sister crossovers?

How does elevated rates of precocious sister centromere seperation effect oogenesis?

Are selfish elements affecting the evolution of meiosis -- and thus recombination rates (from a functional standpoint).

Keep generating those testable hypotheses (cite).


#blah

<5. Model prediction
The models above which predict directional selection on the recombination landscape in males: SACE and Spindle based selection  are most parsimonious for the models review in this paper. The reduction principle wouldn't predict males diverge in genome wide rates and two locus modifier doesn't predict males gwRR would be higher than females.>

<3.Two-locus (protect for drivers)
This second modifer model was persented in [@brandvain2012scrambling] the basics of the model are that females will have a longer genetic map and more centromere biased recombination landscape compared to males.

A strength of this model is that is presents a clear driver / source of differential fitness (ie meiotic drivers can decrease fitness/fertility). This model also accounts for the general female pattern,

However, this model doesn't explain the borad scale pattern for the male pattern.>


<Also:
	-These results are different from the sex differences because the difference is seen in both the raw and normalized IFD, indicateing the mechanism  / physcial strength (interference strength (signal) mediated through the SC has evolved.>

<Some of the traits related to faster male evolution -- distinguish in which models can redict them: the reverseal of heterochiasmy direction and the correlated evolution of gwRR and interference strength.

**DSB evolved but, don't included as an examined trait, since most evolution based models don't make predictions for this stage of meiosis.**
_the propotion of DSB repaired into crossovers hasn't evolved, the total number of DSBs has evolved_  (following the tethered lop model of DSBs formation -- this would suggest that in the high rec strains, assuming no large difference in the Mb size of the chromosomes, 
during chromatin remodeling in prophase, in the high rec strains the chromosomes have longer axis, allowing a higher number but shorter DNA loops (for SPO11 to make the DSBs). Our results mirror those found in Baier et al [@baier2014]

Our results show that the ratios of DSB : CO 
**range from X to X which is similar to those presented in [@baier2014] (which report X to X) **

These conserved ratios suggest that the increase in DSB numbers are accompanied by proportional increase in crossovers.

This shifts support away from evolution at the CO:NCO decision to an earlier prophase stage such as when the chromatin compaction is set up. **increase in DSBs is driven by the number of DNA loops** and due to the inverse relationship of loop and chromatin axis length, this would result in longer sc lengths (which reflect the underlying chromatin axis length).>


<Petkov paper -- states that the sexual dimorphism for interference strength is due to the chromatin structure (meiotic chromosome organization) with the increased interference strength in males being due to longer chromatin loop lengths.

same pattern in humans (Gruhn et al 2013)
  (exceptions, birds (fish?) plants?) -- examining this is hard due to chromosome size effecs
  - This also assumes that the DNA loop sizes have uniform size along and between chromosomes

2. Short axis / longer DNA loops will have stronger interference
This prediction combines the findings of chromatin compaction and interference differences which are intertwined by the transmission of interference along the chromosome axis and the inverse relationship chromatin compaction (axis length and DNA loop sizes).
In physical units (micron) the counting model fits interference well to the 2D linear space of the axis these measures of interference are conserved across sexes in mouse. This is because the counting units are the BASES of DNA loops which is conserved along the 2D length of chromatin axis. When this space is transformed to account for all of the DNA base pairs which are sequestered into the DNA loops above the axis, the strength of interference will vary depending the on underlying chromatin structure.>

<We focused on the largest axis of variation, the faster evolution of genome wide recombination rates in males. One of our primary objectives was to identify other traits which correlate (could explain the increase).>

<Baier et al found significant differences in sc legnths for 
The tests in our study - show moderate to low significance in the differences in SC lengths. (this is because we have more noise-can't distinguish individual chromosomes)


a study of in house mouse -- using different markers for DSBs and intermediates (markers for the recombination pathway (from DSB to mature crossover))
-different DSB marks, distinct inbred strains two classical lab strains and a wild derived strain from castenus, used FISH for chromosome specific measures

our results are similar in several ways
- correlation of final MLH1 and DBS counts, SC measues correlate with the DSB number (even tho the differnces we observe in our study are -- slight / less significant -- we attribute that to chromosome specific noise )>


<We can't distinguish if this is 3 independent instances of evolution in gwRR OR if the patterns are due to shared standing variation, incomplete lineage sorting, but these points are for describing the general patterns.>
<extends the lenomand focusing on two conditions. first that the stength of epistatsis in dependant on the cis or trans nature. the authors note that this would apply to gene archtechture (and increase with in cis ) -- where the regulatroy region and coding region of a gene evolve (together).  second that the relative interference strengths differnt between male and female.  This can apply to (genes acting in sexual antagonism) or can be mediated via imprinted genes. 
originally proposed/outlined by Trivers [] and refined /formalized in [@lenormand2003]. This model describes a system where gametes are competing or selection is acting in the haploid stage to cause difference in genome wide recombination rates between males and females. This is a modifier model of three loci, where the recombination rate between two loci is determined by a third locus (or modifier)>
<2.  Review the assumptions for measuring CO interference in different ways.
The logical model: Crossover interference is a mechanism of suppressing crossovers, negative relationship between interference strength and gwRR. More crossovers can fit along the chromosomes and results in more crossovers overall.
Examples of empirical results supporting this pattern;  Otto Payseur, (Ruiz-Herrera? Bomblies?)
Quantifing using the gamma distribution: 
Caveats,  This doesn't make since for the general recombination landscape:   i) finite chromosome length and ii) limited range of crossovers per chromosomes (1 to 4).
Alternative interference characterization:  i) the number of crossovers are held constant and ii) transform / generalize the rec landscape, (focus on how the rec landscape would change instead of the genome wide rate) iii) assume crossover assurance.
	-The Goldstein model, models predicts a positive correlation with interference strength and higher recombination rate. A modifier which changes the probability of the recombination landscape having 0 crossovers to 2 crossovers, (driven by crossover assurance)>


#References
