\documentclass[]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\usepackage[margin=1in]{geometry}
\usepackage{hyperref}
\hypersetup{unicode=true,
            pdftitle={Discussion Draft},
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{0}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

%%% Use protect on footnotes to avoid problems with footnotes in titles
\let\rmarkdownfootnote\footnote%
\def\footnote{\protect\rmarkdownfootnote}

%%% Change title format to be more compact
\usepackage{titling}

% Create subtitle command for use in maketitle
\newcommand{\subtitle}[1]{
  \posttitle{
    \begin{center}\large#1\end{center}
    }
}

\setlength{\droptitle}{-2em}

  \title{Discussion Draft}
    \pretitle{\vspace{\droptitle}\centering\huge}
  \posttitle{\par}
    \author{}
    \preauthor{}\postauthor{}
    \date{}
    \predate{}\postdate{}
  

\begin{document}
\maketitle

We performed hypothesis testing of models from the literature and
adapted here fit the main results from this study. We choose to examine
three modifier models and two functional and cell physiology models
(table X). It is a bit challenging to test all of these models together
since they were built with difference parameters and built to describe
different patterns of recombination variation. The modifier models were
built to explain heterochiasmy variation between sexes and the COM
functional model was built to describe the interference pattern and
difference in the recombination landscape between the sexes. In cases
where the models can't be extended to predicting results, we list as
`NA'.

\section{Reversed heterochiasmy
direction}\label{reversed-heterochiasmy-direction}

The common direction of heterochiasmy is female biased, however male
biased heterochiasmy examples are not especially rare. Our results are
novel from previous examples of male biased heterochiasmy species
because the panel of strains have much shorter evolutionary distances.
This context gives us insights to sex-specific evolution of
recombination rates.

The prediction from two of the modifier models, haploid selection and
SACE, state that males will generally evolve lower recombination rates.
Male biased heterochiasmy is predicted under the two locus modifier
model when XXXX.

There are no prediction for male biased heterochiasmy evolution
direction under the COM , but it should be noted that this model was not
designed to incorporate this type of data. This model was design to
explain the positive interference pattern for of the recombination
landscape by incorporating a cell mechanics of rapid prophase movements
(RPM) that are thought to facilitate homolog pairing in many species.

Under the spindle based model faster male evolution in male genome wide
recombination rates is primarily driven the strength of the spindle
assembly checkpoint (SAC) .

The genes and pathways are will known, we refer the reader to (Lane and
Kauppi 2019).

The transition from metaphase to anaphase is irreversible, so all
bivalents in the cell have to be aligned correctly to prevent
anueploidy.

Unstable bivalents are detected on the MI spindle, the SAC delays the
cell cycle which potentially allows time for lose bivalents to correctly
attach to the MT spindle. If the `unattached' persists apoptosis will be
triggered. Genetic variants which increase the number of unstable
bivalents, will be selected against in favor for genetic variants
causing faster stabilization of bivalents on the MI spindle. Thus the
SAC can lead to stabilizing selection on the chromosome structure.

Sperm have lower anueploidy rates compared to oocytes (other lines of
evidence) indicating that the `strength' of the SAC varies between
oogenesis and spermatogenesis. In sperm single unpaired bivalents can
trigger the SAC. While in females more oocytes progress through meiosis
even with unpaired bivalents (refereed to as leaky SAC).

These differences in SAC strength have been connected to conserved
features of gametogenesis; centrosome spindle and cell volume (cite)
which result in differing morphology of the metaphase microtubule
spindle.

Directional selection could be the result in spermatogenesis if the
fitness landscape for `chromosome structure' is altered via genetic
architecture changes or GxE effects. (in oogenesis the leaky SAC has
weak selection coefficient)

\section{Greater between cell
variation}\label{greater-between-cell-variation}

Our results of greater between cell variation in females, these results
are supported by human and mouse (Gruhn, Lynn Kong). (Lynn et al. 2002).

These data are also scant, between cell variance isn't often measured
especially for both sexes. (given they require single cell observations
for both sexes).

While the modifier models parameterize sexual dimorphism in the strength
of selection which can shape the variance in a trait these types of
models make predictions for variance between individuals but not within
individuals where the genetics are constant. (selection on penetrance).

In the spindle based selection model, the source of between cell
variance is attributed to difference strength of checkpoints. As
mentioned above, there is sexual dimorphism in the stringency of the
spindle assembly checkpoint (SAC). And these are linked to conserved
features of gametogenesis.

The opposite dynamics of the purifying selection for the `strict' SAC in
spermatocytes is relaxed selection from a `permissive' SAC. As mentioned
above a fundamental sex difference in the metaphase spindle is the
presence of centrioles in spermatocytes which anchors and nucleates the
microtubulue spindle at the two cell poles. While oocytes lack
centrioles which results in a difuse network of microtubules with
mulitple microtubule organizing centers (aMTOCs) (Schuh Ellenburg 2007).
This or larger cell volumne may lead/be a potnetial source of the leaky
SAC in oocytes. Leaky SAC, relaxed selection on the anaphase to
metaphase transition could increase the between cell variance for
crossover number

\section{Typical landscape}\label{typical-landscape}

Our results on the broad scale pattern of crossover placement are
supported by the literature. As reviewed in (Sardell and Kirkpatrick
2020), sexual dimorphism in the broad scale recombination landscape is a
highly conserved trait.

The \textbf{two locus modifier} and \textbf{SACE} model make predictions
for sexual dimorphism in the broad scale recombination landscape, for
diminishing the effect of drive systems and maintaining larger
chromosome blocks respectively.

The two mechanical models also both predict sexual dimorphism in the
recombination landscape for distinct reasons. The COM model (chromosome
pairing based model), predicts the sex differences in the positioning of
crossovers is due to a combination of the length of the axis and
intensity of the rapid prophase movements (RPMs) (Hultén 2011, Rubin,
Macaisne, and Huynh (2020)).

Under the \textbf{spindle based model} we hypothesize that the sexual
dimorphism in the recombination landscapes hinges on sex differences on
the requirements for chromosome cohesion in late meiosis I. The
irreversible process of the metaphase to anaphase transition is
initiated by the protyltic decay of the sister cohesion connecting
homologs (cite). The number and placement of crossovers alter the
distribution of sister cohesion and the resulting `chromosome structure'
when bivalents are aligned on the metaphase plate (cite). The male
crossover telomere bias could be adaptive in order to synchronize (or
minimize) the timing of homologs separating at metaphase I. Differing
lengths of the timing of cell cycle between oogenesis and
spermatogenesis may impose the the different selective pressures. Faster
spermatogenesis may select for synchronization of the separation
homologs or faster separatation at the metaphase to anaphase transition.
While in oogenesis, the slower cell cycle and prolonged arrest may
require chromosome structures with greater stability on the MI spindle.

\section{Chromatin Structure}\label{chromatin-structure}

Our results from musculus\textsuperscript{MSM} and
musculus\textsuperscript{PWD} show that the SC length and chromatin
compaction is uncoupled from the direction of heterochiasmy. These
results slightly depart from predictions which nominate chromatin
organization (axis length) as the primary driver of heterochiasmy
(Petkov et al. 2007). While the relationship may hold for more
heterochiasmy patterns, but these results indicate that rapid male
specific evolution in gwRR, is through a different trait than chromatin
compaction).

Only the COM model predicts sexual dimorphism in chromatin structure
will be longer in oocytes due to greater cell volume. This prediction /
model might fit broader pattern such as in At where pollen is the larger
cell and has longer axis length (Cahoon and Libuda 2019).

\section{Evolution of Interference}\label{evolution-of-interference}

\textbf{we find support that interference strength has evolved in the
two groups of male strains. There is a small number of positive
correlations between genome wide recombination rate and interference
strength in the literature.}

Interference is a fundamental aspect of the recombination landscape. --
genome-wide recombination rate and chromatin compaction

Some basics of the constraints bounds of interference are known, yet
there are still many unknowns regarding the relationship of genome wide
recombination rate and interference strength.

Our results support an opposite pattern predicted from the `logical
model', a positive correlation between interference strength and genome
wide recombination rates; The within sex comparison of two breeds of
cattle with different genome wide recombination rates (Ma et al. 2015),
between lab and wild mice of Peromyscus leucopus, (Peterson, Miller, and
Payseur 2019).

Logically a negative correlation is expected; increasing the number of
crossovers across chromosomes would most logically be done by more
densely spacing crossovers along chromosomes and decreasing interference
strength. This pattern has empirical support from the most species (Otto
and Payseur 2019) and fits well with the fundamental relationship
between the SC area or axis length, the physical upper limit for the
number of crossovers. \textbf{although it is not particulary strong in
our results} We argue that this is due to difference in scale (genome
wide vs single chromosome)

Theoretical models haven't really considered the evolution of
interference strength, neither the haploid selection or two locus
modifier model can not be applied to evolution of interference strength.
While the SACE modifier model does not explicitly model evolution of
interference strength we note that a logical outcome of the main
prediction of maintaining larger chromosome blocks in males, would be a
landscape with stronger interference strength. The COM model predicts
that interference and the recombination landscape arises from known
oscillatory movements during prophase, it lacks evolutionary based
predictions. We propose the spindle based selection model would support
the evolution of interference strength in the positive direction via
modulation of the amount of sister cohesion connecting homologs
(figure).

Models from Goldstein et al (review in (Otto and Payseur 2019) suggest
that if this pattern is widespread interference evolves whenever
increased recombination rates evolve. Perhaps a distinguishing feature
of models which come to this finding is that the number of crossovers is
kept constant. The space across multiple loci (veller?) or between
multiple crossovers increases in a positive manner with genome wide
recombination rates. Given that the empirical range of crossovers per
chromosome is quite small (1-3 (Otto and Payseur 2019, Stapley et al.
(2017)) and the obligate crossover rule, the assumption of constraining
the number of crossovers per chromosome fits well with empirical data.

Biological exceptions to interference: experimentally increasing
crossover number through mutants or fusion chromosomes (Celegans,
(plants, fungi with negative interference.

\section{REVIEW}\label{review}

We propose that the spindle based selection model is the most
parsimounious with our results and known recombination patterns from the
literautre.

\begin{itemize}
\item
  It matches fine scale differences in how DSB hotspots had different
  sex specific strengths with varied due to sex, epigenetc landscape and
  the gene architecture (Brick et al)
\item
  incorporates know cellular / molecular mechanisms with a very common
  pattern sexual dimorphic recombination rate landscape.
\item
  extends modifer model to explain a chromosome based / level pattern
  \emph{weaknesses}, and ways the model diverges from our results
\item
  the simulation present in the paper are too weak to support the rapid
  spread of such a modifer system (selection coefficients are too weak
  to support the evolution -- spread of such a modifer.)
\item
  Other ways this model departs from our results, -Always predicting
  male will be lower recombination rates (or risk breaking up the
  epistatsis blocks) -- which doesn't fit with our results of faster
  male specific evolution in gwRR. However / while not explicitaly
  arameterized or predicted in this model, -- this model predicts larger
  chuncks of chromosomes will segregate together (as outlined in Veller
  et al) our results of stronger male interference and stronger
  interference in high rec strains have the same consequences as
  predited from this modle
\end{itemize}

\section{Future Steps}\label{future-steps}

Be aware of sex-specific and sex average biases. Some cytlogy models
(Celegans) mostly focus on oogenesis and genetic maps are often from
hermaphrodites. (lack of data from one sex). Recognize the value of
Chromosome level measures and incorporating chromosome behavior into
models.

Borader communication across sub-fields of cell physiology and
biophysics for understanding recombination rate evolution. Utilize
mitosis systems given the conservation of the molecular pathway. THink
of meiosis as a program.

Dark matter of the genome. We can't really explude that some of the
recombination rate variation is due to strucutreal variation (more of
less genome). (since everything is mapped to B6). (also maybe there are
crossovers in the centromeres and telomeres -- which might escape
detected in linkage maps)

what are the rates of sister-sister crossovers?

How does elevated rates of precocious sister centromere seperation
effect oogenesis?

Are selfish elements affecting the evolution of meiosis -- and thus
recombination rates (from a functional standpoint).

Keep generating those testable hypotheses (cite).

\section{blah}\label{blah}

The SAC 'read's that kinetochores (the movement machinery built on top
of the centromere) are connected to the MT of the spindle originating
from the correct pole of the cell. (tension is )

centromeres have single and opposite pole connection (synthellic?) and
that the sister cohesion connecting the homologs -- sets up the tension
-- opposes the pull from each of the poles.

Efficient normal SAC -- works by pausing entry into anaphase or triggers
apoptosis -- if a unpaired / tension-lacking tetrad is detected (sperm
SAC is finer tuned (can detect 1-2 tetrads lacking tension -- ), while
the female SAC (has a higher threshold for being triggered (4-5?) (Land
Kauppi,

(relaxed selection - / inefficient SAC -- in females -- fits with higher
rate of achiasmata bivalents (there might be alternative mechanisms
facilitating achiasmate segregation)

Mechanisms for sex differences in the strength of the SAC are the cell
volumne (diffusion of signal molecules) and the centrosome being present
at the spindle have been proposed in the literature. (testing / proving
the mechanism for the new optima is for the high recombining group in
males is out of the scope of this paper, but we propose some hypothesis
are suggested below. but the evolutionary pattern fits that of
directional.

\section*{References}\label{references}
\addcontentsline{toc}{section}{References}

\hypertarget{refs}{}
\hypertarget{ref-cahoonLibuda2019}{}
Cahoon, Cori K, and Diana E Libuda. 2019. ``Leagues of Their Own:
Sexually Dimorphic Features of Meiotic Prophase I.'' \emph{Chromosoma}.
Springer, 1--16.

\hypertarget{ref-hulten2011_COM}{}
Hultén, Maj A. 2011. ``On the Origin of Crossover Interference: A
Chromosome Oscillatory Movement (Com) Model.'' \emph{Molecular
Cytogenetics} 4 (1). Springer: 10.

\hypertarget{ref-laneKauppi2019}{}
Lane, Simon, and Liisa Kauppi. 2019. ``Meiotic Spindle Assembly
Checkpoint and Aneuploidy in Males Versus Females.'' \emph{Cellular and
Molecular Life Sciences} 76 (6). Springer: 1135--50.

\hypertarget{ref-lynn2002}{}
Lynn, Audrey, Kara E Koehler, LuAnn Judis, Ernest R Chan, Jonathan P
Cherry, Stuart Schwartz, Allen Seftel, Patricia A Hunt, and Terry J
Hassold. 2002. ``Covariation of Synaptonemal Complex Length and
Mammalian Meiotic Exchange Rates.'' \emph{Science} 296 (5576). American
Association for the Advancement of Science: 2222--5.

\hypertarget{ref-ma2015_cattle}{}
Ma, Li, Jeffrey R O'Connell, Paul M VanRaden, Botong Shen, Abinash
Padhi, Chuanyu Sun, Derek M Bickhart, et al. 2015. ``Cattle Sex-Specific
Recombination and Genetic Control from a Large Pedigree Analysis.''
\emph{PLoS Genetics} 11 (11). Public Library of Science.

\hypertarget{ref-ottoPaysuer2019}{}
Otto, Sarah P, and Bret A Payseur. 2019. ``Crossover Interference:
Shedding Light on the Evolution of Recombination.'' \emph{Annual Review
of Genetics} 53. Annual Reviews: 19--44.

\hypertarget{ref-peterson2019}{}
Peterson, April L, Nathan D Miller, and Bret A Payseur. 2019.
``Conservation of the Genome-Wide Recombination Rate in White-Footed
Mice.'' \emph{Heredity} 123 (4). Nature Publishing Group: 442--57.

\hypertarget{ref-petkov2007}{}
Petkov, Petko M, Karl W Broman, Jin P Szatkiewicz, and Kenneth Paigen.
2007. ``Crossover Interference Underlies Sex Differences in
Recombination Rates.'' \emph{Trends in Genetics} 23 (11). Elsevier:
539--42.

\hypertarget{ref-rubin2020mixing}{}
Rubin, Thomas, Nicolas Macaisne, and Jean-René Huynh. 2020. ``Mixing and
Matching Chromosomes During Female Meiosis.'' \emph{Cells} 9 (3).
Multidisciplinary Digital Publishing Institute: 696.

\hypertarget{ref-sardell_sex_2020}{}
Sardell, Jason M., and Mark Kirkpatrick. 2020. ``Sex Differences in the
Recombination Landscape.'' \emph{The American Naturalist} 195 (2):
361--79. doi:\href{https://doi.org/10.1086/704943}{10.1086/704943}.

\hypertarget{ref-stapley_variation_2017}{}
Stapley, Jessica, Philine G. D. Feulner, Susan E. Johnston, Anna W.
Santure, and Carole M. Smadja. 2017. ``Variation in Recombination
Frequency and Distribution Across Eukaryotes: Patterns and Processes.''
\emph{Philosophical Transactions of the Royal Society B: Biological
Sciences} 372 (1736): 20160455.
doi:\href{https://doi.org/10.1098/rstb.2016.0455}{10.1098/rstb.2016.0455}.


\end{document}
