---
title: "Discussion Draft"
date: ''
output:
  html_document:
    df_print: paged
    toc: yes
    toc_depth: '3'
    toc_float: yes
  pdf_document: default
  word_document:
    toc: yes
    toc_depth: '3'
bibliography: refs_Main_Report.bib
---


```{r format.Notes, include=FALSE}
#table, wite by all of the columns and rows

#over-view of the synthesis of hypothesis testing
#(this is how we're synthsizing / testing / the framework were using to compare the results to the literature)

#constrast on model types
#-evolution
#-models accounting for chromosome based traits

#HetC traits
# 1. typical landscape
# 2. female variance
# 3. chromatin organization

#faster male evolution traits
# 1. Reversed Direction of Heterochiasmy
# 2. correlated gwRR increase and interference strength

#go through each traits and point out which models support each of the traits
```

other COM, / nucleus membrane citations (Harper, Golubovskaya, & Cande, 2004; Naranjo & Corredor, 2008; Scherthan et al., 1996; Zickler & Kleckner, 2016).

telomere guided movements  citations: (Bass et al.,
2000; Chacon, Delivani, & Tolic, 2016; Curtis, Lukaszewski, & Chrzastek, 1991; Ding, Yamamoto, Haraguchi, & Hiraoka, 2004; Gerton &
Hawley, 2005; Lee, Conrad, & Dresser, 2012; Lefrancois, Rockmill,
Xie, Roeder, & Snyder, 2016; Page & Hawley, 2003)

Repair of DSB - CO
(Anderson & Stack, 2005; Bass et al., 2000; Brown et al., 2005; Croft & Jones, 1989; Higgins, Osman, Jones, & Franklin, 2014; Klutstein & Cooper, 2014, Lukaszewski, 1997; Pratto et al., 2014; Viera, Santos, & Rufas, 2009;
Xiang, Miller, Ross, Alvarado, & Hawley, 2014).

We performed hypothesis testing of models from the literature and adapted here fit the main results from this study. We choose to examine three modifier models and two functional and cell physiology models (table X). It is a bit challenging to test all of these models together since they were built to describe different patterns of recombination variation. The three modifier models were built to explain variation in genome wide recombination rates between sexes (heterochiasmy), while the COM model was built to describe the interference pattern and difference in the recombination landscape between the sexes. In cases where the models can't be extended to predicting results, we list a 'NA' for the prediction.


# Reversed heterochiasmy direction

```{r reversed.HetC, echo=FALSE}

#our results of reversed hetC are novel due to the short evolutionary disances across the panel of mouse strains

#spindle selection modle points

# -- counter points, whats the consequence of reverse heterochiasmy (rapid male evolution in gwRR), 
# -- sex-average / species average of gwRR is higher (for these strains)
# -- 

```

Female biased heterochiasmy is the most common pattern observed, however male biased heterochiasmy is not especially rare [@brandvain2012scrambling]. Our results of male biased heterochiasmy is novel due to the shorter evolutionary distances between the examined strains.

The direction of heterochiasmy under the two locus modifier model is dependent on female specific recombination modifier suppresses or increases recombination, if the modifier is linked to the driving locus, and the meiotic stage (reduction or equatorial division) that the driver acts (see Table 1 of @brandvain2012scrambling).

Under the spindle based model, faster male evolution of the genome wide recombination rate, is primarily driven by a more efficient spindle assembly checkpoint (SAC) acting on chromosome structures at metaphase I alignment. <We refer the reader to [@laneKauppi2019] for more details on the SAC pathway.> The SAC prevents anueploidy by ensuring all bivalents are correctly attached to the microtubulue spindle before starting the irreversible metaphase-to-anaphase transition (cite).  A stable chromosome structure, known as bi-orientation, requires, i) kinetochores attached to opposite poles of the cells and ii) at least one crossover to create tension across a proportion of sister cohesion (cite). Unstable bivalents on the meiosis I spindle, cause the SAC to delay anaphase and allow time for the bivalent to stabilize on the microtubule spindle. If they fail to attach, the SAC triggers apoptosis. Genetic variants which increase the number of unstable bivalents (such as those affecting recombination rate or cohesion), will be selected against in favor for genetic variants causing faster stabilization of bivalents on the MI spindle. Multiple lines of evidence indicate that the effectiveness of SAC in spermatogenesis is greater compared to oogenesis (LaneKauppi, Subrahman). 
These differences in SAC strength have been connected to conserved features of gametogenesis; centrosome spindle and cell volume (cite). This model nominates conserved differences in in the context of meiosis between males and females as the primary source of distinct evolutionary trajectories in genome wide recombination rates.


Regardless of if these hypotheses turn out to be true, our results extend previous reports (dumont) of rapid male-specific evolution in multiple subspecies and strains of house mouse.

whether these instances of evolution are due to seperat independent events or segregating genetic variation within the population requires further study.


These results may hold importance to animal breeding programs -- may suggest that even for species with female biased heterochiasmy, selecting for higher male specific genome wide recombination rates may be the most efficient way to increase the sex-averaged recombination rates. This could be important application for animal breeding programs (Battagin et al 2016).



# Greater between cell variation


```{r within.mouse.var, include=FALSE}
#review results, other data

#2. Greater between cell variation in females, leaky SAC in oocytes can cause more between cell variation within an individual.

#2a. relaxed selection on chromosome structure in oocyte / leaky SAC, can lead to larger variance in crossover number between cells. The opposite dynamics of the purifying selection for 'strict' SAC in spermatocytes.

#?2b. (leaky SAC could be adaptive/prevent to selfish genetic elements (Lampson))?

# consequence of greater female variance -- (between cell)
```


Our results of greater between cell variation in females are supported by empiracal results from human and mouse [@lynn2002, @gruhn2013].

While the modifier models parameterize sexual dimorphism in the strength of selection which can shape the variance in a trait these types of models make predictions  for variance between individuals but not within individuals where the genetics are constant.  

In the spindle based selection model, the less efficient SAC in oocytes leads to relaxed selection on the metaphase I chromosome structure. This will result in increased variance in chromosome structure and crossover number per cell, not just across strains, but across cells within individuals. Oocytes at metaphase I, have an acentriolar spindle comprised of  diffuse network of microtubules with multiple microtubule organizing centers (Schuh Ellenburg 2007). This is another way the spindle based selection model is linked to conserved sex differences in gametogenesis (cite).

<stabilization -- bi-oriented, robust SAC response, SAC acts to correctly time metaphase to anaphase transition -- only after correct alignment and bi-orientation of bivalents in the metaphase cell, -- spindle dynamics much different due to asymmetrical division Mogessie et al 2018, Mogessie Schuh --actin and MT are apart of oocyte spindle>

<Bennabi et al 2016 --spindle assembly inoocytes, sun and kim 2011 -- SAC and regulators, Kolano et al 2012 --error prone oocytes, Manil-Segalen -- forces through MT breaking chromosomes, Mogessie et al 2018 -- meiotic spindle (big) review>

0. Reguardless of whether these models and hypothese are true if the pattern of females having more across cell variation in gwRR

1. It has a consequence of fitting with distinct evolutionary streategies between the sexes of females diversifying the genetics their offspring and males limiting the shuffling of successful combinations (Trivers).



# Typical landscape

```{r typical.landscape.outline, include=FALSE}

#<spindle model -- the telomeric placement of crossovers, minimizes the amount of sister cohesion connecting homologs>
#3. Sexual dimorphism in broad scale recombination landscape,  telomere bias vs uniform recombination landscapes fit with purifying selection and relaxed selection respectively.
#3a. There are sex differences in demand for chromosome cohesion at MI, the trait / variable causing variation in fitness landscape across chromosome structures might be the amount of sister-cohesion connecting homologs.  In oocytes the cell cycle is slower and arrested in some species, this likely selects for chromosome structures that have stability on the spindle. (required to start with more since the cohesion isn't replenished. The optimal chromosome structure might have more sister cohesion, however given the leaky SAC, selection may not be effective.

#3c. Logic of number and placement modulating this area of sister cohesion and chromosome structure (I plan to make a figure to illustrate)
#-For single crossover acrocentric chromosome the sister cohesion distal to the crossover connects homologs. This area reduced when crossover position shifted to telomere (each end of chromosomes). Two crossover acrocentric chromosome the area is distal from the first until the second crossover.

#3d. Metaphase to anaphase transition proceeds by proteolytic decay of sister cohesion on 'arms'.
#- The telomere bias could have an effect of synchronizing the separation of all the bivalents within a cell.

#0. we find the conserved pattern of the typical landscape 
#this trait is a highly conserved aspect of heterochiasmy 
#the exceptions include marsupials which form sex chromosome chains during meiosis -- indicating a potential change to the meiotic program (functional selection)
#it's not known if a similar -- evolutionary force in the other exceptions to the typical landscape, pig and tomatoe.

#regaurdless
#As reviewed in sardel kirkptarick - diviation for a uniform background is generally due to the male recombination landscape.

# these pattern due to (different recombination landscapes along chromosomes can have an impact on the pattern of Fst (Hanel et al 2018)
# additionally other patterns of nucleotide and recombination rates could vary across

#co rate defines the physical scale of linked selection around a locus

# evolutionary genomics from the broad scale heterogenity
#linked selection, (maynarid smith and haig 1974)
#background selection (only with negtive allele), genetic draft (for positive selective mutations)
#gene flow barrier (directional selection is keeping two populations from merging)
# patterns of Fst (Hanel et al, but see Kartej?)

# 


# source could be DSB formation along chromosome - and /or biases in DSB repair 

#CO suppresion, elevated DSB (most repaired as NCO) -- elevated DsB (Brick)


```

Our results on the broad scale pattern of crossover placement are supported by the literature. As reviewed in [@sardell_sex_2020], sexual dimorphism in the broad scale recombination landscape is a highly conserved trait. The two locus modifier and SACE model predicts sexual dimorphism in the broad scale recombination landscape, for diminishing the effect of drive systems and maintaining larger chromosome blocks under linkage in males respectively.  

The two mechanical models both predict sexual dimorphism in the recombination landscape for distinct reasons. The COM model, predicts the sex differences in the recombination landscape via interference is due to a combination of the length of the axis and differing strengths of chromosome movements during the pairing process [@hulten2011_COM, @rubin2020mixing].  

Under the spindle based model we hypothesize that the sexual dimorphism in the recombination landscapes hinges on sex differences on the requirements for chromosome cohesion in late meiosis I. The irreversible process of the metaphase to anaphase transition is initiated by the protyltic decay of the sister cohesion connecting homologs (cite). The number and placement of crossovers alter the distribution of sister cohesion and the resulting chromosome structure when bivalents are aligned and bi-oriented on the metaphase plate (cite). Different timing of cell cycle between oogenesis and spermatogenesis may impose the the different selective pressures on sister cohesion. Faster spermatogenesis may select for synchronization of the separation homologs. While in oogenesis, the slower cell cycle and multiple arrest stages may require chromosome structures with greater stability on the MI spindle.

<The male crossover telomere bias could be adaptive in order to synchronize (or minimize) the timing of homologs separating at metaphase I>

If these hypotheses concerning evolution of these patterns **are not true**  there's still predictions that can be made given the consequences of these patterns. The consequences of this difference in the typical landscape as reviewed in [@sardell_sex_2020]. For the sex averaged landscape along chromosomes, any diviation from uniform placement is generally due to male specific patterns.

There's also the prediction that females have active CO supression near telomeres (variation) -- geiven the evidence in (female mice) that show a higher elevation of DSBs at the telomere (halldorsson?).


# Chromatin Structure


```{r chromatin.organization, include=FALSE}

#supporting results from lit
#chromatin organization

#models
# none of the models make predictions for the axis length (chromatin organization)
# only COM predicts that sexual dimorphism in axis length is due to cell vol

# consequences of longer female axis - 
# (normalzed interference strength due to axis differences / Chromtain loop length)

# significance, consequences
# weak evidence of sex differences in chromatin organization, mice-humans have a similar direction -- but At opposite -- not much empiracal evidence. (more data needed)

# a consequence of these sex differences for the chromatin organization is -- has an outcome of different effects on interference strenth measured in genetic map space -- for a given chromosome, chromatin organization of longer axis will result in shorter loops (emenating off the central axis)


#4.ppl should care -- this is an important aspect of -- how sexual dimorphism in recombination /linkage maps has been charaecterized -- but due to a lack of cytological evidence in both sexes /linking this to chromatin organization has only been done for a handful of species (mice, human, At? Dourad)

```


Our results from musculus^MSM^ and musculus^PWD^ show that the SC length and chromatin compaction is uncoupled from the direction of heterochiasmy. These results slightly depart from predictions which nominate chromatin organization (axis length) as the primary driver of heterochiasmy [@petkov2007]. While the relationship may hold for more general heterochiasmy patterns across broad scale comparisons, these results indicate that rapid male specific  evolution in gwRR, is through a different trait than chromatin compaction. 

<The results of sexual dimorphism in chromatin structure is supported by the literature, especially for mammals [@lynn2002, @gruhn2013], But there's generally a lack of data for other systems see [@cahoonLibuda2019]>  

Only the COM model predicts sexual dimorphism in chromatin structure will be longer in oocytes due to greater cell volume. This prediction model might fit broader pattern such as in *Arabidopsis thalania* where pollen is the larger cell and has longer axis length [@cahoonLibuda2019].  


(also outlined in Hanel et al 2018 -- which connects to typical landscape)

A consequence of sexual dimorphism chromatin organization is that interference strength quantified in base pairs will be sexually dimorphic. There are two aspects that define this relationship. First the density of tethering of loops to the axis is highly conserved (Zick Kleck), meaning variability of the axis length of a given chromosome comes from variation in the loop sizes (Z n K).  Second, the signal of crossover interference is mediated through the chromosome axis or SC (kleck Dernberg). Thus the chromatin organization with larger loops (and shorter axis) will have stronger crossover interference when quantified in base pairs, since more base pairs are sequestered in the chromatin loops per unit of axis or SC. Empirical data confirms that interference strength measured in micrometer along the axis is conserved between sexes while interference strength quantified from genetic map is sexually dimorphic, comes from mice [@petkov2007], humans (Gruhn, Lynn). However, due to lack of relevant cytology data in both sexes it is not clear if the many examples of sexual dimorphism in linkage maps is due to chromatin organization.


# Evolution of Interference


```{r evo.interference, include=FALSE, echo=FALSE}

#models that predict this pattern (CONTROL FOR CO NUMBER -- PRIORITIZE THE WEIGHT /PROB of CO ASSURANCE)
#the pattern is driven in part by short IFD (when short IFD removed in reduced data set -- the low strain mean goes up, but they are still shrter)

#**The rapid evolution of the genome wide rate in our high rec strains is PARTLY due to more enrichment of 2CO which have distinct rec landscape from the 2COs in low rec strains.**

#5. Interference evolution shows a positive correlation with gwRR
#5a. Stronger interference for two crossover chromosomes, increases the amount of sister-cohesion connecting the homologs.
#5b. Under spindle selection model, the fitness landscape of optimal chromosome structure may have changed in the high recombining strains resulting in more 2CO, with larger IFD (sister cohesion area).

# consequence of rapid evolution of normalized interference strength,
# in the 3 strains -- larger linkage blocks?
# but what if the rec pattern is repeated?
```


Examples of evolution of interference strength are rare and may be complicated by changes in the underlying karyotype structure (Segura et al 2014). The general pastern of empirical measures of interference strength is that it has a negative correlation to the genome wide recombination rate [@ottoPaysuer2019]. This fits the logic of the chromosome axis acting as the 'real estate' for DSBs and aligns with known molecular mechanisms (cite). Two examples of the opposite direction, a positive correlation of interference strength and genome-wide recombination rate, involve descriptions of observations at the single bivalent level; the between lab-raised and wild mice of *Peromyscus leucopus* from [@peterson2019] or large effects on genome wide recombination rates across a short evolutionary differences; the within-sex comparison of two breeds of cattle with different genome wide recombination rates [@ma2015_cattle]. We propose that there is a difference in resolution and power between these two groups of empirical results with the positive and negative relationship between genome wide recombination rates and interference strength.


We propose that the **spindle based selection model** can explain the positive correlation and evolution of stronger interference strength via selection on the amount of sister cohesion connecting homologs (figure X, table X). The evolution of genome wide recombination rates in our study is driven by a transition of recombination landscapes from single crossovers to two crossovers. All else being equal, interference strength would be expected to be equal or even weaker. Our results show that the two crossovers are spaced further apart when SC length is controlled for and chromosome size effects are minimized, in the strain with more two crossover chromosomes. Assuming that chromatin compaction between pachtyene and metaphase is uniform along chromosomes, an outcome of this further spacing of two foci results in an increased area of sister cohesion connecting homologs (Figure X). This model gives rise to the possibility that the chromosome structure at metaphase is the target of selection  which is achieved by changing the average recombination landscape across bivalents within the cell. This interpretation has potential to fit with hypotheses mentioned above for explaining other results regarding the evolution of the recombination landscape.  


The **COM** model predicts that interference and the recombination landscape arises from known oscillatory movements during prophase, however it lacks evolutionary based predictions. While the **SACE** modifier model does not explicitly model evolution of interference strength we note that a logical outcome of the main prediction of maintaining larger chromosome blocks in males, would be a landscape with stronger interference strength. 


We have focused on the model involving direct selection on the recombination pathway, that is the impact on the meiotic process on an individual, but we make note on models centering on indirect selection or fitness outcomes of offspring which could also explain our results. Models from Goldstein et al, reviewed in [@ottoPaysuer2019] and Veller 2017, predict that interference strength evolves whenever increased recombination rates evolve. The simulations under this model indicate that the space across multiple loci or between multiple crossovers increases in a positive manner with the genome wide recombination rate. A distinguishing feature of their model is that the number of crossovers per chromosome is kept constant. The range of crossovers per chromosome is quite small, 1-3, for the majority of chromosomes [@ottoPaysuer2019, @stapley_variation_2017]. Constraining the number of crossovers per chromosome in models may fit empirical data better than those where recombination rate across an abstracted genetic space is unconstrained.


None of these models may be appropriate due to X Y and  Z caveats. The consequence of this pattern of greater interference strength is ---
 consequence of rapid evolution of normalized interference strength, larger linkage blocks.
 
 
#Future steps

importance of evolutionary distinct trajectories, measure both sexes -- do evolutionary comparisons within sex.

wild samples, short times scales.

single cell measures

chromatid interference 

integrate physical models -- centromere and kinetochore



Diversity of systems for overcoming limitations

1. identifying indiviual (autosomes) Chromosomes across cells: FISH (diffecult) or karyotypes with identifyable chromosomes (either into groups or largest)(birds, zfish?, peromyscus, Rbt mice, bovids?, humans, ), translocations (robertsonian translocations), 


2. non-model orgs can be used to study meiosis -- look at the weird ones (protezoa)



```{r future.step.plan, echo=FALSE, include=FALSE}
#still preliminary
#sex-specific and sex average biases. Some cytology models (Celegans) mostly focus on oogenesis  and genetic maps are often from hermaphrodites. (lack of data from one sex). Recognize the value of Chromosome level measures and incorporating chromosome behavior into models.
#Broader communication across sub-fields of cell physiology and biophysics for understanding recombination rate evolution. Utilize mitosis systems given the conservation of the molecular pathway. Think of meiosis as a program.

#Dark matter of the genome. We can't really exclude that some of the recombination rate variation is due to structural variation (more of less genome). (since everything is mapped to B6). (also maybe there are crossovers in the centromeres and telomeres -- which might escape detected in linkage maps)

#what are the rates of sister-sister crossovers?
#How does elevated rates of precocious sister centromere separation effect oogenesis?

#Are selfish elements affecting the evolution of meiosis -- and thus recombination rates (from a functional standpoint).

#Keep generating those testable hypotheses (cite).  
#- chromatin interference
```


<a partial pattern seen in Arabidopsis thalinia (Drouaud et al 2007) however there is no SC based measure of interference and low number of double crossovers in the female specific linkage map. It also could be that the greater error in regulation of crossovers may have driven the longer female axis to ensure increase the probability of successful maturation of DSB into crossover chiasmata>  


#References
