---
title: "MLH1 Results Outline"
date: "`r format(Sys.time(), '%d %B %Y')`"
output:
  html_document:
    df_print: paged
    toc: yes
    toc_depth: '5'
    toc_float: yes
  pdf_document: default
bibliography: refs_Main_Report.bib
---

# Results

```{r setup, echo=FALSE, include = FALSE}
library(knitr)
library(ggplot2)
#library(pwr)
library(plyr)
library(lattice)
library(dplyr)
library(raster)
library(lme4)
library(nlme)
library(RLRsim)
library(ggpubr)
library(reshape2)

#setwd("~./MLH1repo/doc/")
setwd("C:/Users/alpeterson7/Documents/MLH1repo/doc/")
#setwd("C:/Users/April/Desktop/MLH1repo")
#load(file="data/MLH1/MLH1_data_setup.RData")

#load main data file
load(file="C:/Users/alpeterson7/Documents/MLH1repo/data/MLH1/MLH1_data_setup_1.5.20.RData")#added batch18

load(file="C:/Users/alpeterson7/Documents/MLH1repo/data/CleanBivData_1.28.20.RData")

#this MLH1 only goes up to batch16!
#load(file="C:/Users/alpeterson7/Documents/MLH1repo/data/MLH1/MLH1_data_setup_12.20.19.RData")#batch18
#load(file="C:/Users/alpeterson7/Documents/MLH1repo/data/MLH1/MLH1_data_setup_12.20.19.RData")#fixing bugs, formating
#load(file="data/MLH1/MLH1_data_setup_12.3.19.RData")#removed exclude mice
#load(file="data/MLH1/MLH1_data_setup_11.12.19.RData")#added batch17
#load(file="data/MLH1/MLH1_data_setup_11.11.19.RData")#deleted DUP image
#load(file="data/MLH1/MLH1_data_setup_9.26.19.RData")#added batch16
#load(file="data/MLH1/MLH1_data_setup_8.29.19.RData") #

#load(file="data/MetaData.RData") #this was replacing most updated data file

#load functions
source("~./MLH1repo/src/CommonFunc_MLH1repo.R")
```



```{r organizing.data, echo=FALSE }
#set the order of another column, based on another variable. consistant order for ploting

#Terminology Note, leptonema, zygonema, and pachynema diplonema diakinesis, are the stage names.
#-ene endings refer to the cells/units within the cell stage? (I think)

#apply this before ploting
MLH1_data <- MLH1_data %>%
  arrange(strain, sex, category, mouse) %>%
  mutate(Original.Name = factor(Original.Name)) #another category that you want the order to match
MLH1_data <- with(MLH1_data, MLH1_data[order(sex, strain),])

# sort your dataframe, by the focal categories
MLH1_data$mouse <- factor(MLH1_data$mouse, levels=unique(MLH1_data$mouse))
#use the above line for when order is out of wak


#count the non-quality measures,  #remove non qualit
#length(MLH1_data[ !(is.na(MLH1_data$quality) | MLH1_data$quality==""), ] )  #15 rows with out quality scores, remove.
#MLH1_data <- MLH1_data[ !(is.na(MLH1_data$quality) | MLH1_data$quality==""), ]

#MLH1_by_F_strain <- MLH1_data[MLH1_data$sex == "female", ]
#MLH1_by_M_strain <- MLH1_data[MLH1_data$sex == "male", ]

#MLH1_data <- MLH1_data %>%
#  arrange(category) %>%
#  mutate(Original.Name = factor(Original.Name)) #another category that you want the order to match

#MLH1_data <- with(MLH1_data, MLH1_data[order(category),])
AP_mouse_table.HQ$var <- as.numeric(AP_mouse_table.HQ$var)

AP_category_table <-  ddply(.data=AP_mouse_table.HQ,
                 .(strain,sex),
                 summarize,
                 sum.ncells = sum(Ncells),
                 mean.MLH1 = mean(mean_co),
                 var.MLH1 = var(mean_co),
                 AV.var_MLH1 = mean(var, na.rm=TRUE)
                                  
)

#there might be an error for changing wd
#write.table(AP_category_table, "~./MLH1repo/strain_table_variances.csv", sep=",", row.names = FALSE)

AP_mouse_table.HQ_target <- AP_mouse_table.HQ[!(is.na(AP_mouse_table.HQ$sex) | AP_mouse_table.HQ$sex==""), ]

remove_strains <- c("TOM","AST","CZECH","CAST","HMI","SPRET","CAROLI", "F1", "PERC")
AP_mouse_table.HQ_target <- AP_mouse_table.HQ_target[ ! AP_mouse_table.HQ_target$strain %in% remove_strains, ]

  
var.plot <- ggplot(AP_mouse_table.HQ_target, aes(y=var, x = sex, color=sex))+geom_jitter()+facet_wrap(~strain)+ggtitle("Plots of mouse Variances for MLH1 counts")+
  scale_color_manual(values=c("royalblue", "limegreen"))+theme_bw()

```


Details of data set cleaning ect is in SetupData.rmd
Effects from different factors are reviewed in Effects.Rmd


```{r basic.numbers, echo=FALSE, eval=FALSE}

kable(table(AP_mouse_table.HQ$category), col.names = c("Category", "mice"), label = "Number of mice by category" )


kable(table(MLH1_data$category), col.names = c("Category","Cells"), label = "Number of Cells by category" )

```


```{r rand.misc, echo=FALSE, include=FALSE, eval=FALSE}
#str(AP_mouse_table_w.Ages)

AP_mouse_table.HQ$var <- as.numeric(AP_mouse_table.HQ$var)

#remove sex.x == na, and strain.x=na or F1
AP_mouse_table.HQ <- AP_mouse_table.HQ[!(is.na(AP_mouse_table.HQ$sex) | AP_mouse_table.HQ$sex==""), ]

AP_mouse_table.HQ <- AP_mouse_table.HQ[!(is.na(AP_mouse_table.HQ$var) | AP_mouse_table.HQ$var==""), ]
#plot variance and mean (facet sex)

AP_mouse_table_w.Ages <- AP_mouse_table_w.Ages[!( AP_mouse_table_w.Ages$strain.x =="F1"), ]

VandMean.mouse <- ggplot(aes(x=mean_co, y = var), data=AP_mouse_table.HQ )+geom_point(aes(color=strain))+facet_wrap(~sex)+ggtitle("mouse MLH1 averages; variance by mean")

cVandMean.mouse <- ggplot(aes(x=mean_co, y = cV), data=AP_mouse_table.HQ )+geom_point(aes(color=strain))+facet_wrap(~sex)+ggtitle("mouse MLH1 averages; cV by means")

str(AP_mouse_table.HQ)

cor(AP_mouse_table.HQ$mean_co[AP_mouse_table.HQ$sex == "female"], AP_mouse_table.HQ$var[AP_mouse_table.HQ$sex == "female"])
#[1] 0.5750399
cor(AP_mouse_table.HQ$mean_co[AP_mouse_table.HQ$sex == "male"], AP_mouse_table.HQ$var[AP_mouse_table.HQ$sex == "male"])
#0.321601

cor(AP_mouse_table.HQ$mean_co[AP_mouse_table.HQ$sex == "female"], AP_mouse_table.HQ$cV[AP_mouse_table.HQ$sex == "female"])
# 0.3515544
cor(AP_mouse_table.HQ$mean_co[AP_mouse_table.HQ$sex == "male"], AP_mouse_table.HQ$cV[AP_mouse_table.HQ$sex == "male"])
#-0.03713536

cor.test(AP_mouse_table.HQ$mean_co[AP_mouse_table.HQ$sex == "male"], 
    AP_mouse_table.HQ$var[AP_mouse_table.HQ$sex == "male"] )
# p-value = 8.623e-09

cor.test(AP_mouse_table.HQ$mean_co[AP_mouse_table.HQ$sex == "male"], 
    AP_mouse_table.HQ$cV[AP_mouse_table.HQ$sex == "male"] )


#remove mice with fewer than 10 cells

AP_mouse_table.HQ.10plus <- AP_mouse_table.HQ[AP_mouse_table.HQ$Ncells >= 10,]

VandMean.mouse.10plus <- ggplot(aes(x=mean_co, y = var), data=AP_mouse_table.HQ.10plus )+geom_point(aes(color=strain))+facet_wrap(~sex)+ggtitle("10 plus mouse MLH1 averages; variance by mean")



cor(AP_mouse_table.HQ.10plus$mean_co[AP_mouse_table.HQ.10plus$sex == "female"], AP_mouse_table.HQ.10plus$cV[AP_mouse_table.HQ.10plus$sex == "female"])


cor.test(AP_mouse_table.HQ.10plus$mean_co[AP_mouse_table.HQ.10plus$sex == "female"], 
    AP_mouse_table.HQ.10plus$var[AP_mouse_table.HQ.10plus$sex == "female"] )



#why did I male just a female table?
female.musc.cells <- MLH1_data[( MLH1_data$sex == "female") & (MLH1_data$subsp == "Musc" ), ]

female.musc.mice.table <- ddply(female.musc.cells, c("sex"), summarise,
                        Nmice = length(unique(mouse)),
                        Ncells  = length(adj_nMLH1.foci),
                        mean_co = as.numeric(format(round(  mean(adj_nMLH1.foci), 3 ), nsmall=3) ),
                        cV = cv(adj_nMLH1.foci),
                        var = format(round(   var(adj_nMLH1.foci),3), nsmall=3),
                        sd   = round(sd(adj_nMLH1.foci), 3),
                        se   = round(sd / sqrt(Ncells), 3)
                        #quality?
)


```



# A. Models for av MLH1 counts foci per cell 


ToDo: add the random effect tests!!


These are the basic Mixed Models we used as a frame work to analyze all the MLH1 count data.

$$mouse .mean CO ~=~ subsp * sex + rand(strain* sex)  + \varepsilon$$

$$mouse .var CO ~=~ subsp * sex + rand(strain* sex) + \varepsilon $$

The mixed model lets us separate effects to see how they effect a mouse mean CO count. The subspecies effect is a proxy for **Divergence** and the random strain effect is a proxy for **Polymorphism**.


Setting up the data set for using the mixed model (base data set, HQ data set)

- only Mus musculus strains with sex matched observations  

- Unorder factors  

- male ages within good range (5 - 12? weeks)

- Mol as a different subsp

- ( Nested strain in subsp?)


*The mouse average table used in this part excluded Q5 cells*

*the MLH1 mean is from mean - adj_MLH1 (males have +1 CO to count for PAR)*

```{r mixed.model.data.setup, echo=FALSE, warning=FALSE}
#https://rcompanion.org/handbook/G_03.html
#testing random effects within lme

#What are the main DFs for this code chunk
#AP_mouse_table_w.Ages   <- used to remove males which are too old (includes outgroups) 
#DF.HetC.MixedModel   <- only Mmusculus
#DF.HetC.MixedModel.HQ <- only mice of certain age, Q5 excluded


#variance is still chr
AP_mouse_table_w.Ages$var <- as.numeric(AP_mouse_table_w.Ages$var)

#use the merged DF to filter by age 
AP_mouse_table_w.Ages$strain <- AP_mouse_table_w.Ages$strain.x
AP_mouse_table_w.Ages$sex <- AP_mouse_table_w.Ages$sex.x
AP_mouse_table_w.Ages$subsp <- AP_mouse_table_w.Ages$subsp.x
#main file, make everything unordered factors
AP_mouse_table_w.Ages <- add_species(AP_mouse_table_w.Ages)

DF.HetC.MixedModel <- AP_mouse_table_w.Ages[AP_mouse_table_w.Ages$species == "M.musculus",]

#remove the extra merged cols
DF.HetC.MixedModel <- subset(DF.HetC.MixedModel, select= -c(sex.y,sex.x,
                            subsp.y,subsp.x,
                    category.y,category.x, strain.x,
                    strain.y, mouse.1  )  )
#DONT NEED CATEGORY


#REMOVE EXTRA STRAINS
DF.HetC.MixedModel <- DF.HetC.MixedModel[DF.HetC.MixedModel$subsp != "Cast",]
#remove musc strains
DF.HetC.MixedModel <- DF.HetC.MixedModel[DF.HetC.MixedModel$strain != "AST",]
DF.HetC.MixedModel <- DF.HetC.MixedModel[DF.HetC.MixedModel$strain != "TOM",]
#remove PERC, MOLF, CZECH, (no females)
DF.HetC.MixedModel <- DF.HetC.MixedModel[DF.HetC.MixedModel$strain != "CZECH",]
DF.HetC.MixedModel <- DF.HetC.MixedModel[DF.HetC.MixedModel$strain != "PERC",]

DF.HetC.MixedModel <- droplevels(DF.HetC.MixedModel)#remove levels not used

#SWITCH TO HQ.DF
#mean_co.HQ  (this might remove some of the MOLF males)
#DF.HetC.MixedModel.HQ <- DF.HetC.MixedModel[DF.HetC.MixedModel$Ncells > 9,]  #this removes the MOLF female

table(DF.HetC.MixedModel$strain, DF.HetC.MixedModel$sex)

#afriad this is removing NA's
DF.HetC.MixedModel.HQ <- DF.HetC.MixedModel[( (DF.HetC.MixedModel$age.weeks < 25) &  (DF.HetC.MixedModel$age.days != 13) ) | is.na(DF.HetC.MixedModel$age.weeks), ] #keeps mice without ages, and removes juvinilles

table(DF.HetC.MixedModel.HQ$strain, DF.HetC.MixedModel.HQ$sex)

#reformate Molosinus subsp
#mean_co.HQ.mol
DF.HetC.MixedModel.HQ$subsp <- ifelse(grepl("MSM", DF.HetC.MixedModel.HQ$strain), "Mol", 
        ifelse(grepl("MOLF", DF.HetC.MixedModel.HQ$strain), "Mol", DF.HetC.MixedModel.HQ$subsp))

# For some reason I changed subsp to nums (wtf)
DF.HetC.MixedModel.HQ$subsp <- ifelse(grepl(1, DF.HetC.MixedModel.HQ$subsp), "Dom", 
        ifelse(grepl(2, DF.HetC.MixedModel.HQ$subsp), "Musc", DF.HetC.MixedModel.HQ$subsp))

DF.HetC.MixedModel.HQ <- DF.HetC.MixedModel.HQ[!(is.na(DF.HetC.MixedModel.HQ$mouse) | DF.HetC.MixedModel.HQ$mouse==""), ]


#remove blank rows (which are these?)
#DF.HetC.MixedModel <- DF.HetC.MixedModel[!(is.na(DF.HetC.MixedModel$mouse) | #DF.HetC.MixedModel$mouse==""), ]
#151 mice, no rows with nas for values

#main df with variables coded as chr
DF.HetC.MixedModel.HQ$subsp <- factor(DF.HetC.MixedModel.HQ$subsp, ordered = FALSE)
DF.HetC.MixedModel.HQ$sex <- factor(DF.HetC.MixedModel.HQ$sex, ordered = FALSE)
DF.HetC.MixedModel.HQ$strain <- factor(DF.HetC.MixedModel.HQ$strain, ordered = FALSE)
DF.HetC.MixedModel.HQ$species <- factor(DF.HetC.MixedModel.HQ$species, ordered = FALSE)

#kable(table(DF.HetC.MixedModel.HQ$sex, DF.HetC.MixedModel.HQ$strain))
```


This is the size of mice used within the Mixed model

`r kable(table(DF.HetC.MixedModel.HQ$sex, DF.HetC.MixedModel.HQ$strain))`

Can I indedate the average number of cells with go into each mouse.?

FOR FINAL ANALYSIS, make sure that MOL is coded as a separate subsp (change all of the main model reports to). For mouse averages of MLH1 counts, only strain with observations for both sexes and are from Mus species. 

subspecies levels: 2(3), Musc and Domestics and Molossinus

strain levels: Don: 3, Musc: 4  Mol: 2

sex: 2 female male  


## Mean CO Count


todo, add the random effect tests, consider changing all to lmer() (those have better random effects)

$$mouse .mean CO ~=~ subsp * sex  + rand(strain*sex) + \varepsilon$$

```{r HQ.MixedModel.1, echo=FALSE, digits = 2}
#table(DF.HetC.MixedModel$strain, DF.HetC.MixedModel$sex)
#mean_co.HQ  (this might remove some of the MOLF males)


HetC.MixedModel.HQ <- lme(mean_co ~ subsp * sex, data=DF.HetC.MixedModel.HQ, random=list(strain=pdDiag(~sex) ) )

as.data.frame( round(  summary(HetC.MixedModel.HQ)$tTable, 6 ) )

ranef(HetC.MixedModel.HQ)
#which is the test for random effects ?!

```


Above is the table of coefficients for MM.1, HQ data set, lme(), random effects strain and strain specific sex effect.  


$$mouse .mean CO ~=~ subsp * sex  + rand(strain*sex) + \varepsilon$$

Above is the table of coefficients for MM.1.mol, molssinus coded as separate subspecies, lme(), random effects strain and strain specific sex effect.  


$$mouse .mean CO ~=~ subsp * sex  + rand(1|strain) + \varepsilon$$

```{r MixedModel.2, echo=FALSE}
#reduced models
Reduced.strain.HQ <- lme(mean_co ~ subsp * sex, data=DF.HetC.MixedModel.HQ, random = ~1|strain)
#summary(Reduced.strain.HQ)

as.data.frame(round(summary(Reduced.strain.HQ)$tTable, 6))
#add a aption to this code chunk
#ranef(Reduced.strain.HQ)



```

Above is the table of coefficients for MM.2, HQ data set, lme(), random effects coded without sex specific effect.
This is model has sex and interaction coefficient is significant.

Can the above model be re-written with lmer so that random effect can be tested?




> MM3

$$mouse .mean CO ~=~ subsp * sex * strain + \varepsilon$$

```{r MixedModel.3, echo=FALSE}
#strain = fixed
lm.strain.fixed.full <- lm(mean_co ~ subsp * sex * strain, data=DF.HetC.MixedModel.HQ)
#library(xtable)

#this table of results is broken -- might be too long -- lm
#lm has different names
as.data.frame(round( summary(lm.strain.fixed.full)$coefficients, 6))

#summary(strain.fixed.full)
#when strain added as fixed, MSM, PWD and WSB are signifcant, subsp*sex is most significant

#non of the 3-way interactions are significant, very puzzeled that WSB is significant and non of ht ehigh Musc are 
```

Above is the table of coefficients for MM.3, HQ data set, lm function, sex, strain and subsp are interacting fixed effects.  

Strain G is significant (p=0.0005), 

interaction terms of sex and strain for (G, MSM, PWD).

How are these models different from the last one -- MM4 is missing subsp, so only sex * strain effects.


> MM.4

$$mouse .mean CO ~=~ sex * strain + \varepsilon$$

```{r MixedModel.4, echo=FALSE}

strain.fixed <- lm(mean_co ~ sex * strain, data=DF.HetC.MixedModel.HQ)

#below table is broken
as.data.frame( round( summary(strain.fixed)$coefficients, 6))

#summary(strain.fixed)
#maybe this WSB effec is just in the HQ data set?
```

Above is the table of coefficients for **MM.4**, HQ data set, lm function, just sex and strain, no subspecies.

Significant factors are 
strain G (p=0.0005), strainMSM and strainPWD (very slight),
sex:G (p=0.001),
sex:MSM, Sex:PWD (very slight p=.05, p=0.3), 


#### ANOVA, mean CO count full vs reduced

Use Anova to compare full vs reduced models
the reduced models will be those missing either, strain, sex, or subsp.

```{r anova.full.v.reduced, echo=FALSE, eval=FALSE}

#strain as fixed
#CONTRAST models
#newnew <- lme(mean_co ~ sex * strain, data=DF.HetC.MixedModel )

HetC.MixedModel.mean_co <- lme(mean_co ~ subsp * sex, data=DF.HetC.MixedModel, random=list(strain=pdDiag(~sex) ) )
R.strain.HetC.MixedModel.mean_co <- lme(mean_co ~ subsp * sex, data=DF.HetC.MixedModel, random = ~1|strain)#musc-male fixed effect is now significant, and male

full.Reduced.HetC.MixedModel.mean_co <- gls(mean_co ~ subsp * sex, data=DF.HetC.MixedModel)

#base vs strain
fullVstrain <- anova(HetC.MixedModel.mean_co, R.strain.HetC.MixedModel.mean_co) #<.0001

#show scientific vs normal
#vv <- fullVstrain$`p-value`
#round(vv[2], 8)
#format(vv[2], scientific=F)

#strain vs no random full.reduce
fullVnon <- anova(HetC.MixedModel.mean_co, full.Reduced.HetC.MixedModel.mean_co) #<.0001

strain.vs.non <- anova(R.strain.HetC.MixedModel.mean_co, full.Reduced.HetC.MixedModel.mean_co) #<.0001

#fullreduce vs strain
#can't distinguish which random effects are more significant -- both tests are significant

fullVreduce22 <- anova(HetC.MixedModel.mean_co, full.Reduced.HetC.MixedModel.mean_co)

#anova for comparing reduced and full models -- 1 of the models should be strain as fixed instead
full2strain <- anova(HetC.MixedModel.HQ, Reduced.strain.HQ)#slightly significant

full.v.fixed <- anova(HetC.MixedModel.HQ, strain.fixed)
#this comparison not useful when there are different fixed effects

#full2noRandom <- anova(HetC.MixedModel.HQ, Reduced.all.HQ) # < .0001 

#strain2noRandom <- anova(Reduced.strain.HQ, Reduced.all.HQ) #NS


#anova tests
#full vs strain
full2strain.mol <- anova(HetC.MixedModel.mol, Reduced.strain.mol)# < .0001
#full vs non
full2noRandom.mol <- anova(HetC.MixedModel.mol, Reduced.all.mol)# < .0001
#strain vs non
strain2noRandom.mol <- anova(Reduced.strain.mol, Reduced.all.mol)# < .0001

anova(Reduced.strain.mol, Reduced.all.mol, HetC.MixedModel.mol)

#base vs mol model (this only makes sense if these are nested models)
anova(HetC.MixedModel.mean_co, HetC.MixedModel.mol)
```



Full model, so significant fixed effects. (For the full model, The proportions of the random effects make sense looking at the average CO data
Models with random effects more significant than nulls (both just strain and no random effects). The models with fewer random effects, the fixed effects, sex and sex*subspec (interaction) become significant.


```{r Base.strain.fixed, echo=FALSE}
#make mixed model with strain as fixed, since there are interesting results
#Consider including these results from all FIXED MODEL

HetC.MM.strain.fixed <- lm(mean_co ~ strain*sex, data= DF.HetC.MixedModel )
#this produces several significant factors

HetC.MM.strain.fixed2 <- lm(mean_co ~ (subsp/strain)*sex, data= DF.HetC.MixedModel )

#summary(HetC.MM.strain.fixed2)
#(WSB and G have the most observations, maybe this is biasing the results (for WSB?))


#summary(HetC.MM.strain.fixed)
#strainKAZ           -1.57392    0.86769  -1.814 0.071773 . 
#strainPWD           -1.98205    0.74191  -2.672 0.008420 ** 
#strainWSB           -3.09482    0.74191  -4.171 5.21e-05 *** 
#sexmale             -2.89186    0.73333  -3.943 0.000125 ***
#strainMSM:sexmale    6.83426    1.16149   5.884 2.69e-08 *** 
#strainPWD:sexmale    7.04433    1.11783   6.302 3.38e-09 ***  
#strainSKIVE:sexmale  4.04919    2.38756   1.696 0.092056 .   
#strainWSB:sexmale    2.85301    1.04317   2.735 0.007024 ** 

#male effect lowers the mean from intercept
#while strain effects 
```


 Summary of mean COs

MM.2 has significant coefficients, sex and sex*subsp interaction.

MM.2.Mol, the sex and interaction term (subsp*sex) are significant.  

MM.4 the strains which are significant are G, PWD and MSM. These 3 strains also have significant interaction (sex*strain) terms.


## Variance in CO count

DECIDE WHICH RESULTS TO DISPLAY.

in addition to the mean gwRR, the variance is also an important metric to describe 

Within animal variance in CO count / genome wide RR is important because it could shape constraints (and determine how effective stabilizing selection can act (ref))



RW's main qtl paper used the within animal variance, 
"We mapped QTL for
MLH1 count variance within individuals, applying an approach
to deconvolve the mean-variance relationship (see
Materials and Methods)."

above method remove / accounts for mean-variance correlation -

The fig5 also has a histogram of within animal variances in MLH1 counts (For F2s, with the parents as reference) (I could do this for males and females)

QTL on chrm 14, decreased variance in GG genotype, compared to hetZ and WSB (I think this also raised the mean)

@article{wang2017genetics,
  title={Genetics of genome-wide recombination rate evolution in mice from an isolated Island},
  author={Wang, Richard J and Payseur, Bret A},
  journal={Genetics},
  volume={206},
  number={4},
  pages={1841--1852},
  year={2017},
  publisher={Genetics Soc America}
}

references for phenotypic variance connected to robustness to environment, increase in developmental consistancy
Waddington 1942, Rutherford and Lindquist 2998, Mackay and Lyman 2005, Levy and Siegal 2008)

phenotypic variance leading to adaptation - 
(Losick and desplan 2008, Beaumont 2009, Eldar and Eloqitz 2010)

Previous vQTLs (Shen etal 2012, Hulse and Cai 2013, Mulder et al 2016)

meiotic Checkpoints for ensuring consistancy (Hochwagen and Amon 2006, Li and Schimenti 2007, Jaramillo-Lambert 2010 and Kauppi 2013)

#### MLH1 variance Figures

Not sure why I have this section -- think this should be into a supplemental doc or something.

```{r variance.boxplots, echo=FALSE, eval=FALSE}
#how did I make this 
#Species.COcount.mixed <- lmer(nMLH1.foci ~ species + quality + (1|mouse), data=MLH1.hq.lab, REML = TRUE)
#Species.COcount.mixed.sum <- summary(Species.COcount.mixed)
#SpeciesCOcount.fix.est <- drop1(Species.COcount.mixed, test="Chisq") 
#SpeciesCOcount.Rand.est <- exactRLRT(Species.COcount.mixed)


#plot of variance
var.boxplot <- ggplot(DF.HetC.MixedModel.HQ, aes(x = as.factor(mouse), y = var, fill= strain,alpha=0.5))+ geom_point()+
 ggtitle("Boxplots of MLH1 distributions") + theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
 ylim( c((min(MLH1_data$adj_nMLH1.foci)-3) ), (max(MLH1_data$adj_nMLH1.foci)+3)) +   scale_fill_manual(values=colors_of_strains)
var.boxplot <- var.boxplot   + facet_wrap(sex ~ strain, scales="free")
#this plot shows variance as points/single values


#plot by sex / seperately
# HQ data DF.HetC.MixedModel.HQ

var.male <- ggplot(DF.HetC.MixedModel.HQ[DF.HetC.MixedModel.HQ$sex == "male",], aes(x = as.factor(mouse), y = var, fill= strain,alpha=0.5))+ geom_point()+
 ggtitle("Male variance levels") + theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  scale_fill_manual(values=colors_of_strains)+  ylim( c( 5, 20) )

var.male <- var.male   + facet_wrap( ~ strain, scales="free")



cV.male <- ggplot(DF.HetC.MixedModel.HQ[DF.HetC.MixedModel.HQ$sex == "male",], aes(x = as.factor(mouse), y = cV, fill= strain,alpha=0.5))+ geom_point()+
 ggtitle("Male cV levels") + theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  scale_fill_manual(values=colors_of_strains)+  ylim( c( 5, 20) )

cV.male <- cV.male   + facet_wrap( ~ strain, scales="free")


```


```{r variance.hist, echo=FALSE, eval=FALSE}

#make series for total dataset?
Var.Hist.total.mice <- ggplot(DF.HetC.MixedModel, aes(x = var, fill=sex)) + 
  geom_histogram(binwidth = 1, alpha=.6)  +
  ggtitle("Histograms of within mouse variance, all mice")
Var.Hist.total.mice <- Var.Hist.total.mice   + facet_wrap(~ strain, scales="free")
Var.Hist.total.mice

#DF.HetC.MixedModel.HQ
Var.Hist <- ggplot(DF.HetC.MixedModel.HQ, aes(x = var, fill=sex)) + 
  geom_histogram(binwidth = 1, alpha=.6)  + 
  ggtitle("Histograms of within mouse variance, 10 cell min")
Var.Hist <- Var.Hist   + facet_wrap(~ strain, scales="free")
Var.Hist


Var.Hist.15cells <- ggplot(DF.HetC.MixedModel.HQ[DF.HetC.MixedModel.HQ$Ncells >14,], aes(x = var, fill=sex)) + 
  geom_histogram(binwidth = 1, alpha=.6)  +   ggtitle("Histograms of within mouse variance, 15 cell min")
Var.Hist.15cells <- Var.Hist.15cells   + facet_wrap(~ strain, scales="free")
Var.Hist.15cells



```


A general pattern is that the female data has higher within mouse variance (both var and cV). Some strains more than others. I think the dissection date needs to be tested as a predictor factor.  It might be tricky to add the dissection date metric. 
Test if dissection date is more significant predictor than strain (for females)
Also test how the number of 0CO bivalents per cell affects mouse with variance.

(LEW females with top 3 variance are all from different dissection dates)

I don't have enough mice with >15 cells for a complete idea of the distributions, but comparing single mice with enough good quality cells show be 


```{r cv.plots, echo=FALSE, eval=FALSE}

cV.Hist.total.mice <- ggplot(DF.HetC.MixedModel, aes(x =cV , fill=sex)) + 
  geom_histogram(binwidth = 1, alpha=.6)  +
  ggtitle("cV of within mouse variance, all mice")
cV.Hist.total.mice <- cV.Hist.total.mice   + facet_wrap(~ strain, scales="free")
cV.Hist.total.mice

#DF.HetC.MixedModel.HQ
cV.Hist <- ggplot(DF.HetC.MixedModel.HQ, aes(x = cV, fill=sex)) + 
  geom_histogram(binwidth = 1, alpha=.6)  + 
  ggtitle("cV of within mouse variance, 10 cell min")
cV.Hist <- cV.Hist   + facet_wrap(~ strain, scales="free")
cV.Hist


```


### Modeling Variance

(use the same models are mean_co, )


1.	HetC.MixedModel.HQ <- lme(mean_co ~ subsp*sex, data=DF.HetC.MixedModel.HQ, random=list(strain=pdDiag(~sex) ) )  

2.	Reduced.strain.HQ <- lme(mean_co ~ subsp * sex, data=DF.HetC.MixedModel.HQ, random = ~1|strain)  

3.	lm(mean_co ~ subsp * sex * strain, data=DF.HetC.MixedModel.HQ)  

4.	strain.fixed <- lm(mean_co ~ sex * strain, data=DF.HetC.MixedModel.HQ)  





$$mouse .cV CO ~=~ subsp * sex + rand(strain*sex)+ \varepsilon$$

```{r cV.model.setup, echo=FALSE}
#modeling within mouse variance

DF.HetC.MixedModel.HQ$var <-  as.numeric(DF.HetC.MixedModel.HQ$var)

#HQ var.CO
HetC.MixedModel.HQ.var_co <- lme(cV ~ subsp * sex, data=DF.HetC.MixedModel.HQ, random=list(strain=pdDiag(~sex) ) )

#summary(HetC.MixedModel.HQ.var_co)

as.data.frame( round(summary(HetC.MixedModel.HQ.var_co)$tTable, 6))

#ranef(HetC.MixedModel.HQ.var_co)
```




$$mouse .cV CO ~=~ subsp * sex + rand(1|strain)+ \varepsilon$$

```{r cV.models, echo=FALSE}
HQjust.strain.var <- lme(var ~ subsp * sex, data=DF.HetC.MixedModel.HQ, random = ~1|strain) 

as.data.frame( round(summary(HQjust.strain.var)$tTable, 6))

#summary(HQjust.strain.var_co)
```









$$mouse .var CO ~=~ subsp * sex  + \varepsilon$$
```{r base.var.mol, echo=FALSE, eval=FALSE}
HetC.MixedModel.mol.var_co <- lme(var ~ subsp * sex, data=DF.HetC.MixedModel.HQ.mol, random=list(strain=pdDiag(~sex) ) )

as.data.frame( round(summary(HetC.MixedModel.mol.var_co)$tTable, 6))

```




$$mouse .var CO ~=~ subsp * sex + rand(strain * sex) + \varepsilon$$
```{r var.model.1, echo=FALSE, eval=FALSE}
#main cV.CO
HetC.MixedModel.varco <- lme(var ~ subsp * sex, data=DF.HetC.MixedModel, random=list(strain=pdDiag(~sex) ) )

as.data.frame( round(summary(HetC.MixedModel.varco)$tTable, 6))

#summary(HetC.MixedModel.cVco)
#ranef(HetC.MixedModel.cVco)
```


```{r var.contrast, echo=FALSE, eval=FALSE}
#contrast models
# review this logic again


full2strain.CV <- anova(HetC.MixedModel.varco, just.strain.var_co)#0.9997

full2noRandom.CV <- anova(HetC.MixedModel.varco, full.Reduced.var_co) #0.3012

strain2noRandom.CV <- anova(just.strain.var_co, full.Reduced.var_co)#0.1213
#for var, p = 0.0219
```


#### Summary variance

The general pattern across all the models, is that sex is a significant effect for all models. The sex-subsp interaction effect p values were larger and only significant for one of the models.

When Molossinus is coded as a separate subspecies, then interaction terms are no longer significant.  (subsp *sex )  interaction?


<!--
For later analysis.  Not sure what these are good for.

```{r Mixed.models.pooled.cells, echo=FALSE, eval=FALSE, include=FALSE}

# check if the main mixed model results hold up when all cells are pooled

#pooled.cells
DF.HetC.MixedModel.pooled <- MLH1_data[MLH1_data$species == "M.musculus",]
DF.HetC.MixedModel.pooled <- DF.HetC.MixedModel.pooled[DF.HetC.MixedModel.pooled$subsp != "Cast",]
#remove musc strains
DF.HetC.MixedModel.pooled <- DF.HetC.MixedModel.pooled[DF.HetC.MixedModel.pooled$strain != "AST",]
DF.HetC.MixedModel.pooled <- DF.HetC.MixedModel.pooled[DF.HetC.MixedModel.pooled$strain != "TOM",]

#remove PERC, MOLF, CZECH,
DF.HetC.MixedModel.pooled <- DF.HetC.MixedModel.pooled[DF.HetC.MixedModel.pooled$strain != "CZECH",]
DF.HetC.MixedModel.pooled <- DF.HetC.MixedModel.pooled[DF.HetC.MixedModel.pooled$strain != "MOLF",]
DF.HetC.MixedModel.pooled <- DF.HetC.MixedModel.pooled[DF.HetC.MixedModel.pooled$strain != "PERC",]


#sex only has 2 levels and this breaks...
model.pooled <- lme(nMLH1.foci ~ subsp * sex, data=DF.HetC.MixedModel.pooled, random=list(strain=pdDiag(~sex) ) )

#pull out values / other things
summary(model.pooled)# indeces for this object: [1]-structure
ranef(model.pooled)

model.pooled$coefficients

#pooled cells,  DF.HetC.MixedModel.pooled
DF.HetC.MixedModel.pooled <- droplevels(DF.HetC.MixedModel.pooled)
DF.HetC.MixedModel.pooled <- DF.HetC.MixedModel.pooled[!(is.na(DF.HetC.MixedModel.pooled$mouse) | DF.HetC.MixedModel.pooled$mouse==""), ]

#consider changing to ordered = FALSE, for the DFs used for MM
DF.HetC.MixedModel.pooled$sex <- factor(DF.HetC.MixedModel.pooled$sex, ordered = TRUE, levels =c("male", "female") )

DF.HetC.MixedModel.pooled$strain<- factor(DF.HetC.MixedModel.pooled$strain, ordered = TRUE, levels =c( "WSB", "G", "LEW", "PWD", "MSM", "SKIVE", "KAZ") )

DF.HetC.MixedModel.pooled <- droplevels(DF.HetC.MixedModel.pooled)


```


- The MSM strain effect is 0, but the sex*strain for molossinus is 7 (this seems very high). This means there is 0 variance added to mean_CO to mean due to being MSM. Does this mean the model is very good at predicting MSM mean_CO. The fixed strain - sex effects are for musc are large. 
This will likely change with MOLF (male molf are low, so smaller fixed effect and )

- I think adding Mol, moves the highest strain into it's own category, so the coefficient is large.
In the first model MSM has largest random effect, since it's so far from WSB female.

-adding MOLF also doesn't have significant factors


changed all to unordered factors.
all the variables are coded as ordered factors.
When the variables are kept factors, and unordered the names appear! yay!

Character assigned variables.  The values are now quite different. Male, sex coefficient is negative. The fixed sex coefficient is the sex effect independent of subspecies or strain.
-->

## Summary of Mixed Model Analyses

**mouse average MLH1 count**

Multiple models were tested to test for potential evo models -- for the mouse average CO counts.  (there were a mix of results -- and nuanced)

First full Mixed model -- no significant fixed factors --(the random effect might be significant).  


The second model were strain was (nested and random), the fixed effects, sex and interaction of (subsp and sex), are much more significant. The coefficients indicate, males in general have 1 less in average, and the musc and molf subsp have ~3 more on average.


**Variance effect**


** Do these patterns hold across Quality bins? **


# Figure 1

Main figure, MLH1 strain avs on top, chrm proportion bar plot on bottom

```{r first.boxplots, echo=FALSE, fig.align='center', warning=FALSE}

#the below ordering works!! keep it
MLH1_data$strain<- factor(MLH1_data$strain, levels =c("G", "LEW","WSB",
              "PWD","CZECH","SKIVE",
              "KAZ","TOM","AST",
              "MSM","MOLF",
              "CAST","HMI","SPRET","SPIC", "other"), order=T )
#order the data frame
MLH1_data <- with(MLH1_data, MLH1_data[order(sex, strain),])


#never make boxplot with mouse on X. mouse level boxplots should be faceted by subspecies 
#can't re-order the miceagg
#ff <- ggplot(MLH1_data[order(MLH1_data$strain), ], 
#             aes(x = as.factor(mouse), y = adj_nMLH1.foci, fill= strain ))+geom_boxplot()+ ggtitle("Boxplots of MLH1 distributions by mouse") + theme(axis.title.x=element_blank(),
#        axis.text.x=element_blank(),
#        axis.ticks.x=element_blank()) +scale_fill_manual(values=c("#56B4E9", "cadetblue", "lightblue", #"coral4","coral1", "#E69F00","red", "red4", "red3","purple","grey1", "grey", 
#        "grey","grey","grey",  "grey","grey","grey" )) +
# ylim( c((min(MLH1_data$adj_nMLH1.foci)-3) ), (max(MLH1_data$adj_nMLH1.foci)+5)) + facet_wrap(species~ sex, scales="free")
#don't show M.musculus by mouse
#scale_fill_manual(values=c("#56B4E9", "cadetblue", "lightblue", "coral4","coral1", "#E69F00","red", "red3", "red4","orange","yellow", "yellow3", 
 #       "green","green4","grey",  "grey","grey","grey" )) +

#order MLH1_data by strain!!
#colors_of_strains, colors .. 
#there is no order to 'mouse' so it looks shitty

justMuscu <- ggplot(MLH1_data[MLH1_data$species == "M.musculus", ])+
  geom_boxplot(aes(x = as.factor(strain), y = adj_nMLH1.foci, fill= strain,alpha=0.5) )+ ggtitle("Boxplots of MLH1 distributions") + theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
 ylim( c((min(MLH1_data$adj_nMLH1.foci)-3) ), (max(MLH1_data$adj_nMLH1.foci)+3)) +   scale_fill_manual(values=colors_of_strains)
#facet_grid(~sex, scales="free_x")+scale_color_manual(values=colors_of_strains)+

justMuscu <- justMuscu   + facet_wrap(~ sex, scales="free")

#scales="free_y"
#this is ok except for the set placement of mice on x axis
#if the full dataset was ordered correctly, this would kinda fix it


#boxplots by strain
ss <- ggplot(MLH1_data, aes(x = as.factor(strain), y = adj_nMLH1.foci, fill= strain )) + geom_boxplot(aes(alpha=0.5))  + ggtitle("Boxplots of MLH1 distributions by strain") + theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+
  ylim( c( (min(MLH1_data$adj_nMLH1.foci)-3), (max(MLH1_data$adj_nMLH1.foci)+5)) )+scale_fill_manual(values=colors_of_strains)

ss <- ss   + facet_wrap(~ sex, scales="free")

#these plot currently look bad
```


```{r table.for.plot, echo=FALSE}
#why do I add in a different table?

AP_strain.table.HQ <- ddply(MLH1_data, c("strain", "sex"), summarise,
                        Nmice =length(unique(mouse)),
                        Ncells  =length(nMLH1.foci),
                        mean_co =as.numeric(format(round(mean(nMLH1.foci), 3 ), nsmall=3) ),
                        cV = cv(nMLH1.foci, na.rm = TRUE),
                        var = as.numeric(format(round(   var(nMLH1.foci, na.rm = TRUE),3), nsmall=3)), 
                        sd   = round(sd(nMLH1.foci, na.rm = TRUE), 3),
                       #Standard error (mice vs cells)
                        se  = round(sd / sqrt(Ncells), 3)
    )

```


```{r Fig1, echo=FALSE, warning=FALSE}

Fig1_v1_strain <- ggplot(AP_strain.table.HQ, aes(y=mean_co, x=strain, color=strain, shape=sex))+ 
  ylab("Average MLH1 Foci per Cell")+

  geom_point(aes(shape = sex), size=3, position=position_dodge(.5))+
  geom_errorbar(aes(ymin=mean_co-(se*2), ymax=mean_co+(se*2) ), width=.3, position = position_dodge(.5), size=1.001)+
  
  ylim(c(18, 35))+
 scale_color_manual(values=colors_of_strains)+ 
  theme(legend.position="none",
  panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
  panel.background = element_blank(), axis.line=element_line(colour = "black"),
  axis.title.x=element_blank(),axis.text.x=element_blank(),axis.ticks.x=element_blank())+
#remove y labfor male 
  #segments
  annotate("segment", x=.75, xend=4.5, y=19.5, yend=19.5, colour="black", size=1)+
  #Musc
  annotate("segment", x= 5.5, xend=10.2, y=19.5, yend=19.5, colour="black", size=1)+
  #mol
  annotate("segment", x=11, xend=12.25, y=19.5, yend=19.5, colour="black", size=1)+
  #cast
  annotate("segment", x=13, xend=14, y=19.5, yend=19.5, colour="black", size=1)+
  #outgroups
  annotate("segment", x=14.5, xend=16, y=19.5, yend=19.5, colour="black", size=1)+
  #label
  annotate("text", x = c(3,8.5,10,12, 14), #list of all x points 
           y = c(18.5,18.5,33,18.5, 18.5),
           
   label = c("M. m. domesticus","M. m. musculus", "M. m. molossinus", "M. m. castaneus", "Outgroups"),
           fontface = 'italic', hjust = 0.5, size=3.5)


Fig1_v1_strain
```


#### Proportion of chrm classes

make a smaller propoertion chrm block --

These results are meant to compare the proportions of bivalents with 0,1,2 or 3 chromosomes.
Most of the variation in gwRR across strains in is due to more 2COs at the 'expense' / trade off of 1COs. (the gwRR for most mice is low/close to the minimum)


This is also another way to score heterochiasmy (that gets around the sex chrms, i think)

The data below are from hand measured cells (not from bivalent data)
generate smaller data set of chrm proportions (0CO, 1CO,2CO etc)
10 cells from each musc category. **Add Molf female**


```{r chrm_prop_data, echo=FALSE}
#shut off eval for now because of wd / path error
#chrm.class.DF = read.csv("~./MLH1repo/data/ChrmClassProportions.csv", header = TRUE)
#chrm.class.DF = read.csv("C:/Users/alpeterson7/Documents/MLH1repo/data/ChrmClassProportions.csv", header = TRUE)

chrm.class.DF = read.csv("C:/Users/alpeterson7/Documents/MLH1repo/data/ChrmClassProportions2.csv", header = TRUE)

#this excel file is ~100 cells (10 per category) where I manually verified the number of chrom classes. (this file can also be used to quantify error for CO counts)

chrm.class.DF$fileName <- (chrm.class.DF$Original.Name)
chrm.class.DF <- add_mouse(chrm.class.DF)
chrm.class.DF <- add_strain(chrm.class.DF)
chrm.class.DF <- add_sex(chrm.class.DF)
chrm.class.DF <- add_category(chrm.class.DF)
#chrm.class.DF <- add_euth_date(meta.data.file2)
#chrm.class.DF <- add_age(meta.data.file2)

#table(chrm.class.DF$category)
#str(chrm.class.DF)

#summarize by category
proportion_stats_category  <- ddply(.data=chrm.class.DF, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                  tot_biv  = sum(biv.sum.check,  na.rm = TRUE),

                 n0CO = sum(n0CO, na.rm = TRUE),
                  n1CO = sum(n1CO,na.rm = TRUE),
                  n2CO = sum(n2CO,na.rm = TRUE),
                  n3CO = sum(n3CO,na.rm = TRUE),
                  n4CO = sum(n4CO,na.rm = TRUE),
                 
                 per0CO = sum(n0CO) / tot_biv,                 
                 per1CO= sum(n1CO)/ tot_biv,
                 per2CO= sum(n2CO)/ tot_biv,
                 per3CO= sum(n3CO)/ tot_biv,
                 per4CO= sum(n4CO)/ tot_biv,
                 
                 mean.MLH1.foci = mean(MLH1.sum.check),
                 mean.MLH1.foci2 = mean(nMLH1.foci),
                 meanQ = mean(quality)

)
#this is the table that needs melting
#but is also easy to redo by hand

#write.table(proportion_stats_category, "~./MLH1repo/prop_category.csv", sep=",", row.names = FALSE)
```


```{r plot.chrm.prop.melt, echo=FALSE}

#add subspecies to these values
melt.Chrm.Prop.table = read.csv("~./MLH1repo/data/ChrmClassProportions2_melt.csv", header = TRUE)
melt.Chrm.Prop.table <- melt.Chrm.Prop.table[c(1:5)]

#add the subsps

melt.Chrm.Prop.table$strain<- factor(melt.Chrm.Prop.table$strain, levels =c("LEW",
                      "WSB","G", "TOM","AST","KAZ", "SKIVE","PWD","MSM", "MOLF"), order=T )



melt.Chrm.Prop.table <- with(melt.Chrm.Prop.table, melt.Chrm.Prop.table[order(strain),])

chrm.class.barplot <- ggplot(data = melt.Chrm.Prop.table, aes(y = value, x = strain, fill=as.factor(type)) ) + geom_col(position = "dodge")+facet_grid(sex~strain, scales = "free")+ggtitle("Chrm Proportions from Hand Measured Data")+ylim(c(0,1))

#need a second DF / the first df
#+ annotate() text 
#annotate()text

#proportion_stats_category,  tot_biv, ncells

#add number of cells and bivalents in the corner


chrm.class.barplot
```


```{r test.Chrm.Prop, echo=FALSE}


#this first one is the 'first' MLH1 count
#which is this variable

#oopp2 <- lm( mean.MLH1.foci2 ~ per0CO+ per1CO +per2CO +per3CO, data = proportion_stats_category)
#summary(oopp2)
#none of the classes are significant

#mean is the new calculated-- mean 0CO is significant and intercept
#oopp <-  lm( mean.MLH1.foci ~ per0CO+ per1CO +per2CO +per3CO, data = #proportion_stats_category)


#summary(oopp)
#not sure if these are inter-correlated
#I don't know why 0COs and the intercept are significant..
#0CO and 1CO are significant when only chrm classes are included (and intercept)


#no longer significant when used
#lm regression indicates that the proportions aren't significant contributers to meanMLH1 -- but this could be due to error
#after added more MSM data, tot_biv is not significant
#0CO is now more significant, and 2CO is sightly significant


#2CO proportion is the most significant, 1CO is second most significat, 3CO doesn't have enough power - total bivalents is also significant, there might be a bias

#MSM needs more bialent observations
#use melt to put all propotions 
#display / visualize / compare proportions
#the table needs to be melted, because it's split by category

#Need to run the chrm proportions thru melt code.

chrm.class.DF$perBiv0 <-chrm.class.DF$n0CO/chrm.class.DF$biv.sum.check
chrm.class.DF$perBiv1<-chrm.class.DF$n1CO/chrm.class.DF$biv.sum.check
chrm.class.DF$perBiv2<-chrm.class.DF$n2CO/chrm.class.DF$biv.sum.check
chrm.class.DF$perBiv3<-chrm.class.DF$n3CO/chrm.class.DF$biv.sum.check
chrm.class.DF$perBiv4<-chrm.class.DF$n4CO/chrm.class.DF$biv.sum.check

#chrm.class.DF$biv.sum.check
```

High musc males have half 1CO and 2CO bivalents.
For following up on the biv stuff, find ways to compare the 1CO to 2COs in Dom and Musc. (sis-coten, length, distance from centromere).

Add p.values of tests for the how the proportions differ.


*cast female data came from another publication, add this ref (Kohler?)

The bottom lines and subspecies points don't exactly match up.

**add in cell images like the other draft figures.**


# Rapid male specific evolution

list out the points which support rapid male specific evolution for MLH1 rates.



number above bars are the raw counts for bivalent classes.

Potential figure 2 (eventually). Try to make the figure taller.
-- add more data to make sample sizes more equal.
-- add space to indicate where female data is missing

SKIVE male has lower proportion of 2CO compared to the hand measured, but it still looks like it's more than other strains.
What's up with Lew females? there's lots of 2COs, also WSB female.

This type of plot is another way to score heterochiasmy that gets around the sex chromosomes, because the data come from samples of bivalents and not averages of cells. If these data are random samples of bivalents, the X 


ToDo: chisq tests for difference in proportions, 
- within strain across sex,
- males, across high and low groups
- look at mouse level proportions



## DMC1 Results Spermatocytes

These data are meant to test if CO precursors, double strand breaks (DSBs), are significant predictors for MLH1 count variation. Simply, do mice with more DSBs also have more COs?

ToDo:
  - attempt to supplement data with more KAZ DMC1 cell observations.
  - change all the bad variable names

```{r DMC1.setup, echo=FALSE}
#load DMC1 csv
DMC1.DF <- read.csv("~./DMC1/newJuviQuant.csv", header = TRUE)
DMC1.DF$fileName <- DMC1.DF$Original.Name
DMC1.DF <- add_mouse(DMC1.DF)
DMC1.DF <- add_strain(DMC1.DF)
DMC1.DF <- add_category(DMC1.DF)
DMC1.DF <- add_subsp(DMC1.DF)
DMC1.DF <- DMC1.DF[DMC1.DF$stage != "P",]

#make DMC1 table
DMC1.mean.obs.L <- ddply(.data=DMC1.DF[DMC1.DF$stage == "L",],
                 .(category),
                 summarize, 
                 mean.count = mean( total.count)
)
DMC1.mean.obs.L$stage <- "L"

DMC1.mean.obs.Z <- ddply(.data=DMC1.DF[DMC1.DF$stage == "Z",],
                 .(category),
                 summarize, 
                 mean.count = mean( total.count)
)
DMC1.mean.obs.Z$stage <- "Z"

#these tables are merged latter

```

### Testing Normality Assumptions

```{r normality, echo=FALSE, warning=FALSE}
#histograms for each

#- check that distributions are close to normal
#- check for equal variance.

DMC1.histogram <- ggplot(DMC1.DF, aes(total.count, fill = stage)) + geom_histogram(alpha = 0.4)+facet_wrap(~strain)+ ggtitle("DMC1 histogram")

DMC1.histogram
#checking equal variance, https://www.statmethods.net/stats/anovaAssumptions.html
#bartlet's test, Levine's test
```

The counts look close to normal.

Potentially figure 2. Using a function to display the p-vales for mean comparison

```{r DMC1.plot, echo=FALSE, warning=FALSE}
#think of better color scheme
#make the jitter col thinner or make the stacked dots.
#remember that box-plots add outlier dots.
#change y labels

#experimenting with ggboplot
everyday <- ggboxplot(DMC1.DF[DMC1.DF$stage != "P",], x = "strain", y = "total.count",
          color = "stage", palette = "jco",
          add = "jitter", short.panel.labs = FALSE)
  #library gpubr has functions to add in sig/test bars.

DMC1.boxplot <- ggplot(aes(x=strain, y=total.count, color=stage), data=DMC1.DF[DMC1.DF$stage != "P",])+
  geom_boxplot(aes(alpha=.5))+geom_jitter()+ggtitle("DMC1 averages in juvenile mice")
  DMC1.boxplot
  #this will add neat bars, but still need to figure out what groups ect it's actually using
#THE stat_compare function adds a arm thing and pvalue for 
#  stat_compare_means(comparisons = my_comparisons)

#status - legend, expression y-axis, facet, facet_wrap SYMBOL
#legend - stage. y=total count, FACet strain


#add column to distinguish high vs low rec strains
DMC1.DF$rec_group <- ifelse(grepl("WSB", DMC1.DF$strain), "Low",
                          ifelse(grepl("G",DMC1.DF$strain), "Low",
                                 
              ifelse(grepl("KAZ", DMC1.DF$strain), "Low",
            ifelse(grepl("PWD", DMC1.DF$strain), "High",
               ifelse(grepl("MSM", DMC1.DF$strain), "High","")))))
DMC1.DF$rec_group <- as.factor(DMC1.DF$rec_group)

                   
# annotation table with adjusted pvals and y-position of the labels
anno_df = compare_means(total.count ~ rec_group, group.by = "stage", data = DMC1.DF[DMC1.DF$stage != "P",]) %>% mutate(y_pos = 300)

DMC1.bxlt_High.to.Low <- ggplot(DMC1.DF[DMC1.DF$stage != "P",], aes(x=rec_group, y=total.count)) + 
  geom_boxplot(position=position_dodge()) + 
  geom_point(aes(color=stage), position=position_jitterdodge()) + 
  facet_wrap(~stage, nrow=1) + 
  ggsignif::geom_signif(
    data=anno_df, 
    aes(xmin=group1, xmax=group2, annotations=p.adj, y_position=y_pos), 
    manual=TRUE
  )+ggtitle("Comparison of DMC1 means across high and low recombing groups")
#this is kinda the plot I'm enivioning 
DMC1.bxlt_High.to.Low

#I want a test between 1) high vs low cell stages, 
#I think a better way to display would be
#facet by stage, x is the strain
#compare means across high vs low  (this level isn't defined)

#example from here
#https://github.com/kassambara/ggpubr/issues/65


#https://www.r-bloggers.com/add-p-values-and-significance-levels-to-ggplots/

```

Above figure shows p-values for comparison of means across the cell stages.

Foci of DMC1 were scored from Leptotene and Zygotene spermatocytes of juvenile mice (12-14-18 days). One mouse represents each strain. The main comparison to examine is that between the high (PWD and MSM) and low recombining (WSB, G, KAZ) strains.

ToDo think of ways to add in the compared means for the high and low groups (across cell stage)
how similar are G-WSB or PWD-MSM to each other?



## DMC1 Permutations with equal sampling

The number of observations are not quite equal across strains and cell stages, so I ran permutations sampling 25 cells from the high and low recombining groups of cells. Below are the distributions of p values for permuted t-tests from sampling 25 cells from the pooled high and low recombining groups. Permutations were done for both cell stages. The actual p value is in red. 

The observed p value does not fall outside of the permuted distribution with assumes equal cell sampling.

```{r DMC1.subsampling, echo=FALSE, eval=FALSE}

#recreate the DMC1 data by subsampling ~9 or lowest cell number from 
# the 2 groups

#make the 2 pools
High.L.pool <- DMC1.DF$total.count[( (DMC1.DF$stage == "L") & ( ( DMC1.DF$strain == "MSM") | (DMC1.DF$strain == "PWD") ) )  ]
High.Z.pool <- DMC1.DF$total.count[( (DMC1.DF$stage == "Z") & ( ( DMC1.DF$strain == "MSM") | (DMC1.DF$strain == "PWD") ) )  ]

Low.L.pool <- DMC1.DF$total.count[( (DMC1.DF$stage == "L") & ( ( DMC1.DF$strain == "WSB") | (DMC1.DF$strain == "G") | (DMC1.DF$strain == "KAZ") ) )]
Low.Z.pool <- DMC1.DF$total.count[( (DMC1.DF$stage == "Z") & ( ( DMC1.DF$strain == "WSB") | (DMC1.DF$strain == "G") | (DMC1.DF$strain == "KAZ") ) )]


#permute drawing / subsampling and performing the t-test
#put the p.value results into a DF!


#permut.SC.cv.diff  <- as.data.frame(
#sample(THE.DF, SAMPLE.SIZE, REPLACE?)

#permuting samples -- and re-licating t-tests

#nrep <- 1000

#permuting samples to replicate the t.test across cell stages from high and low with equal sample sizes.
bunchOpvals.L <- replicate(1000, 
    # t.test
    t.test( sample(High.L.pool, 25, replace = FALSE),
            sample(Low.L.pool, 25, replace = FALSE), var.equal = FALSE )$p.value
          )
bunchOpvals.L <- as.data.frame(bunchOpvals.L)
bunchOpvals.L$stage <- "L"

bunchOpvals.Z <- replicate(1000, 
    # t.test
    t.test( sample(High.Z.pool, 25, replace = FALSE),
            sample(Low.Z.pool, 25, replace = FALSE), var.equal = FALSE)$p.value
          )

bunchOpvals.Z <- as.data.frame(bunchOpvals.Z)
bunchOpvals.Z$stage <- "Z"

#could merge the lists of p-values
replicated.Ttests <- cbind(bunchOpvals.L, bunchOpvals.Z)
#have to melt, the rbinded dataframe to get the cols 'stage' and 'pvale'


#ASSUMPTIONS; normality, equal variance

#plot the lists of pvals in histogram? compare to the observed pvalue to get a sense as of how much sample size shifts the pvalues.

plot.pvals.L <- ggplot(bunchOpvals.L, aes(bunchOpvals.L) )+geom_histogram(bins = 50)+ ggtitle("1000 rep t-tests sample size 25, High vs Low Leptotene")+geom_vline(xintercept=ttest.HighLow.L$p.value, linetype="dotted", color="red")


plot.pvals.Z <- ggplot(bunchOpvals.Z, aes(bunchOpvals.Z) )+geom_histogram(bins = 50)+ ggtitle("1000 rep t-tests sample size 25, High vs Low Zygotene")+geom_vline(xintercept=ttest.HighLow.Z$p.value, linetype="dotted", color="red")
plot.pvals.L
plot.pvals.Z


#The t.test( ) function produces a variety of t-tests. Unlike most statistical packages, the default assumes #unequal variance and applies the Welsh df modification.
#n use the var.equal = TRUE
#https://www.statmethods.net/stats/ttest.html
```


```{r DMC1_MLH1.cor, echo=FALSE}
#make category averages from AP_mouse_table.HQ
mean.MLH1.obs <- ddply(.data=AP_mouse_table.HQ,
                 .(category),
                 summarize, 
                 mean.MLH1 = mean(mean_co),
                 var.MLH1 = var(mean_co)
)

DMC1.target.strains <- subset(mean.MLH1.obs, category == "G male" |
                   category == "WSB male" |
                     category == "KAZ male" |
                  category == "MSM male" |
                    category == "PWD male")
#attach this mini DF to the DMC1 data

DMC1.mean.obs.L$mean.count.L <- DMC1.mean.obs.L$mean.count
DMC1.mean.obs.Z$mean.count.Z <- DMC1.mean.obs.Z$mean.count


DMC1table <- merge(DMC1.mean.obs.L, DMC1.mean.obs.Z, by="category")

DMC1table.main <- merge(DMC1.target.strains, DMC1table, by="category")

MLH1.DMC1.L_cor <- cor(DMC1table.main$mean.MLH1, DMC1table.main$mean.count.L, method = c("pearson"))#MLH1.DMC1.Z_cor

MLH1.DMC1.Z_cor <- cor(DMC1table.main$mean.MLH1, DMC1table.main$mean.count.Z, method = c("pearson"))#0.2793237

```

> At what stage are DMC1 foci numbers most correlated with MLH1?

The correlation with MLH1 and leptotene cells is `r MLH1.DMC1.L_cor`.

The correlation with MLH1 and zygotene cells is `r MLH1.DMC1.Z_cor`.


## Total SC Results

insert male total sc results here


# Female patterns

Nuanced
there is strain specific evolution for females ... 


# Sexual dimorphism and heterochiasmy

total SC length / SC compaction and telomere bias in positioning are conserved / consistently sexually dimorphism (in these mice?)  for most species

make the heterochiasmy plot, with female strain averages adjusted for X
(measure variance in CO counts for the top 5 female whole cell bivalents)
(variance in )
how does this compare to variance of mid 5, bottom 5?


# Predictions / tests for selection on Bivalent structure



# Blank


```{r trackcurBivData, echo=FALSE}
#list / tally of current curated Bivdata
Cur_biv = read.csv("~./MLH1repo/data/CurrentCuratedDataset_13.1.20.csv", header = TRUE)

colnames(Cur_biv) <- c('mouse', 'blank','0','1')


AP_mouse_table_w.Ages_HQ <- AP_mouse_table_w.Ages[ -c(9:12, 14, 18, 22:24) ]

AP_mouse_table_w.Ages_HQ <- add_sex(AP_mouse_table_w.Ages_HQ)
AP_mouse_table_w.Ages_HQ <- add_strain(AP_mouse_table_w.Ages_HQ)
AP_mouse_table_w.Ages_HQ <- add_category(AP_mouse_table_w.Ages_HQ)
AP_mouse_table_w.Ages_HQ <- add_subsp(AP_mouse_table_w.Ages_HQ)
AP_mouse_table_w.Ages_HQ <- add_species(AP_mouse_table_w.Ages_HQ)

#only M.musculus
AP_mouse_table_w.Ages_HQ <- AP_mouse_table_w.Ages_HQ[AP_mouse_table_w.Ages_HQ$species == "M.musculus",]
#REMOVE EXTRA STRAINS
AP_mouse_table_w.Ages_HQ <- AP_mouse_table_w.Ages_HQ[AP_mouse_table_w.Ages_HQ$subsp != "Cast",]
#remove musc strains
AP_mouse_table_w.Ages_HQ <- AP_mouse_table_w.Ages_HQ[AP_mouse_table_w.Ages_HQ$strain != "AST",]
AP_mouse_table_w.Ages_HQ <- AP_mouse_table_w.Ages_HQ[AP_mouse_table_w.Ages_HQ$strain != "TOM",]
#remove PERC, MOLF, CZECH, (no females)
AP_mouse_table_w.Ages_HQ <- AP_mouse_table_w.Ages_HQ[AP_mouse_table_w.Ages_HQ$strain != "CZECH",]
AP_mouse_table_w.Ages_HQ <- AP_mouse_table_w.Ages_HQ[AP_mouse_table_w.Ages_HQ$strain != "PERC",]
#156 mice (from 193)

#some mice are missing 13oct14

#match the MM dataset
AP_mouse_table_w.Ages_HQ <- AP_mouse_table_w.Ages_HQ[AP_mouse_table_w.Ages_HQ$Ncells > 9,]
AP_mouse_table_w.Ages_HQ <- AP_mouse_table_w.Ages_HQ[AP_mouse_table_w.Ages_HQ$age.weeks < 26,]
#126 mice

#AP_mouse_table_w.Ages_HQ <- AP_mouse_table_w.Ages_HQ[!(is.na(AP_mouse_table_w.Ages_HQ$mouse) | DF.HetC.MixedModel.HQ$mouse==""), ]

#mice missing from curated list
mice.within.BivData <- AP_mouse_table_w.Ages_HQ[ AP_mouse_table_w.Ages_HQ$mouse %in% Cur_biv$mouse, ]#54 mice in (14 mice have enteries but 0 curated biv)

#(14 mice need at least some curation

cur.data.mice <- Cur_biv[ Cur_biv$mouse %in% AP_mouse_table_w.Ages_HQ$mouse, ]

mice.missing.bivData <- AP_mouse_table_w.Ages_HQ[ ! AP_mouse_table_w.Ages_HQ$mouse %in% Cur_biv$mouse, ]#72 (includes mice not in MM)
#72 mice that I need to upload/run the algorithm on their images

#mice missing SC pass == '1'
colnames(Cur_biv) <- c("mouse","blank","SC.fail", "SC.pass")
NeedsCur <- Cur_biv[Cur_biv$SC.fail == 0,]

NeedsCur <- add_strain(NeedsCur)
write.table(NeedsCur, "~./MLH1repo/data/needs_cur.csv", sep=",",
            row.names = TRUE)


#44% of mice are missing curation

#write this table
#write.table(OG.cURATE.table.mouse, "~./MLH1repo/data/CurrentCuratedDataset_13.1.20.csv", sep=",",
#            row.names = TRUE)

mice.missing.bivData <- mice.missing.bivData[!(is.na(mice.missing.bivData$mouse) | mice.missing.bivData$mouse==""), ]
write.table(mice.missing.bivData, "~./MLH1repo/data/MissingMice_13.1.20.csv", sep=",",
            row.names = TRUE)


```



```{r barlett.test, echo=FALSE}

#bartlett.test(total.count~strain, data=DMC1.DF[DMC1.DF$stage == "L",])#L (don't work with single KAZ L cell)
bartlett.test(total.count~strain, data=DMC1.DF[DMC1.DF$stage == "Z",])#Z

```

Evidence for non-equal variance across strains for zygotene cells.

Ideas for post-hoc tests, remove strains 1 at a time and test variance. Test variance for L cells without KAz.


From ANOVA analysis strain is significant for Leptotene cells but not Zygotene cells. (but take these results with a grain of salt before assumptions are tested.)

```{r DMC1_anova, echo=FALSE, warning=FALSE}
#1. ANOVA, is there a difference in DMC1 counts across strains?
mod.L <- lm(total.count ~ strain, data = DMC1.DF[DMC1.DF$stage == "L",])
mod.Z <- lm(total.count ~ strain, data = DMC1.DF[DMC1.DF$stage == "Z",])

l.anova <- anova(mod.L) #0.0002701 ***
z.anova <- anova(mod.Z) #NS

l.anova
z.anova
```


```{r DMC1_t.tests, echo=FALSE, warning=FALSE}
#2. t-tests for the DMC1 counts 
#need to account for sample sizes when comparing t.tests (KAZ and G don't have equal across )
#best case scnario -- more microscoping to get ~20 cells for each stage by strain combo


#L vs Z (all species)
#jj <- t.test(DMC1.male.data$total.count[DMC1.male.data$stage == "Z"], DMC1.male.data$total.count[DMC1.male.data$stage == "L"] )
#p-value = 1.032e-05
#not a valid interpertation


#high vs low group (L)
ttest.HighLow.L <- t.test(DMC1.DF$total.count[( (DMC1.DF$stage == "L") & ( ( DMC1.DF$strain == "MSM") | (DMC1.DF$strain == "PWD") ) )  ],
               DMC1.DF$total.count[( (DMC1.DF$stage == "L") & ( ( DMC1.DF$strain == "WSB") | (DMC1.DF$strain == "G") | (DMC1.DF$strain == "KAZ") ) )]    )
# p-value =  0.002145


#high vs low group (Z)
ttest.HighLow.Z <- t.test(DMC1.DF$total.count[( (DMC1.DF$stage == "Z") & ( ( DMC1.DF$strain == "MSM") | (DMC1.DF$strain == "PWD") ) )  ],
               DMC1.DF$total.count[( (DMC1.DF$stage == "Z") & ( ( DMC1.DF$strain == "WSB") | (DMC1.DF$strain == "G") | (DMC1.DF$strain == "KAZ") ) )]    )
# p-value = 0.6628


#dom vs musc (L) (2 subsp vs)
ttest.subsp.L <- t.test(DMC1.DF$total.count[(DMC1.DF$subsp == "Dom") & (DMC1.DF$stage == "L")], DMC1.DF$total.count[(DMC1.DF$subsp == "Musc") & (DMC1.DF$stage == "L")] )
# p-value = 0.003016

#dom vs musc (Z)
ttest.subsp.Z <- t.test(DMC1.DF$total.count[(DMC1.DF$subsp == "Dom") & (DMC1.DF$stage == "Z")], DMC1.DF$total.count[(DMC1.DF$subsp == "Musc") & (DMC1.DF$stage == "Z")] )
#p-value = 0.09588

#KAZ vs Dom (Z) (only 1 L cell)
ttest.KAZ.Dom.z <- t.test(DMC1.DF$total.count[(DMC1.DF$subsp == "Dom") & (DMC1.DF$stage == "Z")], 
      DMC1.DF$total.count[(DMC1.DF$strain == "KAZ") & (DMC1.DF$stage == "Z")] )

#p-value = 0.004148
#Kaz Z are higher than Dom
```

Divide the Musculus strains into high and low recombining groups to characterize the difference in distributions. The p-values from the t.tests of the high vs low groups, indicate that the high recombining are significantly higher for the L cells (p value = `r ttest.HighLow.L$p.value`) while the Z cells, there is not a significant difference across the high and low groups (value = `r ttest.HighLow.Z$p.value`).





# Note on Heterochiasmy Definition

I present heterochiasmy as a comparison of oocyte to spermatocyte MLH1 counts, but the sex chromosomes/bivalents complicate this comparison.
In females the XX bivalent is indistinguishable from the autosomes. To the meiotic recombination machinery, it is an autosome and has a similar REC landscape. 

Whereas in spermatocytes the XY bivalent is visually distinct and any MLH1 where not included in the count). (I note if the and Y are paired, which they are at a high rate). The XY pair triggers a response to un-paired chromosomes and only has MLH1 foci within the PAR (the the tips of X and Y). 


To make a more equivalent comparison I will estimate which bivalent is the XX in oocytes, and subtract that average REC from the category average of each strain.

1. Compile full-cell data from females (all 20 bivalents measured)
2. Look at the SC length -ranked data, extract the 3rd longest estimate average REC for this bivalent, 
3. check how variable the REC is across the 1st,2nd,4th, and 5th are.


According to mouse genome website, the X is the 3rd largest chromosome by total amount of DNA (Mb).



```{r HetC.rates, echo=FALSE, include=FALSE, warning=FALSE}
#calculate the initial HetC levels from best quality mice and cells (compare both levels (cell and mouse average levels))
#subset MLH1_DF, remove non passing mice (as diagnoised from plots)
#remove quality 5 cells

#outgroups
#SPIC
HetC.SPIC <- MLH1_data[(MLH1_data$strain == "SPIC")&(MLH1_data$quality < 5),]
#unique(HetC.SPIC$mouse)
#non-passing mice
SPIC.remove.mice <- c("30may18_SPIC_f1", "8jun18_SPIC_m2")
HetC.SPIC <- HetC.SPIC[ ! HetC.SPIC$mouse %in% SPIC.remove.mice, ]
#calq sex mean
SPIC.mean.MLH1.obs <- ddply(.data=HetC.SPIC, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                 mean.MLH1 = mean(adj_nMLH1.foci),
                 var.MLH1 = var(adj_nMLH1.foci)
)
SPIC.HETC.val <- SPIC.mean.MLH1.obs$mean.MLH1[1] / SPIC.mean.MLH1.obs$mean.MLH1[2]

#SPRET
HetC.SPRET <- MLH1_data[(MLH1_data$strain == "SPRET")&(MLH1_data$quality < 5),]
#unique(HetC.SPRET$mouse)
#non-passing mice
#SPRET.remove.mice <- c("", "")
#HetC.SPRET <- HetC.SPRET[ ! HetC.SPRET$mouse %in% SPRET.remove.mice, ]
#calq sex mean
SPRET.mean.MLH1.obs <- ddply(.data=HetC.SPRET, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                 mean.MLH1 = mean(adj_nMLH1.foci),
                 var.MLH1 = var(adj_nMLH1.foci)
)

SPRET.HETC.val <- SPRET.mean.MLH1.obs$mean.MLH1[1] / SPRET.mean.MLH1.obs$mean.MLH1[2]

#CAROLI
HetC.CAROLI <- MLH1_data[(MLH1_data$strain == "CAROLI")&(MLH1_data$quality < 5),]
#unique(HetC.CAROLI$mouse)
#non-passing mice
#SPRET.remove.mice <- c("", "")
#HetC.SPRET <- HetC.SPRET[ ! HetC.SPRET$mouse %in% SPRET.remove.mice, ]
#calq sex mean
CAROLI.mean.MLH1.obs <- ddply(.data=HetC.CAROLI, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                 mean.MLH1 = mean(adj_nMLH1.foci),
                 var.MLH1 = var(adj_nMLH1.foci)
)

#cast
HetC.CAST <- MLH1_data[(MLH1_data$strain == "CAST")&(MLH1_data$quality < 5),]
#unique(HetC.CAST$mouse)
#non-passing mice
#X.remove.mice <- c("", "")
#HetC.SPRET <- HetC.SPRET[ ! HetC.SPRET$mouse %in% SPRET.remove.mice, ]
#calq sex mean
CAST.mean.MLH1.obs <- ddply(.data=HetC.CAST, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                 mean.MLH1 = mean(adj_nMLH1.foci),
                 var.MLH1 = var(adj_nMLH1.foci)
)
CAST.HETC.val <- CAST.mean.MLH1.obs$mean.MLH1[1] / CAST.mean.MLH1.obs$mean.MLH1[2]

#HMI
HetC.HMI <- MLH1_data[(MLH1_data$strain == "HMI")&(MLH1_data$quality < 5),]
#unique(HetC.HMI$mouse)
#non-passing mice
#X.remove.mice <- c("", "")
#HetC.SPRET <- HetC.SPRET[ ! HetC.SPRET$mouse %in% SPRET.remove.mice, ]
#calq sex mean
HMI.mean.MLH1.obs <- ddply(.data=HetC.HMI, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                 mean.MLH1 = mean(adj_nMLH1.foci),
                 var.MLH1 = var(adj_nMLH1.foci)
)

#dom

#WSB
HetC.WSB <- MLH1_data[(MLH1_data$strain == "WSB")&(MLH1_data$quality < 5),]
#unique(HetC.WSB$mouse)
#non-passing mice
WSB.remove.mice <- c("8oct14_WSB_f1", "18nov17_WSB_f5", "16jun15_WSB_f3","10mar15_WSB_m2")
HetC.WSB <- HetC.WSB[ ! HetC.WSB$mouse %in% WSB.remove.mice, ]
#calq sex mean
WSB.mean.MLH1.obs <- ddply(.data=HetC.WSB, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                 mean.MLH1 = mean(adj_nMLH1.foci),
                 var.MLH1 = var(adj_nMLH1.foci)
)

WSB.HETC.val <- WSB.mean.MLH1.obs$mean.MLH1[1] / WSB.mean.MLH1.obs$mean.MLH1[2]


#LEW
HetC.LEW <- MLH1_data[(MLH1_data$strain == "LEW")&(MLH1_data$quality < 5),]
unique(HetC.LEW$mouse)
#non-passing mice
LEW.remove.mice <- c("4jan17_LEW_f2")
HetC.LEW <- HetC.WSB[ ! HetC.LEW$mouse %in% LEW.remove.mice, ]

#calq sex mean
LEW.mean.MLH1.obs <- ddply(.data=HetC.LEW, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                 mean.MLH1 = mean(adj_nMLH1.foci),
                 var.MLH1 = var(adj_nMLH1.foci)
)

LEW.HETC.val <- LEW.mean.MLH1.obs$mean.MLH1[1] / LEW.mean.MLH1.obs$mean.MLH1[2]


#G
HetC.G <- MLH1_data[(MLH1_data$strain == "G")&(MLH1_data$quality < 5),]
#unique(HetC.G$mouse)
#non-passing mice
G.remove.mice <- c("8jun15_G_f3","8jun15_G_f4")
HetC.G <- HetC.G[ ! HetC.G$mouse %in% G.remove.mice, ]

#calq sex mean
G.mean.MLH1.obs <- ddply(.data=HetC.G, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                 mean.MLH1 = mean(adj_nMLH1.foci),
                 var.MLH1 = var(adj_nMLH1.foci)
)
G.HETC.val <- G.mean.MLH1.obs$mean.MLH1[1] / G.mean.MLH1.obs$mean.MLH1[2]


#PERC
HetC.PERC <- MLH1_data[(MLH1_data$strain == "PERC")&(MLH1_data$quality < 5),]
#unique(HetC.PERC$mouse)
#non-passing mice
#G.remove.mice <- c("8jun15_G_f3","8jun15_G_f4")
#HetC.G <- HetC.PERC[ ! HetC.PERC$mouse %in% PERC.remove.mice, ]

#calq sex mean
PERC.mean.MLH1.obs <- ddply(.data=HetC.PERC, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                 mean.MLH1 = mean(adj_nMLH1.foci),
                 var.MLH1 = var(adj_nMLH1.foci)
)
G.HETC.val <- G.mean.MLH1.obs$mean.MLH1[1] / G.mean.MLH1.obs$mean.MLH1[2]

#MUSC

#PWD
HetC.PWD <- MLH1_data[(MLH1_data$strain == "PWD")&(MLH1_data$quality < 5),]
#unique(HetC.PWD$mouse)
#non-passing mice
PWD.remove.mice <- c("1apr15_PWD_f2","8oct14_PWD_f2","8oct14_PWD_f5")
HetC.PWD <- HetC.PWD[ ! HetC.PWD$mouse %in% PWD.remove.mice, ]

#calq sex mean
PWD.mean.MLH1.obs <- ddply(.data=HetC.PWD, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                 mean.MLH1 = mean(adj_nMLH1.foci),
                 var.MLH1 = var(adj_nMLH1.foci)
)
PWD.HETC.val <- PWD.mean.MLH1.obs$mean.MLH1[1] / PWD.mean.MLH1.obs$mean.MLH1[2]

# CZECH
HetC.CZECH <- MLH1_data[(MLH1_data$strain == "CZECH")&(MLH1_data$quality < 5),]
#unique(HetC.CZECH$mouse)

#non-passing mice
#PWD.remove.mice <- c("1apr15_PWD_f2","8oct14_PWD_f2","8oct14_PWD_f5")
#HetC.PWD <- HetC.PWD[ ! HetC.PWD$mouse %in% PWD.remove.mice, ]

#calq sex mean
CZECH.mean.MLH1.obs <- ddply(.data=HetC.CZECH, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                 mean.MLH1 = mean(adj_nMLH1.foci),
                 var.MLH1 = var(adj_nMLH1.foci)
)

CZECH.HETC.val <- CZECH.mean.MLH1.obs$mean.MLH1[1] / CZECH.mean.MLH1.obs$mean.MLH1[2]


# SKIVE
HetC.SKIVE <- MLH1_data[(MLH1_data$strain == "SKIVE")&(MLH1_data$quality < 5),]
#unique(HetC.SKIVE$mouse)

#non-passing mice
#PWD.remove.mice <- c("1apr15_PWD_f2","8oct14_PWD_f2","8oct14_PWD_f5")
#HetC.PWD <- HetC.PWD[ ! HetC.PWD$mouse %in% PWD.remove.mice, ]

#calq sex mean
SKIVE.mean.MLH1.obs <- ddply(.data=HetC.SKIVE, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                 mean.MLH1 = mean(adj_nMLH1.foci),
                 var.MLH1 = var(adj_nMLH1.foci)
)

SKIVE.HETC.val <- SKIVE.mean.MLH1.obs$mean.MLH1[1] / SKIVE.mean.MLH1.obs$mean.MLH1[2]



# KAZ
HetC.KAZ <- MLH1_data[(MLH1_data$strain == "KAZ")&(MLH1_data$quality < 5),]
#unique(HetC.KAZ$mouse)

#non-passing mice
#PWD.remove.mice <- c("1apr15_PWD_f2","8oct14_PWD_f2","8oct14_PWD_f5")
#HetC.PWD <- HetC.PWD[ ! HetC.PWD$mouse %in% PWD.remove.mice, ]

#calq sex mean
KAZ.mean.MLH1.obs <- ddply(.data=HetC.KAZ, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                 mean.MLH1 = mean(adj_nMLH1.foci),
                 var.MLH1 = var(adj_nMLH1.foci)
)
KAZ.HETC.val <- KAZ.mean.MLH1.obs$mean.MLH1[1] / KAZ.mean.MLH1.obs$mean.MLH1[2]


#TOM
HetC.TOM <- MLH1_data[(MLH1_data$strain == "TOM")&(MLH1_data$quality < 5),]
#unique(HetC.TOM$mouse)

#non-passing mice
#PWD.remove.mice <- c("1apr15_PWD_f2","8oct14_PWD_f2","8oct14_PWD_f5")
#HetC.PWD <- HetC.PWD[ ! HetC.PWD$mouse %in% PWD.remove.mice, ]

#calq sex mean
TOM.mean.MLH1.obs <- ddply(.data=HetC.TOM, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                 mean.MLH1 = mean(adj_nMLH1.foci),
                 var.MLH1 = var(adj_nMLH1.foci)
)
TOM.HETC.val <- TOM.mean.MLH1.obs$mean.MLH1[1] / TOM.mean.MLH1.obs$mean.MLH1[2]



#AST
HetC.AST <- MLH1_data[(MLH1_data$strain == "AST")&(MLH1_data$quality < 5),]
#unique(HetC.AST$mouse)

#non-passing mice
#PWD.remove.mice <- c("1apr15_PWD_f2","8oct14_PWD_f2","8oct14_PWD_f5")
#HetC.PWD <- HetC.PWD[ ! HetC.PWD$mouse %in% PWD.remove.mice, ]

#calq sex mean
AST.mean.MLH1.obs <- ddply(.data=HetC.AST, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                 mean.MLH1 = mean(adj_nMLH1.foci),
                 var.MLH1 = var(adj_nMLH1.foci)
)
AST.HETC.val <- AST.mean.MLH1.obs$mean.MLH1[1] / AST.mean.MLH1.obs$mean.MLH1[2]


#Molosinus

#MSM
HetC.MSM <- MLH1_data[(MLH1_data$strain == "MSM")&(MLH1_data$quality < 5),]
unique(HetC.MSM$mouse)

#non-passing mice
MSM.remove.mice <- c("12sep16_MSM_f3")
HetC.MSM <- HetC.MSM[ ! HetC.MSM$mouse %in% MSM.remove.mice, ]

#calq sex mean
MSM.mean.MLH1.obs <- ddply(.data=HetC.MSM, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                 mean.MLH1 = mean(adj_nMLH1.foci),
                 var.MLH1 = var(adj_nMLH1.foci)
)

MSM.HETC.val <- MSM.mean.MLH1.obs$mean.MLH1[1] / MSM.mean.MLH1.obs$mean.MLH1[2]


#MOLF
HetC.MOLF <- MLH1_data[(MLH1_data$strain == "MOLF")&(MLH1_data$quality < 5),]
unique(HetC.MOLF$mouse)

#non-passing mice
MOLF.remove.mice <- c("27oct18_MOLF_m3")
HetC.MOLF <- HetC.MOLF[ ! HetC.MOLF$mouse %in% MOLF.remove.mice, ]

#calq sex mean
MOLF.mean.MLH1.obs <- ddply(.data=HetC.MOLF, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                 mean.MLH1 = mean(adj_nMLH1.foci),
                 var.MLH1 = var(adj_nMLH1.foci)
)

MOLF.HETC.val <- MOLF.mean.MLH1.obs$mean.MLH1[1] / MOLF.mean.MLH1.obs$mean.MLH1[2]

```



```{r HetC.plot, echo=FALSE, warning=FALSE}
#HETC plot

#set up data

HetC_strain_table.M <- ddply(MLH1_data[(MLH1_data$quality < 5)&(MLH1_data$sex == "male"),], c("strain"), summarise,
                         Nmice = length(unique(mouse)),
                         Ncells  = length(adj_nMLH1.foci),
                         mean_co = format(round(  mean(adj_nMLH1.foci), 3 ), nsmall=3),
                         var = format(round(   var(nMLH1.foci),3), nsmall=3),
                         sd   = round(sd(adj_nMLH1.foci), 3),
                         se   = round(sd / sqrt(Ncells), 3),
                         cV = round( (as.numeric(sd) / as.numeric(mean_co) ),3)
)
#
#HetC_strain_table.M <- HetC_strain_table.M[ -c(7, 4, 10,11,12, 14, 17, 18,19), ]

HetC_strain_table.F <- ddply(MLH1_data[(MLH1_data$quality < 5)&(MLH1_data$sex == "female"),], c("strain"), summarise,
                         Nmice = length(unique(mouse)),
                         Ncells  = length(adj_nMLH1.foci),
                         mean_co = format(round(  mean(adj_nMLH1.foci), 3 ), nsmall=3),
                         var = format(round(   var(nMLH1.foci),3), nsmall=3),
                         sd   = round(sd(adj_nMLH1.foci), 3),
                         se   = round(sd / sqrt(Ncells), 3),
                         cV = round( (as.numeric(sd) / as.numeric(mean_co) ),3)
)
#HetC_strain_table.F <- HetC_strain_table.F[ -c(11), ]

#remove the 'other'

HetC_strain_table.M$sex <- "male"
HetC_strain_table.F$sex <- "female"

HetC_strain_table <- merge(HetC_strain_table.M, HetC_strain_table.F, by= "strain")

HetC_strain_table <- add_subsp(HetC_strain_table)

#HetC_strain_table$subsp <- c("Dom", "Dom", "Dom", "Musc", "Musc","Musc","Musc", "Cast", "Outgroup","Outgroup")

HetC_strain_table$mean_co.x <- as.numeric(HetC_strain_table$mean_co.x)
HetC_strain_table$mean_co.y <- as.numeric(HetC_strain_table$mean_co.y)
#str(HetC_strain_table)

Het.C.plot <- ggplot(data= HetC_strain_table, aes(x= mean_co.x, y=mean_co.y, color = subsp, label=strain ))+ geom_point(aes(size=2)) + xlim(c(20,35)) + ylim(c(20, 35)) + geom_abline(slope = 1, intercept = 0)+labs(x="male MLH1 mean",y="female MLH1 mean")+ggtitle("Heterochiasmy Plot")+
  scale_color_manual(values=c("royalblue", "limegreen",  "tomato", "slategray2", "plum", "grey")) + geom_text(data = HetC_strain_table, aes(label=HetC_strain_table$strain, x=mean_co.x, y= mean_co.y,  hjust = 0, nudge_x = 0.09, nudge_y = 0.09) )


Het.C.plot
#this plot is missing several 

```


There is now MOLF, which has female biased hetC
3 of my Musc strains have male biased patter; SKIVE, PWD and MSM.
1 of the musc strains has female biased heterochiasmy, KAZ. 



```{r mouse.specific.Qplot, echo=FALSE, message=FALSE, results='hide', warning=TRUE, include=FALSE, eval=FALSE}
#### Distributions by mouse
#There are code chunks that make a big dfs of CO count by quality score plot. The #code has been commented out, and moved to it's own doc.

#remove this block since it has it's own src/script

### to integrate 'pass'/fail
### this should be in setup data

#mice that don't have female data will break the chain-loop
#MOLF female, HMI, CZECH female, CAROLI female, Tom male

#instead of adding fake data, remove the categories from list

#filler_data_MOLF.f = c("BatchX", "1jan19_fake_MOLF_f1_sp1", "99999", "3","20","20","","","","","","1jan19_fake_MOLF_f1_sp1_rev.tif",
#                       "fake_MOLF_f1", "MOLF female", "MOLF","female","Musc", "20","")
#filler_data_HMI.f = c("BatchX", "1jan19_fake_HMI_f1_sp1", "99999", "3","20","20","","","","","","1jan19_fake_HMI_f1_sp1_rev.tif",
#                       "fake_HMI_f1", "HMI female", "HMI","female","Cast", "20","")
#filler_data_CZECH.f = c("BatchX", "1jan19_fake_CZECH_f1_sp1", "99999", "3","20","20","","","","","","1jan19_fake_CZECH_f1_sp1_rev.tif",
#                       "fake_CZECH_f1", "CZECH female", "CZECH","female","Musc", "20","")
#filler_data_CAROLI.f = c("BatchX", "1jan19_fake_CAROLI_f1_sp1", "99999", "3","20","20","","","","","","1jan19_fake_CAROLI_f1_sp1_rev.tif",
#                       "fake_CAROLI_f1", "CAROLI female", "CAROLI","female","Caroli", "20","")
#MLH1_data_table <- rbind(MLH1_data, filler_data_MOLF.f,filler_data_HMI.f,filler_data_CZECH.f,filler_data_CAROLI.f)#error from dataset thing

pass_mice <- c("28sep18_CZECH_m3","26oct18_MOLF_m1","22jun15_G_m2")


#this loop makes a quant_status col for a list of mice
for(i in 1:length(MLH1_data$mouse)){
  MLH1_data$quant_status[i] <- ifelse( is.element(MLH1_data$mouse[i], pass_mice), "pass", NA)
}


df_list <- split(MLH1_data, (MLH1_data$category))
#remove sections which are in category but have no data
#this is where you remove the non-data data!
df_list <- df_list[ -c(7,12,14,18,20,21,22,24,26,28,30) ]

#7 PERC, 12 MOLF female, 
#

#df list, is a list of df's which were split() from BIG df by category
df_list 
o=0
i=0

MLH1_data$quant_status <- as.factor(MLH1_data$quant_status)

#make a new empty df. create a ddply thing with mean calculations (this is like a list for matching up later)
for(i in 1:(length(df_list))){
 # only include data with quality values, then create a ddply for specific sub-df that calqs mean by each mouse
  df_list[[i]] <- df_list[[i]][!(is.na(df_list[[i]]$quality) | df_list[[i]]$quality==""), ]
  sub_mouse_means <- ddply(df_list[[i]], .(mouse), summarise,
                      mouse_nMLH1 = mean(adj_nMLH1.foci)
  )
# this loop is going though each row of dataframes within the df.list and assigning the mouse mean
  for(o in 1:length(df_list[[i]]$Original.Name)){
    df_list[[i]]$mouse_mean_MLH1[o] <- sub_mouse_means$mouse_nMLH1[ (df_list[[i]]$mouse[o] == sub_mouse_means$mouse) ]
        }
}


plot_list = list()
for (h in 1:length(df_list)) {
  pf <- ggplot(data = df_list[[h]], aes(colour = factor(Batch)), width = 0.25)+ 
    geom_jitter(aes(y= adj_nMLH1.foci, x= quality)) + 
    geom_rect(data = df_list[[h]], aes(fill=as.factor(quant_status)), alpha=0.5, xmin=-100, xmax=100, ymin=-100, ymax=100)+
    facet_wrap(~ mouse, drop = TRUE) + 
    geom_hline(aes(yintercept = mouse_mean_MLH1), color="red") +scale_fill_manual(values=c(NA, "red"), breaks=NULL) +    geom_hline(yintercept = c( mean(df_list[[h]]$adj_nMLH1.foci) ), color="black" )
  #theme(plot.background = element_rect(fill = "green")) #theme for indiviual mice, 
  #make an empty ggplot? for missing data
  
  plot_list[[h]] =   pf
}

#orig (I think this is the same as above)
#plot_list = list()
#for (h in 1:length(df_list)) {
#  pf <- ggplot(df_list[[h]],(aes(y= adj_nMLH1.foci, x= quality)))
#  pf <- pf +geom_rect(data = df_list[[h]], aes(fill=quant_status), xmin=-100, xmax=100, ymin=-100, ymax=100, alpha=0.005)

#    pf <- pf + geom_jitter(width = 0.25, aes(colour = factor(Batch)) )
#  pf <- pf + geom_hline(yintercept = c( mean(df_list[[h]]$adj_nMLH1.foci) ), color="black" )
#  pf <- pf + facet_wrap(~ mouse) + geom_hline(aes(yintercept = mouse_mean_MLH1), color="red") + 
    
 #   scale_fill_manual(values=alpha(c("red"), .005), breaks=NULL) +

    #theme for indiviual mice, 
#theme(plot.background = element_rect(fill = "green"))
#  plot_list[[h]] =   pf
#}
#can't change the alpha/transparency of the red rec

#aes must be same length ---   #geom_point(stat = "identity") +
# the last one is correct....?

#so plot list is some wride dataframe, but indeces 1:12 in plot list are ggplots
```

```{r show.plots, echo=FALSE, message=FALSE,results='hide', warning=FALSE, include=FALSE, eval=FALSE}
geom_hline(aes(yintercept = med, group = gr), colour = 'red')
#no plot is made that goes in the list -- so it can't be shown
for (gg in 1:length(plot_list) ){
  show(plot_list[gg])
 # invisible(lapply(obj, function(x) plot(x,main="some plot")))
}

#this stops at MSM
```

The mouse specific scatter plots aren't show here because there are too bulky. These plots are in a different document.

Making all of these scatter plots, allows us to look at the whole distributions of the data for each mouse. The distance of the red line from the black could be a indicator of slides or mice with slide specific technical noise.










```{r HetC.plot22, echo=FALSE,  fig.width=10, fig.height=5, fig.caption="Sex-Specific MLH1 average", eval=FALSE}


no_female <- c("HMI", "SKIVE", "CZECH", "MOLF", "CAROLI", "TOM","AST")

MixedModel_CO_data <- AP_mouse_table_w.Ages[AP_mouse_table_w.Ages$strain.x != "other",]
CO_data_HetC_w.old <- AP_mouse_table_w.Ages[AP_mouse_table_w.Ages$strain.x != "other",]
CO_data_HetC_w.old <- CO_data_HetC_w.old[ ! CO_data_HetC_w.old$strain.x %in% no_female, ]
CO_data_HetC_w.old <- CO_data_HetC_w.old[!(is.na(CO_data_HetC_w.old$Nmice) | CO_data_HetC_w.old$Nmice==""), ]

MixedModel_CO_data <- MixedModel_CO_data[MixedModel_CO_data$age.weeks < 15,]

MixedModel_CO_data <- MixedModel_CO_data[!(is.na(MixedModel_CO_data$mouse) | MixedModel_CO_data$mouse==""), ]


MixedModel_CO_data <- MixedModel_CO_data[ ! MixedModel_CO_data$strain.x %in% no_female, ]

kuku <- ggplot(MixedModel_CO_data, aes(y=mean_co, x=mouse, color=sex.y))+geom_point()+facet_wrap(~strain.x)

bubu <- ggplot(data=CO_data_HetC_w.old)+geom_point(aes(y=mean_co, x=mouse, color=sex.x))+facet_wrap(~strain.x)

#add titles
#use multiplot
#remove X axis

multiplot(kuku,bubu, cols=2)
```

# Trash / Extra

<!--

```{r chrm.pro.Curated.BivData, echo=FALSE, eval=FALSE}


#this chunk migth need to be deleted // I think everything is already done within the BivData
#this is a second mode of 

#bivalent data might be too big..
#load(file="~./MLH1repo/data/CleanBivData.RData")
#Curates_BivData -- hand.foci.count!

#For each mouse I need to count up
#the number of bivalent classes are not counted up

chrm.propotion.category <- ddply(.data=Curated_BivData, 
                 .(category),
                 summarize, 
                 ncells = length(unique(fileName)),
                 nmice = length(unique(mouse)),
                  tot_biv  = length(Obj.ID),

                 n0CO = length(which(hand.foci.count == 0)),
                  n1CO = length(which(hand.foci.count == 1)),
                  n2CO = length(which(hand.foci.count == 2)),
                  n3CO = length(which(hand.foci.count == 3)),
                  #n4CO = sum(n4CO),
                 
                 per0CO = sum(n0CO) / tot_biv,                 
                 per1CO= sum(n1CO)/ tot_biv,
                 per2CO= sum(n2CO)/ tot_biv,
                 per3CO= sum(n3CO)/ tot_biv )
                # per4CO= sum(n4CO)/ tot_biv  )



Chrm.proportions_Biv.Data <- melt(chrm.propotion.category, c(
  "per0CO", "per1CO", "per2CO", "per3CO"), 
  id=c("category", "tot_biv") )
colnames(Chrm.proportions_Biv.Data) <- c("category","total_biv","Chrm.Type","value") 

# I think I need to make 2 melt tables -- for count and proportions
Chrm.numbers_Biv.Data <- melt(chrm.propotion.category, c("n0CO", "n1CO", "n2CO", "n3CO" ), 
  id=c("category", "tot_biv") )
colnames(Chrm.numbers_Biv.Data) <- c("category","total_biv", "Chrm.Type", "biv_count") 

Chrm.prop_Biv.Data <- cbind(Chrm.proportions_Biv.Data, Chrm.numbers_Biv.Data[,4])
colnames(Chrm.prop_Biv.Data) <- c("category","total_biv", "Chrm.Type","value", "biv_count") 

Chrm.prop_Biv.Data <-Chrm.prop_Biv.Data[(!is.na(Chrm.prop_Biv.Data$value)),]

Chrm.prop_Biv.Data$sex <- ifelse(grepl("female", Chrm.prop_Biv.Data$category), "female", ifelse(grepl("male", Chrm.prop_Biv.Data$category), "male", "" ))

Chrm.prop_Biv.Data$strain <- ifelse(grepl("WSB", Chrm.prop_Biv.Data$category), "WSB", ifelse(grepl("G", Chrm.prop_Biv.Data$category), "G",
ifelse(grepl("LEW", Chrm.prop_Biv.Data$category), "LEW",                                                                                                           
      ifelse(grepl("KAZ", Chrm.prop_Biv.Data$category), "KAZ",           
        ifelse(grepl("CZECH", Chrm.prop_Biv.Data$category), "CZECH", 
        ifelse(grepl("SKIVE", Chrm.prop_Biv.Data$category), "SKIVE",
        ifelse(grepl("PWD", Chrm.prop_Biv.Data$category), "PWD",  
        ifelse(grepl("MSM", Chrm.prop_Biv.Data$category), "MSM", "" ))))))))

Chrm.prop_Biv.Data$strain<- factor(Chrm.prop_Biv.Data$strain,levels =c("WSB","G","LEW",
                       "KAZ","CZECH", "SKIVE","PWD","MSM"), order=T )
#no mol yet

Chrm.prop_Biv.Data <- with(Chrm.prop_Biv.Data, Chrm.prop_Biv.Data[order(strain),])

#need to make list of positions

chrm_prop.BivData_plot <- ggplot(data = Chrm.prop_Biv.Data, aes(y = value, x = strain, fill=as.factor(Chrm.Type)) ) + geom_col(position = "dodge")+ggtitle("Chrm Proportions from BivData")+ facet_wrap(~sex, scales = "free", ncol = 1 )+ylim(c(0,1)) + labs(x="", y='Proportions') +   geom_text(aes(label = biv_count), position = position_dodge(0.99))
  
chrm_prop.BivData_plot

```

-->


# References