% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  pdftitle={Sex-Specific Evolution of the Meiotic Recombination Rate},
  pdfauthor={April L. Peterson, Bret Payseur},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage[margin=1in]{geometry}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering
\usepackage{titling}

\pretitle{%
  \begin{center}
  \LARGE
  \includegraphics[width=4cm,height=6cm]{logo.png}\\[\bigskipamount]
}
\posttitle{\end{center}}
\usepackage{booktabs}
\usepackage{longtable}
\usepackage{array}
\usepackage{multirow}
\usepackage{wrapfig}
\usepackage{float}
\usepackage{colortbl}
\usepackage{pdflscape}
\usepackage{tabu}
\usepackage{threeparttable}
\usepackage{threeparttablex}
\usepackage[normalem]{ulem}
\usepackage{makecell}
\usepackage{xcolor}

\title{Sex-Specific Evolution of the Meiotic Recombination Rate}
\author{April L. Peterson, Bret Payseur}
\date{2020-06-14}

\begin{document}
\maketitle

\hypertarget{abstract}{%
\section{ABSTRACT}\label{abstract}}

Although meiotic recombination is required for successful gametogenesis
in most species that reproduce sexually, the rate of crossing over
varies among individuals. \textbf{define genome-wide recombination
rate}.

Differences in recombination rate between females and males are perhaps
the most striking form of this variation. Existing data fail to directly
address the extent to which recombination experiences similar
evolutionary pressures. To fill this gap, we measured meiotic
recombination in both sexes for a panel of house mouse cross three
subspecies. Using inbred strains and single cell immunohistochemistry
allowed us to place sex-specific observations within the same genetic
background and meiotic context. Our results indicate highly discordant
evolutionary patterns in the two sexes. Whereas male recombination rates
show evidence of rapid evolution over short evolutionary timescales,
female recombination rates measured in the same strains are mostly
static. These results strongly indicate that house mouse has two genome
wide recombination rates which display distinct evolutionary
trajectories.

\hypertarget{introduction}{%
\section{INTRODUCTION}\label{introduction}}

\hypertarget{importance-of-recombination-rate}{%
\subsubsection{0. Importance of recombination
rate}\label{importance-of-recombination-rate}}

Meiosis can be reduced to the expression of (2n -\textgreater{} 4n
-\textgreater{} 2n -\textgreater{} 1n) which tracks the duplication of a
diploid genome into haploid cell products. The meiotic program relies on
crossovers and the process of recombination to ensure the correct
separation of chromosomes. The total number of COs per cell (at the 4n
stage) is equal to the genome wide recombination rate (gwRR).
\textbf{The genome wide recombination rate is a cell based metric for
genome wide recombination rate is tightly connected to organisms fitness
through correct chromosome segregation and fertility.}

\textbf{at the fine scale / in regards to genetic variation}, In the
indriect manner, the recombination rate regulates populations' response
to selection, and determine the fate of novel mutations by transferring
beneficial mutations onto novel genetic backgrounds or by breaking
linkage of negative mutations from beneficial genetic backgrounds
(cite). This process shapes the genomic patterns of genetic variation,
with high recombination areas of genome having more nucleotide variation
while areas of low recombination have decreased genetic variation (Begun
Aquadro, Nachman, Payseur).

\textbf{but back to the genome-wide recombination rate}, The process of
chromosome segregation is integral to the process of producing haploid
cells and can be considered a constraint in regards to the evolution.
When the genome wide recombination rate is framed within the single cell
context it is closely connected to the contraints which are imposed at
the single cell level. Reflecting the importance of it's role in
chromosome segreation is that the number of haploid pairs is strong
predictor for genome wide RR across large species(taxa) / divergence
scales (Otto and Payseur, 2019; Stapley et al., 2017). (the number of
crossovers per chromosome are highly conserved)\textless However, whie
the rate of recombination is highly variable, the count of crossovers
per chromosome is conserved between 1 to 3 (cite), suggesting that such
thresholds/constraints may be behind the overall patterns.\textgreater{}

\textbf{connection of single cell based gwRR to thresholds imposed at
the single cell level} (why is gwRR the better scale for evolutionary
patterns?)

\hypertarget{understanding-levels-of-variation-in-gwrr}{%
\subsubsection{1. Understanding levels of variation in
gwRR}\label{understanding-levels-of-variation-in-gwrr}}

Because we know the meiotic programs, we have expectations for where the
thresholds(constraints) of the gwRR the rate will fall. First, the lower
threshold is defined as an obligate crossover per homolog pair which
promotes proper chromosome segregation (Nagaoka et al., 2012). second
(while there is less evidence,), the upper threshold maybe
regulated/shaped/affected due to rates of ectopic recombination or the
programmed DNA damage preceding crossovers act as constraints to the
upper threshold of meiotic recombination (cite). Yet, Within these
thresholds the genome wide recombination rate still varies across
species and indiviuals. which is the subject of an open question in the
field. \textbf{the gwRR is the best metric for since both of these
constraints are thought to act at the cell wide level.}

\textbf{Sex is one of the most notable axes along which individuals
vary}. Sexual dimorphism in recombination rates is called heterochiasmy
(Lenormand, 2003). An understanding of how sex shapes the evolution of
recombination cannot be achieved with available data. Comprehensive
comparisons of female and male recombination rates usually come from
outbred populations humans (Kong et al., 2004,, 2008, 2014; Halldorsson
et al., 2019), dog, cattle (Ma et al., 2015; Shen et al., 2018), (other
sheep) sheep (Johnston et al., 2016), and mouse (CC cite) in which the
role of sex is confounded with the contributions of genetic variation.
Under anisogamy where a species has distinct gametes, there is no first
principle which would predict the evolution of sexually dimorphic
recombination rates. Yet heterochiasmy, is commonly observed in
dioecious species, suggesting that other meiotic traits which
distinguish the gametes and their meiotic programs, for example
symmetrical vs asymmetrical cell division, may impose selection for
sexually dimorphic recombination rates. Although it is clear that the
relationship between female and male recombination rates can differ
among species, comparisons between and within closely related species
are missing. Direct contrasts between females and males across a common,
diverse set of genomic backgrounds would reveal whether the
recombination rate evolves differently in the sexes.

\textless It's important to note, -- that the genome wide recombination
rate can be decomposed -- and connected to related traits (which also
show conserved patterns of sexual dimorphism\textgreater{}

\textbf{understanding heterochiasmy and sexual dimorphism in meiosis
have advanced since first being documented (morgan 1914) -- more than
just the direction -- more evidence is accumulating that there are
conserved patterns of sexual dimorphism in meiotic recombinatio -
related to recombination rate}. Recently two specific features have been
highlighted as conserved patterns of heterochiasmy:

Cahoon and Libuda (2019) show In species with meiotic chromatin (SC axis
length) quantified for both sexes, longer chromosome axis generally has
higher recombination rates. Sardell and Kirkpatrick (2020), A survey of
51 species shows there are conserved broad scale recombination landscape
differences. Generally males have telomere-bias crossover placement and
females have more uniform placement.

These observations raise the question of conservation in sexually
patterns are maintained with the gwRR and it's decomposed traits

\textbf{remember these points kinda distract from gwRR aspect}

\hypertarget{the-house-mouse-is-a-great-model-for-uncovering-evolutionary-patterns-at-a-short-timescale}{%
\subsubsection{2. The House Mouse is a great model for uncovering
evolutionary patterns at a short
timescale}\label{the-house-mouse-is-a-great-model-for-uncovering-evolutionary-patterns-at-a-short-timescale}}

The House mouse complex arose from a recent radiation providing an
opportunity to interrogate natural variation at short evolutionary
scales, potentially narrowing the mutational space for identifying
variants which would cause gwRR variation. Add divergence ranges.
divergence spanning 0.5 mya to 5 mya.

Classical lab strains of mice have generated extensive knowledge central
to meiosis and outcomes on recombination through extensive studies on
the genes involved in the meiotic recombination pathway (cite) and
previous crosses for understanding the genetic architecture of
recombination rate variation ({\textbf{???}}); Murdoch et al. (2010);
Wang and Payseur (2017){]}.

\textbf{Wild derived inbred strains generate the best comparison of
females and males, besides the sex chromosomes, the mouse for each
genome are almost identical}. Inbred strains enable one of the most
direct comparison of both female and male -- versions of meiosis
--enable use to conclude that sex is a primary factor --

Additionally due to their global distribution allows us to compare
samples of natural diversity from a broad geographic range. (wild
derived inbred strains -- will capture genetic diversity not represented
in classical lab strains.)

While there are strains with Robertsonian translocations, in house mouse
and related murid species it is possible to assemble sets of strains
with identical karyotypes, 20 pairs of acrocentric chromosomes.

House mouse is well suited for single cell cytology approaches. The
single cell level allows integration of data at a closer connection to
the molecular pathway and meiotic program. In addition to the
quantification of gwRR. This study quantifies precursors to crossovers,
double strand breaks (DSBs), and meiotic chromosome morphology across
stages of meiosis which would not be accessible to with genetic linkage
data and crosses.

\hypertarget{what-we-accomplished-in-this-paper}{%
\subsubsection{3. What we accomplished in this
paper}\label{what-we-accomplished-in-this-paper}}

For the first time in many of these strains, We report a rare, direct,
evolutionary comparison of recombination rate in females and males for
this short evolutionary scale. We use rare strains from a broad range of
geographic locations of the species territory.

The gwRR is made up of crossovers occurring on individual chromosomes
within cells. Thus we quantified meiotic chromosome morphology (SC
length) and placement of crossovers to comprise an approximate picture
of the recombination landscape.

Our results indicate rapid male specific evolution of gwRR occurred in a
subset of house mouse lines. We observed up to a 30\% difference
(translating into approximately \textasciitilde7 more crossovers per
cell), a surprising amount considering the short evolutionary time
scale.

\hypertarget{materials-and-methods}{%
\section{MATERIALS AND METHODS}\label{materials-and-methods}}

\hypertarget{mice}{%
\subsubsection{Mice}\label{mice}}

We used a panel of wild-derived inbred strains of house mice (\emph{Mus
musculus}) and related murid species to profile natural genetic
variation in recombination (Table 1). Our survey included 5 strains from
\emph{M. m. musculus}, 4 strains from \emph{M. m. domesticus}, 2 strains
from \emph{M. m. molossinus}, 2 strains from \emph{M. m. castaneus}, and
1 strain each from \emph{M. spicilegus} and \emph{M. spretus}.
\textless removed \emph{M. caroli}\textgreater{} We subsequently denote
strains by their abbreviated subspecies and name
(e.g.~\emph{domesticus\textsuperscript{WSB}}). Mice were housed at
dedicated, temperature-controlled facilities in the UW-Madison School of
Medicine and Public Health, with the exception of mice from Gough
Island, which were housed in a temperature-controlled facility in the
UW-Madison School of Veterinary Medicine. Mice from an inbred strain of
Gough Island mice were sampled after \textbf{XX} generations of
brother-sister mating. All mice were provided with ad libitum food and
water. Procedures followed protocols approved by IACUC.

\hypertarget{tissue-collection-and-immunohistochemistry}{%
\subsubsection{Tissue Collection and
Immunohistochemistry}\label{tissue-collection-and-immunohistochemistry}}

The same dry-down spread technique was applied to both spermatocytes and
oocytes, following (Peters et al., 1997), with adjustment for volumes.
Spermatocyte spreads were collected and prepared as described in
(Peterson et al., 2019). The majority of mice used for MLH1 counts were
between 5 and 12 weeks of age. Juvenile mice between 12 and 15 days of
age were used for DMC1 counts. Both ovaries were collected from embryos
(16-21 embryonic days) or neonates (0-48 hours after birth). Whole
testes were incubated in 3ml of hypotonic solution for 45 minutes.
Decapsulated ovaries were incubated in 300ul of hypotonic solution for
45 minutes. Fifteen microliters of cell slurry (masticated gonads) were
transferred to 80ul of 2\% PFA solution. Cells were fixed in this
solution and dried in a humid chamber at room temperature overnight. The
following morning, slides were treated with a Photoflow wash (Kodak,
cite). Slides were stored at -20*C if not stained immediately. To
visualize the structure of meiotic chromosomes, we used antibody markers
for the centromere (CREST) and lateral element of the synaptonemal
complex (SC) (SYCP3). Crossovers (COs) were visualized as MLH1 foci.
Double strand breaks (DSBs) were visualized as DMC1 foci. The staining
protocol followed (Anderson et al., 1999) and (Koehler et al., 2002).
Antibody staining and slide blocking were performed in 1X antibody
dilution buffer (ADB) (normal donkey serum (Jackson ImmunoResearch), 1X
PBS, bovine serum albumin (Sigma), and Triton X-100 (Sigma)). Following
a 30-minute blocking wash in ABD, each slide was incubated with 60ul of
a primary antibody master mix for 48 hours at 37*C. The master mix
recipe contained polyclonal anti-rabbit anti-MLH1 (Calbiochem; diluted
1:50) or anti-rabbit anti-DMC1 (mix of DMC1), anti-goat polyclonal
anti-SYCP3, (Abcam; diluted 1:50), and anti-human polyclonal antibody to
CREST (Antibodies, Inc; diluted 1:200) suspended in ADB. Slides were
washed twice in 50ml ADB before the first round of secondary antibody
incubation for 12 hours at 37*C. Alexa Fluor 488 donkey anti-rabbit IgG
(Invitrgoen, location; diluted to 1:100) and Coumarin AMCA donkey
anti-human IgG (Jackson ImmunoResearch; diluted to 1:200) were suspended
in ADB. The last incubation of Alexa Fluor 568 donkey anti-goat
(Invitrogen; diluted 1:100) was incubated at 1:100 for 2 hours at 37* C.
Slides were fixed with Prolong Gold Antifade (Invitrogen) for 24 hours
after a final wash in 1x PBS.

\hypertarget{image-processing}{%
\subsubsection{Image Processing}\label{image-processing}}

Images were captured using a Zeiss Axioplan 2 microscope with AxioLab
camera and AxioVision software (Zeiss, Cambridge, UK). Preprocessing,
including cropping, noise reduction, and histogram adjustments, was
performed using Photoshop (v13.0). Image file names were anonymized
before manual scoring of MLH1 foci or DMC1 foci using Photoshop.

\hypertarget{analysis}{%
\subsubsection{Analysis}\label{analysis}}

\textless new 1/n\textgreater{}

To estimate the number of crossovers across the genome, we counted MLH1
foci. MLH1 foci were counted in cells with intact and complete
karyotypes (19 acrocentric bivalents and XY for spermatocytes; 20
acrocentric bivalents for oocytes) and distinct MLH1 foci. A quality
score ranging from 1 (best) to 5 (worst) was assigned to each cell based
on visual appearance of staining and spread of bivalents. Cells with a
score of 5 were excluded from the final analysis. Distributions of MLH1
count per cell were visually inspected for normality (Supplemental
Figure 1). MLH1 foci located on the XY in spermatocytes were excluded
from counts. In addition to MLH1 counts, we measured several traits to
further characterize the recombination landscape. To estimate the number
of double-strand breaks, a minority of which lead to crossovers, mean
DMC1 foci per cell was quantified for a single male from each of a
subset of strains (\emph{molossinus\textsuperscript{MSM}},
\emph{musculus\textsuperscript{PWD}},
\emph{domesticus\textsuperscript{WSB}}, and
\emph{domesticus\textsuperscript{G}} ). SC morphology and CREST foci
number were used to stage spermatocytes as early zygotene or late
zygotene.

\textless new 2/n\textgreater{} To measure bivalent SC length, two image
analysis algorithms were used. The first algorithm estimates the total
(summed) SC length across bivalents for individual cells (Wang et al.
(2019)). The second algorithm estimates the SC length of individual
bivalents (Peterson et al. (2019)). Both algorithms apply a
`skeletonizing' transformation to synapsed chromosomes that produces a
single, pixel-wide `trace' of the bivalent shape. Total SC length per
cell was quantified from pachytene cell images (Wang et al. (2019)).

To reduce algorithmic errors in SC isolation, outliers were visually
identified at the mouse level and removed from the data set. Mouse
averages were calculated from cell-wide total SC lengths in 3,204 out of
3,884 cells with MLH1 counts. SC length of individual bivalents was
quantified in pachytene cell images (Peterson et al. (2019)). The DNA
CrossOver algorithm (Peterson et al. (2019)) isolates single,
straightened bivalent shapes, returning SC length, location of MLH1
foci, and location of CREST (centromere) foci. The algorithm
substantially speeds the accurate measurement of bivalents, but it
sometimes interprets overlapping bivalents as single bivalents. In our
data set, average proportions of bivalents per cell isolated by the
algorithm ranged from 0.48 (\emph{molossinus\textsuperscript{MSM}} male)
to 0.72 (\emph{musculus\textsuperscript{KAZ}} female). From the total
set of pachytene cell images, 10,222 bivalent objects were isolated by
the algorithm. Following a manual curation, 9,575 single-bivalent
observations remained. The accuracy of the algorithm is high compared to
hand measures after this curation step (Peterson et al., 2019). The
curated single bivalent data supplements our cell-wide MLH1 count data
with MLH1 foci counts for single bivalents. Proportions of bivalents
with the same number of MLH1 foci were compared across strains using a
chi-square test.

\textless new 3/n\textgreater{} To account for confounding effects of
sex chromosomes from pooled samples of bivalents, we also considered a
reduced data set including only bivalents with SC lengths below the 2nd
quartile in cells with at least 17 of 20 single bivalent measures. This
``short bivalent'' data set included the four or five shortest bivalents
and excluded the X bivalent in oocytes. A total of 704 short bivalents
were isolated from 102 oocytes and 43 spermatocytes. Although this
smaller data set has decreased power, it offers a more comparable set of
single bivalents to compare between the sexes. A ``long bivalent'' data
set was formed from those bivalents above the 4th quartile in SC lengths
per cell. A total of 709 long bivalents were isolated from 102 oocytes
and 43 spermatocytes.

\textless new 4/n\textgreater{} To examine crossover interference, the
distance (in SC units) between MLH1 foci (inter-focal distance;
IFD\textsubscript{raw}) was measured for those single bivalents
containing two MLH1 foci. A normalized measure of interference
(IFD\textsubscript{norm}) was computed by dividing
IFD\textsubscript{raw} by SC length on a per-bivalent basis.

\textless new 5/n\textgreater{} We used a series of statistical models
to interpret patterns of variation in the recombination traits we
measured (Table 2). We used mouse average as the dependent variable in
all analyses. We first constructed a linear mixed model (M1) using
lmer() from the lmer4 package (Bates et al., 2015) in R (v3.5.2) (Team,
2015). In this model, strain was coded as a random effect, with
significance evaluated using a likelihood ratio test (using exactRLRT()
from RLRsim (Scheipl et al., 2008)). Subspecies, sex, and their
interaction were coded as fixed effects, with significance evaluated
using a chi-square test comparing the full and reduced models (drop1()
and anova()) (Bates et al., 2015). As each observation was at the level
of a single mouse (mouse average), it was uniquely coded within the
dataset, nesting was implicit, each strain only occurs within one
strain, and was not explicitly coded in our linear and mixed models. We
used the subspecies effect to quantify divergence between subspecies and
the (random) strain effect to quantify variation within subspecies in a
sex-specific manner. In separate analyses using model M1, we considered
mouse averages as dependent variables for each of the following traits:
MLH1 count per cell, total SC length per cell, single bivalent SC length
per cell, IFD\textsubscript{raw}, IFD\textsubscript{norm}, and average
MLH1 position (for single-focus bivalents). Four additional lnear models
containing only fixed effects (M2-M5) (Table 2) were used to further
investigate results obtained from model M1.

\hypertarget{results}{%
\section{RESULTS}\label{results}}

\hypertarget{genome-wide-recombination-rate-evolves-differently-in-females-and-males}{%
\subsection{Genome-wide recombination rate evolves differently in
females and
males}\label{genome-wide-recombination-rate-evolves-differently-in-females-and-males}}

\textless new 1/n\textgreater{}

We used counts of MLH1 foci per cell to estimate genome-wide
recombination rates in 14 wild-derived inbred strains sampled from three
subspecies of house mice (\emph{M. m. domesticus}, \emph{M. m. musculus}
and \emph{M. m. molossinus} ) and three additional species of Mus
(\emph{M. spretus}, \emph{M. spicilegus} , and \emph{M. caroli}). Mean
MLH1 focus counts for 186 mice were quantified from an average of 21.97
spermatocytes per male (for a total of 1,714 spermatocytes) and 18
oocytes per female (for a total of 1,427 oocytes).

\textless new 2/n\textgreater{}

Graphical comparisons reveal sex-specific dynamics to the evolution of
genome-wide recombination rate (Figure 1A). First, MLH1 focus counts
differ between females and males in most strains. Second, the difference
in counts between the sexes varies among strains. Although most strains
show more MLH1 foci in females, two strains
(\emph{musculus\textsuperscript{PWD}} and
\emph{molossinus\textsuperscript{MSM}}) exhibit higher counts in males.
In females, numbers of MLH1 foci are evenly distributed around the
sex-wide mean of approximately 25 (Figure 1B). In stark contrast, males
largely separate into two groups of strains with high numbers (near 30)
and low numbers (near 23) of foci (Figure 1C). Strain mean MLH1 focus
counts from females and males are uncorrelated (Spearman's \(\rho\) =
0.08 - 0.08; p = 0.84 -- 0.84) across the set of strains.

\textless new 3/n\textgreater{}

To further partition variation in recombination rate, we fit a series of
linear models to mean MLH1 focus counts from 137 house mice (Table 2;
detailed results available in Supplemental Table X). Strain, sex,
subspecies, and sex*subspecies each affect MLH1 focus count in a linear
mixed model (M1; strain (random effect): p \textless{} 10-4 --
\ensuremath{10^{-4}}; sex: p = 6.2x10-4 --
\ensuremath{4.95\times 10^{-6}}; subspecies: p = 1.1x10-3 -- 0;
subspecies*sex: p = 2x10-4 -- \ensuremath{2.1\times 10^{-4}}).

The effect of subspecies is no longer significant in a model treating
all factors as fixed effects \textbf{(M2; p = 0.09 -- )}, highlighting
strain and sex as salient variables. Two strains exhibit particularly
strong effects on MLH1 focus count (M3;
\emph{domesticus\textsuperscript{G}} p = \ensuremath{2.19\times 10^{-6}}
; \emph{domesticus\textsuperscript{LEW}} p = 0.02), with sex-strain
interactions involving three strains (M3;
\emph{domesticus\textsuperscript{G}} p = 0;
\emph{molossinus\textsuperscript{MSM}} p = 0 ;
\emph{musculus\textsuperscript{PWD}} p =
\ensuremath{5.31\times 10^{-4}}).

In separate analyses of males (M4; n = 80 - 71), three strains
disproportionately shape MLH1 focus count (as observed in Figure 1C):
\emph{musculus\textsuperscript{PWD}} (p =
\ensuremath{3.08\times 10^{-7}} ; effect = 6.11 ),
\emph{molossinus\textsuperscript{MSM}} (p =
\ensuremath{3.52\times 10^{-9}}; effect = 7.02), and
\emph{musculus\textsuperscript{SKIVE}} (p =
\ensuremath{7.59\times 10^{-4}}; effect = 4.04 ). These three strains
point to substantial evolution in the genome-wide recombination rate in
spermatocytes; we subsequently refer to them as ``high-recombination''
strains.

In females (M4; n= 76 - 76), four strains affect MLH1 focus count:
\emph{domesticus\textsuperscript{G}} (p =
\ensuremath{1.16\times 10^{-5}}; effect = 3.28),
\emph{molossinus\textsuperscript{MSM}} (p =
\ensuremath{2.97\times 10^{-5}}; effect = 2.98),
\emph{domesticus\textsuperscript{LEW}} (p = 0.03 ; effect = 1.7), and
\emph{musculus\textsuperscript{PWD}} (p = 0.08; effect = 1.16). Strain
effect sizes in females are modest in magnitude compared to those in
males.

Together, these results demonstrate that the genome-wide recombination
rate evolves in a highly sex-specific manner.

\hypertarget{synaptonemal-complexes-are-longer-in-females}{%
\subsection{Synaptonemal complexes are longer in
females}\label{synaptonemal-complexes-are-longer-in-females}}

\textless new 1/n\textgreater{} The variation in sex differences in
recombination we discovered provided an opportunity to determine whether
sex differences in chromatin compaction, as measured by the length of
the synaptonemal complex (SC), are reversed when heterochiasmy is
reversed. In all strains except \emph{musculus\textsuperscript{SKIVE}},
females have longer SCs than males, whether SC length was estimated as
the total length across bivalents or as the length of short bivalents
(t-tests; all p \textless{} 0.05, except short bivalents in
\emph{musculus\textsuperscript{SKIVE}}, p = 0.11 -- 0.11`).

Among short bivalents (to which the female X bivalent does not
contribute), female to male ratios of mouse mean SC length range from
1.26 - 1.26 (\emph{musculus\textsuperscript{PWD}}) to 1.49 - 1.49
(\emph{domesticus\textsuperscript{WSB}}) across strains. That females
have longer SCs is further supported by models that include covariates,
which identify sex as the most consistently significant effect for total
SC length (M1: p = 7.16\^{}\{-20\} -- \ensuremath{1.2\times 10^{-31}} ;
M2: p = 5.33\^{}\{-4\} -- \ensuremath{1.95\times 10^{-8}}; M3: p =0.05
-- 0.01 ) and short bivalent length (M1: p = 1.27\^{}\{-11 --
\ensuremath{1.27\times 10^{-11}}\}; M2: p = 1.98\^{}\{-7\} --
\ensuremath{1.98\times 10^{-7}}; M3: p = 1.98\^{}\{-7\} --
\ensuremath{1.98\times 10^{-7}}).

The existence of some subspecies and strain effects (M1 subspecies*sex:
p = 0.03 -- 0.03; M2: strain-\emph{musculus\textsuperscript{SKIVE}} :
p=0.02 -- 0.02 ; \emph{musculus\textsuperscript{SKIVE}}*male: p = 0.08
-- 0.08) further indicates that SC length has evolved among strains and
among subspecies.

In summary, two approaches for measuring SC length demonstrate that
females have longer SCs (chromosome axes), even in strains in which
males recombine more. This pattern implies that in high-recombination
strains, spermatocytes have less space than oocytes in which to position
additional crossovers.

\textbf{double check that these are the only sig factors for all the
models}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{females-and-males-differ-in-crossover-positions-and-crossover-interference}{%
\subsection{Females and males differ in crossover positions and
crossover
interference}\label{females-and-males-differ-in-crossover-positions-and-crossover-interference}}

\textless new 1/n\textgreater{} We used normalized positions of MLH1
foci along bivalents with a single focus to compare crossover location
while controlling for differences in SC length. In all strains, MLH1
foci tend to be closer to the telomere in males (mean normalized
position in males: 0.68--0.68; mean normalized position in females: 0.56
-- 0.56; t-test; p = 3.84\^{}\{-23\} --
\ensuremath{3.84\times 10^{-23}}). Sex is also the strongest determinant
of MLH1 focus position in the models we tested (M1: p = 7.79\^{}\{-26 --
\ensuremath{7.79\times 10^{-26}}\}; M2: p = 5.39\^{}\{-8\} --
\ensuremath{5.39\times 10^{-8}}; M3: p = 5.39\^{}\{-8\} --
\ensuremath{5.39\times 10^{-8}}).

\textless new 2/n\textgreater{} Males have longer normalized mean
inter-focal distances (IFD\textsubscript{norm}) than females in seven
out of eight strains (t-tests; p \textless{} 0.05), with only
\emph{musculus\textsuperscript{KAZ}} showing no difference (p = 0.33 --
0.33). Examination of IFD\textsubscript{norm} distributions indicates
that females are centered at approximately 50\% and show a slight
enrichment of low (\textless25\%) values, whereas males are enriched for
higher values. Models treating IFDnorm as the dependent variable support
the inference of stronger interference in males, with sex being the most
significant variable (M1: p = 3.11\^{}\{-12\} --
\ensuremath{3.11\times 10^{-12}} ; M2: p = 0.01 -- 0.01; M3: p = 0.01 --
0.01). In contrast, there is no clear signal of sex differences in raw
mean inter-focal distances (IFD\textsubscript{raw}) across the full set
of strains, whether they are considered separately or together.
Visualization of normalized MLH1 foci positions on bivalents with two
crossovers (Figure 4C) further suggests that interference distances vary
more in females than in males, and that males display a stronger
telomeric bias in the placement of the distal crossover.

In summary, controlling for differences in SC length (chromatin
compaction) indicates that interference is stronger in males, whereas
interference on the physical scale is similar in the sexes.

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{evolution-of-genome-wide-recombination-rate-is-dispersed-across-bivalents-associated-with-double-strand-break-number-and-connected-to-crossover-interference}{%
\subsection{Evolution of genome-wide recombination rate is dispersed
across bivalents, associated with double-strand break number, and
connected to crossover
interference}\label{evolution-of-genome-wide-recombination-rate-is-dispersed-across-bivalents-associated-with-double-strand-break-number-and-connected-to-crossover-interference}}

\textless new 1/n\textgreater{} We used the contrast between males from
high-recombination strains and males from low-recombination strains to
identify features of the recombination landscape associated with
evolutionary transitions in the genome-wide recombination rate. We
considered proportions of bivalents with different numbers of
crossovers, double-strand break number, SC length, and crossover
positioning.

\textless new 2/n\textgreater{} Ninety-six percent of single bivalents
in our pooled dataset (n = 9,576 -- 9,575 ) have either one or two MLH
foci (Figure 3). The proportions of single-focus (1CO) bivalents
vs.~double-focus (2CO) bivalents distinguish high-recombination strains
from low-recombination strains (Figure 3). High-recombination strains
are enriched for 2CO bivalents at the expense of 1CO bivalents:
proportions of 2CO bivalents are 0.33 -- 0.33 in
\emph{musculus\textsuperscript{SKIVE}}, 0.44 -- 0.44 in
\emph{musculus\textsuperscript{PWD}} , and 0.51 -- 0.51 in
\emph{molossinus\textsuperscript{MSM}} (Supplemental Figure 3).
Following patterns in the genome-wide recombination rate, male
\emph{musculus\textsuperscript{PWD}} and male
\emph{molossinus\textsuperscript{MSM}} have 2CO proportions that are
more similar to each other than to strains from their own subspecies
(chi-square tests; \emph{musculus\textsuperscript{PWD}}
vs.~\emph{molossinus\textsuperscript{MSM}}: p = 0.37 -- 0.37 ;
\emph{musculus\textsuperscript{PWD}}
vs.~\emph{musculus\textsuperscript{KAZ}}: p = 1.23\^{}\{-31\} --
\ensuremath{1.23\times 10^{-31}}; \emph{molossinus\textsuperscript{MSM}}
vs.~\emph{molossinus\textsuperscript{MOLF}}: p = 2.65\^{}\{-6\} --
\ensuremath{2.82\times 10^{-6}} ). These results demonstrate that
evolution of the genome-wide recombination rate reflects changes in
crossover number across multiple bivalents.

\textless new 3/n\textgreater{} To begin to localize evolution of
genome-wide recombination rate to steps of the recombination pathway, we
counted DMC1 foci in prophase spermatocytes as markers for double-strand
breaks (DSBs). DMC1 foci were counted in a total of 76 -- 76 early
zygotene and 75 -- 75 late zygotene spermatocytes from two
high-recombination strains (\emph{musculus\textsuperscript{PWD}} and
\emph{molossinus\textsuperscript{MSM}}) and three low-recombination
strains (\emph{musculus\textsuperscript{KAZ}},
\emph{domesticus\textsuperscript{WSB}}, and
\emph{domesticus\textsuperscript{G}}). High-recombination strains have
significantly more DMC1 foci than low-recombination strains in early
zygotene cells (t-test; p \textless{} 10\^{}\{-6\} --
\ensuremath{10^{-6}} ). In contrast, the two strain groups do not differ
in DMC1 foci in late zygotene cells (t-test; p = 0.66 -- 0.66 ). Since
DSBs are repaired as either COs or non-crossovers (NCOs), the ratio of
MLH1 foci to DMC1 foci can be used to estimate the proportion of DSBs
designated as COs. High-recombination and low-recombination strains do
not differ in the MLH1/DMC1 ratio, whether DMC1 foci were counted in
early zygotene cells or late zygotene cells (t-test; p \textgreater{}
0.05). These results raise the possibility that the evolution of
genome-wide recombination rate is primarily determined by processes that
precede the CO/NCO decision, at least in house mice.

\textless new 4/n\textgreater{} Total SC length only partially
differentiates high-recombination strains from low-recombination strains
(Figure 5). Whereas high-recombination strains as a group have
significantly greater total SC length than low-recombination strains
(t-test; p = 0.01 -- 0.02 ), separate tests within subspecies show that
the two strain categories differ within \emph{M. m. molossinus} (p =
0.02 -- \ensuremath{4.71\times 10^{-4}} ) but not within \emph{M. m.
musculus} (p = 0.4 -- 0.65 ). Additionally, mouse means for the reduced
(short and long) bivalent datasets do not differ between
high-recombination and low-recombination strains (t-test; short: p =
0.88 -- 0.88 ; long: p = 0.19 -- 0.19 ).

Write results for M4 and M5 here total SC M4 results

Musc subspecies very \textbf{\emph{, Mol }} (both ++) G, LEW, MSM SKIVE,
KAZ (- -)

M5 (almost all strains are significant) G, LEW, PWD, MSM MOLF, SKIVE,
KAZ, TOM, AST, CZECH

In a model with total SC length as the dependent variable (\emph{M2?}),
two subspecies effects are significant (\emph{M. m. musculus} p =
2.33\^{}\{-6\} -- 0.65, \emph{M. m. molossinus}, p = 21\^{}\{-6\} --
\ensuremath{4.71\times 10^{-4}}). In models with SC lengths of short and
long bivalents as dependent variables, several subspecies and strain
effects reach significance (p \textless{} 0.05), but they are not
consistent across models. Collectively, these results reveal that
evolution of SC length is not strongly associated with evolution of
genome-wide recombination rate in house mice.

\textless new 5/n\textgreater{} In summary, evolution of the genome-wide
recombination rate in males is connected to double-strand break number
and crossover interference, but not to SC length and crossover position
(on single-crossover bivalents).

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{discussion}{%
\section{DISCUSSION}\label{discussion}}

\textless new 1/n\textgreater{} By comparing recombination rates in
females and males from the same diverse set of genetic backgrounds, we
isolated sex as a primary factor in the evolution of this fundamental
meiotic trait. Recombination rate differences are more pronounced in
males than females. Because inter-strain divergence times are identical
for the two sexes, this observation demonstrates that the genome-wide
recombination rate evolves faster in males, at least in house mice. More
generally, recombination rate divergence is decoupled in females and
males. These disparities are remarkable given that recombination rates
for the two sexes were measured in identical genomic backgrounds (other
than the presence/absence of the Y chromosome). Our results provide the
strongest evidence yet that the genome-wide recombination rate follows
distinct evolutionary trajectories in males and females.

\textless new 2/n\textgreater{} At the genetic level, the sex-specific
evolution we documented indicates that some mutations responsible for
divergence in recombination rate have dissimilar phenotypic effects in
females and males. A subset of the genetic variants associated with
genome-wide recombination rate within populations of humans (Kong et
al., 2004,, 2008, 2014; Halldorsson et al., 2019), Soay sheep (Johnston
et al., 2016), and cattle (Ma et al., 2015; Shen et al., 2018) appear to
show sex-specific properties, including opposite effects in females and
males. Furthermore, inter-sexual correlations for recombination rate are
weak in humans (Fledel-Alon et al., 2011) and Soay sheep (Johnston et
al., 2016). Crosses between the strains we surveyed could be used to
identify and characterize the genetic variants responsible for
recombination rate evolution in house mice (Dumont and Payseur, 2011;
Wang et al., 2019; Wang and Payseur, 2017). These variants could
differentially affect females and males at any step in the recombination
pathway. Although our DMC1 profiling was limited to males from a small
number of strains (for practical reasons), our findings suggest that
mutations that determine the number of double-strand breaks contribute
to sex-specific evolution in the recombination rate. A study of two
classical inbred strains and one wild-derived inbred strain of house
mice also found a positive association between crossover number and
double-strand break number in males (Baier et al., 2014).

\textless new 3/n\textgreater{} Another implication of our results is
that the connection between recombination rate and fitness differs
between males and females. Little is known about whether and how natural
selection shapes recombination rate in nature (Dapper and Payseur, 2017;
Ritz et al., 2017). Samuk et al. (2020) recently used a quantitative
genetic test to conclude that an 8\% difference in genome-wide
recombination rate between females from two populations of
\emph{Drosophila pseudoobscura} was caused by natural selection.
Applying similar strategies to species in which both sexes recombine,
including house mice, would be a logical next step to understanding the
sex-specific evolution of recombination rate.

\textless new 4/n\textgreater{} Population genetic models have been
built to explain sexual dimorphism in the number and placement of
crossovers, which is a common phenomenon (Brandvain and Coop, 2012;
Sardell and Kirkpatrick, 2020). Modifier models predicted that lower
recombination rates in males will result from haploid selection
(Lenormand, 2003) or sexually antagonistic selection on coding and
cis-regulatory regions of genes (Sardell and Kirkpatrick, 2020). Another
modifier model showed that meiotic drive could stimulate female-specific
evolution of the recombination rate (Brandvain and Coop, 2012). Although
these models fit the conserved pattern of sex differences in crossover
positions, they do not readily explain our observations of sex-specific
evolution in the genome-wide recombination rate. In particular, the
alternation across strains in which sex has more crossovers is
unexpected.

\textless new 5/n\textgreater{} We propose an alternative interpretation
of our findings based on the cell biology of gametogenesis. During
meiosis, achieving a stable chromosome structure requires the attachment
of kinetochores to opposite poles of the cell and at least one crossover
to create tension across the sister chromosome cohesion distal to
chiasmata (Lane and Kauppi, 2019; VanVeen and Hawley, 2003). The spindle
assembly checkpoint (SAC) prevents aneuploidy by ensuring that all
bivalents are correctly attached to the microtubule spindle
(``bi-oriented'') before starting the metaphase-to-anaphase transition
via the release of the sister cohesion holding homologs together (Lane
and Kauppi, 2019). Hence, selection seems likely to favor mutations that
optimize the process of bi-orientation and chromosome separation,
thereby prohibiting the SAC from delaying the cell cycle or triggering
apoptosis. Multiple lines of evidence indicate that the SAC is more
effective in spermatogenesis than in oogenesis (Lane and Kauppi, 2019),
perhaps due to the presence of the centrosome spindle (So et al., 2019)
and larger cell volume (Kyogoku and Kitajima, 2017) in oocytes. The
higher stringency of the SAC during spermatogenesis suggests that
selection will be better at removing mutations that interfere with
bi-orientation in males than in females. Therefore, faster male
evolution of the genome-wide recombination rate could be driven by the
more stringent SAC acting on chromosome structures at the metaphase I
alignment.

\textless new 5/n\textgreater{} Our SAC model is consistent with other
features of our data. We showed that widespread sex differences in
broad-scale crossover positioning (Sardell and Kirkpatrick, 2020) apply
across house mice, even in lineages where the direction of heterochiasmy
is reversed. The number and placement of crossovers affects the area of
sister chromosome cohesion distal to crossovers which needs to be
released for the first reductional chromosome segregation (Dumont and
Desai, 2012; Lane and Kauppi, 2019; Subramanian and Hochwagen, 2014;
VanVeen and Hawley, 2003). Faster spermatogenesis may select for
synchronization of the separation across all homologs within the cell
(\textbf{Kudo} ), whereas in oogenesis, the slower cell cycle and
multiple arrest stages may require chromosome structures with greater
stability on the MI spindle, especially for those organisms that undergo
dictyate arrest (Lee, 2019).

\textless new 6/n\textgreater{} We propose that the SAC model also can
explain the correlated evolution of stronger crossover interference and
higher genome-wide recombination rate in male house mice. Our results
show that crossovers are spaced further apart in strains enriched for
double-crossover bivalents when SC length is taken into account and
chromosome size effects are minimized. Assuming chromatin compaction
between (prophase) pachytene and metaphase is uniform along bivalents,
this increased spacing is expected to expand the area for sister
cohesion to connect homologs and may improve the fidelity of chromosomal
segregation. While the SAC model postulates direct fitness effects of
interference, a modifier model predicted that indirect selection on
recombination rate -- via its modulation of offspring genotypes -- can
strengthen interference as well (Goldstein et al., 1993).

\textless new 7/n\textgreater{} Regardless of the underlying mechanism,
our results provide a rare demonstration that crossover interference can
diverge over short evolutionary timescales. The notion that stronger
interference can co-evolve with higher genome-wide recombination rate is
supported by differences between breeds of cattle (Ma et al. (2015)) and
differences between populations of white-footed mice (Peterson et al.,
2019). In contrast, mammalian species with stronger interference tend to
exhibit lower genome-wide recombination rates (Segura et al. (2013),
Otto and Payseur (2019)). Collectively, these patterns suggest that
inferences about the evolutionary dynamics of interference depend on the
timescale under consideration.

\textless new 8/n\textgreater{} Our findings further reveal that
evolution of the genome-wide recombination rate does not require major
changes in the degree of chromatin compaction. Female house mice
consistently show longer SCs, even in strains with more recombination in
males. Studies in mice (Lynn et al., 2002, p. @petkov2007) and humans
(Gruhn et al., 2013, p. @tease2004) suggest that chromosomal axes are
longer (and DNA loops are shorter) in females than males. Some authors
have suggested that conserved sex differences in crossover positioning
(more uniform placement in females) and interference strength (stronger
interference in males) could be due to looser chromatin packing of the
meiotic chromosome structure in females (Haenel et al., 2018; Petkov et
al., 2007). A cellular model designed to explain interference attributes
sexual dimorphism in chromatin structure to greater cell volumes and
oscillatory movements of telomeres and kinetochores in oocytes (Hultén,
2011). More recent work connects the sparser recombination landscape to
sex differences in the crossover repair pathway (Wang et al., 2017).

\textless new 9/n\textgreater{} Our conclusions are accompanied by
several caveats. First, MLH1 foci only identify interfering crossovers
(Holloway et al., 2008). Although most crossovers belong to this class
\textbf{(REF)}, our approach likely underestimated genome-wide
recombination rates. Evolution of the number of non-interfering
crossovers is a topic worth examining. A second limitation is that our
investigation of crossover locations was confined to the relatively low
resolution possible with immunofluorescent cytology. Positioning
crossovers with higher resolution could reveal additional evolutionary
patterns. Finally, the panel of inbred lines we surveyed may not be
representative of recombination rate variation within and between
subspecies of house mice. We considered most available wild-derived
inbred lines, but house mice have a broad geographic distribution.
Nevertheless, we expect our primary conclusion that recombination rate
evolves in a sex-specific manner to be robust to geographic sampling
because differences between females and males exist for the same set of
inbred strains.

\textless new 10/n\textgreater{} While the causes of sex differences in
recombination remain mysterious Lenormand et al. (2016), our conclusions
have implications for a wide range of recombination research. For
biologists uncovering the cellular and molecular determinants of
recombination, our results suggest that mechanistic differences between
the sexes could vary by genetic background. For researchers charting the
evolutionary trajectory of recombination, our findings indicate that
sex-specific comparisons are crucial. For theoreticians building
evolutionary models of recombination, different fitness regimes and
genetic architectures in females and males should be considered.
Elevating sex as a primary determinant of recombination would be a
promising step toward integrating knowledge of cellular mechanisms with
evolutionary patterns to understand recombination rate variation in
nature.

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{references}{%
\section*{REFERENCES}\label{references}}
\addcontentsline{toc}{section}{REFERENCES}

\hypertarget{refs}{}
\leavevmode\hypertarget{ref-anderson1999}{}%
Anderson LK, Reeves A, Webb LM, Ashley T. 1999. Distribution of crossing
over on mouse synaptonemal complexes using immunofluorescent
localization of mlh1 protein. \emph{Genetics} \textbf{151}:1569--1579.

\leavevmode\hypertarget{ref-baier2014}{}%
Baier B, Hunt P, Broman KW, Hassold T. 2014. Variation in genome-wide
levels of meiotic recombination is established at the onset of prophase
in mammalian males. \emph{PLoS genetics} \textbf{10}.

\leavevmode\hypertarget{ref-lme4}{}%
Bates D, Mächler M, Bolker B, Walker S. 2015. Fitting linear
mixed-effects models using lme4. \emph{Journal of Statistical Software}
\textbf{67}:1--48.
doi:\href{https://doi.org/10.18637/jss.v067.i01}{10.18637/jss.v067.i01}

\leavevmode\hypertarget{ref-brandvain2012scrambling}{}%
Brandvain Y, Coop G. 2012. Scrambling eggs: Meiotic drive and the
evolution of female recombination rates. \emph{Genetics}
\textbf{190}:709--723.

\leavevmode\hypertarget{ref-CahoonLibuda2019}{}%
Cahoon CK, Libuda DE. 2019. Leagues of their own: Sexually dimorphic
features of meiotic prophase i. \emph{Chromosoma} 1--16.

\leavevmode\hypertarget{ref-DapperPayseur2017}{}%
Dapper AL, Payseur BA. 2017. Connecting theory and data to understand
recombination rate evolution. \emph{Philosophical Transactions of the
Royal Society B: Biological Sciences} \textbf{372}:20160469.

\leavevmode\hypertarget{ref-dumont2011}{}%
Dumont BL, Payseur BA. 2011. Genetic analysis of genome-scale
recombination rate evolution in house mice. \emph{PLoS genetics}
\textbf{7}.

\leavevmode\hypertarget{ref-dumontDesai2012}{}%
Dumont J, Desai A. 2012. Acentrosomal spindle assembly and chromosome
segregation during oocyte meiosis. \emph{Trends in cell biology}
\textbf{22}:241--249.

\leavevmode\hypertarget{ref-fledel2011}{}%
Fledel-Alon A, Leffler EM, Guan Y, Stephens M, Coop G, Przeworski M.
2011. Variation in human recombination rates and its genetic
determinants. \emph{PloS one} \textbf{6}.

\leavevmode\hypertarget{ref-goldstein1993}{}%
Goldstein DB, Bergman A, Feldman MW. 1993. The evolution of
interference: Reduction of recombination among three loci.
\emph{Theoretical population biology} \textbf{44}:246--259.

\leavevmode\hypertarget{ref-gruhn2013}{}%
Gruhn JR, Rubio C, Broman KW, Hunt PA, Hassold T. 2013. Cytological
studies of human meiosis: Sex-specific differences in recombination
originate at, or prior to, establishment of double-strand breaks.
\emph{PloS one} \textbf{8}.

\leavevmode\hypertarget{ref-haenel2018}{}%
Haenel Q, Laurentino TG, Roesti M, Berner D. 2018. Meta-analysis of
chromosome-scale crossover rate variation in eukaryotes and its
significance to evolutionary genomics. \emph{Molecular ecology}
\textbf{27}:2477--2497.

\leavevmode\hypertarget{ref-halldorsson2019}{}%
Halldorsson BV, Palsson G, Stefansson OA, Jonsson H, Hardarson MT,
Eggertsson HP, Gunnarsson B, Oddsson A, Halldorsson GH, Zink F, others.
2019. Characterizing mutagenic effects of recombination through a
sequence-level genetic map. \emph{Science} \textbf{363}:eaau1043.

\leavevmode\hypertarget{ref-holloway2008mus81}{}%
Holloway JK, Booth J, Edelmann W, McGowan CH, Cohen PE. 2008. MUS81
generates a subset of mlh1-mlh3--independent crossovers in mammalian
meiosis. \emph{PLoS genetics} \textbf{4}.

\leavevmode\hypertarget{ref-hulten2011_COM}{}%
Hultén MA. 2011. On the origin of crossover interference: A chromosome
oscillatory movement (com) model. \emph{Molecular cytogenetics}
\textbf{4}:10.

\leavevmode\hypertarget{ref-johnston2016_soay}{}%
Johnston SE, Bérénos C, Slate J, Pemberton JM. 2016. Conserved genetic
architecture underlying individual recombination rate variation in a
wild population of soay sheep (ovis aries). \emph{Genetics}
\textbf{203}:583--598.

\leavevmode\hypertarget{ref-koehler2002}{}%
Koehler KE, Cherry JP, Lynn A, Hunt PA, Hassold TJ. 2002. Genetic
control of mammalian meiotic recombination. I. Variation in exchange
frequencies among males from inbred mouse strains. \emph{Genetics}
\textbf{162}:297--306.

\leavevmode\hypertarget{ref-Kong2004}{}%
Kong A, Barnard J, Gudbjartsson DF, Thorleifsson G, Jonsdottir G,
Sigurdardottir S, Richardsson B, Jonsdottir J, Thorgeirsson T, Frigge
ML, others. 2004. Recombination rate and reproductive success in humans.
\emph{Nature genetics} \textbf{36}:1203--1206.

\leavevmode\hypertarget{ref-Kong2014}{}%
Kong A, Thorleifsson G, Frigge ML, Masson G, Gudbjartsson DF, Villemoes
R, Magnusdottir E, Olafsdottir SB, Thorsteinsdottir U, Stefansson K.
2014. Common and low-frequency variants associated with genome-wide
recombination rate. \emph{Nature genetics} \textbf{46}:11.

\leavevmode\hypertarget{ref-Kong2008}{}%
Kong A, Thorleifsson G, Stefansson H, Masson G, Helgason A, Gudbjartsson
DF, Jonsdottir GM, Gudjonsson SA, Sverrisson S, Thorlacius T, others.
2008. Sequence variants in the rnf212 gene associate with genome-wide
recombination rate. \emph{Science} \textbf{319}:1398--1401.

\leavevmode\hypertarget{ref-kyogoku2017}{}%
Kyogoku H, Kitajima TS. 2017. Large cytoplasm is linked to the
error-prone nature of oocytes. \emph{Developmental cell}
\textbf{41}:287--298.

\leavevmode\hypertarget{ref-LaneKauppi2019}{}%
Lane S, Kauppi L. 2019. Meiotic spindle assembly checkpoint and
aneuploidy in males versus females. \emph{Cellular and molecular life
sciences} \textbf{76}:1135--1150.

\leavevmode\hypertarget{ref-Lee2019}{}%
Lee J. 2019. Is age-related increase of chromosome segregation errors in
mammalian oocytes caused by cohesin deterioration? \emph{Reproductive
Medicine and Biology}.

\leavevmode\hypertarget{ref-lenormand2003}{}%
Lenormand T. 2003. The evolution of sex dimorphism in recombination.
\emph{Genetics} \textbf{163}:811--822.

\leavevmode\hypertarget{ref-lenormand2016}{}%
Lenormand T, Engelstädter J, Johnston SE, Wijnker E, Haag CR. 2016.
Evolutionary mysteries in meiosis. \emph{Philosophical Transactions of
the Royal Society B: Biological Sciences} \textbf{371}:20160001.

\leavevmode\hypertarget{ref-lynn2002}{}%
Lynn A, Koehler KE, Judis L, Chan ER, Cherry JP, Schwartz S, Seftel A,
Hunt PA, Hassold TJ. 2002. Covariation of synaptonemal complex length
and mammalian meiotic exchange rates. \emph{Science}
\textbf{296}:2222--2225.

\leavevmode\hypertarget{ref-ma2015_cattle}{}%
Ma L, O'Connell JR, VanRaden PM, Shen B, Padhi A, Sun C, Bickhart DM,
Cole JB, Null DJ, Liu GE, others. 2015. Cattle sex-specific
recombination and genetic control from a large pedigree analysis.
\emph{PLoS genetics} \textbf{11}.

\leavevmode\hypertarget{ref-murdoch2010}{}%
Murdoch B, Owen N, Shirley S, Crumb S, Broman KW, Hassold T. 2010.
Multiple loci contribute to genome-wide recombination levels in male
mice. \emph{Mammalian Genome} \textbf{21}:550--555.

\leavevmode\hypertarget{ref-nagaoka2012}{}%
Nagaoka SI, Hassold TJ, Hunt PA. 2012. Human aneuploidy: Mechanisms and
new insights into an age-old problem. \emph{Nature Reviews Genetics}
\textbf{13}:493--504.

\leavevmode\hypertarget{ref-ottoPaysuer2019}{}%
Otto SP, Payseur BA. 2019. Crossover interference: Shedding light on the
evolution of recombination. \emph{Annual review of genetics}
\textbf{53}:19--44.

\leavevmode\hypertarget{ref-peters_1997}{}%
Peters AH, Plug AW, Vugt MJ van, De Boer P. 1997. SHORT COMMUNICATIONS A
drying-down technique for the spreading of mammalian meiocytes from the
male and female germline. \emph{Chromosome research} \textbf{5}:66--68.

\leavevmode\hypertarget{ref-peterson2019}{}%
Peterson AL, Miller ND, Payseur BA. 2019. Conservation of the
genome-wide recombination rate in white-footed mice. \emph{Heredity}
\textbf{123}:442--457.

\leavevmode\hypertarget{ref-petkov2007}{}%
Petkov PM, Broman KW, Szatkiewicz JP, Paigen K. 2007. Crossover
interference underlies sex differences in recombination rates.
\emph{Trends in Genetics} \textbf{23}:539--542.

\leavevmode\hypertarget{ref-Ritz2017}{}%
Ritz KR, Noor MA, Singh ND. 2017. Variation in recombination rate:
Adaptive or not? \emph{Trends in Genetics} \textbf{33}:364--374.

\leavevmode\hypertarget{ref-samuk2020}{}%
Samuk K, Manzano-Winkler B, Ritz KR, Noor MA. 2020. Natural selection
shapes variation in genome-wide recombination rate in drosophila
pseudoobscura. \emph{Current Biology}.

\leavevmode\hypertarget{ref-sardell_sex_2020}{}%
Sardell JM, Kirkpatrick M. 2020. Sex differences in the recombination
landscape. \emph{The American Naturalist} \textbf{195}:361--379.
doi:\href{https://doi.org/10.1086/704943}{10.1086/704943}

\leavevmode\hypertarget{ref-RLRsim}{}%
Scheipl F, Greven S, Kuechenhoff H. 2008. Size and power of tests for a
zero random effect variance or polynomial regression in additive and
linear mixed models. \emph{Computational Statistics \& Data Analysis}
\textbf{52}:3283--3299.

\leavevmode\hypertarget{ref-segura2013}{}%
Segura J, Ferretti L, Ramos-Onsins S, Capilla L, Farré M, Reis F,
Oliver-Bonet M, Fernández-Bellón H, Garcia F, Garcia-Caldés M, others.
2013. Evolution of recombination in eutherian mammals: Insights into
mechanisms that affect recombination rates and crossover interference.
\emph{Proceedings of the Royal Society B: Biological Sciences}
\textbf{280}:20131945.

\leavevmode\hypertarget{ref-Shen2018_cattle}{}%
Shen B, Jiang J, Seroussi E, Liu GE, Ma L. 2018. Characterization of
recombination features and the genetic basis in multiple cattle breeds.
\emph{BMC genomics} \textbf{19}:304.

\leavevmode\hypertarget{ref-So2019}{}%
So C, Seres KB, Steyer AM, Mönnich E, Clift D, Pejkovska A, Möbius W,
Schuh M. 2019. A liquid-like spindle domain promotes acentrosomal
spindle assembly in mammalian oocytes. \emph{Science}
\textbf{364}:eaat9557.

\leavevmode\hypertarget{ref-stapley_variation_2017}{}%
Stapley J, Feulner PGD, Johnston SE, Santure AW, Smadja CM. 2017.
Variation in recombination frequency and distribution across eukaryotes:
Patterns and processes. \emph{Philosophical Transactions of the Royal
Society B: Biological Sciences} \textbf{372}:20160455.
doi:\href{https://doi.org/10.1098/rstb.2016.0455}{10.1098/rstb.2016.0455}

\leavevmode\hypertarget{ref-subramanian2014}{}%
Subramanian VV, Hochwagen A. 2014. The meiotic checkpoint network:
Step-by-step through meiotic prophase. \emph{Cold Spring Harbor
perspectives in biology} \textbf{6}:a016675.

\leavevmode\hypertarget{ref-Rstudio}{}%
Team R. 2015. RStudio: Integrated Development Environment for R.

\leavevmode\hypertarget{ref-tease2004}{}%
Tease C, Hulten M. 2004. Inter-sex variation in synaptonemal complex
lengths largely determine the different recombination rates in male and
female germ cells. \emph{Cytogenetic and genome research}
\textbf{107}:208--215.

\leavevmode\hypertarget{ref-vanVeen2003}{}%
VanVeen JE, Hawley RS. 2003. Meiosis: When even two is a crowd.
\emph{Current Biology} \textbf{13}:R831--R833.

\leavevmode\hypertarget{ref-wang2019_SC}{}%
Wang RJ, Dumont BL, Jing P, Payseur BA. 2019. A first genetic portrait
of synaptonemal complex variation. \emph{PLoS genetics}
\textbf{15}:e1008337.

\leavevmode\hypertarget{ref-Wang2017island}{}%
Wang RJ, Payseur BA. 2017. Genetics of genome-wide recombination rate
evolution in mice from an isolated island. \emph{Genetics}
\textbf{206}:1841--1852.

\leavevmode\hypertarget{ref-wang2017inefficient}{}%
Wang S, Hassold T, Hunt P, White MA, Zickler D, Kleckner N, Zhang L.
2017. Inefficient crossover maturation underlies elevated aneuploidy in
human female meiosis. \emph{Cell} \textbf{168}:977--989.

\end{document}
