---
title: 'Sex-Specific Evolution of the Meiotic Recombination Rate'
date: '`r Sys.Date()`'
author: "April L. Peterson, Bret Payseur"
output:
  html_document:
    toc: yes
    toc_depth: '5'
    toc_float: yes
  pdf_document:
    toc: yes
    toc_depth: '5'
  word_document:
    toc: yes
    toc_depth: '5'
always_allow_html: yes
bibliography: refs_Main_Report.bib
---


```{r set.up, echo=FALSE, include=FALSE}

#library(tidyverse)
library(knitr)
library(ggplot2)
#library(pwr)

library(dplyr)
library(plyr)
library(lattice)
library(raster)
library(lme4)
library(nlme)
library(RLRsim)
library(ggpubr)
library(reshape2)
library(patchwork)
library(png)
library(grid)
library(gridBase)
library(gridExtra)
library(dplyr)
library(cowplot)
library(kableExtra)
library(xlsx)

#set wd
#load datafile

options(digits=2)

#setwd("~.")#switch over to here()


load(file="~./MLH1repo/data/Results_setup_5.21.20.RData") #Results_setup_5.14.20.RData

#load(file="~./MLH1repo/data/Results_setup_5.14.20.RData") #Results_setup_5.14.20.RData

#load(file="~./MLH1repo/data/Results_setup_4.21.20.RData")

#load(file="~./MLH1repo/data/CleanBivData_6.3.20.RData")

#
#load(file="~./MLH1repo/data/total_SC_30.12.19.RData")

#skeletonize - this is the total sc file
#SC.skel = read.csv("~./MLH1repo/data/SCskel_output_feb20.csv", header = TRUE, strip.white = TRUE)


#load functions
source("~./MLH1repo/src/CommonFunc_MLH1repo.R")
```

# INTRODUCTION

# MATERIALS AND METHODS


### MICE

We used a panel of wild-derived inbred strains of house mice ( _Mus musculus_ ) and related murids to profile natural genetic variation in recombination (Table 1). Our survey included 5 strains from _M. m. musculus_, 4 strains from _M. m. domesticus_, 2 strains from _M. m. molossinus_, 2 strains from _M. m. castaneus_, and 1 strain each from _M. caroli_, _M. spicilegus_, and _M. spretus_. We subsequently denote strains by their abbreviated subspecies and name (e.g. _domesticus^WSB^_).  Mice were housed at dedicated, temperature controlled facilities in the UW-Madison School of Medicine and Public Health, with the exception of mice from Gough Island, which were housed in a temperature-controlled facility in the UW-Madison School of Veterinary Medicine. Mice from an inbred strain of Gough Island mice were sampled after XX generations of brother-sister mating. All mice were provided with ad libitum food and water. All procedures followed protocols approved by IACUC.


### Tissue Collection and Immunohistochemistry

*new* - The same dry-down spread technique was applied to both spermatocytes and oocytes based on (@peters_1997), with adjustment for volumes. Spermatocyte spreads were collected and prepared as described in [@peterson2019]. The majority of mice used MLH1 quantification were between 5 and 12 weeks of age. Juvenile mice between 12 and 15 days of age were used for DMC1 quantification. Both ovaries were collected from embryos (16-21 embryonic days) or neonates (0-48 hours after birth). Whole testes were incubated in 3ml of hypotonic solution for 45 minutes. Decapsulated ovaries were incubated in 300ul of hypotonic solution for 45 minutes. Fifteen microliters of cell slurry (masticated gonads) were transferred to 80ul of 2% PFA solution. Cells were fixed in this solution and dried in a humid chamber at room temperature overnight. The following morning, slides were treated with a Photoflow wash (Kodak, cite). Slides were stored at -20 \* C if not stained immediately.
To visualize the structure of meiotic chromosomes we used antibody markers for the centromere (CREST) and lateral element of the synaptonemal complex (SC) (SYCP3). Crossovers (COs) were visualized as MLH1 foci. Double strand breaks (DSBs) were visualized as DMC1 foci. The staining protocol followed [@anderson1999] and [@koehler2002]. Antibody staining and slide blocking were performed in 1X antibody dilution buffer (ADB) (normal donkey serum (Jackson ImmnuoResearch), 1X PBS, bovine serum albumin (Sigma), and Triton X-100 (Sigma)).
Following a 30-minute blocking wash in ABD, each slide was incubated with 60ul of a primary antibody master mix for 48 hours at 37\* C. The master mix recipe contained polyclonal anti-rabbit anti-MLH1 (Calbiochem; diluted 1:50) or anti rabbit anti-DMC1) (mix of DMC1), anti-goat polyclonal anti-SYCP3, (Abcam; diluted 1:50), and anti-human polyclonal antibody to CREST (Antibodies, Inc; diluted 1:200) suspended in ADB. Slides were washed twice in 50ml ADB before the first round of secondary antibody incubation for 12 hours at 37\* C. Alexa Fluor 488 donkey anti-rabbit IgG (Invitrgoen, location; diluted to 1:100) and Coumarin AMCA donkey anti-human IgG (Jackson ImmunoResearch; diluted to 1:200) were suspended in ADB. The last incubation of Alexa Fluor 568 donkey anti-goat (Invitrogen; diluted 1:100) was incubated at 1:100 for 2 hours at 37\* C. Slides were fixed with Prolong Gold Antifade (Invitrogen) for 24 hours after a final wash in 1x PBS.



### Image Processing


new - Images were capture using a Zeiss Axioplan 2 microscope with AxioLab camera and AxioVision software (Zeiss, Cambridge, UK). Preprocessing, including cropping, noise reduction, and histogram adjustments, was performed using Photoshop (v13.0). Image file names were anonymized before manual scoring of MLH1 or DMC1 foci using photoshop.


### Analysis

*new-* To estimate the number of crossovers across the genome, we counted MLH1 foci. MLH1 foci were counted in cells with intact and complete karyotypes (19 acrocentric bivalents and XY for spermatocytes; 20 acrocentric bivalents for oocytes) and distinct MLH1 foci. A quality score ranging from 1 (best) to 5 (worst) was assigned to each cell based on visual appearance of staining and spread of bivalents. Cells with a score of 5 were excluded from the final analysis. Distributions of MLH1 count per cell were visually inspected for normality (supplemental figure ). MLH1 foci located on the XY in spermatocytes were excluded from counts. In addition to MLH1 counts, we measured several traits to further characterize the recombination landscape. To estimate the number of double-strand breaks, a minority of which lead to crossovers, mean DMC1 foci per cell was quantified for a single male from a subset of strains ( _musculus^PWD^_, _musculus^MSM^_, _domesticus^WSB^_, and _domesticus^G^_ ). SC morphology and CREST foci number was used to stage spermatocytes as early zygotene or late zygotene.

To measure bivalent SC length, two image analysis algorithms were used. The first algorithm estimates the total (summed) SC length across bivalents for individual cells (@wang2019_SC). The second algorithm estimates the SC length of individual bivalents (@peterson2019). Both algorithms apply a ‘skeletonizing’ transformation to synapsed chromosomes that produces a single, pixel-wide ‘trace’ of the bivalent shape. 

Total SC length per cell was quantified from pachtyene cell images [@wang2019_SC]. To reduce algorithmic errors in SC isolation, outliers were visually identified at the mouse level and removed from the data set. Mouse averages were calculated from cell-wide total SC lengths in `r formatC( length(  (Total.SC_DF  %>% filter(species == "M.musculus")  )$Random.Name), format="d", big.mark=",")` -- 3,371 out of `r formatC( length(  (MLH1_data %>% filter(species == "M.musculus") )$Random.Name), format="d", big.mark=",")`-- 4,143 cells with MLH1 counts.

SC length of individual bivalents was quantified in pachytene cell images [@peterson2019]. The DNA CrossOver algorithm [@peterson2019] isolates single, straightened bivalent shapes, returning SC length, location of MLH1 foci, and location of CREST (centromere) foci. The algorithm substantially speeds the accurate measurement of bivalents, but it sometimes interprets overlapping bivalents as single bivalents. In our data set, average proportions of bivalents per cell isolated by the algorithm ranged from 
`r min(algorithm.stat.category$mean_pass.rate)` -- 0.51 ( _molossinus^MSM^_ male) to`r max(algorithm.stat.category$mean_pass.rate)` -- 0.72 ( _musculus^KAZ^_ female). From the total set of cell images, `r formatC( length(unique(cleanCurated_BivData %>% filter(SC.pass == "1"))$fileName ), format="d", big.mark=",")` -- 10,458 bivalent objects were isolated by the algorithm. Following a manual curation, `r formatC( length(unique(cleanCurated_BivData %>% filter(hand.foci.count != ""))$Obj.ID ), format="d", big.mark=",")` -- 9,829 single-bivalent observations remained. The accuracy of the algorithm is high compared to hand measures after this curation step [@peterson2019]. The curated single bivalent data supplements our cell wide MLH1 count data with MLH1 foci counts for single bivalents. The proportions of bivalents with the same number of MLH1 foci where compared across strains by a chi-squared test using prop.test().

To account for confounding effects of sex chromosomes from pooled samples of bivalents, we also considered a reduced data set including only bivalents with SC lengths below the 2nd quartile in cells with at least 17 of 20 single bivalent measures. This “short bivalent” data set included the four or five shortest bivalents, and excluded the -- X bivalent in oocytes. A total of `r sum(Short.biv_mouse_table$nobs )` -- 678 short bivalents were isolated from `r sum(  (Short.biv_mouse_table  %>% filter(sex=="female") )$ncells)` -- 103 oocytes and `r sum(  (Short.biv_mouse_table  %>% filter(sex=="male") )$ncells)` -- 37 spermatocytes. Although this smaller data set has decreased power, it offers a more comparable set of single bivalents to compare between the sexes. A “long bivalent” data set was formed from those bivalents above the 4th quartile in SC lengths per cell. `r sum(Long.biv_mouse_table$n_obs)`  number of long bivalents were isolated from `r sum(  (Long.biv_mouse_table  %>% filter(sex=="female") )$ncells)` oocytes and xx `r sum(  (Long.biv_mouse_table  %>% filter(sex=="male") )$ncells)` spermatocytes. 

To examine crossover interference, the distance (in SC units) between MLH1 foci (inter-focal distance; IFD~raw~) was measured for those single bivalents containing two MLH1 foci. A normalized measure of interference (IFD~norm~) was computed by dividing IFDraw by SC length on a per-bivalent basis.

We used a series of statistical models to interpret patterns of variation in the recombination traits we measured (Table 2). We first constructed a mixed model (M1) using the lmer() from the lmer4 package (cite) in R (v3.5.2) [@Rstudio]. In this model, strain was coded as a random effect, with significance evaluated using a likelihood ratio test (using exactRLRT()). Subspecies, strain, and their interaction were coded as fixed effects,  with significance evaluated using a chi-square test comparing the full and reduced models (drop1() and anova()). *discribe the hiearchal nature*. strains nested within subspecies -- nesting by default, 

We used the subspecies effect to quantify divergence between subspecies and the (random) strain effect to quantify variation within subspecies in a sex-specific manner. In separate analyses, we considered mouse averages as dependent variables for each of the following traits: MLH1 count per cell, variance in MLH1 count per cell, coefficient of variation in MLH1 count per cell, total SC length per cell, single bivalent SC length per cell, IFD~raw~, IFD~norm~, and average MLH1 position (for single-focus bivalents). Four additional models containing only fixed effects (M2-M5) (Table 2) were used to further investigate results obtained from the initial mixed model. 


# RESULTS

```{r FOR.PLOT.organizing.data, echo=FALSE, message=FALSE, warning=FALSE}
#set the order of another column, based on another variable. consistant order for ploting

#apply this before ploting
MLH1_data <- MLH1_data %>%
  arrange(strain, sex, category, mouse) %>%
  mutate(Original.Name = factor(Original.Name)) #another category that you want the order to match
MLH1_data <- with(MLH1_data, MLH1_data[order(sex, strain),])

# sort your dataframe, by the focal categories
MLH1_data$mouse <- factor(MLH1_data$mouse, levels=unique(MLH1_data$mouse))
#use the above line for when order is out of wak

#MLH1_data <- with(MLH1_data, MLH1_data[order(category),])
Mouse.Table_HQ$var <- as.numeric(Mouse.Table_HQ$var)

AP_mouse_table.HQ_target <- Mouse.Table_HQ[!(is.na(Mouse.Table_HQ$sex) | Mouse.Table_HQ$sex==""), ]

remove_strains <- c("TOM","AST","CZECH","CAST","HMI","SPRET","CAROLI", "F1", "PERC")
AP_mouse_table.HQ_target <- AP_mouse_table.HQ_target[ ! AP_mouse_table.HQ_target$strain %in% remove_strains, ]

  
var.plot <- ggplot(AP_mouse_table.HQ_target, aes(y=var, x = sex, color=sex))+geom_jitter()+facet_wrap(~strain)+ggtitle("Plots of mouse Variances for MLH1 counts")+
  scale_color_manual(values=c("royalblue", "limegreen"))+theme_bw()

```

# Figure 1

```{r MLH1.MIXED.MODEL.setup, echo=FALSE, warning=FALSE, eval=TRUE, message=FALSE}

#DF.HetC.MixedModel.HQ
#this blck is duplicated in setupData.rm
#https://rcompanion.org/handbook/G_03.html
#testing random effects within lme

DF.HetC.MixedModel <- Mouse.Table_HQ[Mouse.Table_HQ$species == "M.musculus",]

#AP_mouse_table excluide Qual5 cells
DF.HetC.MixedModel.HQ <- Mouse.Table_HQ
DF.HetC.MixedModel.HQ <- add_species(DF.HetC.MixedModel.HQ)

#main df with variables coded as chr
DF.HetC.MixedModel.HQ$subsp <- factor(DF.HetC.MixedModel.HQ$subsp, ordered = FALSE)
DF.HetC.MixedModel.HQ$sex <- factor(DF.HetC.MixedModel.HQ$sex, ordered = FALSE)
DF.HetC.MixedModel.HQ$strain <- factor(DF.HetC.MixedModel.HQ$strain, ordered = FALSE)
#DF.HetC.MixedModel.HQ$species <- factor(DF.HetC.MixedModel.HQ$species, ordered = FALSE)

#kable(table(DF.HetC.MixedModel.HQ$sex, DF.HetC.MixedModel.HQ$strain))

all.cell.MM.HQ <- ddply(.data=DF.HetC.MixedModel.HQ,
                 .(sex, strain),
                 summarize,
                 all.cells = sum(Ncells)
)

```

```{r table.for.plot, echo=FALSE, warning=FALSE, message=FALSE}
#why do I add in a different table?
#this move this to the setup file
#MLH1_data  %>% filter(quality < 5)  -- low quality cells removed

#pooled
strain.table.HQ_Fig.og <- ddply( MLH1_data  %>% filter(quality < 5), c("subsp","strain", "sex"), summarise,
                        Nmice =length(unique(mouse)),
                        Ncells  =length(nMLH1.foci),
                        mean_co =as.numeric(format(round(mean(nMLH1.foci), 3 ), nsmall=3) ),
                        cV = cv(nMLH1.foci, na.rm = TRUE),
                        var = as.numeric(format(round(   var(nMLH1.foci, na.rm = TRUE),3), nsmall=3)), 
                        sd   = round(sd(nMLH1.foci, na.rm = TRUE), 3),
                       #Standard error (mice vs cells)
                        se  = round(sd / sqrt(Ncells), 3)
                       
                       #range_mouse_means = range()
                   #consider adding crhm proportions?
    )

#
strain.table.HQ_Fig.og <- add_subsp(strain.table.HQ_Fig.og)
strain.table.HQ_Fig.og <- add_species(strain.table.HQ_Fig.og)

strain.table.HQ_Fig <-  strain.table.HQ_Fig.og %>% filter(strain != "F1" & strain != "PERC" &
                       strain != "TOM" &  strain != "AST" & strain != "CZECH" & strain != "CAROLI" & strain != "HMI")

#adjust order of plot here
strain.table.HQ_Fig$strain <- factor(strain.table.HQ_Fig$strain,levels =c(
     "MSM", "MOLF",   
  "PWD", "SKIVE",  "KAZ",
  "WSB","G","LEW","PERC",
                     "CAST",
                         "SPRET","SPIC" ), order=T )
```


```{r Fig1.top, echo=FALSE, warning=FALSE, message=FALSE}
#Mouse.Table_HQ
#change the segements and text!!

Fig1_v1_strain <- ggplot(strain.table.HQ_Fig, aes(y=mean_co, x=strain, color=strain, shape=sex))+ 
  ylab("Average MLH1 Foci per Cell")+

  geom_point(aes(shape = sex), size=3, position=position_dodge(.5))+

    geom_errorbar(aes(ymin=mean_co-(se*2), ymax=mean_co+(se*2) ), width=.3, position = position_dodge(.5), size=1)+
  
  ylim(c(18, 35))+
 scale_color_manual(values=colors_of_strains)+ 

    theme(legend.position="none",
  panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
  panel.background = element_blank(), axis.line=element_line(colour = "black"),
  axis.title.y=element_blank(),
  axis.title.x=element_blank(),axis.text.x=element_blank(),axis.ticks.x=element_blank())+
#remove y labfor male 
 
   #mol
  annotate("segment", x=.75, xend=2.3, y=19.5, yend=19.5, colour="black", size=1)+
  
  #Musc,
  annotate("segment", x= 2.75, xend=5.2, y=30, yend=30, colour="black", size=1)+
  
  #dom
  annotate("segment", x=5.8, xend=8.5, y=19.5, yend=19.5, colour="black", size=1)+
  
  #cast
  annotate("segment", x=8.8, xend= 9.3, y=30, yend=30, colour="black", size=1)+

    #spret
  annotate("segment",x=9.5, xend=10.2, y=19.5, yend=19.5, colour="black", size=1)+
  
  #spic
  annotate("segment",x=10.5, xend=11.2, y=30, yend=30, colour="black", size=1)+
  
  #label
    annotate("text", x = c(1.6,3.9,6.3,8,10, 10.3), #list of all x points 
           y = c(18.5, 31, 18.5, 31, 18.5, 31),
           
   label = c("M. m. molossinus","M. m. musculus", "M. m. domesticus", "M. m. castaneus", "M. spretus", "M. spicilegus"),
           fontface = 'italic', hjust = 0.5, size=3.5)

```

```{r female.show.plots, echo=FALSE, fig.width = 4, fig.height=2.5, warning=FALSE, message=FALSE}

#MLH1_data  %>% filter(quality < 5)
#mouse avs

female_mouse.av <- Mouse.Table_HQ.OG %>% filter(sex == "female") %>% filter(species == "M.musculus")

female_mouse.av99 <- Mouse.Table_HQ %>% filter(sex == "female") %>% filter(species == "M.musculus")

female_mouse.av <- add_sex(female_mouse.av)
female_mouse.av <- add_strain(female_mouse.av)
female_mouse.av <- add_subsp(female_mouse.av)
female_mouse.av <- add_species(female_mouse.av)

female_mouse.av$strain <- factor(female_mouse.av$strain, ordered = FALSE, levels =c( "MSM", "MOLF",
  "PWD", "SKIVE", "KAZ",
  
  "WSB", "G", "LEW",
                                                                         "CAST", 
                                                                   "SPRET", "SPIC") )

female.specific.plot <- ggplot(female_mouse.av, aes(x = as.factor(strain), y = mean_co, color= strain))+ geom_boxplot()+  geom_jitter(aes( alpha=0.5), width = .2 )+ ggtitle("")+scale_color_manual(values=colors_of_strains)+ylim(c(17,34))+
  theme(legend.position="none",
      
        axis.title.y =element_blank(), 
      axis.text.x = element_blank(), axis.title.x = element_blank(), axis.ticks.x = element_blank(),
  panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
  panel.background = element_blank(), axis.line=element_line(colour = "black"))+geom_hline(yintercept = 20)

#+theme(axis.text.x = element_text(angle = 45, hjust = 1))
```

```{r male.show.plots, echo=FALSE, fig.width = 4, fig.height=2.5, warning=FALSE, message=FALSE}
#try customize the plot size

male_mouse.avs_for.analysis$strain <- factor(male_mouse.avs_for.analysis$strain, ordered = FALSE, levels =c( "MSM", "MOLF",
  "PWD", "SKIVE", "KAZ","TOM","AST","CZECH",
  
  "WSB", "G", "LEW","PERC",
                                                                         "CAST", "HMI",
                                                                   "SPRET", "SPIC") )

male.plot <- ggplot( male_mouse.avs_for.analysis, aes(x = as.factor(strain), y = mean_co, color= strain))+geom_jitter( alpha=0.5, width = .2) +geom_boxplot()+ ggtitle("")+scale_color_manual(values=colors_of_strains)+ylim(c(17,34))+
 
   theme(legend.position="none", axis.text.x=element_blank(), axis.title.x = element_blank(), axis.ticks.x = element_blank(),  axis.title.y = element_blank(), 
  panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
  panel.background = element_blank(), axis.line=element_line(colour = "black"))+geom_hline(yintercept = 19)

#horizontal line = 20

```

```{r MLH1.cell.examples, echo=FALSE, message=FALSE, warning=FALSE}

male.MLH1.cell.ex <- readPNG('~/MLH1repo/doc/figureFiles/MLH1.male.cell_ex.png')

female.MLH1.cell.ex <- readPNG('~/MLH1repo/doc/figureFiles/MLH1.female_cell.ex.png')

empty.DF <- data.frame(col1 = c(1,3), col2 = c(1,3) )

#Cell GGPLOT

maleMLH1.cell.obj <- ggplot(empty.DF, aes(x = col1, y = col2)) +
  background_image(male.MLH1.cell.ex) + theme(axis.title.x=element_blank(),
     axis.text.x=element_blank(),axis.ticks.x=element_blank(),axis.title.y=element_blank(),                                  axis.text.y=element_blank(),axis.ticks.y=element_blank() )


femaleMLH1.cell.obj <- ggplot(empty.DF, aes(x = col1, y = col2)) +
  background_image(female.MLH1.cell.ex) + theme(axis.title.x=element_blank(),
     axis.text.x=element_blank(),axis.ticks.x=element_blank(),axis.title.y=element_blank(),                                  axis.text.y=element_blank(),axis.ticks.y=element_blank() )

```

```{r Figure1.Main.plot.three.plots, echo=FALSE, warning=FALSE, fig.width=7, fig.cap='*Figure 1 Mean MLH1 count distributions by strain.* A) Strain averages of MLH1 counts per cell. B) Female specific MLH1 count distributions. C)	Male specific MLH1 counts per cell.', message=FALSE}


MLH1.Figure1  <- ggdraw() +
  draw_plot(Fig1_v1_strain, x = .02, y = .55, width=1, height=.45)+
  
  draw_plot(female.specific.plot, x=0, y=0, width=.5,height= .55)+
  draw_plot(femaleMLH1.cell.obj, x=.35, y=.35, width=.18, height= .18)+
  
  draw_plot(male.plot, x=.5, y = 0,  width = .5, height = .55)+
  draw_plot(maleMLH1.cell.obj, x=.85, y = .35,  width = .19, height=.19)+
 
  draw_plot_label(label = c("A", "B", "C"), size = 14,
                  x = c(0, .01, .51), y = c(.99, .55, 0.55) )

MLH1.Figure1
#add the cell images

#ggsave("~/MLH1repo/doc/Figure1.png")
```


## B Genome-wide recombination rate evolves differently in females and males


```{r MLH1.summ.stats, echo=FALSE}

#strain.table.HQ_Fig.og  all strains
#strain.table.HQ_Fig  extra strains removed

main.table <- strain.table.HQ_Fig.og  %>% filter(strain != "F1")

main.table <- add_subsp(main.table)
main.table <- add_species(main.table)

```


```{r female.strain.table, echo=FALSE}
female_mouse.avs_for.analysis <- female_mouse.avs_for.analysis %>% filter(strain != "CAST")
#female strain table
#female_mouse.avs_for.analysis
F.strain.table <- ddply(female_mouse.avs_for.analysis, c("strain"), summarise,
                       
                        Nmice = length(unique(mouse)),
                        
                        total.cells  = sum(Ncells, na.rm = TRUE),
                        
                        mouse.mean_co = mean(mean_co, na.rm=TRUE),
                       
                        mean.mouse_cV = mean(cV, na.rm = TRUE),
                        
                        mean.mouse_var = mean(var, na.rm = TRUE ), 
                        
                        mean.mouse_sd   = mean(sd, na.rm = TRUE ),
                       
                        #Standard error (mice vs cells)
                        mean.mouse_se  = mean(se, na.rm=TRUE)
    )
#F.strain.table

#F.strain.table

# I don't know why most of the table is blank
```

```{r mouse.var.plots.female, echo=FALSE, fig.width = 4, fig.height=2.5, include=FALSE, eval=FALSE}

#variance plots
musc.female.VAR.plot <- ggplot( (female_mouse.avs_for.analysis %>% filter(sex== "female") %>% filter(subsp == "Musc") ) , aes(color= strain))+geom_jitter(aes(x = as.factor(strain), y =var, color= strain, alpha=0.5), width = .2 )+ ggtitle("Musc female of within mouse MLH1 distributions")+scale_color_manual(values=colors_of_strains)
musc.female.VAR.plot

Mol.female.VAR.plot <- ggplot( (female_mouse.avs_for.analysis %>% filter(subsp == "Mol") ), aes(color= strain))+geom_jitter(aes(x = as.factor(strain), y = var, color= strain, alpha=0.5), width = .2 )+ ggtitle("Mol female of within mouse MLH1 distributions")+scale_color_manual(values=colors_of_strains)
Mol.female.VAR.plot

Dom.female.VAR.plot <- ggplot(Dom.female_mouse.av, aes(color= strain))+geom_jitter(aes(x = as.factor(strain), y = var, color= strain), width = .2 )+ ggtitle("Dom female\nwithin mouse  MLH1 distributions") + scale_color_manual(values=colors_of_strains)
Dom.female.VAR.plot

```

```{r male.strain.table, echo=FALSE}


M.strain.table <- ddply(male_mouse.avs_for.analysis, c("strain"), summarise,
                        Nmice =length(unique(mouse)),
                        
                        totNcells  = sum(Ncells, na.rm=TRUE),
                        
                        mean.mouse_mean_co = mean(mean_co, na.rm=TRUE),
                       
                        mean.mouse_cV = mean(cV, na.rm = TRUE),
                        
                        mean.mouse_var = mean(var, na.rm = TRUE ), 
                        
                        mean.mouse_sd   = mean(sd, na.rm=TRUE),
                       
                        #Standard error (mice . cells)
                        mean.mouse_se  = mean(se, na.rm=TRUE)
    )
#M.strain.table
#same issue as the female table .. ugh
```

```{r mouse.var.plots.male, echo=FALSE, fig.width = 4, fig.height=2.5, include=FALSE, include=FALSE, eval=FALSE}

#variance plots
musc.male.VAR.plot <- ggplot(Musc.male_mouse.av, aes(color= strain))+geom_jitter(aes(x = as.factor(strain), y =var, color= strain, alpha=0.5), width = .2 )+ ggtitle("VAR Musc male of within mouse MLH1 distributions")+scale_color_manual(values=colors_of_strains)


Mol.male.VAR.plot <- ggplot(Mol.male_mouse.av, aes(color= strain))+geom_jitter(aes(x = as.factor(strain), y = var, color= strain, alpha=0.5), width = .2 )+ ggtitle("VAR Mol male of within mouse MLH1 distributions")+scale_color_manual(values=colors_of_strains)


Dom.male.VAR.plot <- ggplot(Dom.male_mouse.av, aes(color= strain))+geom_jitter(aes(x = as.factor(strain), y = var, color= strain, alpha=0.5), width = .2 )+ ggtitle("VAR Dom male ofwithin mouse  MLH1 distributions")+scale_color_manual(values=colors_of_strains)

```


```{r MAIN.TABLE1, echo=FALSE, eval=TRUE}
#write the full table to a supplemental table
#show the head (or shortened)
#kable(table(AP_mouse_table.HQ$category), col.names = c("Category", "mice"), label = "Number of mice by category" )
#kable(strain.table.HQ_Fig, "latex")

df.strain.table <- strain.table.HQ_Fig
#kable(df.strain.table)
#pander(df.strain.table)

#kable(strain.table.HQ_Fig, col.names = c("Subspecies", "Strain","Sex", "Number of Mice", "Number of Cells", "Mean MLH1 count per cell", "cV", "Var", "sd", "se"), caption = "Strain Level Summary Statistics" )  %>%
#  kable_styling() %>%
#  scroll_box(width = "800px", height = "200px")

#kable(table(MLH1_data$category), col.names = c("Category","Cells"), label = "Number of Cells by category" )

#write excel
#write.xlsx2(strain.table.HQ_Fig, "Main_MLH1_table.xlsx", sheetName = "Sheet1", 
#  col.names = TRUE, row.names = FALSE, append = FALSE)

```



We used counts of MLH1 foci per cell to estimate genome-wide recombination rates in `r length(unique(  (main.table  %>% filter(species == "M.musculus") )$strain) )` -- 14 wild-derived inbred strains sampled from three subspecies of house mice ( _M. m. domesticus_, _M. m. musculus_ and _M. m. molossinus_ ) and two additional species of Mus ( _M. spretus_ and _M. spicilegus_ ). Mean MLH1 foci counts for `r length(unique(  (main.table  %>% filter(species == "M.musculus") )$mouse) )` -- 161 mice were quantified from an average of `r mean( (Mouse.Table_HQ %>% filter(sex == "male") )$Ncells, na.rm = T)` -- 22.0 spermatocytes per male (for a total of `r mean( (Mouse.Table_HQ %>% filter(sex == "male") )$Ncells, na.rm = T)` -- 1,867 spermatocytes) and -- 18.5 oocytes per female (for a total of `r formatC(sum( (main.table %>% filter(species == "M.musculus") %>% filter(sex == "female"))$Ncells), format="d", big.mark=",")` -- 1,409 oocytes).
Graphical comparisons reveal sex-specific dynamics to the evolution of genome-wide recombination rate (Figure 1A). First, MLH1 focus counts differ between females and males in most strains. Second, the relationship between MLH1 foci in the sexes varies among strains. Although most strains show more MLH1 foci in females, three strains - _musculus^PWD^_, _musculus^SKIVE^_, and _molossinus^MSM^_ - exhibit higher MLH1 focus counts in males. In females, numbers of MLH1 foci are evenly distributed around the sex-wide mean of approximately 25 (Figure 1B). In stark contrast, males largely separate into two groups of strains with high numbers (near 30) and low numbers (near 23 MLH1 foci per cell) of foci (Figure 1C). 


To further situate variation in recombination rate within an evolutionary framework, we fit a series of statistical models including subspecies, strain, and sex, to mean MLH1 focus counts from `r length(DF.HetC.MixedModel.HQ$mean_co)` -- 187 mice. We began with a full mixed model (M1; see Materials and Methods), showing that strain (random effect p < 10^{-6} -- p < `r 1/10000`), sex (p = 1.5510^{-8} -- `r MLH1.MMlmer_results$sex.results`), subspecies (p=1.7210^{-4} -- `r MLH1.MMlmer_results$sub.sp`), and subspecies*sex (p = 3.110^{-5} -- `r MLH1.MMlmer_results$interaction.pval` ) each significantly affect MLH1 focus count. After a model including all factors as fixed effects (M2) revealed only weak contributions of subspecies, we focused on additional models designed to illuminate the role of strain and sex. A model with only these two variables (M3) identified two strains with particularly strong effects on MLH1 focus count:  _musculus^MSM^_ (p = 3.9910^{-6} -- `r summ.MLH1.M3.strain.fixed$coefficients[,4][6]`) and _domesticus^G^_  (p = 1.0410^{-6} -- `r summ.MLH1.M3.strain.fixed$coefficients[,4][3]` ). In addition, two strains exhibit strain-by-sex interactions: _molossinus^MSM^_  (p = 1.2610^{-4} -- `r summ.MLH1.M3.strain.fixed$coefficients[,4][13]` ) and _musculus^PWD^_ (p = 3.8610^{-4} -- `r summ.MLH1.M3.strain.fixed$coefficients[,4][12]` ).

We next fit models separately for  `r length(male_mouse.avs_for.analysis$mouse)` -- 192 males and `r length(female_mouse.avs_for.analysis$mouse)` -- 144 females (M4). In the male dataset, three strains affect MLH1 focus count (as observed in Figure 1C): _musculus^PWD^_ (p = 6.3110^{-8} -- `r sum_MLH1.male.glm.M4$coefficients[,4][8]` ; effect = `r sum_MLH1.male.glm.M4$coefficients[8]` -- 6.1 foci), and _musculus^SKIVE^_ (p = 0--`r sum_MLH1.male.glm.M4$coefficients[,4][10]` ; effect = 3.8-- `r sum_MLH1.male.glm.M4$coefficients[10]` ), and _molossinus^MSM^_  (p=2.4210^{-12} -- `r sum_MLH1.male.glm.M4$coefficients[,4][9]`; effect = `r sum_MLH1.male.glm.M4$coefficients[9]` -- 7.0).

These three strains point to substantial evolution in the genome-wide recombination rate in spermatocytes; we subsequently refer to them as “high-recombination” strains. Analysis of the female dataset identifies four strains with effects on recombination rate: _domesticus^G^_ (p = 2.510^{-6} -- `r sum_MLH1.female.spc.glm.M2$coefficients[2,][4]`),  *why is mol so different*  _molossinus^MSM^_  (p = 6.2410^{-6} -- `r sum_MLH1.female.spc.glm.M2$coefficients[5,][4]`), _domesticus^LEW^_ (p = 0.01 -- `r sum_MLH1.female.spc.glm.M2$coefficients[3,][4]` ), and _musculus^PWD^_ (p= 0.02 -- `r sum_MLH1.female.spc.glm.M2$coefficients[4,][4]` ). Strain effect sizes in females are modest in magnitude (ranging from 1 to 4 foci) compared to those in males. 
Together, these results demonstrate that the genome-wide recombination rate evolves in a highly sex-specific manner. 


## B Synaptonemal complexes are longer in females


*new* -- Our survey of recombination rate across house mice provides an opportunity to determine whether sex differences in chromatin compaction (SC length) are reversed when heterochiasmy is reversed. In all strains except _musculus^SKIVE^_, females have longer SCs than males, whether SC length was estimated as the total length across bivalents or as the length of short bivalents (t-test; p < 0.05 -- ). In the dataset of short bivalents (to which the female X bivalent does not contribute), female to male ratios of mouse mean SC length range from 1.15 -- ( _musculus^MSM^_) to 1.49 -- ( _domesticus^WSB^_) across strains. That females have longer SCs is further supported by models that include covariates, which identify sex as the most consistently significant effect (p < 0.05 -- ). The existence of some subspecies and strain effects (p < 0.05 -- ) further indicates that SC length has evolved among strains and among subspecies. In summary, two approaches for measuring SC length demonstrate that females have longer SCs (chromosome axes), even in strains in which males have more MLH1 foci per cell. This pattern further suggests that in high-recombination strains, spermatocytes have less space in which to position additional crossovers than do oocytes.


*old? -- * Counting MLH1 foci in multiple oocytes for each female and multiple spermatocytes for each male allowed us to examine determinants of the within-mouse variance in recombination rate. To do this, we considered the same models as above, but replaced mean MLH1 foci count with within-mouse variance in MLH1 foci count per cell as the dependent variable. Sex is the only variable that significantly affects recombination rate in both the mixed model (M1) (p < `r 1/1000000`) and general linear models (M2) (p = `r MLH1var_M2_sum$coefficients[,4][9]`) and M3 (p = `r round(glmMLH1Var_M3_sum$coefficients[,4][2], 5)`). 

In general, females have almost twice as much variance in MLH1 foci per cell compared to males (Figure 1). Since estimates of within-mouse variance may be more susceptible to technical error from the staining protocol, we repeated the analyses using a subset of cells with higher quality scores (quality score 1 and 2, see Materials and Methods).  The results are similar: sex is the strongest effect (M1 p < `r 1/1000000`; M2 p = `r  round(sum.MLH1var_M2_Q12$coefficients[,4][4], 5)`;  also M3 p = `r MLH1var_M3_Q12_sum$coefficients[,4][2]`). When both quality-curated and full datasets are considered, strain does not significantly and consistently affect variance in MLH1 foci count per cell in either sex. These results suggest that within-mouse variance in recombination rate evolves independently of mean recombination rate.

<LmerMLH1_M1_Q12_VAR_results> <Bret's comment for below, are there differences between the curated and the full dataset? -- Q1 and full data set>



## B Females and males differ in crossover positions and crossover interference


# triangles

```{r double.IFD.triangle.plots, echo=FALSE}

#good function, saves to file, can't reproduce in rmd ..width=650, height=500
#big loop for automating the IFD triangles
only.male.strains <- c("F1", "PERC", "CZECH", "AST", "TOM", "HMI", "SPRET", "CAROLI", "other")
auto_curate_DF <- Curated_BivData[ ! Curated_BivData$strain %in% only.male.strains, ]

df_strain_list <- split(auto_curate_DF, (auto_curate_DF$strain))
df_strain_list <- df_strain_list[ -c(4,10,11,12,13,14,15,16,17,18,19) ]
#make a new empty df. create a ddply thing with mean calculations (this is like a list for matching up later)

#this works
top.triangle <- function(my.y, my.x, my.data) {
  plot(my.y ~ my.x, data= my.data, axes=FALSE, ylab='', xlab="", 
       main='', xlim=c(0, 1), ylim=c(0, 1), xaxt="n", yaxt="n", asp=1,
       col =  colors_of_strains[current.strain])
  
  mtext(current.strain, 3, 5, font=2, cex=1.3, adj=.95)
  mtext("Position.2.Female", 2, .75)
  mtext("Position.1.Female", 3, 2)
  
  axis(side=2, las=1, pos=0)
  axis(side=3, las=1, pos=1)
  lines(0:1, 0:1)
}

bottom.triangle <- function(my.y, my.x, my.data){
  points(my.x ~ my.y, data=my.data, xpd=TRUE, 
          col =  colors_of_strains[current.strain])
  
  mtext("Position.2.Male", 1, 1.5, at=mean(par()$usr[1:2]) + x.dist)
  mtext("Position.1.Male", 4, 3, padj=par()$usr[1] +  7)
  x.at <- axisTicks(par()$usr[1:2], 0) + x.dist
  axis(side=1, las=1, pos=0, at=x.at, 
       labels=F, xpd=TRUE)
  mtext(seq(0, 1, .2), 1, 0, at=x.at)
  axis(4, las=1, pos=1 + x.dist)
  lines(0:1 + x.dist, 0:1, xpd=TRUE)
}

x.dist <- .1
RP <- list()

for(i in 1:(length(df_strain_list))){
  #print(i)
  current.strain <- as.character(df_strain_list[[i]]$strain[1])
  
  #declare file for saving png
  #png(paste0("~./MLH1repo/doc/figureFiles/IFD.triangle_", current.strain, ".png"),  width=650, height=500, res=100)
  
  plot.new()
  
  my.data = (Curated_BivData %>% filter(strain == current.strain) %>% filter(hand.foci.count == 2) )
    
  my.x.top = (Curated_BivData %>% filter(strain == current.strain) %>% filter(sex == "female") %>% filter(hand.foci.count == 2) )$PER_Foci_1 
  
  my.y.top = (Curated_BivData %>% filter(strain == current.strain) %>% filter(sex == "female") %>% filter(hand.foci.count == 2) )$PER_Foci_2
  
  my.x.bottom = (Curated_BivData %>% filter(strain == current.strain) %>% filter(sex == "male") %>% filter(hand.foci.count == 2) )$PER_Foci_1  
  
  my.y.bottom = (Curated_BivData %>% filter(strain == current.strain) %>% filter(sex == "male") %>% filter(hand.foci.count == 2) )$PER_Foci_2
  
  #adjusting this mar lowers the margins
  op <- par(mar=c(.6,.6, .6, .6) + 0.1, oma=c(2, 0, 0, 2))
  
  top.triangle(my.y.top, my.x.top, my.data)
  
  bottom.triangle(my.y.bottom+x.dist, my.x.bottom, my.data) #x.dist

  par(op)
  RP[[i]] <- recordPlot()
  dev.off()
}
```


```{r Q1.IFD_load.triangles, echo=FALSE, message=FALSE, warning=FALSE, fig.cap = "Distribution of normalized foci positions (IFDs) for double crossover bivalents across sexes. Female and male patterns are displayed in the top and bottom triangle respectively. Each point reflects a single double crossover bivalent, with the normalized positions of foci reflected on each axis."}

plot1.WSB <- readPNG('~/MLH1repo/doc/figureFiles/IFD.triangle_WSB.png')
plot2.LEW <- readPNG('~/MLH1repo/doc/figureFiles/IFD.triangle_LEW.png')
plot3.G <- readPNG('~/MLH1repo/doc/figureFiles/IFD.triangle_G.png')
plot4.KAZ <- readPNG('~/MLH1repo/doc/figureFiles/IFD.triangle_KAZ.png')
plot5.SKIVE <- readPNG('~/MLH1repo/doc/figureFiles/IFD.triangle_SKIVE.png')
plot6.PWD <- readPNG('~/MLH1repo/doc/figureFiles/IFD.triangle_PWD.png')
plot7.MSM <- readPNG('~/MLH1repo/doc/figureFiles/IFD.triangle_MSM.png')


# grid.arrange(rasterGrob(plot1.WSB), rasterGrob(plot2.LEW), rasterGrob(plot3.G),
#             rasterGrob(plot4.KAZ), rasterGrob(plot5.SKIVE), rasterGrob(plot6.PWD),
#             rasterGrob(plot7.MSM), 
#             ncol=3)

#load all the files / arrange the plots from above 
#WSB.triangle.png <- readPNG("~./MLH1repo/doc/figureFiles/test.triangle_WSB.png")
```

```{r Q1.F1.pos.LONG.SHORT.BIV_scatter, echo=FALSE}
#real.long.bivData
Q1.short.biv_scatr_box   <- ggplot( (real.short.bivData %>% filter(hand.foci.count == 1) ), aes(x=sex, y=PER_Foci_1, color=strain) )+geom_jitter( )+ geom_boxplot(aes(y= PER_Foci_1 ),alpha=.3 ) + scale_color_manual(values=colors_of_strains)+facet_wrap(~subsp, scales = "free")+ylim(c(0,1.1))+ggtitle("Mouse av. \nNormalized 1CO Foci pos \nShort Biv")


Q1.long.biv_scatr_box   <- ggplot((real.long.bivData %>% filter(hand.foci.count == 1) ), aes(x=sex, y=PER_Foci_1, color=strain) )+geom_jitter( )+ geom_boxplot(aes(y= PER_Foci_1 ),alpha=.3 ) + scale_color_manual(values=colors_of_strains)+facet_wrap(~subsp, scales = "free")+ylim(c(0,1.1))+ggtitle("Mouse av. \nNormalized 1CO Foci pos \nLong Biv")

```

```{r Q1.mouse.av.F1.pos.LONG.BIV_scatter, echo=FALSE, eval=FALSE}
#REAL.short_mouse_table
#REAL.long_mouse_table
#mouse.av.long.biv.table

Q1.av.long.biv_scatr_box   <- ggplot(REAL.long_mouse_table  , aes(x=sex, y=mean_F1pos_PER, color=strain) )+geom_jitter( )+ geom_boxplot(aes(y= mean_F1pos_PER ),alpha=.3 ) + scale_color_manual(values=colors_of_strains)+facet_wrap(~subsp, scales = "free")+ylim(c(0,1.1))+ggtitle("Mouse av. \nNormalized 1CO Foci pos \nLong Biv")

Q1.av.short.biv_scatr_box   <- ggplot(REAL.short_mouse_table, aes(x=sex, y=mean_F1pos_PER, color=strain) )+geom_jitter( )+ geom_boxplot(aes(y= mean_F1pos_PER ),alpha=.3 ) + scale_color_manual(values=colors_of_strains)+facet_wrap(~subsp, scales = "free")+ylim(c(0,1.1))+ggtitle("Mouse av. \nNormalized 1CO Foci pos \nShort Biv")
```


*new* -- We used normalized positions of MLH1 foci along bivalents with a single focus to compare crossover location while controlling for differences in SC length. In all strains, MLH1 foci tend to be closer to the telomere in males (mean normalized position in males: 0.68 -- `r mean( (mouse.level.BivData_table %>% filter(sex == "male") %>% filter(species == "M.musculus") )$norm.first.foci.pos, na.rm = TRUE )`; mean normalized position in females: `r mean( (mouse.level.BivData_table %>% filter(sex == "female") %>% filter(species == "M.musculus") )$norm.first.foci.pos, na.rm = TRUE )` -- 0.56; t-test; p = 2.9210^{-22} -- `r Q1_1CO.pos_t.test.FULL$p.value` ). Sex is also the strongest determinant of MLH1 focus position in the models we tested (M1: p = 1.2610^{-25} -- `r lmer.Q1.nrm1CO_M1_results$sex.results` ; M2: p = 1.3310^{-7} -- `r summry.Q1.glm_F1pos_M2$coefficients[,4][8]`; M3: p = 1.3310^{-7} -- `r summry.Q1.glm_1CO.pos_M3$coefficients[,4][8]`).


Males have longer normalized mean inter-focal distances (IFD~norm~) than females in seven out of eight strains (t-tests; p < 1.4910^{-12} --), with only _musculus^KAZ^_ showing no difference (t-test; p = 0.33 -- `r Q1.IFD.raw_FULL_t.test$p.value` ). Examination of IFD~norm~ distributions indicates that female values are centered at approximately 50% and show a slight enrichment of low (<25%) values, whereas males are enriched for higher values. Analysis of models treating IFD~norm~ as the dependent variable supports the inference of stronger interference in males, with sex being the most significant variable (M1: p = 6.7410^{-14} --`r Q1.IFD.PER_M1_lmer.results$sex.results` ; M2: p = 0.01 -- `r summry.glm.Q1_IFD.PER.M2$coefficients[,4][4]`, M3: p = 0.01 -- `r summry.glm.Q1_IFD.PER.M3$coefficients[,4][2]`). In contrast, there is no clear signal of sex differences in raw mean inter-focal distances (IFD~raw~) across the full set of strains, whether they are considered separately or together. Visualization of normalized MLH1 foci positions on bivalents with two crossovers (Figure 4C) further suggests that interference distances vary more in females than in males and that males display a stronger telomeric bias in the placement of the distal crossover. In summary, controlling for differences in SC length (chromatin compaction) using IFDnorm indicates that interference is stronger in males, whereas consideration of IFD~raw~ shows that the sexes exhibit a similar level of interference on the physical scale. 


## B Evolution of genome-wide recombination rate is dispersed across bivalents, associated with double-strand break number, and connected to crossover interference

turn off all these DMC1 tables -- to makes sure the right DFs are loaded in from the setup scripts.


```{r DMC1.boxplot.plot, echo=FALSE, warning=FALSE}

#change order

DMC1.DF$strain<- factor(DMC1.DF$strain, levels =c(
                      "WSB","G", "KAZ","PWD","MSM"), order=T )


DMC1.boxplot <- ggplot(aes(x=strain, y=total.count, color=stage), data=DMC1.DF[DMC1.DF$stage != "P",])+
  geom_boxplot(alpha=.5) +geom_jitter() +
  
    labs(y = "DMC1 count per Cell")+
  
  scale_x_discrete( labels = c( expression( italic( domesticus^WSB) ), expression(  italic( domesticus^G) ), expression( italic( musculus^KAZ) ),
 
     expression( italic(musculus^PWD)), expression( italic( molossinus^MSM) ) ) ) +

  scale_color_discrete(name = "", labels = c("Early Zygotene", "Late Zygotene"))+scale_color_manual(values=c("#E69F00", "#0072B2"))+
                        
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),legend.position="top",
        panel.background = element_blank(), axis.line = element_line(colour = "black"), axis.title.x = element_blank())

#"#E69F00" orange
#"#0072B2" darker blue

```


```{r Main.plot.DMC1, echo=FALSE, fig.width=6.5, fig.height=5, fig.cap="*Figure 2 male DSB estimates* A) Example early zygotene spermatocyte spread, B) Example late zygotene spermatocyte spread. Green foci show DMC1. C)	Distribution of DMC1 counts per cell  by strains.", message=F, warning=F}

png.DMC1.early.cell <- readPNG('~/MLH1repo/doc/figureFiles/cell.examples/DMC1.early/good.early.cell.png')
png.DMC1.late.cell <- readPNG('~/MLH1repo/doc/figureFiles/cell.examples/DMC1.late/good.late.cell.png')

empty.DF <- data.frame(col1 = c(1,3), col2 = c(1,3) )

#EMPTY GGPLOT
Early.cell.obj <- ggplot(empty.DF, aes(x = col1, y = col2)) +
  background_image(png.DMC1.early.cell) + theme(axis.title.x=element_blank(),
                         axis.text.x=element_blank(),axis.ticks.x=element_blank(),axis.title.y=element_blank(),
                                  axis.text.y=element_blank(),axis.ticks.y=element_blank() )

Late.cell.obj <- ggplot(empty.DF, aes(x = col1, y = col2)) +
  background_image(png.DMC1.late.cell) + theme(axis.title.x=element_blank(),
                         axis.text.x=element_blank(),axis.ticks.x=element_blank(),axis.title.y=element_blank(),
                                  axis.text.y=element_blank(),axis.ticks.y=element_blank() )

#cowplot

DMC1.Figure2  <- ggdraw() +
  draw_plot(Early.cell.obj, x=0, y=.5, width=.4,height= .45)+
  draw_plot(Late.cell.obj, x=.5, y=.5,  width = .4, height = .45)+
  draw_plot(DMC1.boxplot, x=0, y = 0,  width =1, height=.55)+
 
  draw_plot_label(label = c("A", "B", "C"), size = 14,
                  x = c(0, .49, 0), y = c(1,1, 0.55) )+
#WSB
  draw_line(
    x = c(0.12, .23),
    y = c( 0.01, .01),
    color = colors_of_strains['WSB'], size=1
  ) +
  #G
  draw_line(
    x = c( 0.31, .42),
    y = c(0.01, 0.01),
    color = colors_of_strains['G'],size=1
  ) +
  #KAZ
  draw_line(
    x = c(0.49, .57),
    y = c(0.01, 0.01),
    color = colors_of_strains['KAZ'], size=1
  ) +
  #PWD
  draw_line(
    x = c(0.64, .73),
    y = c(0.01, .01),
    color = colors_of_strains['PWD'], size=1
  ) +
  #MSM
  draw_line(
    x = c(0.80, .91),
    y = c(0.01, .01),
    color = colors_of_strains['MSM'], size=1
  )
DMC1.Figure2
#three.DMC1.plot <- { Early.cell.obj + Late.cell.obj  } / DMC1.boxplot + plot_layout(widths = 1)

#three.DMC1.plot + plot_annotation(
#  title = '',
#  subtitle = '') + plot_annotation(tag_levels = 'A')

#ggsave("~/MLH1repo/doc/Figure2.png")

```


```{r algorithm_performance_stats, echo=FALSE}

#number of mice, number of cells
#mean number of isolate bivs per cell,
#use these two metrics
#cell.level.BivData_table$boxs.IDd 
#cell.level.BivData_table$nboxes_passed

#calq the mean rate by category

cell.level.BivData_table$pass.rate <- cell.level.BivData_table$nboxes_passed / cell.level.BivData_table$boxs.IDd 

algorithm.stat.mouse <- ddply(cell.level.BivData_table, c("mouse"), summarise,
                        n_obs = length( unique(fileName) ),
                         mean_pass.rate = mean(pass.rate)
                         )

cell.level.BivData_table <- add_sex(cell.level.BivData_table)

algorithm.stat.category <- ddply(cell.level.BivData_table, c("sex", "strain"), summarise,
                        n_obs = length(unique(fileName)),
                        mean_pass.rate = mean(pass.rate)
                         )

#showing the table..
#data.frame(table( droplevels( Curated_BivData  %>%  filter(hand.foci.count  == 0 | hand.foci.count  == 1 |                                       hand.foci.count  == 2 | hand.foci.count  == 3 | hand.foci.count  == 4 ) )$category ) )
```


```{r Chrm.props.plot2, echo=FALSE, fig.width=8, fig.cap="*Figure 3* Chromosome Class Proportions. proportions calculated from pooled single bivalent data by strain."}

#barplots for proportions of chrm classes

#consider adding this the the setup block (could have done this to category level table)
mouse.level.BivData_table$per.0CO <- mouse.level.BivData_table$n0CO / mouse.level.BivData_table$tot.biv

mouse.level.BivData_table$per.1CO <- mouse.level.BivData_table$n1CO / mouse.level.BivData_table$tot.biv

mouse.level.BivData_table$per.2CO <- mouse.level.BivData_table$n2CO / mouse.level.BivData_table$tot.biv

mouse.level.BivData_table$per.3CO <- mouse.level.BivData_table$n3CO / mouse.level.BivData_table$tot.biv

mouse.level.BivData_table$per.4CO <- mouse.level.BivData_table$n4CO / mouse.level.BivData_table$tot.biv


MELT_Chrm.Proportion <- melt(mouse.level.BivData_table[,c(
        "mouse", "Nmice", "Ncells", "boxs.IDd", "nboxes_passed",  "nbiv",               "tot.biv", "mean_SC", "var.SC_length", "sd.SC_length", "se.SC_length",        "mean.SC_1CO", "mean.SC_2CO", "mean.SC_3CO", "mean.SC_4CO", "mean_1CO.pos",  "mean_1CO.pos.PER", "norm.first.foci.pos", "mean_IFD.2CO_ABS", "mean_IFD.2CO_PER", "mean_IFD_ABS", "mean_IFD_PER", "mean.siscoten", "mean.siscoten_1CO", "mean.siscoten_2CO", "mean.siscoten_3CO", "mean.telo.dist", "mean.cent.dist",     "mean.sis.co.ten", "sex", "strain", "subsp", "category", "Rec.group", "species",
                                                     
               #leave out the Foci
                "per.0CO", "per.1CO", "per.2CO", "per.3CO", "per.4CO")], 
               
  id=c("mouse", "Nmice", "Ncells", "boxs.IDd", "nboxes_passed",  "nbiv",               "tot.biv", "mean_SC", "var.SC_length", "sd.SC_length", "se.SC_length",        "mean.SC_1CO", "mean.SC_2CO", "mean.SC_3CO", "mean.SC_4CO", "mean_1CO.pos",  "mean_1CO.pos.PER", "norm.first.foci.pos", "mean_IFD.2CO_ABS", "mean_IFD.2CO_PER", "mean_IFD_ABS", "mean_IFD_PER", "mean.siscoten", "mean.siscoten_1CO", "mean.siscoten_2CO", "mean.siscoten_3CO", "mean.telo.dist", "mean.cent.dist",     "mean.sis.co.ten", "sex", "strain", "subsp", "category", "Rec.group", "species"))

MELT_Chrm.Proportion <- MELT_Chrm.Proportion[(!is.na(MELT_Chrm.Proportion$value)),]
MELT_Chrm.Proportion <- MELT_Chrm.Proportion[!(is.na(MELT_Chrm.Proportion$value) | MELT_Chrm.Proportion$value==0), ]

#no filename for add mouse, strain subsp
MELT_Chrm.Proportion <- add_category(MELT_Chrm.Proportion)
MELT_Chrm.Proportion <- add_sex(MELT_Chrm.Proportion)
MELT_Chrm.Proportion <- add_strain(MELT_Chrm.Proportion)
MELT_Chrm.Proportion <- add_species(MELT_Chrm.Proportion)

#then this DF has to be summarized into categories
chrm.class.category_MELT <- ddply(MELT_Chrm.Proportion, c("strain","sex", "variable"), summarise,
                             mean.percent.type = mean(value)
                             #sum is the wrong!
                             #used average percentage, might consider a different summary
)

#can't apply function use this
chrm.class.category_MELT$subsp <- ifelse(grepl("WSB", chrm.class.category_MELT$strain), "Dom", 
             ifelse(grepl("G", chrm.class.category_MELT$strain), "Dom",
                 ifelse(grepl("LEW", chrm.class.category_MELT$strain), "Dom", 
                ifelse(grepl("PERC", chrm.class.category_MELT$strain), "Dom",
                                                            
          ifelse(grepl("CAST", chrm.class.category_MELT$strain), "Cast",
              ifelse(grepl("HMI", chrm.class.category_MELT$strain), "Cast",
                                                                                 
                                       ifelse(grepl("MSM", chrm.class.category_MELT$strain), "Mol", #added Mol as distinct subsp class
                                      ifelse(grepl("MOLF", chrm.class.category_MELT$strain), "Mol",    
            ifelse(grepl("PWD", chrm.class.category_MELT$strain), "Musc", 
                     ifelse(grepl("CZECH", chrm.class.category_MELT$strain), "Musc", 
                 ifelse(grepl("AST", chrm.class.category_MELT$strain), "Musc",     
                      ifelse(grepl("TOM", chrm.class.category_MELT$strain), "Musc",  
                   ifelse(grepl("SKIVE", chrm.class.category_MELT$strain), "Musc",
                        ifelse(grepl("KAZ", chrm.class.category_MELT$strain), "Musc",
                                                                                                                                              
             ifelse(grepl("SPRET", chrm.class.category_MELT$strain), "Spretus",
               ifelse(grepl("SPIC", chrm.class.category_MELT$strain), "Spic",  "other"))))))))))))))))
  

chrm.class.category_MELT$subsp <- factor(chrm.class.category_MELT$subsp, ordered = TRUE,levels =c( "Dom",                    "Cast",
                                                                 "Musc",
                                                                 "Mol",
                                                                 "Spretus","Spic",
                                                                 "other") )
#ggplot

chrm.class.barplot <- ggplot( data = (chrm.class.category_MELT %>% filter(strain != "CZECH")),
                             aes(y = mean.percent.type, x = strain, fill=as.factor(variable)) ) + geom_col(position = "dodge")+ggtitle("")+ylim(c(0,1)) + facet_wrap(sex~subsp, scales = "free")+ theme(legend.position="top", legend.title = element_blank(), 
                                                                                                           axis.title.y = element_blank(), axis.text.x = element_text(size=9), axis.title.x = element_blank() )+scale_fill_discrete(name = "Chromosome Class", labels =c('0CO','1CO','2CO','3CO','4CO') )

kk <- addSmallLegend(chrm.class.barplot)

```

```{r Chrm.Prop.misc.calulations, echo=FALSE}

total.biv.all <- sum( MELT_Chrm.Proportion$n0CO, MELT_Chrm.Proportion$n1CO, MELT_Chrm.Proportion$n2CO, MELT_Chrm.Proportion$n3CO, MELT_Chrm.Proportion$n4CO)

```

```{r Q1_SC.biv.short_ttest, echo=FALSE}

#run t.tests for the mouse av. for reduced single bivalent datasets

#WSB
Q1_mus.av_shortSC_t.test.WSB <- t.test((Short.biv_mouse_table %>% filter(sex == "male") %>% filter(strain == "WSB") )$mean_SC,
                           (Short.biv_mouse_table %>% filter(sex == "female") %>% filter(strain == "WSB") )$mean_SC)

#G
Q1_mus.av_shortSC_t.test.G <- t.test((Short.biv_mouse_table %>% filter(sex == "male") %>% filter(strain == "G") )$mean_SC,
                           (Short.biv_mouse_table %>% filter(sex == "female") %>% filter(strain == "G") )$mean_SC)

#LEW
Q1_mus.av_shortSC_t.test.LEW <- t.test((Short.biv_mouse_table %>% filter(sex == "male") %>% filter(strain == "LEW") )$mean_SC,
                           (Short.biv_mouse_table %>% filter(sex == "female") %>% filter(strain == "LEW") )$mean_SC)

#PWD
Q1_mus.av_shortSC_t.test.PWD <- t.test((Short.biv_mouse_table %>% filter(sex == "male") %>% filter(strain == "PWD") )$mean_SC,
                           (Short.biv_mouse_table %>% filter(sex == "female") %>% filter(strain == "PWD") )$mean_SC)

#KAZ
Q1_mus.av_shortSC_t.test.KAZ <- t.test((Short.biv_mouse_table %>% filter(sex == "male") %>% filter(strain == "KAZ") )$mean_SC,
                           (Short.biv_mouse_table %>% filter(sex == "female") %>% filter(strain == "KAZ") )$mean_SC)


#SKIVE - this value used in the text
#only one SKVIE female, this is breaking the test
#Q1_mus.av_shortSC_t.test.SKIVE <- t.test(
  
#      (Short.biv_mouse_table %>% filter(sex == "male") %>% filter(strain == "SKIVE") )$mean_SC,
#      (Short.biv_mouse_table %>% filter(sex == "female") %>% filter(strain == "SKIVE") )$mean_SC
#      )


```


```{r Q1SC_short.biv_dif, echo=FALSE, eval=TRUE}

#calq the different in short.biv means across sexes
#Long.biv_mouse_table

#WSB
Q1SC.mus.av.short.biv_DIFF_WSB <- mean( (Short.biv_mouse_table %>% filter(sex == "female") %>% filter(strain == "WSB") )$mean_SC) / mean(  (Short.biv_mouse_table %>% filter(sex == "male") %>% filter(strain == "WSB") )$mean_SC )

#LEW
Q1SC.mus.av.short.biv_DIFF_LEW <- mean( (Short.biv_mouse_table %>% filter(sex == "female") %>% filter(strain == "LEW") )$mean_SC) / mean(  (Short.biv_mouse_table %>% filter(sex == "male") %>% filter(strain == "LEW") )$mean_SC )

#G
#Q1SC.mus.av.short.biv_DIFF_G <- mean( (Q1.real.short_mouse_table %>% filter(sex == "female") %>% filter(strain == "G") )$mean_SC) / mean(  (Q1.real.short_mouse_table %>% filter(sex == "male") %>% filter(strain == "G") )$mean_SC )

#PWD
Q1SC.mus.av.short.biv_DIFF_PWD <- mean( (Short.biv_mouse_table %>% filter(sex == "female") %>% filter(strain == "PWD") )$mean_SC) / mean(  (Short.biv_mouse_table %>% filter(sex == "male") %>% filter(strain == "PWD") )$mean_SC )

#KAZ
Q1SC.mus.av.short.biv_DIFF_KAZ <- mean( (Short.biv_mouse_table %>% filter(sex == "female") %>% filter(strain == "KAZ") )$mean_SC) / mean(  (Short.biv_mouse_table %>% filter(sex == "male") %>% filter(strain == "KAZ") )$mean_SC )

#SKIVE
Q1SC.mus.av.short.biv_DIFF_SKIVE <- mean( (Short.biv_mouse_table %>% filter(sex == "female") %>% filter(strain == "SKIVE") )$mean_SC) / mean(  (Short.biv_mouse_table %>% filter(sex == "male") %>% filter(strain == "SKIVE") )$mean_SC )

#MSM
Q1SC.mus.av.short.biv_DIFF_MSM <- mean( (Short.biv_mouse_table %>% filter(sex == "female") %>% filter(strain == "MSM") )$mean_SC) / mean(  (Short.biv_mouse_table %>% filter(sex == "male") %>% filter(strain == "MSM") )$mean_SC )


#this bivData ratio is the same as mouse average
Q1SC.BivData.short.biv_DIFF_MSM <- mean( (Short.biv_mouse_table %>% filter(sex == "female") %>% filter(strain == "MSM") )$chromosomeLength) / mean(  (Short.biv_mouse_table %>% filter(sex == "male") %>% filter(strain == "MSM") )$chromosomeLength )

```



# Fig 3


```{r Q1.SC.show.short.biv, echo=FALSE, eval=TRUE}

#Short.biv_mouse_table

Q1.SC.short.mouse_scatter_boxplot <- ggplot( Short.biv_mouse_table, aes(y = mean_SC, x = sex, color=strain)) + geom_jitter()+ geom_boxplot(alpha=.2) +scale_color_manual(values=colors_of_strains)+facet_wrap(~subsp, scales="free")+ylim(c(20,115) )+ggtitle("")+  
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"),  legend.position="none", axis.title.x = element_blank(), axis.text.x = element_text(size=7), axis.title.y = element_text(size = 10), axis.text.y = element_text(size=9) )+
  
  labs(y = "Short SC Length")

ggsave("~/MLH1repo/doc/Figure3_Q1.SC.short.mouse_scatter_boxplot.png")

```


```{r Q1.tot.SC.plot, echo=FALSE, warning=FALSE}
# fig.width = 4, fig.height=2.5

Q1.totSC.mouse.av.table <-  mouse.total.skel %>% filter(species == "M.musculus") %>% filter(subsp != "Cast")
  
Q1.totSC.mouse.av.table$strain <- factor(Q1.totSC.mouse.av.table$strain,levels =c(
   "WSB","G","LEW","PERC", 
  "PWD", "SKIVE",  "KAZ",
  "MSM", "MOLF"), order=T )

Q1.totSC.mouse.av.table <- with(Q1.totSC.mouse.av.table, Q1.totSC.mouse.av.table[order(sex, strain),])

#order the strains
Q1.totSC.mouse.av.table <- Q1.totSC.mouse.av.table %>%
  arrange(strain) %>%
  mutate(mouse = factor(mouse))

Q1.totSC.mouse.av.table <- with(Q1.totSC.mouse.av.table, Q1.totSC.mouse.av.table[order(sex, strain),])


Q1.totSC.mouse.av.table$subsp2 <- 
   ifelse(grepl("Dom", Q1.totSC.mouse.av.table$subsp), "M. m. domesticus", 
        ifelse(grepl("Musc",Q1.totSC.mouse.av.table$subsp), "M. m. musculus",
            ifelse(grepl("Mol",Q1.totSC.mouse.av.table$subsp), "M. m. molossinus",   
               "other")))


#change this plot to look like the short bivalent plot -- boxplots + mouse points

Tot.SC_mouse.av_ALL <- ggplot(data= Q1.totSC.mouse.av.table, aes(y = mean.skel, x=sex, color=strain))+
  scale_color_manual(values=colors_of_strains)+
    facet_wrap(~subsp, scales="free", ncol = 3) + 
  
  geom_jitter()+ geom_boxplot(alpha=.2)+ 
  #geom_hline(data =totSC.strain.av.table, aes(yintercept=mean.skel) )+
  
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black"),
    axis.title.x=element_blank(), axis.ticks.x=element_blank(),
axis.text.x = element_text(size=7), 
legend.position="none", axis.text.y = element_text(size=9), axis.title.y = element_text(size = 10) ) +
  
  labs(y = "Total SC Length")+ylim(c(1000,2600 ))+ggtitle("")

ggsave("~/MLH1repo/doc/Figure3_Tot.SC_mouse.av_ALL.png")

```


```{r Fig3.load.icons, echo=FALSE}
# load in the icons

all.chrm.yellow <- readPNG('~/MLH1repo/doc/figureFiles/all_yellow_binary.png')
single.chrm.yellow <- readPNG('~/MLH1repo/doc/figureFiles/single_chrm_yellow.png')

empty.DF <- data.frame(col1 = c(1,3), col2 = c(1,3) )

#EMPTY GGPLOT
all.chrm.icon <- ggplot(empty.DF, aes(x = col1, y = col2)) +
  background_image(all.chrm.yellow) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),
    axis.title.x=element_blank(),                       axis.text.x=element_blank(),axis.ticks.x=element_blank(),axis.title.y=element_blank(),
                                  axis.text.y=element_blank(),axis.ticks.y=element_blank() )


single.chrm.icon <- ggplot(empty.DF, aes(x = col1, y = col2)) +
  background_image(single.chrm.yellow) + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),
    axis.title.x=element_blank(),
   axis.text.x=element_blank(),axis.ticks.x=element_blank(),axis.title.y=element_blank(),
                                  axis.text.y=element_blank(),axis.ticks.y=element_blank() )

```




```{r Q1.Two.SC.plots, echo=FALSE, warning=FALSE, fig.width=8, fig.cap= 'caption'}

Q2.SC_dif.Figure3  <- ggdraw() +
  
  draw_plot(Q1.SC.short.mouse_scatter_boxplot, x=0, y=.38, width=.5, height= .6)+
  draw_plot(Tot.SC_mouse.av_ALL, x=.475, y=.38,  width = .5, height = .6)+

  draw_plot(single.chrm.icon, x=.44, y=.79,  width = .1, height = .1)+   
  draw_plot(all.chrm.icon, x=.88, y=.79,  width = .1, height = .1)+

  
  #drawing triangles in the corner
  #top right-triangle
  draw_line(
    x = c(.1, .2,  .1, .1 ), #keep the x values consistant
    y = c( .25, .25,  .15, .25),
    color = "blue", size=.75
  ) +
  
  #bottom right-triangle
  draw_line(
    x = c( .1, .2, .2, .1 ),
    y = c( .1,  .2, .1, .1),
    color = "black", size=.75
  ) +
  
  
  draw_plot_label(label = c("A", "B", "C"), size = 14,
                  x = c(0, .49, 0), y = c(1,1, 0.33) )

```


Make a figure for the diagram.  Import the straightened bivalents.. Load in the wsb double triangle. 


```{r Q1.total_SC.table, echo=FALSE}

totSC.strain.av.table <-  mouse.total.skel[,c("subsp","strain", "sex","nmice","ncells","mean.skel"),  drop=T]

#head of table
#see the DMC1 code

#kable(totSC.strain.av.table, row.names = F, caption = "caption") %>%
#  kable_styling() %>%
#  scroll_box(width = "800px", height = "200px")

#add variance
#write.xlsx(totSC.strain.av.table, file = "Results_Table.xlsx", sheetName = "Total.SC.results", append = TRUE, row.names = F)

```


```{r Q1.total.SC_t.test, echo=FALSE}

#run t-test for male female avergage for each strain

#ALL
Q1_tot.sc_t.test.all <- t.test((mouse.total.skel %>% filter(sex == "male") %>% filter(species == "M.musculus") )$mean.skel,
                           (mouse.total.skel %>% filter(sex == "female") %>% filter(species == "M.musculus")  )$mean.skel)


#dom
Q1_tot.sc_t.test.DOM <- t.test((mouse.total.skel %>% filter(sex == "male") %>% filter(subsp == "Dom") )$mean.skel,
                           (mouse.total.skel %>% filter(sex == "female") %>% filter(subsp == "Dom") )$mean.skel)

#WSB
Q1_tot.sc_t.test.WSB <- t.test((mouse.total.skel %>% filter(sex == "male") %>% filter(strain == "WSB") )$mean.skel,
                           (mouse.total.skel %>% filter(sex == "female") %>% filter(strain == "WSB") )$mean.skel)
#G
Q1_tot.sc_t.test.G <- t.test((mouse.total.skel %>% filter(sex == "male") %>% filter(strain == "G") )$mean.skel,
                           (mouse.total.skel %>% filter(sex == "female") %>% filter(strain == "G") )$mean.skel)
#LEW
Q1_tot.sc_t.test.LEW <- t.test((mouse.total.skel %>% filter(sex == "male") %>% filter(strain == "LEW") )$mean.skel,
                           (mouse.total.skel %>% filter(sex == "female") %>% filter(strain == "LEW") )$mean.skel)

#Musc
Q1_tot.sc_t.test.MUSC <- t.test((mouse.total.skel %>% filter(sex == "male") %>% filter(subsp == "Musc") )$mean.skel,
                           (mouse.total.skel %>% filter(sex == "female") %>% filter(subsp == "Musc") )$mean.skel)

#PWD
Q1_tot.sc_t.test.PWD <- t.test((mouse.total.skel %>% filter(sex == "male") %>% filter(strain == "PWD") )$mean.skel,
                           (mouse.total.skel %>% filter(sex == "female") %>% filter(strain == "PWD") )$mean.skel)

#KAZ
Q1_tot.sc_t.test.KAZ <- t.test((mouse.total.skel %>% filter(sex == "male") %>% filter(strain == "KAZ") )$mean.skel,
                           (mouse.total.skel %>% filter(sex == "female") %>% filter(strain == "KAZ") )$mean.skel)

#SKIVE
#Q1_tot.sc_t.test.SKIVE <- t.test((totSC.mouse.av.table %>% filter(sex == "male") %>% filter(strain == "SKIVE") )$mean.skel,
#           (totSC.mouse.av.table %>% filter(sex == "female") %>% filter(strain == "SKIVE") )$mean.skel, paired = FALSE, alternative = #"greater")
#cell level total SC
#MLH1.merge.Skel   $skel_size
#error not enought female observations, ONLY 1 female SKIVE mouse

Q1_tot.scBIV_t.test.SKIVE <- t.test((Total.SC_DF %>% filter(sex == "male") %>% filter(strain == "SKIVE") )$skel_size,
           (Total.SC_DF %>% filter(sex == "female") %>% filter(strain == "SKIVE") )$skel_size   )


#MSM
Q1_tot.sc_t.test.MSM <- t.test((mouse.total.skel %>% filter(sex == "male") %>% filter(strain == "MSM") )$mean.skel,
                           (mouse.total.skel %>% filter(sex == "female") %>% filter(strain == "MSM") )$mean.skel)

```


```{r Q1.totSC_male.first.comp, echo=FALSE}

Q1.male.Dom.Musc.t.test <- t.test( ( mouse.total.skel %>% filter(sex == "male") %>% filter(subsp == "Dom") )$mean.skel,
(mouse.total.skel %>% filter(sex == "male") %>% filter(subsp == "Musc") )$mean.skel)

Q1.male.Dom.Mol.t.test <- t.test( ( mouse.total.skel %>% filter(sex == "male") %>% filter(subsp == "Dom") )$mean.skel,
(mouse.total.skel %>% filter(sex == "male") %>% filter(subsp == "Mol") )$mean.skel)
```


```{r Q1.M.lmer_totalSC, echo=FALSE, warning=FALSE, message=FALSE}

Q1.lmer_totSC.mouse.av.table <- mouse.total.skel %>% filter(strain != "AST" & strain != "TOM" & strain != "CZECH" & strain != "MOLF")

#str(Q1.lmer_totSC.mouse.av.table)

#unorder factors
Q1.lmer_totSC.mouse.av.table$subsp <- factor(Q1.lmer_totSC.mouse.av.table$subsp, ordered = FALSE)
Q1.lmer_totSC.mouse.av.table$sex <- factor(Q1.lmer_totSC.mouse.av.table$sex, ordered = FALSE)
Q1.lmer_totSC.mouse.av.table$strain <- factor(Q1.lmer_totSC.mouse.av.table$strain, ordered = FALSE)
Q1.lmer_totSC.mouse.av.table$species <- factor(Q1.lmer_totSC.mouse.av.table$species, ordered = FALSE)


#TOTAL SC AREA
lmer.Q1_mus.av_totSC <- lmer(mean.skel ~ subsp * sex + (1|strain), data=Q1.lmer_totSC.mouse.av.table, REML = TRUE) 

lmer.Q1_mus.av_totSC_R.sex <- lmer(mean.skel ~ subsp + (1|strain), data=Q1.lmer_totSC.mouse.av.table, REML = TRUE) 

lmer.Q1_mus.av_totSC_R.subsp <- lmer(mean.skel ~ sex + (1|strain), data=Q1.lmer_totSC.mouse.av.table, REML = TRUE) 


lmer.Q1_mus.av_totSC_interaction  <-drop1(lmer.Q1_mus.av_totSC, test="Chisq" )$`Pr(Chi)`[2]

lmer.Q1_mus.av_totSC_sex  <-anova(lmer.Q1_mus.av_totSC, lmer.Q1_mus.av_totSC_R.sex)$`Pr(>Chisq)`[2]

lmer.Q1_mus.av_totSC_subp  <- anova(lmer.Q1_mus.av_totSC, lmer.Q1_mus.av_totSC_R.subsp)$`Pr(>Chisq)`[2]

lmer.Q1_mus.av_totSC_strain  <- exactRLRT(lmer.Q1_mus.av_totSC)$p.value 

#as.data.frame(round(summary(lmer.SCLength_all_MM.M1)$coefficients, 6))#these Dont return p values for the fixed effects

Q1.lmer.totSC_results <- data.frame(sub.sp = lmer.Q1_mus.av_totSC_subp, 
sex.results =  lmer.Q1_mus.av_totSC_sex, 
               interaction.pval = lmer.Q1_mus.av_totSC_interaction,  rand.pval = 
lmer.Q1_mus.av_totSC_strain)

```


```{r Q1.M1_short_biv_lmer, echo=FALSE, warning=FALSE, message=FALSE}

lmer.Q1_SCLength_short_M1 <- lmer(mean_SC ~ subsp * sex + (1|strain), data=Short.biv_mouse_table, REML = TRUE) 

lmer.Q1_SCLength_short_M1_R.sex <- lmer(mean_SC ~ subsp + (1|strain), data=Short.biv_mouse_table, REML = TRUE) 

lmer.Q1_SCLength_short_M1_R.subsp <- lmer(mean_SC ~ sex + (1|strain), data=Short.biv_mouse_table, REML = TRUE) 

Q1lmer.MM_SC_short_interaction  <-drop1(lmer.Q1_SCLength_short_M1, test="Chisq" )$`Pr(Chi)`[2]
Q1lmer.MM_SC_short_sex  <-anova(lmer.Q1_SCLength_short_M1, lmer.Q1_SCLength_short_M1_R.sex)$`Pr(>Chisq)`[2]
Q1lmer.MM_SC_short_subp  <- anova(lmer.Q1_SCLength_short_M1, lmer.Q1_SCLength_short_M1_R.subsp)$`Pr(>Chisq)`[2]
Q1lmer.MM_SC_short_strain  <- exactRLRT(lmer.Q1_SCLength_short_M1)$p.value 

#as.data.frame(round(summary(lmer.SCLength_all_MM.M1)$coefficients, 6))#these Dont return p values for the fixed effects

Q1lmer.MM_SC_short_results <- data.frame(sub.sp = Q1lmer.MM_SC_short_subp, 
sex.results =  Q1lmer.MM_SC_short_sex, 
               interaction.pval = Q1lmer.MM_SC_short_interaction,  rand.pval = 
Q1lmer.MM_SC_short_strain)


#Q1lmer.MM_SC_short_results

```


```{r Q2.SC.short.long.setup, echo=FALSE}

#table(male.long.biv$strain)
#the Long biv dataset is quite unbalance (esp MSM)
male.long.biv <- real.long.bivData %>% filter(sex== "male")

#mouse av for long biv
Male.cell.av.long.biv.table <- ddply(male.long.biv, c("fileName"), summarise,
                             
                        n_obs = length(Obj.ID),
                         mean.SC = mean(chromosomeLength)
                         )
Male.cell.av.long.biv.table <- add_mouse(Male.cell.av.long.biv.table)
Male.cell.av.long.biv.table <- add_strain(Male.cell.av.long.biv.table)
Male.cell.av.long.biv.table <- add_subsp(Male.cell.av.long.biv.table)

Male.cell.av.long.biv.table$strain <- factor(Male.cell.av.long.biv.table$strain, ordered = FALSE)
Male.cell.av.long.biv.table$strain <- factor(Male.cell.av.long.biv.table$strain, ordered = FALSE)
Male.cell.av.long.biv.table$strain <- factor(Male.cell.av.long.biv.table$strain, ordered = FALSE)


Q2mouse.av.long.biv.table <- ddply(male.long.biv, c("mouse"), summarise,
                        n_obs = length(Obj.ID),
                         mean.SC = mean(chromosomeLength),
                        
                         #mean.first.1CO
                        norm.first.foci.pos =  mean(PER_Foci_1[which(hand.foci.count == 1)])
                         )

Q2mouse.av.long.biv.table <- add_strain(Q2mouse.av.long.biv.table)
Q2mouse.av.long.biv.table <- add_subsp(Q2mouse.av.long.biv.table)

Q2mouse.av.long.biv.table$strain <- factor(Q2mouse.av.long.biv.table$strain, ordered = FALSE)
Q2mouse.av.long.biv.table$subsp <- factor(Q2mouse.av.long.biv.table$subsp, ordered = FALSE)

#divide musc males into groups
Q2mouse.av.long.biv.table$Rec.group <- ifelse(grepl("PWD", Q2mouse.av.long.biv.table$strain), 1, 
                                  ifelse(grepl("MSM", Q2mouse.av.long.biv.table$strain), 1,
                                  ifelse(grepl("SKIVE", Q2mouse.av.long.biv.table$strain), 1,
                                           0)))

Q2mouse.av.long.biv.table$Rec.group.exclu <- ifelse(grepl("PWD", Q2mouse.av.long.biv.table$strain), 1, 
                                ifelse(grepl("MSM", Q2mouse.av.long.biv.table$strain), 1, 0))


male.short.biv <- real.short.bivData %>% filter(sex== "male")

Q2mouse.av.short.biv.table <- ddply(male.short.biv, c("fileName"), summarise,
                             
                        n_obs = length(Obj.ID),
                         mean.SC = mean(chromosomeLength),
                        
                        #mean.first.1CO
                        norm.first.foci.pos =  mean(PER_Foci_1[which(hand.foci.count == 1)]
                         ) ) 

Q2mouse.av.short.biv.table <- add_mouse(Q2mouse.av.short.biv.table)
Q2mouse.av.short.biv.table <- add_strain(Q2mouse.av.short.biv.table)
Q2mouse.av.short.biv.table <- add_subsp(Q2mouse.av.short.biv.table)

Q2mouse.av.short.biv.table$strain <- factor(Q2mouse.av.short.biv.table$strain, ordered = FALSE)
Q2mouse.av.short.biv.table$subsp <- factor(Q2mouse.av.short.biv.table$subsp, ordered = FALSE)

Q2mouse.av.short.biv.table$Rec.group <- ifelse(grepl("PWD", Q2mouse.av.short.biv.table$strain), 1, 
ifelse(grepl("MSM", Q2mouse.av.short.biv.table$strain), 1,
                                  ifelse(grepl("SKIVE", Q2mouse.av.short.biv.table$strain), 1,
                                           0)))

Q2mouse.av.short.biv.table$Rec.group.exclu <- ifelse(grepl("PWD", Q2mouse.av.short.biv.table$strain), 1, 
                                ifelse(grepl("MSM", Q2mouse.av.short.biv.table$strain), 1, 0))
```

```{r Q2.show.SC.length, echo=FALSE, warning=FALSE, eval=FALSE}

Curated_BivData99 <- Curated_BivData[Curated_BivData$hand.foci.count < 5,]
Curated_BivData99 <- Curated_BivData99[(!is.na(Curated_BivData99$sex)),]

#adjust order of plot here
Curated_BivData99$strain <- factor(Curated_BivData99$strain,levels =c(
  "WSB","G","LEW","PERC",  
"PWD", "SKIVE",  "KAZ",
     "MSM", "MOLF",   
                     "CAST",
                         "SPRET","SPIC" ), order=T )

Q2.SC_BivData <- Curated_BivData %>% filter(sex=="male") %>% filter( hand.foci.count < 5)

#nested boxplots, pooled Bivalents
#x axis for hand.focicount -- good for seperating the points --- but it pools 

#prediction for Dom strains: low variation 
Q2.SC.length_Nested_boxplots_Dom <- ggplot(Q2.SC_BivData[Q2.SC_BivData$subsp == "Dom",], aes(y = chromosomeLength, x=strain, fill=as.factor(hand.foci.count) ))+geom_boxplot(alpha=0.5)+ geom_jitter(alpha = .1)+
  ggtitle("Bivalent SC Lengths by Chrm Class \n Dom")+facet_wrap(subsp~sex, scales="free")+ylim(c(25,160))+theme(legend.position="none") + scale_color_manual(values=color_chrm_class)

Q2.SC.length_Nested_boxplots_Dom2 <- ggplot(Q2.SC_BivData[Q2.SC_BivData$subsp == "Dom",], aes(y = chromosomeLength, x=as.factor(hand.foci.count), fill=as.factor(hand.foci.count) ))+geom_boxplot(aes(x=strain),alpha=0.5)+ geom_jitter(alpha = .1)+
  ggtitle("Bivalent SC Lengths by Chrm Class \nDom")+facet_wrap(subsp~sex, scales="free")+ylim(c(25,160))+theme(legend.position="none") +scale_color_manual(values=color_chrm_class)

Q2.SC.length_Nested_boxplots_Musc2 <- ggplot(Q2.SC_BivData[Q2.SC_BivData$subsp == "Musc",], aes(y = chromosomeLength, x=as.factor(hand.foci.count), fill=as.factor(hand.foci.count) ))+geom_boxplot( aes(x=strain), alpha=0.5)+geom_jitter(alpha=.2)+
  ggtitle("Bivalent SC Lengths by Chrm Class \nMusc")+facet_wrap(subsp~sex, scales="free")+ylim(c(25,160))+theme(legend.position="none") +scale_color_manual(values=color_chrm_class)


Q2.SC.length_Nested_boxplots_Mol <- ggplot(Q2.SC_BivData[Q2.SC_BivData$subsp == "Mol",], aes(y = chromosomeLength, x=as.factor(hand.foci.count), fill=as.factor(hand.foci.count) ))+geom_boxplot(aes(x=strain),alpha=0.5)+geom_jitter(alpha=.3)+  ggtitle("Bivalent SC Lengths by Chrm Class\n Mol")+facet_wrap(subsp~sex, scales="free")+ylim(c(25,160))+theme(legend.position="none")+scale_color_manual(values=color_chrm_class)

#+theme(legend.position="none") 
```


```{r Q2.SC.long.biv_plots, echo=FALSE, eval=FALSE}

#REAL.long_mouse_table
#REAL.short_mouse_table

#real.long.bivData
#real.short.bivData

shoo.long.biv <- ggplot( (real.long.bivData %>% filter(sex == "male") ), aes(y=chromosomeLength, x=strain, color=strain))+geom_jitter()+facet_wrap(~subsp, scales = "free")+scale_color_manual(values=colors_of_strains)+ylim(c(75,140))+geom_boxplot(alpha=.2) + ggtitle("Single Biv \nMale Long Biv SC Lengths")


#cell.avlong.biv <- ggplot(cell.av.long.biv.table, aes(y=mean.sc, x=strain, color=strain))+geom_jitter()+facet_wrap(~subsp, scales = "free")+scale_color_manual(values=colors_of_strains)+ylim(c(75,140))+geom_boxplot(alpha=.2) + ggtitle("Cell av. \nMale Long Biv SC Lengths")


mouse.shoo.long.biv <- ggplot( Long.biv_mouse_table, aes(y=mean.sc, x=strain, color=strain))+geom_jitter()+facet_wrap(~subsp, scales = "free")+scale_color_manual(values=colors_of_strains)+ylim(c(75,140))+geom_boxplot(alpha=.2) + ggtitle("Mouse av. \nMale Long Biv SC Lengths")

                      
#cell.avlong.biv <- ggplot( cell.av.long.biv.table, aes(y=mean.sc, x=strain, color=strain))+geom_jitter()+facet_wrap(~subsp, scales = "free")+scale_color_manual(values=colors_of_strains)+ylim(c(75,140))+geom_boxplot(alpha=.2) + #ggtitle("Cell av. \nMale Long Biv SC Lengths")

```


```{r Q2_tot.sc_plot, echo=FALSE, warning=FALSE}

mouse.av_totsc_male <- ggplot(data= mouse.total.skel %>% filter(sex == "male"), aes(y = mean.skel, x=mouse, color=strain))+ geom_point()+scale_color_manual(values=colors_of_strains)+
  facet_wrap(Rec.group~subsp, scales="free")+
  theme(axis.title.x=element_blank(),  axis.text.x=element_blank(), axis.text.y = element_text(size=9), axis.ticks.x=element_blank(), legend.position = c(0.8, 0.2) ) + ylim(c(1200,1850 ) )+labs(y="Mouse Mean Total SC")

mouse.av_totsc_male <- addSmallLegend(mouse.av_totsc_male)

```


```{r Q2.SC_MLH1.by.totalSC, echo=FALSE, warning=FALSE, fig.height=3, fig.width=8, fig.cap="Variation in total SC area per cell across strains and Relationship between total sc and mean MLH1 count per cell across male mice. A)	 Mean total SC area per mouse across subspecies and recombination groups. High- and low-recombining groups are indicated by 1 and 0 respectively. (Horizontal lines indicate strain averages for total SC area.)B) Mouse averaged rates of MLH1 and total SC per cell", echo=FALSE}

#make plot of mouse averages for MLH1 and total SC
bep <- merge(  male_mouse.avs_for.analysis,   mouse.total.skel)

X.by.texas <- ggplot(bep, aes(x=mean.skel, y=mean_co, color=strain))+geom_point()+
  scale_color_manual(values=colors_of_strains)+theme(legend.position="none")+labs(x="Mouse Mean Total SC", y="Mouse Mean MLH1 count")

#plot_grid(mouse.av_totsc_male, X.by.texas, nrow=1, labels=c('A', 'B')) #Or labels="AUTO"

```


```{r Q2totalSC_checkdates, echo=FALSE, eval=FALSE}

#investigating potential skew / technical effects in the algorithm

Neg.control <- ggplot(bep, aes(x=mean.skel, y=mean.bin, color=strain))+geom_point()+
  scale_color_manual(values=colors_of_strains)+ggtitle("Total SC by mean CO\nMouse Means")
#not sure what my expectations for this plot are, 

median(bep$oldest.image.date) #"2017-08-03"
#there might be an issue with the algorithm performance 
#given the parrelle diagonal lines -- 
#this might be due to the dates imaged  2017-02-07

bep$date.cut.off <- ifelse( bep$oldest.image.date < "2017-02-07", 0, 1 )

Neg.control.find.date <- ggplot(bep, aes(x=mean.skel, y=mean.bin, color=strain))+geom_point()+
  scale_color_manual(values=colors_of_strains)+ggtitle("Total SC by mean CO\nMouse Means")+facet_wrap(~date.cut.off)

#there's evidence that the algorithm might have some biased effects -- 
#that will be investigated later.

```


```{r Q2.totSC.ttest, echo=FALSE}
#totalSC high and low rec group

Q2tot.sc_t.test_High.Low <- t.test((mouse.total.skel %>% filter(sex == "male") %>% filter(Rec.group == "0") )$mean.skel,
                (mouse.total.skel %>% filter(sex == "male") %>% filter(Rec.group == "1") )$mean.skel)
#p-value = 0.007

#mol -- not enough Mol observations
#Q2tot.sc_t.test_High.Low_Mol <- t.test((mouse.total.skel %>% filter(sex == "male") %>% filter(Rec.group == "0") %>% filter(subsp == "Mol") )$mean.skel,
#         (mouse.total.skel %>% filter(sex == "male") %>% filter(Rec.group == "1") %>% filter(subsp == "Mol") )$mean.skel)

#musc
Q2tot.sc_t.test_High.Low_Musc <- t.test((mouse.total.skel %>% filter(sex == "male") %>% filter(Rec.group == "0") %>% filter(subsp == "Musc"))$mean.skel,
                (mouse.total.skel %>% filter(sex == "male") %>% filter(subsp == "Musc") %>% filter(Rec.group == "1") )$mean.skel)

```


```{r Q2.short.long.ttest, echo=FALSE}

#long and short 
#mouse.av.short.biv.table
#Q2mouse.av.short.biv.table
Q2short.biv_t.test_High.Low <- t.test((Q2mouse.av.short.biv.table %>% filter(Rec.group == "0") )$mean.SC,
                                      (Q2mouse.av.short.biv.table %>% filter(Rec.group == "1") )$mean.SC)

#p-value = 0.9

#mouse.av.long.biv.table
Q2long.biv_t.test_High.Low <- t.test((Q2mouse.av.long.biv.table  %>% filter(Rec.group == "0") )$mean.SC,
                          (Q2mouse.av.long.biv.table  %>% filter(Rec.group == "1") )$mean.SC)
#p-value = 0.2


#mUSC-short
Q2.short_t.test_High.Low.musc <- t.test((Q2mouse.av.short.biv.table  %>% filter(Rec.group == "0") %>% filter(subsp == "Musc") )$mean.SC,
                          (Q2mouse.av.short.biv.table  %>% filter(Rec.group == "1") %>% filter(subsp == "Musc"))$mean.SC)

#mol-short
#Q2.short_t.test_High.Low.mol <- t.test((Q2mouse.av.short.biv.table  %>% filter(Rec.group == "0") %>% filter(subsp == "Mol") )$mean.SC,
#                          (Q2mouse.av.short.biv.table  %>% filter(Rec.group == "1") %>% filter(subsp == "Mol"))$mean.SC)


#musc-long
Q2.long_t.test_High.Low.musc <- t.test((Q2mouse.av.long.biv.table  %>% filter(Rec.group == "0") %>% filter(subsp == "Musc") )$mean.SC,
                          (Q2mouse.av.long.biv.table  %>% filter(Rec.group == "1") %>% filter(subsp == "Musc"))$mean.SC)

#mol-long
#Q2.long_t.test_High.Low.mol <- t.test((Q2mouse.av.long.biv.table  %>% filter(Rec.group == "0") %>% filter(subsp == "Mol") )$mean.SC,
#                          (Q2mouse.av.long.biv.table  %>% filter(Rec.group == "1") %>% filter(subsp =="Mol"))$mean.SC)

#not enought observations 

```


```{r Q2.SC_HvL_allBiv, echo=FALSE}

#High-Low_All-BivData

Q2.all.BIV_t.test_HL <- t.test((Curated_BivData %>%filter(sex == "male")%>%filter(Rec.group == "0") )$chromosomeLength,
              (Curated_BivData %>%filter(sex == "male")%>%filter(Rec.group == "1"))$chromosomeLength)


#short
Q2.BIV_short_t.test_HL <- t.test((real.short.bivData %>%filter(sex == "male")%>%filter(Rec.group == "0") )$chromosomeLength,
              (real.short.bivData %>%filter(sex == "male")%>%filter(Rec.group == "1"))$chromosomeLength)

#long
Q2.BIV_long_t.test_HL <- t.test((real.long.bivData %>%filter(sex == "male")%>%filter(Rec.group == "0") )$chromosomeLength,
              (real.long.bivData %>%filter(sex == "male")%>%filter(Rec.group == "1"))$chromosomeLength)

```


```{r Q2.SC_across.subsp_tot.SC, echo=FALSE}

#Dom-Musc
Q2.totSC_t.test_DomMusc <- t.test((mouse.total.skel  %>% filter(subsp == "Dom") )$mean.skel,
                          (mouse.total.skel  %>%  filter(subsp == "Musc"))$mean.skel)
#p-value = 0.5

#Dom-Mol
Q2.totSC_t.test_DomMol <- t.test((mouse.total.skel  %>% filter(subsp == "Dom") )$mean.skel,
                          (mouse.total.skel  %>%  filter(subsp == "Mol"))$mean.skel)
#p-value = 0.1
```



```{r Q2.SC_across.subsp_short.long_biv, echo=FALSE}

#DOM-MUSC-ALL
Q2.all.BIV_t.test_DomMusc <- t.test((Curated_BivData %>%filter(subsp == "Dom")%>%filter(sex == "male") )$chromosomeLength,
              (Curated_BivData %>% filter(subsp == "Musc")%>%filter(sex == "male"))$chromosomeLength)
#p-value = 4e-13

#DOM-MOL-ALL
Q2.all.BIV_t.test_DomMol <- t.test((Curated_BivData%>%filter(subsp == "Dom")%>%filter(sex == "male") )$chromosomeLength,
              (Curated_BivData %>% filter(subsp == "Mol")%>%filter(sex == "male"))$chromosomeLength)
#p-value = 6e-10

#DOM-musc-long
Q2.longBIV_t.test_DomMusc <- t.test((real.long.bivData%>%filter(subsp == "Dom")%>%filter(sex == "male") )$chromosomeLength,
              (real.long.bivData %>% filter(subsp == "Musc")%>%filter(sex == "male"))$chromosomeLength)
#p-value = 0.2

#DOM-musc-short
Q2.shortBIV_t.test_DomMusc <- t.test((real.short.bivData  %>% filter(subsp == "Dom")%>%filter(sex == "male") )$chromosomeLength,
              (real.short.bivData %>% filter(subsp == "Musc")%>%filter(sex == "male"))$chromosomeLength)
#p-value = 0.8


#DOM-mol-long
Q2.longBIV_t.test_DomMol <- t.test((real.long.bivData  %>% filter(subsp == "Dom")%>%filter(sex == "male") )$chromosomeLength,
                          (real.long.bivData %>% filter(subsp == "Mol")%>%filter(sex == "male"))$chromosomeLength)
#p-value = 0.4

#DOM-mol-short
Q2.shortBIV_t.test_DomMol <- t.test((real.short.bivData  %>% filter(subsp == "Dom")%>%filter(sex == "male") )$chromosomeLength,
                          (real.short.bivData %>% filter(subsp == "Mol")%>%filter(sex == "male"))$chromosomeLength)
#p-value = 0.06

```


```{r Q2.glm.SC.long, echo=FALSE}

#this dataset is all males
glmQ2_long.biv_M4 <- glm(mean_SC ~ subsp * strain, data= (Long.biv_mouse_table %>% filter(sex == "male") )  )
sum.glmQ2_long.biv_M4 <- summary(glmQ2_long.biv_M4)


glmQ2_long.biv_M4 <- glm(mean_SC ~ strain, data= (Long.biv_mouse_table %>% filter(sex == "male") ))
sum.glmQ2_long.biv_M4 <- summary(glmQ2_long.biv_M4)
#round(sum.glmQ2_long.biv_M2$coefficients[,c(1,4)], 6)

glmQ2_long.biv_M4.2_DOM <- glm(mean_SC ~ strain, data= (Long.biv_mouse_table %>% filter(sex == "male") %>% filter(subsp == "Dom") )  )

sum.glmQ2_long.biv_M4.2_DOM <- summary(glmQ2_long.biv_M4.2_DOM)

#MUSC
glmQ2_long.biv_M4.2_MUSC <- glm(mean_SC ~ strain, data= 
  ( Long.biv_mouse_table %>% filter(sex == "male") %>% filter(subsp == "Musc"))  )
sum.glmQ2_long.biv_M4.2_MUSC <- summary(glmQ2_long.biv_M4.2_MUSC)

#MOL can't be done -- only 1 mouse for each MSM and MOLF
#glmQ2_long.biv_M4.2_MOL <- glm(mean_SC ~ strain, data=  ( Long.biv_mouse_table %>% filter(sex == "male") %>%  filter(subsp == "Mol") )  )

#sum.glmQ2_long.biv_M4.2_MOL <- summary(glmQ2_long.biv_M4.2_MOL)


#Dom
glm_SC.male_Dom_long.biv <- glm(mean_SC ~ strain, data= Long.biv_mouse_table[Long.biv_mouse_table$subsp == "Dom",])
#Musc
glm_SC.male_Musc_long.biv <- glm(mean_SC ~ strain, data= Long.biv_mouse_table[Long.biv_mouse_table$subsp == "Musc",])
#summary(glm_SC.male_Musc_long.biv)
#Mol
glm_SC.male_Mol_long.biv <- glm(mean_SC ~ strain, data= Long.biv_mouse_table[Long.biv_mouse_table$subsp == "Mol",])

#pairwise.t.test( ( Q2mouse.av.long.biv.table %>% filter(subsp == "Dom")%>% filter(subsp == "Dom") )$mean.SC, ( Q2mouse.av.long.biv.table %>% filter(subsp == "Dom") )$strain )

#pairwise.t.test( ( Q2mouse.av.long.biv.table %>% filter(subsp == "Dom") )$mean.SC, ( Q2mouse.av.long.biv.table %>% filter(subsp == "Dom") )$strain, p.adj = "bonf", pool.sd = FALSE)

#pairwise.t.test( ( Q2mouse.av.long.biv.table %>% filter(subsp == "Musc") )$mean.SC, ( Q2mouse.av.long.biv.table %>% filter(subsp == "Musc") )$strain )
```


```{r Q2.glm.SC.short, echo=FALSE}
#Q2mouse.av.short.biv.table
#mouse.av.short.biv.table

glmQ2_short.biv_M1 <- glm(mean.SC ~ subsp*strain, data= Q2mouse.av.short.biv.table)
sum.glmQ2_short.biv_M1 <- summary(glmQ2_short.biv_M1)

#round(summary(sum.glmQ2_short.biv_M1)$coefficients[,c(1,4)], 6)

glmQ2_short.biv_M2 <- glm(mean.SC ~ strain, data= Q2mouse.av.short.biv.table)
glmQ2_short.biv_M2 <- summary(glmQ2_short.biv_M2)

#DOM
glmQ2_short.biv_M2_DOM <- glm(mean.SC ~ strain, data= ( Q2mouse.av.short.biv.table %>% filter(subsp == "Dom") )  )
sum.glmQ2_short.biv_M2_DOM <- summary(glmQ2_short.biv_M2_DOM)

#MUSC
glmQ2_short.biv_M2_MUSC <- glm(mean.SC ~ strain, data= ( Q2mouse.av.short.biv.table %>% filter(subsp == "Musc") )  )
sum.glmQ2_short.biv_M2_MUSC <- summary(glmQ2_short.biv_M2_MUSC)

#MOL
#within subspecies

#pairwise.t.test( ( Q2mouse.av.short.biv.table %>% filter(subsp == "Dom") )$mean.SC, ( Q2mouse.av.short.biv.table %>% filter(subsp == "Dom") )$strain, p.adj = "bonf", pool.sd = FALSE)

#pairwise.t.test( ( Q2mouse.av.short.biv.table %>% filter(subsp == "Musc") )$mean.SC, ( Q2mouse.av.short.biv.table %>% filter(subsp == "Musc") )$strain, p.adj = "bonf", pool.sd = FALSE)

```


```{r Q2.SC_class.rec.group_setup, echo=FALSE, include=FALSE}
#REMOVE THIS, IT uses total bivalent data


#rerun the models for datasets -- deviding by chrm class and rec group
Curated_BivData$Rec.group <- ifelse(grepl("PWD male", Curated_BivData$category), 1, 
                            ifelse(grepl("MSM male", Curated_BivData$category), 1,
                             ifelse(grepl("SKIVE male", Curated_BivData$category), 1, 0)) )


Q2.SC_BivData_low.rec <- Curated_BivData %>% filter(sex=="male") %>% filter( hand.foci.count < 5)%>% filter(Rec.group == 0 )
Q2.SC_BivData_high.rec <- Curated_BivData %>% filter(sex=="male") %>% filter( hand.foci.count < 5) %>% filter(Rec.group == 1 )

#calq_mouse av
mouse.av.SC.low.rec <- ddply(Q2.SC_BivData_low.rec, c("mouse"), summarise,
                        n_obs = length(Obj.ID),
                        n_cells = length(unique(fileName) ),
                        mean.SC.total = mean(chromosomeLength),

                    mean.SC_0CO = mean(chromosomeLength[which(hand.foci.count == 0)], na.rm = TRUE),
                  n_0CO =  sum((hand.foci.count == 0), na.rm = TRUE ),
                  
                   mean.SC_1CO = mean(chromosomeLength[which(hand.foci.count == 1)], na.rm = TRUE),
                    n_1CO = sum((hand.foci.count == 1), na.rm = TRUE),
                  
                   mean.SC_2CO = mean(chromosomeLength[which(hand.foci.count == 2)], na.rm = TRUE),
                   n_2CO = sum((hand.foci.count == 2), na.rm = TRUE),
                  
                  mean.SC_3CO = mean(chromosomeLength[which(hand.foci.count == 3)], na.rm = TRUE),
                  n_3CO = sum((hand.foci.count == 3), na.rm = TRUE)
                                                
                         )

mouse.av.SC.low.rec <- add_strain(mouse.av.SC.low.rec)
mouse.av.SC.low.rec <- add_subsp(mouse.av.SC.low.rec)

mouse.av.SC.low.rec$strain <- factor(mouse.av.SC.low.rec$strain, ordered = FALSE)
mouse.av.SC.low.rec$subsp <- factor(mouse.av.SC.low.rec$subsp, ordered = FALSE)


#calq_mouse av
mouse.av.SC.high.rec <- ddply(Q2.SC_BivData_high.rec, c("mouse"), summarise,
                        n_obs = length(Obj.ID),
                         mean.SC = mean(chromosomeLength)
                         )
mouse.av.SC.high.rec <- add_mouse(mouse.av.SC.high.rec)
mouse.av.SC.high.rec <- add_strain(mouse.av.SC.high.rec)
mouse.av.SC.high.rec <- add_subsp(mouse.av.SC.high.rec)

mouse.av.SC.high.rec$strain <- factor(mouse.av.SC.high.rec$strain, ordered = FALSE)
mouse.av.SC.high.rec$add_subsp <- factor(mouse.av.SC.high.rec$subsp, ordered = FALSE)
```




*new* -- We used the contrast between males from high-recombination strains and males from low-recombination strains to identify features of the recombination landscape associated with evolutionary transitions in the genome-wide recombination rate. We considered proportions of bivalents with different numbers of crossovers, double-strand breaks, SC length, and crossover positions, including crossover interference.
Ninety-six percent of single bivalents in our pooled dataset (n = `r formatC( sum( Chrm.prop_strain_table$n0CO, Chrm.prop_strain_table$n1CO, Chrm.prop_strain_table$n2CO, Chrm.prop_strain_table$n3CO, Chrm.prop_strain_table$n4CO ), format="d", big.mark=",")` -- 34,982) have either one or two MLH foci (Figure 3). The proportions of single-focus (1CO) bivalents vs. double-focus (2CO) bivalents distinguish high-recombination strains from low-recombination strains (Figure 3). 

High-recombination strains are enriched for 2CO bivalents at the expense of 1CO bivalents: proportions of 2CO bivalents are `r chrm.prop.male.high1$n2CO[chrm.prop.male.high1$strain == "SKIVE"] / chrm.prop.male.high1$nbiv[chrm.prop.male.high1$strain == "SKIVE"]` -- 0.33 in _musculus^SKIVE^_, `r chrm.prop.male.high1$n2CO[chrm.prop.male.high1$strain == "PWD"] / chrm.prop.male.high1$nbiv[chrm.prop.male.high1$strain == "PWD"]` -- 0.44 in _musculus^PWD^_ , and `r chrm.prop.male.high1$n2CO[chrm.prop.male.high1$strain == "MSM"] / chrm.prop.male.high1$nbiv[chrm.prop.male.high1$strain == "MSM"]` -- 0.53 in _molossinus^MSM^_ (*SUP* Figure 3). Following patterns in the genome-wide recombination rate, male musculusPWD and male _molossinus^MSM^_ have 2CO proportions that are more similar to each other than to strains from their own subspecies (chi-square tests; _musculus^PWD^_ vs. _musculus^KAZ^_: p = 3.1510^{-33} -- `r PWD.KAZ.2CO.prop.test$p.value`; _molossinus^MSM^_ vs. _molossinus^MOLF^_: p = 4.7210^{-13} -- `r MSM.MOLF.2CO.prop.test$p.value`). PWD vs MS: p = `r PWD.MSM.2CO.prop.test$p.value`. **Add the extra test PWD-MSM**  These results demonstrate that evolution of the genome-wide recombination rate reflects changes in crossover number on multiple bivalents.  


To begin to localize evolution of genome-wide recombination rate to steps of the recombination pathway, we counted DMC1 foci in prophase spermatocytes as markers for double-strand breaks (DSBs). DMC1 foci were counted in a total of 76 -- `r sum( (DMC1table.main$ncells.x) )`  early zygotene and 76 -- `r sum( (DMC1table.main$ncells.y) )` late zygotene spermatocytes from three low-recombination strains ( _musculus^KAZ^_, _domesticus^WSB^_ , and  _domesticus^G^_) and two high-recombination strains ( _musculus^PWD^_ and _molossinus^MSM^_). High-recombination strains have significantly more DMC1 foci than low-recombination strains in early zygotene cells (t-test; p < 10^{-6} --  p < `r 1/1000000` ). In contrast, the two strain groups do not differ in DMC1 foci in late zygotene cells (t-test; p = 0.66 -- `r ttest.HighLow.Z.pval` ). Since DSBs are repaired as either COs or non-crossovers (NCOs), the ratio of MLH1 foci to DMC1 foci can be used to estimate the proportion of DSBs designated as COs. High-recombination and low-recombination strains do not differ in the MLH1/DMC1 ratio, whether DMC1 foci were counted in early zygotene cells or late zygotene cells (t-test; p > 0.05). These results raise the possibility that the evolution of genome-wide recombination rate is primarily determined by processes that precede the CO/NCO decision.


Although there is a positive correlation between total SC length and MLH1 foci when both traits are measured as mouse means (Spearman’s r = 0.48 --`r cor( mouse.total.skel$mean.skel, mouse.total.skel$mean.MLH1, method = c("spearman") )`; p = 2.2410^{-10} -- `r cor.test(mouse.total.skel$mean.skel, mouse.total.skel$mean.MLH1, method=c("spearman") )$p.value` ), total SC length only partially differentiates high-recombination strains from low-recombination strains (Figure 5). **Check if this is correct for all strains**. Whereas high-recombination strains as a group have significantly greater total SC length than low-recombination strains (t-test; p = 0.01 -- `r Q2tot.sc_t.test_High.Low$p.value`), separate tests within subspecies show that the two strain categories differ within _M. m. molossinus_ (p = 0.03 -- r Q2tot.sc_t.test_High.Low_Mol -- p.value ) but not within M. m. musculus (p = 0.87 -- `r Q2tot.sc_t.test_High.Low_Musc$p.value`). 

**Q2.SC.reduced** Q2short.biv_t.test_High.Low
Additionally, mouse means for the reduced (short and long) bivalent datasets do not differ between high-recombination and low-recombination strains (t-test; short: p = 0.88 -- `r Q2short.biv_t.test_High.Low$p.value` ; long: p = 0.18 -- `r Q2long.biv_t.test_High.Low$p.value`). In a model with **total SC length** as the dependent variable, two subspecies effects are significant ( _M. m. musculus_ p = 1.2410^{-6} -- `r sum.Q2.glm_mus.av_totSC_M4[2,4]` , _M. m. molossinus_, p = 10^{-6} -- `r sum.Q2.glm_mus.av_totSC_M4[3,4]` ). In models with SC lengths of short and long bivalents as dependent variables, several subspecies and strain effects reach significance (p < 0.05 -- ), but they are not consistent across models. Collectively, these observations reveal that evolution of the genome-wide recombination rate is not strongly associated with evolution of SC length.  

**Q2-CO.pos**  Q2.CO.pos t-test
High-recombination and low-recombination strains do not differ in MLH1 focus positions on bivalents with a single focus, either for the full dataset (t-test; p = 0.24 -- `r Q2.1COpos_Ttest$p.value` ) or within subspecies (t-test; _M. m. musculus_: p = 0.41 -- `r Q2.1COpos_mus.av_Musc_ttest$p.value` ; _M. m. molossinus_ : p = 0.07 -- r Q2.1COpos_mus.av_Mol_ttest$p.value ). **While domesticusWSB and _molossinus^MOLF^_ exhibit strain effects in a model with normalized crossover position as the dependent variable** (Q2.COpos), these two strains do not differ in MLH1 count. These results show that evolution of crossover position on chromosomes with single crossovers is decoupled from evolution of the genome-wide recombination rate, at least at this scale of resolution.

**Q2_IFD**
High-recombination strains have greater inter-focal distances than low-recombination strains (t-test; IFD~norm~: p = 7.7410^{-7} -- `r t.test_mouse.av.IFD_PER$p.value` ; IFD~raw~: p = 8.7810^{-6} -- `r t.test_mouse.av.IFD_ABS$p.value`). This pattern holds within _M. m. musculus_ (t-test; IFD~norm~: p = 2.0410^{-5} -- `r  t.test_mouse.av.IFD_PER_MUSC$p.value`; IFD~raw~: p = 1.9410^{-4} -- `r  t.test_mouse.av.IFD_ABS_MUSC$p.value`), with a trend in the same direction within _M. m. molossinus_ (IFD~norm~: p= 0.17 -- `r t.test_mouse.av.IFD_PER_MOL$p.value` ; IFD~raw~: p = 0.08 -- `r t.test_mouse.av.IFD_ABS_MOL$p.value`). Similar results are observed with models treating IFD~raw~ and IFD~norm~ as dependent variables: only effects associated with high-recombination strains are significant (p < 0.05). That IFD~raw~ and IFD~norm~ show similar patterns eliminates variation in SC lengths (and bivalent sizes) as the primary explanation. The main distinction in IFD~norm~ distributions between high-recombination and low-recombination strains is an enrichment of IFD~norm~ values under 30% in low-recombination strains (Figure 6). The frequency of IFDnorm values that fall below 30% ranges from 8.2% ( _domesticus^G^_) to 16% ( _musculus^KAZ^_) in low-recombination strains, whereas high-recombination strains all show frequencies below 4% (0%, 1.3%, and 3.3% for _musculus^SKIVE^_, _molossinus^MSM^_, and _musculus^PWD^_, respectively). These results indicate that evolution of the genome-wide recombination rate is accompanied by the evolution of crossover interference.  

In summary, evolution of the genome-wide recombination rate in males is connected to double-strand break number and crossover interference, but not (consistently) to SC length and crossover position (on single-crossover bivalents).



# B[Reserve for chapter 3?] Within-individual variation in the genome-wide recombination rate is higher in females

round(sum.MLH1var_M2_Q12$coefficients[,4][4], 5)
**MLH1var**  M1 - LmerMLH1_CV_results, M2 MLH1var_M2_sum_coE, M3  M4 glmMLH1Var_M3_sum.Coef

Counting MLH1 foci in multiple oocytes for each female and in multiple spermatocytes for each male allowed us to examine determinants of variation in recombination rate within mice. To do this, we considered the same models as above, but replaced mean MLH1 focus count (for a mouse) with within-mouse variance in MLH1 focus count (for a mouse) as the dependent variable. Sex is the only variable that affects MLH1 focus count in all models (M1: p < 10^{-6} --`r LmerMLH1_CV_results$sex.results` ; M2: p = 0.03 -- `r` ; M3: p = 0.03 -- ``).  

In general, females have almost twice as much inter-cellular variance in MLH1 foci compared to males (Figure 1). Since estimates of within-mouse variance may be more susceptible to technical error from staining, we repeated the analyses using a subset of cells with high quality scores (1 or 2; Materials and Methods). The results are similar: sex is the strongest effect (M1 p < 10^{-6}; M2 p = 2.310^{-4} -- ; M3 p = 2.2810^{-4} -- ). When both quality-curated and full datasets are considered, strain does not significantly and consistently affect variance in MLH1 focus count in either sex. These results show that sex is a primary determinant of inter-cellular variance in the genome-wide recombination rate across diverse genetic backgrounds.

,
 Q12 MLH1.var -- M1 -LmerMLH1_M1_Q12_VAR_results -- LmerMLH1_M1_Q12_VAR_results,  M2 -sum.MLH1var_M2_Q12.co_eff,  M3 -MLH1var_M3_Q12_sum

# IFD double triangle


```{r Comp_Ratios, echo=FALSE, eval=FALSE}

#calq ratios of belo 30

top.male <-  G.Bivalents.DF %>% filter(sex== "male") %>%  filter(hand.foci.count== 2) %>%filter(IFD1_PER <= .3) 

bottom.male <-  G.Bivalents.DF %>% filter(sex== "male") %>%  filter(hand.foci.count== 2) 

ratio.male <- length(top.male$Obj.ID)/ length(bottom.male$Obj.ID)


top.female <-  G.Bivalents.DF %>% filter(sex== "female") %>%  filter(hand.foci.count== 2) %>% filter(IFD1_PER <= .3)

bottom.female <-  G.Bivalents.DF %>% filter(sex== "female")%>%  filter(hand.foci.count== 2)

ratio.female <- length(top.female$Obj.ID)/ length(bottom.female$Obj.ID)

#what is the fastest way to calq all these ratios?  also what's the fastst way to get them into a DF / table?


#put all the DFs in a list, for each DF in the list -- apply the filters for the nom and denominator
MOLF.Bivalents.DF <- Curated_BivData %>% filter(strain == "MOLF")


DF.strain.list <- list(
  WSB.Bivalents.DF, G.Bivalents.DF, LEW.Bivalents.DF, 
  PWD.Bivalents.DF, KAZ.Bivalents.DF, SKIVE.Bivalents.DF,
  MSM.Bivalents.DF, MOLF.Bivalents.DF
  )

entr_DF <- data.frame(strain=c("WSB", "G", "LEW", "PWD", "KAZ", "SKIVE", "MSM", "MOLF"), 
        total.2CO_female=c(0,0,0,0,0,0,0,0), total.2CO_male=c(0,0,0,0,0,0,0,0), below30_male = c(0,0,0,0,0,0,0,0), below30_female = c(0,0,0,0,0,0,0,0),     
          female.ratio = c(0,0,0,0,0,0,0,0), male.ratio = c(0,0,0,0,0,0,0,0))
  
i = 1
for(x in (DF.strain.list) ){
  
  #print(head())
  
  #these filters won't apply to the indexed DF
  top.male =  x %>% filter(sex== "male") %>%  filter(hand.foci.count== 2) %>%filter(IFD1_PER <= .3)

  bottom.male <-  x %>% filter(sex== "male") %>%  filter(hand.foci.count== 2)

  entr_DF$total.2CO_male[i] <- length(bottom.male$Obj.ID)
  entr_DF$below30_male[i] <- length(top.male$Obj.ID)
  entr_DF$male.ratio[i] <- length(top.male$Obj.ID)/ length(bottom.male$Obj.ID)
  
  top.female <-  x %>% filter(sex== "female") %>%  filter(hand.foci.count== 2) %>% filter(IFD1_PER <= .3)
  
  bottom.female <-  x %>% filter(sex== "female")%>%  filter(hand.foci.count== 2)

  
  entr_DF$total.2CO_female[i] <- length(bottom.female$Obj.ID)
  entr_DF$below30_female[i] <- length(top.female$Obj.ID)
  entr_DF$female.ratio[i] <- length(top.female$Obj.ID)/ length(bottom.female$Obj.ID)
  
  i = i+1
  
  }
#entr_DF

entr_DF$sex.dif <- entr_DF$female.ratio - entr_DF$male.ratio

```

```{r Q1.IFD.long.short.setup, echo=FALSE, eval=FALSE}

#real.long.bivData
#real.short.bivData

Q1.IFD.long <- ddply(real.long.bivData, c("mouse"), summarise,
                             
                        n_obs = length(Obj.ID),
                         mean.SC = mean(chromosomeLength),
            
                      meanIFD2CO_raw = mean(IFD1_ABS[which(hand.foci.count == 2)], na.rm = TRUE),
                     meanIFD2CO_norm = mean(IFD1_PER[which(hand.foci.count == 2)], na.rm = TRUE)
                         
                     )

Q1.IFD.long <- add_sex(Q1.IFD.long)
Q1.IFD.long <- add_strain(Q1.IFD.long)
Q1.IFD.long <- add_subsp(Q1.IFD.long)

Q1.IFD.long$strain <- factor(Q1.IFD.long$strain, ordered = FALSE)
Q1.IFD.long$subsp <- factor(Q1.IFD.long$subsp, ordered = FALSE)



Q1.IFD.short <- ddply(real.short.bivData, c("mouse"), summarise,
                             
                        n_obs = length(Obj.ID),
                         mean.SC = mean(chromosomeLength),
            
                      meanIFD2CO_raw = mean(IFD1_ABS[which(hand.foci.count == 2)], na.rm = TRUE),
                     meanIFD2CO_norm = mean(IFD1_PER[which(hand.foci.count == 2)], na.rm = TRUE)
                         
                     )

Q1.IFD.short <- add_sex(Q1.IFD.short)
Q1.IFD.short <- add_strain(Q1.IFD.short)
Q1.IFD.short <- add_subsp(Q1.IFD.short)

Q1.IFD.short$strain <- factor(Q1.IFD.short$strain, ordered = FALSE)
Q1.IFD.short$subsp <- factor(Q1.IFD.short$subsp, ordered = FALSE)

#t.test. mouse av
Q1.IFD.raw_short.ttest_KAZ <-  t.test( (Q1.IFD.short %>% filter(sex == "male") %>% filter(strain =="KAZ" ))$meanIFD2CO_norm,
                                ( Q1.IFD.short %>% filter(sex == "female") %>% filter(strain =="KAZ"))$meanIFD2CO_norm)

  
Q1.IFD.raw_long.ttest_KAZ <- t.test( (Q1.IFD.long %>% filter(sex == "male") %>% filter(strain =="KAZ" ))$meanIFD2CO_norm,
                                ( Q1.IFD.long %>% filter(sex == "female") %>% filter(strain =="KAZ"))$meanIFD2CO_norm)
#both the short and long norm IFD are NS
  

#t.test Biv
Q1.IFD.raw_BIV.long.ttest_KAZ <-  t.test( (real.long.bivData %>% filter(sex == "male") %>% filter(strain =="KAZ" )  %>% filter(hand.foci.count == "2") )$IFD1_PER,
                ( real.long.bivData %>% filter(sex == "female") %>% filter(strain =="KAZ") %>%  filter(hand.foci.count == "2"))$IFD1_PER)

Q1.IFD.raw_BIV.short.ttest_KAZ <-  t.test( (real.short.bivData %>% filter(sex == "male") %>% filter(strain =="KAZ" )  %>% filter(hand.foci.count == "2") )$IFD1_PER,
                ( real.short.bivData %>% filter(sex == "female") %>% filter(strain =="KAZ") %>%  filter(hand.foci.count == "2"))$IFD1_PER)
```



```{r Q2.1CO.show_plots, echo=FALSE,  warning=FALSE, fig.cap="Male distributions of single crossover normalized positions.", fig.width=4}
#Curated_BivData_male

#boxplots of nrm F1
male.bivdata.1CO <- cleanCurated_BivData %>% filter(sex == "male") %>% filter(species =="M.musculus") %>% filter(hand.foci.count == 1)
  
Q2.show.nrmF1 <- ggplot(male.bivdata.1CO, aes(x=strain, y=PER_Foci_1, color=strain) )+geom_jitter(alpha=.23 )+ geom_boxplot(aes(y= PER_Foci_1), alpha=.5 ) + scale_color_manual(values=colors_of_strains)+facet_wrap(~subsp, scales = "free")+ylim(c(0,1.1))+theme(legend.position="none", axis.text.x = element_text(size=9), axis.title.x = element_blank())+labs(y="Foci 1 Relative Position")

```

```{r Q2.1COpos.LONG.BIV_plot, echo=FALSE, warning=FALSE}
#make the same plots as above but with the long biv data set
male.long.biv.1CO <- real.long.bivData %>% filter(sex== "male") %>% filter(species =="M.musculus") %>% filter(hand.foci.count == 1)

#mouse.av.long.biv.table

Q2.show.nrmF1_long.biv <- ggplot(male.long.biv.1CO, aes(x=strain, y=PER_Foci_1, color=strain) )+geom_jitter( )+ geom_boxplot(aes(y= PER_Foci_1 ),alpha=.3 ) + scale_color_manual(values=colors_of_strains)+facet_wrap(~subsp, scales = "free")+ylim(c(0,1.1))+ggtitle("Normalized 1CO Foci pos \nLong Biv")

Mouse.av.Long.biv <- ddply(male.long.biv.1CO, c("mouse"), summarise,
                        Nmice = length(unique(mouse)),
                        Nbiv  = length(Obj.ID),
                       mean.1CO_PER = mean(PER_Foci_1)
                      # mean_CO = mean()
                       )

Mouse.av.Long.biv <- add_strain(Mouse.av.Long.biv)
Mouse.av.Long.biv <- add_subsp(Mouse.av.Long.biv)


Q2.shoooo   <- ggplot(Mouse.av.Long.biv, aes(x=strain, y=mean.1CO_PER, color=strain) )+geom_jitter( )+ geom_boxplot(aes(y= mean.1CO_PER ),alpha=.3 ) + scale_color_manual(values=colors_of_strains)+facet_wrap(~subsp, scales = "free")+ylim(c(0,1.1))+ggtitle("mouse ave \nNormalized 1CO Foci pos \nLong Biv")

```


```{r Q2.1COpos.show.MOL, echo=FALSE}

#(mouse.avs_4MM %>%  filter(sex == "male") %>%  filter(subsp == "Mol") %>% filter(Rec.group == #"0"))$norm.first.foci.pos

mus.av1COpos.plot <- ggplot(data= mouse.level.BivData_table %>%  filter(sex == "male") %>%  filter(subsp == "Mol"),
                      aes(x= mouse, y = norm.first.foci.pos))+geom_point()
#+facet_wrap(~mouse, scales = "free")

single.biv_1COpos.plot_Mol <- ggplot(data= Curated_BivData %>%  filter(sex == "male") %>%  filter(subsp == "Mol") %>%  filter(hand.foci.count == "1") %>%  filter(chromosomeLength < 100),
                      aes(x= fileName, y = PER_Foci_1, color=chromosomeLength))+geom_point()+facet_wrap(~mouse, scales = "free") + ylim(c(0,1))

#+scale_color_manual(values=colors_of_strains)


single.biv_1COpos.plot_Dom <- ggplot(data= Curated_BivData %>% filter(sex == "male") %>%filter(subsp == "Dom") %>%  filter(hand.foci.count == "1"),
                      aes(x= fileName, y = PER_Foci_1, color=strain ))+geom_point()+facet_wrap(~mouse, scales = "free") + ylim(c(0,1))+scale_color_manual(values=colors_of_strains)

#CHRM.length is very significant-- some of the same subsp and strains are sig     
```


```{r Q2.show.IFD, echo=FALSE, warning = FALSE}
#subset
male.biv.2CO <- cleanCurated_BivData %>% filter(sex== "male") %>% filter(species =="M.musculus") %>% filter(hand.foci.count == 2)

male.biv.2CO$Rec.group <- ifelse(grepl("PWD", male.biv.2CO$strain), 1, 
                                    ifelse(grepl("MSM", male.biv.2CO$strain), 1, 
                                           ifelse(grepl("SKIVE", male.biv.2CO$strain), 1, 0)))



biv.IFDs_PER_by.group <- ggplot(male.biv.2CO, aes(y = IFD1_PER, x= strain, color=strain))+geom_boxplot(alpha=.2)+
  geom_jitter(alpha=.2)+
  ggtitle("")+
  scale_color_manual(values=colors_of_strains)+ facet_wrap(Rec.group ~ subsp, scales = "free")+
  theme(legend.position="none", axis.text.x = element_blank(), axis.title.x = element_blank() )  + labs("Normalized IFD ") +
  ylim(c(0,1.1))

#consider adding horizontal line

#to adjust the boxplot width
#https://stackoverflow.com/questions/46174222/ggplot2-incorrect-boxplot-width-when-facetting-with-facets-of-different-scales
#I don't think there's a straightforward way to achieve that in ggplot. You may have better luck plotting each facet as a separate ggplot object (on a subset of the data), then arrange them on a single page, using grid.arrange() from gridExtra for example.

```

```{r Q2.IFD.glm.test, echo=FALSE, warning=FALSE}

#M1.ABS
Q2.M1glm_IFD.raw <- glm(mean_IFD.2CO_ABS ~ subsp * strain, data= mouse.level.BivData_table %>% filter(sex == "male") %>% filter(species == "M.musculus")  )
sum.Q2.M1glm_IFD.raw <- summary(Q2.M1glm_IFD.raw) 

#M1.PER
Q2.M1glm_IFD.PER <- glm(mean_IFD.2CO_PER ~ subsp * strain, data= mouse.level.BivData_table %>% filter(sex == "male") %>% filter(species == "M.musculus") )

sum.Q2.M1glm_IFD.PER <- summary(Q2.M1glm_IFD.PER)
#summary(Q2.M1glm_IFD.PER) 

#M2.raw
Q2.M2glm_IFD.raw <- glm(mean_IFD.2CO_ABS ~ strain, data= mouse.level.BivData_table %>% filter(sex == "male") %>% filter(species == "M.musculus") )
Sum.Q2.M2glm_IFD.raw <- summary(Q2.M2glm_IFD.raw)

Q2.M2glm_IFD.PER <- glm(mean_IFD.2CO_PER ~ strain, data= mouse.level.BivData_table %>% filter(sex == "male") %>% filter(species == "M.musculus") )
sum.Q2.M2glm_IFD.PER <- summary(Q2.M2glm_IFD.PER)
#  MSM strain effect for 2CO PER not significant

#within subspecies
#Dom
Q2.glm.M2_IFD_Dom <- glm(mean_IFD.2CO_ABS ~ strain, data= mouse.level.BivData_table %>% filter(subsp == "Dom") )
#summary(Q2.glm.M2_IFD_Dom)

#Musc
Q2.glm.M2_IFD_Musc <- glm(mean_IFD.2CO_ABS ~ strain, data= mouse.level.BivData_table %>% filter(subsp == "Musc") )
#summary(Q2.glm.M2_IFD_Musc)

#Mol
Q2.glm.M2_IFD_Mol <- glm(mean_IFD.2CO_ABS ~ strain, data= mouse.level.BivData_table %>% filter(subsp == "Mol") )
#summary(Q2.glm.M2_IFD_Mol)


#Dom
Q2.glm.M2_IFD.PER_Dom <- glm(mean_IFD.2CO_PER ~ strain, data= mouse.level.BivData_table %>% filter(subsp == "Dom") )
#summary(Q2.glm.M2_IFD.PER_Dom) 

#Musc
Q2.glm.M2_IFD.PER_Musc <- glm(mean_IFD.2CO_PER ~ strain, data= mouse.level.BivData_table %>% filter(subsp == "Musc") )
#summary(Q2.glm.M2_IFD.PER_Musc) 

#Mol
Q2.glm.M2_IFD.PER_Mol <- glm(mean_IFD.2CO_PER ~ strain, data= mouse.level.BivData_table %>% filter(subsp == "Mol")  )
#summary(Q2.glm.M2_IFD.PER_Mol)

```


# Single rewrite IFD triangle


```{r Q2.double.tri.rewrite, echo=FALSE, warning=FALSE}

#good function, saves to file, can't reproduce in rmd ..width=650, height=500
#big loop for automating the IFD triangles

Q2.males.IFD <- c("F1", "other")
Q2.auto_curate_DF <- cleanCurated_BivData[ ! cleanCurated_BivData$strain %in% Q2.males.IFD, ]

Q2.df_strain_list <- split(Q2.auto_curate_DF, (Q2.auto_curate_DF$strain))


#this works
Q2.bottom.triangle2 <- function(my.y, my.x, my.data){
  points(my.x ~ my.y, data=my.data, xpd=TRUE, 
          col =  colors_of_strains[unique(my.data$strain)])
  
  mtext("Position.2.Male", 1, -2, at= mean(par()$usr[1:2]) + 0)
  
  #the second val will move the x lable when negative
  
  #at should be changing the position
  #above is just changing the text not the numbers
  #first number, on which side of the plot (1=bottom, 2=left, 3=top, 4=right)
  
  #second number - on which MARgin line, starting at 0 counting outwards.
  #where text should be,  at = 0, 1 (bad)
  
  mtext("Position.1.Male", 4, 0, padj=par()$usr[1] + .5) #last value at .5 is good for moving y label
  
  x.at <- axisTicks(par()$usr[1:2], 0) #where the ticks should go
  
  axis(side=1, las=1, pos=0, at=x.at, 
       labels=T, xpd=TRUE)
#changing labels=T, puts the ddefault labels back
  
  #second value should be around .75 (moves label text closer)
  axis(4, las=1, pos=1, at=x.at) #this is the y axis
  lines(0:1, 0:1, xpd=TRUE) #diagonal line
}

x.dist <- .25
RP <- list()

for(i in 1:(length(Q2.df_strain_list))){
  #print(i)
  current.strain <- as.character(Q2.df_strain_list[[i]]$strain[1])
  
  #par(mar = c(2, 2, 2, 2))
  #dev.new(width = 8, height = 4, unit = "in")
  #declare file for saving png
  png(paste0("~./MLH1repo/doc/figureFiles/Q2.IFD.triangle_", current.strain, ".png"),  width= 650, height=500, res = 100) #500 seems like a good size
  plot.new()
  
  my.data = (cleanCurated_BivData %>% filter(strain == current.strain) %>% filter(hand.foci.count == 2) )
  my.x.bottom = (cleanCurated_BivData %>% filter(strain == current.strain) %>% filter(sex == "male") %>% filter(hand.foci.count == 2) )$PER_Foci_1 
  my.y.bottom = (cleanCurated_BivData %>% filter(strain == current.strain) %>% filter(sex == "male") %>% filter(hand.foci.count == 2) )$PER_Foci_2
  
  #adjusting this mar lowers the margins (the edges)
  op <- par(mar=c(.25, .15, .25, .75), oma=c(2, 0, 0, 2))
#(1=bottom, 2=left, 3=top, 4=right)

Q2.bottom.triangle2(my.y.bottom, my.x.bottom, my.data) #x.dist

  par(op)
  RP[[i]] <- recordPlot()
  dev.off()
}

```


```{r Q2.load.bottom.tri, echo=FALSE}

#only load, SKIVE, PWD, MSM
#MOLF, KAZ, LEW
Q2.bottom.LEW <- readPNG('~/MLH1repo/doc/figureFiles/Q2.IFD.triangle_LEW.png')
Q2.bottom.KAZ <- readPNG('~/MLH1repo/doc/figureFiles/Q2.IFD.triangle_KAZ.png')
Q2.bottom.SKIVE <- readPNG('~/MLH1repo/doc/figureFiles/Q2.IFD.triangle_SKIVE.png')
Q2.bottom.PWD <- readPNG('~/MLH1repo/doc/figureFiles/Q2.IFD.triangle_PWD.png')
Q2.bottom.MSM <- readPNG('~/MLH1repo/doc/figureFiles/Q2.IFD.triangle_MSM.png')
Q2.bottom.MOLF <- readPNG('~/MLH1repo/doc/figureFiles/Q2.IFD.triangle_MOLF.png')

empty.DF <- data.frame(col1 = c(1,3), col2 = c(1,3) )

#EMPTY GGPLOT
Q2.LEW.bottom.tri <- ggplot(empty.DF, aes(x = col1, y = col2)) +
  background_image(Q2.bottom.LEW) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),
    axis.title.x=element_blank(),                       axis.text.x=element_blank(),axis.ticks.x=element_blank(),axis.title.y=element_blank(),
                                  axis.text.y=element_blank(),axis.ticks.y=element_blank() )

```


```{r Q2.IFD_load.triangles, echo=FALSE, message=FALSE, warning=FALSE, fig.width=6, fig.cap="male IFD distributions highlighting differences across recombining groups"}




Q2.SC_dif.Figure3  <- ggdraw() +
  
  draw_plot(Q1.SC.short.mouse_scatter_boxplot, x=0, y=.38, width=.5, height= .6)+
  draw_plot(Tot.SC_mouse.av_ALL, x=.475, y=.38,  width = .5, height = .6)+

  draw_plot(single.chrm.icon, x=.44, y=.79,  width = .1, height = .1)+   
  draw_plot(all.chrm.icon, x=.88, y=.79,  width = .1, height = .1)+

  
  #drawing triangles in the corner
  #top right-triangle
  draw_line(
    x = c(.1, .2,  .1, .1 ), #keep the x values consistant
    y = c( .25, .25,  .15, .25),
    color = "blue", size=.75
  )


#draw boxes around the  


```



# Discussion


```{r corMLH1.FM, echo=FALSE}

#needs to be strain
sperman.corr.FM <- cor( ( strain.table.HQ_Fig %>% filter(sex == "female") %>% filter(species == "M.musculus" ) )$mean_co, ( strain.table.HQ_Fig %>% filter(sex == "male") %>% filter(species == "M.musculus" ) )$mean_co, 
                        method = c("spearman") )

sperman.corr.FM.pval <- cor.test( ( strain.table.HQ_Fig %>% filter(sex == "female") %>% filter(species == "M.musculus" ) )$mean_co, ( strain.table.HQ_Fig %>% filter(sex == "male") %>% filter(species == "M.musculus" ) )$mean_co, 
                        method = c("spearman") )
```


By comparing recombination rates in females and males from the same diverse set of genetic backgrounds, we isolated sex as a primary factor in the evolution of this fundamental meiotic trait. Recombination rate differences are more pronounced in males than females. Because inter-strain divergence times are identical for the two sexes, this observation demonstrates that the genome-wide recombination rate evolves faster in males, at least in house mice. More generally, recombination rate divergence is decoupled in males and females (Spearman’s correlation test comparing female and male strain mean MLH1 values for house mice; rho = 0.18 -- `r sperman.corr.FM.pval$estimate`; p = 0.60 -- `r sperman.corr.FM.pval$p.value` ). These disparities are remarkable given that recombination rates for the two sexes were measured in the same genomic backgrounds. Our results provide the strongest evidence yet that the genome-wide recombination rate follows distinct evolutionary trajectories in males and females. Next, we consider several explanations for this phenomenon.
At the genetic level, the sex-specific evolution we documented indicates that some mutations responsible for divergence in recombination rate have dissimilar phenotypic effects in females and males. A subset of the genetic variants associated with genome-wide recombination rate within populations of humans (@Kong2004, @Kong2008, @Kong2014, @halldorsson2019), Soay sheep [@johnston2016_soay], and cattle (@ma2015_cattle, @Shen2018_cattle) appear to show sex-specific properties, including antagonistic effects in females and males. Inter-sexual correlations are weak for recombination rate in humans [@fledel2011] and Soay sheep (@johnston2016_soay). Crosses between the strains we surveyed could be used to identify and characterize the genetic variants responsible for recombination rate evolution in house mice (@dumont2011, @Wang2017island). These variants could differentially affect females and males at any step in the recombination pathway. Although our DMC1 profiling was limited to males from a small number of strains (for practical reasons), our findings suggest that mutations that determine the number of double-strand breaks contributed to sex-specific evolution in the recombination rate.  

Our results also raise the prospect that the connection between recombination rate and fitness differs between males and females. Little is known about whether and how natural selection shapes recombination in nature (@DapperPayseur2017, @Ritz2017). Using a quantitative genetic test, [@samuk2020]  recently discovered that genome-wide recombination rates in females from two populations of _Drosophila pseudoobscura_ diverged in a manner consistent with local adaptation. Applying similar strategies to species in which both sexes recombine, including house mice, would be a logical next step to understanding the sex-specific evolution of recombination rate.
Population genetic models have been built to explain sexual dimorphism in the number and placement of crossovers, which is a common phenomenon (@brandvain2012scrambling, @sardell_sex_2020). Modifier models predicted that lower recombination rates in males will result from haploid selection [@lenormand2003] or sexually antagonistic selection on coding and cis-regulatory regions of genes [@sardell_sex_2020]. Another modifier model showed that meiotic drive could stimulate female-specific evolution of the recombination rate [@brandvain2012scrambling]. Although these models fit the conserved pattern of sex differences in crossover positions, they do not readily explain our observations of sex-specific evolution in the genome-wide recombination rate. In particular, the alternation across strains in which sex has more crossovers is unexpected.  
We propose an alternative interpretation based on the cell biology of gametogenesis. During meiosis, achieving a stable chromosome structure requires the attachment of kinetochores to opposite poles of the cell and at least one crossover to create tension across a proportion of sister cohesion [@LaneKauppi2019]. The spindle assembly checkpoint (SAC) prevents aneuploidy by ensuring that all bivalents are correctly attached to the microtubule spindle (“bi-oriented”) before starting the metaphase-to-anaphase transition [@LaneKauppi2019]. Hence, selection seems likely to favor mutations that optimize the process of bi-orientation, thereby prohibiting the SAC from delaying the cell cycle or triggering apoptosis.  

Multiple lines of evidence indicate that the SAC is more effective in spermatogenesis than in oogenesis [@LaneKauppi2019], perhaps due to the presence of the centrosome spindle [@So2019] and higher cell volume [@kyogoku2017] in oocytes. The higher stringency of the SAC during spermatogenesis suggests that selection will be better at removing mutations that interfere with bi-orientation in males than in females. Therefore, faster male evolution of the genome-wide recombination rate could be driven by the stronger SAC acting on chromosome structures at the metaphase I alignment.  

Our SAC model is consistent with other features of our data. We showed that widespread sex differences in broad-scale crossover positioning [@sardell_sex_2020] apply across house mice, even in lineages where the direction of heterochiasmy is reversed. We hypothesize that these sex differences hinge on disparate requirements for chromosome cohesion in late meiosis I. The irreversible process of the metaphase-to-anaphase transition is initiated by the proteolytic decay of the sister cohesion proteins that connect homologs (@LaneKauppi2019, @subramanian2014, @dumontDesai2012). The number and placement of crossovers alter the distribution of sister cohesion and the resulting chromosome structure when bivalents are aligned and bi-oriented on the metaphase plate (@vanVeen2003, @altendorfer2020). Differences in timing of the cell cycle between oogenesis and spermatogenesis could impose contrasting selective pressures on how sister cohesion affects chromosome structure. Faster spermatogenesis may select for synchronization of the separation homologs (cite). While in oogenesis, the slower cell cycle and multiple arrest stages may require chromosome structures with greater stability on the MI spindle, especially for those with dictyate arrest [@Lee2019].
We propose that the SAC model also can explain the correlated evolution of stronger crossover interference and higher genome-wide recombination rate in male house mice. Our results show that crossovers are spaced further apart in strains enriched for double-crossover bivalents when SC length is taken into account and chromosome size effects are minimized. Assuming chromatin compaction between (prophase) pachtyene and metaphase is uniform along bivalents, this increased spacing is expected to expand the area for sister cohesion to connect homologs and improve the fidelity of chromosomal segregation. Although the SAC model postulates direct fitness effects of interference, a modifier model predicts that indirect selection on recombination rate – via its modulation of offspring genotypes – can strengthen interference as well [@goldstein1993].  


Regardless of the underlying mechanism, our results provide a rare demonstration that crossover interference can diverge over short evolutionary timescales. The notion that stronger interference can co-evolve with higher genome-wide recombination rate is supported by differences between breeds of cattle @ma2015_cattle and differences between wild-born and laboratory-raised white-footed mice @peterson2019. In contrast, mammalian species with stronger interference tend to exhibit lower genome-wide recombination rates (@segura2013, @ottoPaysuer2019). Collectively, these patterns suggest that inferences about the evolutionary dynamics of interference depend on the timescale under consideration.  
 
Our findings further reveal that evolution of the genome-wide recombination rate does not require changes in the degree of chromatin compaction. Females consistently showed longer SCs, even in strains with more recombination in males. Studies in mice (@lynn2002, @petkov2007) and humans [@gruhn2013] suggest that chromosomal axes are longer (and DNA loops are shorter) in females than males. A cellular model designed to explain interference attributes sexual dimorphism in chromatin structure to greater cell volumes and oscillatory movements of telomeres and kinetochores in oocytes (@hulten2011_COM).  

Our conclusions are accompanied by several caveats. First, MLH1 foci only identify interfering crossovers **(REF)**. Although most crossovers (XX%; REF) belong to this class, our approach likely underestimated genome-wide recombination rates. Evolution of the number of non-interfering crossovers is a topic worth examining. A second limitation is that our investigation of crossover positions was confined to the relatively low resolution possible with immunofluorescent cytology. Positioning crossovers with higher resolution could reveal additional evolutionary patterns. Finally, the panel of inbred lines we surveyed may not be representative of recombination rate variation within and between subspecies of house mice. We considered most available wild-derived inbred lines, but house mice have a broad geographic distribution. Nevertheless, we expect our conclusion that recombination evolves in a sex-specific manner to be robust to geographic sampling because differences between females and males exist for the same set of inbred strains.  

The causes of sex differences in recombination remain mysterious @lenormand2016. Our conclusions have implications for a wide range of recombination research. For biologists uncovering the cellular and molecular determinants of recombination, our results suggest that mechanistic differences between the sexes could vary by genetic background. For researchers charting the evolutionary trajectory of recombination, our findings indicate that sex-specific comparisons are crucial. For theoreticians building evolutionary models of recombination, different fitness regimes and genetic architectures in females and males should be considered. Elevating sex as a primary determinant of recombination would be a promising step toward integrating knowledge of cellular mechanisms with evolutionary patterns to understand recombination rate variation in nature.


# REFERENCES

