---
title: "MLH1 Results Outline"
date: 
output:
  html_document:
    df_print: paged
    toc: yes
    toc_depth: '5'
    toc_float: yes
  pdf_document: default
  word_document:
    toc: yes
    toc_depth: '5'
bibliography: refs_Main_Report.bib
---


```{r setup, echo=FALSE, include = FALSE}
library(knitr)
library(ggplot2)
#library(pwr)
library(plyr)
library(lattice)
library(dplyr)
library(raster)
library(lme4)
library(nlme)
library(RLRsim)
library(ggpubr)
library(reshape2)
library(patchwork)

options(digits=2)

setwd("~.")#switch over to here()

#MLH1_data_setup_2.24.20.RData
load(file = "~./MLH1repo/data/MLH1/MLH1_data_setup_2.24.20.RData")

#load(file = "C:/Users/alpeterson7/Documents/MLH1repo/data/MLH1/MLH1_data_setup_1.31.20.RData")

#2.2.20
#load(file="C:/Users/April/Documents/MLH1repo/data/CleanBivData_1.27.20.RData")
#load(file="C:/Users/alpeterson7/Documents/MLH1repo/data/CleanBivData_1.27.20.RData")

load(file="~./MLH1repo/data/CleanBivData_19.2.20.RData")

load(file="~./MLH1repo/data/total_SC_30.12.19.RData")

#skeletonize
SC.skel = read.csv("~./MLH1repo/data/SCskel_output_feb20.csv", header = TRUE, strip.white = TRUE)

#load(file="~./MLH1repo/data/MLH1_data_setup_1.5.20.RData")

#setwd("~./MLH1repo/doc/")
#setwd("C:/Users/alpeterson7/Documents/MLH1repo/doc/")
#setwd("C:/Users/April/Documents/MLH1repo")
#load(file="data/MLH1/MLH1_data_setup.RData")

#load main data file
#load(file="C:/Users/April/Documents/MLH1repo/data/MLH1/MLH1_data_setup_1.5.20.RData")#added batch18
#load(file="C:/Users/April/Documents/MLH1repo/data/CleanBivData_1.28.20.RData")
#load(file="C:/Users/alpeterson7/Documents/MLH1repo/data/CleanBivData_1.28.20.RData")

#this MLH1 only goes up to batch16!
#load(file="C:/Users/alpeterson7/Documents/MLH1repo/data/MLH1/MLH1_data_setup_12.20.19.RData")#batch18
#load(file="C:/Users/alpeterson7/Documents/MLH1repo/data/MLH1/MLH1_data_setup_12.20.19.RData")#fixing bugs, formating
#load(file="data/MLH1/MLH1_data_setup_12.3.19.RData")#removed exclude mice
#load(file="data/MLH1/MLH1_data_setup_11.12.19.RData")#added batch17
#load(file="data/MLH1/MLH1_data_setup_11.11.19.RData")#deleted DUP image
#load(file="data/MLH1/MLH1_data_setup_9.26.19.RData")#added batch16
#load(file="data/MLH1/MLH1_data_setup_8.29.19.RData") #

#load(file="data/MetaData.RData") #this was replacing most updated data file

#load functions
source("~./MLH1repo/src/CommonFunc_MLH1repo.R")
```

<!-- Data Set Up Notes -->

```{r organizing.data, echo=FALSE }
#set the order of another column, based on another variable. consistant order for ploting

#Terminology Note, leptonema, zygonema, and pachynema diplonema diakinesis, are the stage names.
#-ene endings refer to the cells/units within the cell stage? (I think)

#apply this before ploting
MLH1_data <- MLH1_data %>%
  arrange(strain, sex, category, mouse) %>%
  mutate(Original.Name = factor(Original.Name)) #another category that you want the order to match
MLH1_data <- with(MLH1_data, MLH1_data[order(sex, strain),])

# sort your dataframe, by the focal categories
MLH1_data$mouse <- factor(MLH1_data$mouse, levels=unique(MLH1_data$mouse))
#use the above line for when order is out of wak


#count the non-quality measures,  #remove non qualit
#length(MLH1_data[ !(is.na(MLH1_data$quality) | MLH1_data$quality==""), ] )  #15 rows with out quality scores, remove.
#MLH1_data <- MLH1_data[ !(is.na(MLH1_data$quality) | MLH1_data$quality==""), ]

#MLH1_by_F_strain <- MLH1_data[MLH1_data$sex == "female", ]
#MLH1_by_M_strain <- MLH1_data[MLH1_data$sex == "male", ]

#MLH1_data <- MLH1_data %>%
#  arrange(category) %>%
#  mutate(Original.Name = factor(Original.Name)) #another category that you want the order to match

#MLH1_data <- with(MLH1_data, MLH1_data[order(category),])
AP_mouse_table.HQ$var <- as.numeric(AP_mouse_table.HQ$var)

AP_category_table <-  ddply(.data=AP_mouse_table.HQ,
                 .(strain,sex),
                 summarize,
                 sum.ncells = sum(Ncells),
                 mean.MLH1 = mean(mean_co),
                 var.MLH1 = var(mean_co),
                 AV.var_MLH1 = mean(var, na.rm=TRUE)
                                  
)

#there might be an error for changing wd
#write.table(AP_category_table, "~./MLH1repo/strain_table_variances.csv", sep=",", row.names = FALSE)

AP_mouse_table.HQ_target <- AP_mouse_table.HQ[!(is.na(AP_mouse_table.HQ$sex) | AP_mouse_table.HQ$sex==""), ]

remove_strains <- c("TOM","AST","CZECH","CAST","HMI","SPRET","CAROLI", "F1", "PERC")
AP_mouse_table.HQ_target <- AP_mouse_table.HQ_target[ ! AP_mouse_table.HQ_target$strain %in% remove_strains, ]

  
var.plot <- ggplot(AP_mouse_table.HQ_target, aes(y=var, x = sex, color=sex))+geom_jitter()+facet_wrap(~strain)+ggtitle("Plots of mouse Variances for MLH1 counts")+
  scale_color_manual(values=c("royalblue", "limegreen"))+theme_bw()

```

```{r misc.setup, echo=FALSE}

colors_of_strains <- c('WSB'= "#56B4E9",'G' ='cadetblue','LEW'= 'lightblue','PERC'= 'blue',
                       #blues
                       
                       'PWD'= 'red2',
                       'SKIVE'= 'coral1','KAZ'= 'coral4', 
                       
                       'TOM'= 'indianred1','AST' = 'tan2',
                       'CZECH'= 'brown1',
                       
                       'MSM'= 'deeppink', 'MOLF'= 'hotpink', #mol are pinks, bc they are fancy mice
                       
                    'CAST'='green','HMI'= 'forestgreen', #greens
                    
                    'SPRET'='gold','SPIC'= 'goldenrod1', #browns? color of mounds?
                    'CAROLI'='gold3',
                    
                    "F1"='grey', "other" = 'grey'
                    )

color_chrm_class <- c( '0' = "#00BFC4", '1' = "#7CAE00", '2' = "#F8766D", '3' = "#C77CFF", '4'= "#F564E3")
#4!!

#show_col(hue_pal()(5))

#add Rec group,

Curated_BivData$Rec.group <- ifelse(grepl("PWD male", Curated_BivData$category), 1, 
                              ifelse(grepl("MSM male", Curated_BivData$category), 1,
                                ifelse(grepl("SKIVE male", Curated_BivData$category), 1, 0)))

Curated_BivData$Rec.group_alt <- ifelse(grepl("PWD male", Curated_BivData$category), 1, 
                              ifelse(grepl("MSM male", Curated_BivData$category), 1, 0))


#apply this before ploting, to display strains in desired order
#MLH1_data <- MLH1_data %>%
#  arrange(strain, sex, category, mouse) %>%
#  mutate(Original.Name = factor(Original.Name)) #another category that you want the order to match
#MLH1_data <- with(MLH1_data, MLH1_data[order(sex, strain),])

# sort your dataframe, by the focal categories
#MLH1_data$mouse <- factor(MLH1_data$mouse, levels=unique(MLH1_data$mouse))


```

```{r MLH1.FEMALE.TABLE, echo=FALSE}
female_mouse.av <- subset(AP_mouse_table_w.Ages, select= -c(sex.y,sex.x,
                            subsp.y,subsp.x,
                    category.y,category.x, strain.x,
                    strain.y, mouse.1  )  )

female_mouse.av <- add_sex(female_mouse.av)
female_mouse.av <- add_strain(female_mouse.av)
female_mouse.av <- add_subsp(female_mouse.av)
female_mouse.av <- add_species(female_mouse.av)

female_mouse.avs_for.analysis <-  female_mouse.av %>% filter(sex== "female") %>% filter(species == "M.musculus") %>% filter(strain != "CZECH") %>% filter(strain != "CAST")

female_mouse.avs_for.analysis$subsp <- factor(female_mouse.avs_for.analysis$subsp, ordered = FALSE)
female_mouse.avs_for.analysis$sex <- factor(female_mouse.avs_for.analysis$sex, ordered = FALSE)
female_mouse.avs_for.analysis$strain <- factor(female_mouse.avs_for.analysis$strain, ordered = FALSE)
female_mouse.avs_for.analysis$species <- factor(female_mouse.avs_for.analysis$species, ordered = FALSE)

Musc.female_mouse.av <-  female_mouse.avs_for.analysis %>% filter(sex== "female") %>% filter(subsp == "Musc")

#Mol
#Mol.female_mouse.av <-  female_mouse.avs_for.analysis %>% filter(sex== "female") %>% filter(subsp == "Mol")

#Dom 
Dom.female_mouse.av <-  female_mouse.avs_for.analysis %>% filter(sex== "female") %>% filter(subsp == "Dom")

```

```{r MLH1.MALE.TABLE, echo=FALSE}
#AP_mouse_table.MLH1_HQ

#find the table that includes the las image data
male_MLH1mouse.av <- AP_mouse_table_w.Ages

male_MLH1mouse.av <- add_sex(male_MLH1mouse.av)
male_MLH1mouse.av <- add_strain(male_MLH1mouse.av)
male_MLH1mouse.av <- add_subsp(male_MLH1mouse.av)
male_MLH1mouse.av <- add_species(male_MLH1mouse.av)

fomo.male_mouse.av <-  male_MLH1mouse.av %>% filter(sex== "male") %>% filter(species == "M.musculus") 

fomo.male_mouse.av$subsp <- factor(fomo.male_mouse.av$subsp, ordered = FALSE)
fomo.male_mouse.av$sex <- factor(fomo.male_mouse.av$sex, ordered = FALSE)
fomo.male_mouse.av$strain <- factor(fomo.male_mouse.av$strain, ordered = FALSE)
fomo.male_mouse.av$species <- factor(fomo.male_mouse.av$species, ordered = FALSE)


Musc.male_mouse.av <-  fomo.male_mouse.av %>% filter(sex== "male") %>% filter(subsp == "Musc")
Mol.male_mouse.av <-  fomo.male_mouse.av %>% filter(sex== "male") %>% filter(subsp == "Mol")
#Dom version 
Dom.male_mouse.av <-  fomo.male_mouse.av %>% filter(sex== "male") %>% filter(subsp == "Dom")
```

```{r MLH1.MIXED.MODEL.setup, echo=FALSE, warning=FALSE, eval=TRUE}
#DF.HetC.MixedModel.HQ

#this blck is duplicated in setupData.rm

#https://rcompanion.org/handbook/G_03.html
#testing random effects within lme

#What are the main DFs for this code chunk
#AP_mouse_table_w.Ages   <- used to remove males which are too old (includes outgroups) 
#DF.HetC.MixedModel   <- only Mmusculus
#DF.HetC.MixedModel.HQ <- only mice of certain age, Q5 excluded

#variance is still chr
#AP_mouse_table_w.Ages$var <- as.numeric(AP_mouse_table_w.Ages$var)

#use the merged DF to filter by age 
#AP_mouse_table_w.Ages$strain <- AP_mouse_table_w.Ages$strain.x
#AP_mouse_table_w.Ages$sex <- AP_mouse_table_w.Ages$sex.x
#AP_mouse_table_w.Ages$subsp <- AP_mouse_table_w.Ages$subsp.x
#main file, make everything unordered factors
#AP_mouse_table_w.Ages <- add_species(AP_mouse_table_w.Ages)

DF.HetC.MixedModel <- AP_mouse_table_w.Ages[AP_mouse_table_w.Ages$species == "M.musculus",]

#remove the extra merged cols
#DF.HetC.MixedModel <- subset(DF.HetC.MixedModel, select= -c(sex.y,sex.x,
#                            subsp.y,subsp.x,
#                    category.y,category.x, strain.x,
#                    strain.y, mouse.1  )  )


#REMOVE EXTRA STRAINS
#DF.HetC.MixedModel <- DF.HetC.MixedModel[DF.HetC.MixedModel$subsp != "Cast",]
#remove musc strains
#DF.HetC.MixedModel <- DF.HetC.MixedModel[DF.HetC.MixedModel$strain != "AST",]
#DF.HetC.MixedModel <- DF.HetC.MixedModel[DF.HetC.MixedModel$strain != "TOM",]
#remove PERC, MOLF, CZECH, (no females)
#DF.HetC.MixedModel <- DF.HetC.MixedModel[DF.HetC.MixedModel$strain != "CZECH",]
#DF.HetC.MixedModel <- DF.HetC.MixedModel[DF.HetC.MixedModel$strain != "PERC",]

#DF.HetC.MixedModel <- droplevels(DF.HetC.MixedModel)#remove levels not used

#AP_mouse_table excluide Qual5 cells
DF.HetC.MixedModel.HQ <- AP_mouse_table.MLH1_HQ

DF.HetC.MixedModel.HQ <- add_species(DF.HetC.MixedModel.HQ)

#main df with variables coded as chr
DF.HetC.MixedModel.HQ$subsp <- factor(DF.HetC.MixedModel.HQ$subsp, ordered = FALSE)
DF.HetC.MixedModel.HQ$sex <- factor(DF.HetC.MixedModel.HQ$sex, ordered = FALSE)
DF.HetC.MixedModel.HQ$strain <- factor(DF.HetC.MixedModel.HQ$strain, ordered = FALSE)
#DF.HetC.MixedModel.HQ$species <- factor(DF.HetC.MixedModel.HQ$species, ordered = FALSE)

#kable(table(DF.HetC.MixedModel.HQ$sex, DF.HetC.MixedModel.HQ$strain))


all.cell.MM.HQ <- ddply(.data=DF.HetC.MixedModel.HQ,
                 .(sex, strain),
                 summarize,
                 all.cells = sum(Ncells)
)

```

```{r MLH1.HetC.rates, echo=FALSE, include=FALSE, warning=FALSE}
#calculate the initial HetC levels from best quality mice and cells (compare both levels (cell and mouse average levels))
#subset MLH1_DF, remove non passing mice (as diagnoised from plots)
#remove quality 5 cells

#outgroups
#SPIC
HetC.SPIC <- MLH1_data[(MLH1_data$strain == "SPIC")&(MLH1_data$quality < 5),]
#unique(HetC.SPIC$mouse)
#non-passing mice
SPIC.remove.mice <- c("30may18_SPIC_f1", "8jun18_SPIC_m2")
HetC.SPIC <- HetC.SPIC[ ! HetC.SPIC$mouse %in% SPIC.remove.mice, ]
#calq sex mean
SPIC.mean.MLH1.obs <- ddply(.data=HetC.SPIC, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                 mean.MLH1 = mean(adj_nMLH1.foci),
                 var.MLH1 = var(adj_nMLH1.foci)
)
SPIC.HETC.val <- SPIC.mean.MLH1.obs$mean.MLH1[1] / SPIC.mean.MLH1.obs$mean.MLH1[2]

#SPRET
HetC.SPRET <- MLH1_data[(MLH1_data$strain == "SPRET")&(MLH1_data$quality < 5),]
#unique(HetC.SPRET$mouse)
#non-passing mice
#SPRET.remove.mice <- c("", "")
#HetC.SPRET <- HetC.SPRET[ ! HetC.SPRET$mouse %in% SPRET.remove.mice, ]
#calq sex mean
SPRET.mean.MLH1.obs <- ddply(.data=HetC.SPRET, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                 mean.MLH1 = mean(adj_nMLH1.foci),
                 var.MLH1 = var(adj_nMLH1.foci)
)

SPRET.HETC.val <- SPRET.mean.MLH1.obs$mean.MLH1[1] / SPRET.mean.MLH1.obs$mean.MLH1[2]

#CAROLI
HetC.CAROLI <- MLH1_data[(MLH1_data$strain == "CAROLI")&(MLH1_data$quality < 5),]
#unique(HetC.CAROLI$mouse)
#non-passing mice
#SPRET.remove.mice <- c("", "")
#HetC.SPRET <- HetC.SPRET[ ! HetC.SPRET$mouse %in% SPRET.remove.mice, ]
#calq sex mean
CAROLI.mean.MLH1.obs <- ddply(.data=HetC.CAROLI, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                 mean.MLH1 = mean(adj_nMLH1.foci),
                 var.MLH1 = var(adj_nMLH1.foci)
)

#cast
HetC.CAST <- MLH1_data[(MLH1_data$strain == "CAST")&(MLH1_data$quality < 5),]
#unique(HetC.CAST$mouse)
#non-passing mice
#X.remove.mice <- c("", "")
#HetC.SPRET <- HetC.SPRET[ ! HetC.SPRET$mouse %in% SPRET.remove.mice, ]
#calq sex mean
CAST.mean.MLH1.obs <- ddply(.data=HetC.CAST, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                 mean.MLH1 = mean(adj_nMLH1.foci),
                 var.MLH1 = var(adj_nMLH1.foci)
)
CAST.HETC.val <- CAST.mean.MLH1.obs$mean.MLH1[1] / CAST.mean.MLH1.obs$mean.MLH1[2]

#HMI
HetC.HMI <- MLH1_data[(MLH1_data$strain == "HMI")&(MLH1_data$quality < 5),]
#unique(HetC.HMI$mouse)
#non-passing mice
#X.remove.mice <- c("", "")
#HetC.SPRET <- HetC.SPRET[ ! HetC.SPRET$mouse %in% SPRET.remove.mice, ]
#calq sex mean
HMI.mean.MLH1.obs <- ddply(.data=HetC.HMI, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                 mean.MLH1 = mean(adj_nMLH1.foci),
                 var.MLH1 = var(adj_nMLH1.foci)
)

#dom

#WSB
HetC.WSB <- MLH1_data[(MLH1_data$strain == "WSB")&(MLH1_data$quality < 5),]
#unique(HetC.WSB$mouse)
#non-passing mice
WSB.remove.mice <- c("8oct14_WSB_f1", "18nov17_WSB_f5", "16jun15_WSB_f3","10mar15_WSB_m2")
HetC.WSB <- HetC.WSB[ ! HetC.WSB$mouse %in% WSB.remove.mice, ]
#calq sex mean
WSB.mean.MLH1.obs <- ddply(.data=HetC.WSB, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                 mean.MLH1 = mean(adj_nMLH1.foci),
                 var.MLH1 = var(adj_nMLH1.foci)
)

WSB.HETC.val <- WSB.mean.MLH1.obs$mean.MLH1[1] / WSB.mean.MLH1.obs$mean.MLH1[2]


#LEW
HetC.LEW <- MLH1_data[(MLH1_data$strain == "LEW")&(MLH1_data$quality < 5),]
unique(HetC.LEW$mouse)
#non-passing mice
LEW.remove.mice <- c("4jan17_LEW_f2")
HetC.LEW <- HetC.WSB[ ! HetC.LEW$mouse %in% LEW.remove.mice, ]

#calq sex mean
LEW.mean.MLH1.obs <- ddply(.data=HetC.LEW, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                 mean.MLH1 = mean(adj_nMLH1.foci),
                 var.MLH1 = var(adj_nMLH1.foci)
)

LEW.HETC.val <- LEW.mean.MLH1.obs$mean.MLH1[1] / LEW.mean.MLH1.obs$mean.MLH1[2]


#G
HetC.G <- MLH1_data[(MLH1_data$strain == "G")&(MLH1_data$quality < 5),]
#unique(HetC.G$mouse)
#non-passing mice
G.remove.mice <- c("8jun15_G_f3","8jun15_G_f4")
HetC.G <- HetC.G[ ! HetC.G$mouse %in% G.remove.mice, ]

#calq sex mean
G.mean.MLH1.obs <- ddply(.data=HetC.G, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                 mean.MLH1 = mean(adj_nMLH1.foci),
                 var.MLH1 = var(adj_nMLH1.foci)
)
G.HETC.val <- G.mean.MLH1.obs$mean.MLH1[1] / G.mean.MLH1.obs$mean.MLH1[2]


#PERC
HetC.PERC <- MLH1_data[(MLH1_data$strain == "PERC")&(MLH1_data$quality < 5),]
#unique(HetC.PERC$mouse)
#non-passing mice
#G.remove.mice <- c("8jun15_G_f3","8jun15_G_f4")
#HetC.G <- HetC.PERC[ ! HetC.PERC$mouse %in% PERC.remove.mice, ]

#calq sex mean
PERC.mean.MLH1.obs <- ddply(.data=HetC.PERC, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                 mean.MLH1 = mean(adj_nMLH1.foci),
                 var.MLH1 = var(adj_nMLH1.foci)
)
G.HETC.val <- G.mean.MLH1.obs$mean.MLH1[1] / G.mean.MLH1.obs$mean.MLH1[2]

#MUSC

#PWD
HetC.PWD <- MLH1_data[(MLH1_data$strain == "PWD")&(MLH1_data$quality < 5),]
#unique(HetC.PWD$mouse)
#non-passing mice
PWD.remove.mice <- c("1apr15_PWD_f2","8oct14_PWD_f2","8oct14_PWD_f5")
HetC.PWD <- HetC.PWD[ ! HetC.PWD$mouse %in% PWD.remove.mice, ]

#calq sex mean
PWD.mean.MLH1.obs <- ddply(.data=HetC.PWD, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                 mean.MLH1 = mean(adj_nMLH1.foci),
                 var.MLH1 = var(adj_nMLH1.foci)
)
PWD.HETC.val <- PWD.mean.MLH1.obs$mean.MLH1[1] / PWD.mean.MLH1.obs$mean.MLH1[2]

# CZECH
HetC.CZECH <- MLH1_data[(MLH1_data$strain == "CZECH")&(MLH1_data$quality < 5),]
#unique(HetC.CZECH$mouse)

#non-passing mice
#PWD.remove.mice <- c("1apr15_PWD_f2","8oct14_PWD_f2","8oct14_PWD_f5")
#HetC.PWD <- HetC.PWD[ ! HetC.PWD$mouse %in% PWD.remove.mice, ]

#calq sex mean
CZECH.mean.MLH1.obs <- ddply(.data=HetC.CZECH, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                 mean.MLH1 = mean(adj_nMLH1.foci),
                 var.MLH1 = var(adj_nMLH1.foci)
)

CZECH.HETC.val <- CZECH.mean.MLH1.obs$mean.MLH1[1] / CZECH.mean.MLH1.obs$mean.MLH1[2]


# SKIVE
HetC.SKIVE <- MLH1_data[(MLH1_data$strain == "SKIVE")&(MLH1_data$quality < 5),]
#unique(HetC.SKIVE$mouse)

#non-passing mice
#PWD.remove.mice <- c("1apr15_PWD_f2","8oct14_PWD_f2","8oct14_PWD_f5")
#HetC.PWD <- HetC.PWD[ ! HetC.PWD$mouse %in% PWD.remove.mice, ]

#calq sex mean
SKIVE.mean.MLH1.obs <- ddply(.data=HetC.SKIVE, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                 mean.MLH1 = mean(adj_nMLH1.foci),
                 var.MLH1 = var(adj_nMLH1.foci)
)

SKIVE.HETC.val <- SKIVE.mean.MLH1.obs$mean.MLH1[1] / SKIVE.mean.MLH1.obs$mean.MLH1[2]



# KAZ
HetC.KAZ <- MLH1_data[(MLH1_data$strain == "KAZ")&(MLH1_data$quality < 5),]
#unique(HetC.KAZ$mouse)

#non-passing mice
#PWD.remove.mice <- c("1apr15_PWD_f2","8oct14_PWD_f2","8oct14_PWD_f5")
#HetC.PWD <- HetC.PWD[ ! HetC.PWD$mouse %in% PWD.remove.mice, ]

#calq sex mean
KAZ.mean.MLH1.obs <- ddply(.data=HetC.KAZ, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                 mean.MLH1 = mean(adj_nMLH1.foci),
                 var.MLH1 = var(adj_nMLH1.foci)
)
KAZ.HETC.val <- KAZ.mean.MLH1.obs$mean.MLH1[1] / KAZ.mean.MLH1.obs$mean.MLH1[2]


#TOM
HetC.TOM <- MLH1_data[(MLH1_data$strain == "TOM")&(MLH1_data$quality < 5),]
#unique(HetC.TOM$mouse)

#non-passing mice
#PWD.remove.mice <- c("1apr15_PWD_f2","8oct14_PWD_f2","8oct14_PWD_f5")
#HetC.PWD <- HetC.PWD[ ! HetC.PWD$mouse %in% PWD.remove.mice, ]

#calq sex mean
TOM.mean.MLH1.obs <- ddply(.data=HetC.TOM, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                 mean.MLH1 = mean(adj_nMLH1.foci),
                 var.MLH1 = var(adj_nMLH1.foci)
)
TOM.HETC.val <- TOM.mean.MLH1.obs$mean.MLH1[1] / TOM.mean.MLH1.obs$mean.MLH1[2]



#AST
HetC.AST <- MLH1_data[(MLH1_data$strain == "AST")&(MLH1_data$quality < 5),]
#unique(HetC.AST$mouse)

#non-passing mice
#PWD.remove.mice <- c("1apr15_PWD_f2","8oct14_PWD_f2","8oct14_PWD_f5")
#HetC.PWD <- HetC.PWD[ ! HetC.PWD$mouse %in% PWD.remove.mice, ]

#calq sex mean
AST.mean.MLH1.obs <- ddply(.data=HetC.AST, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                 mean.MLH1 = mean(adj_nMLH1.foci),
                 var.MLH1 = var(adj_nMLH1.foci)
)
AST.HETC.val <- AST.mean.MLH1.obs$mean.MLH1[1] / AST.mean.MLH1.obs$mean.MLH1[2]


#Molosinus

#MSM
HetC.MSM <- MLH1_data[(MLH1_data$strain == "MSM")&(MLH1_data$quality < 5),]
unique(HetC.MSM$mouse)

#non-passing mice
MSM.remove.mice <- c("12sep16_MSM_f3")
HetC.MSM <- HetC.MSM[ ! HetC.MSM$mouse %in% MSM.remove.mice, ]

#calq sex mean
MSM.mean.MLH1.obs <- ddply(.data=HetC.MSM, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                 mean.MLH1 = mean(adj_nMLH1.foci),
                 var.MLH1 = var(adj_nMLH1.foci)
)

MSM.HETC.val <- MSM.mean.MLH1.obs$mean.MLH1[1] / MSM.mean.MLH1.obs$mean.MLH1[2]


#MOLF
HetC.MOLF <- MLH1_data[(MLH1_data$strain == "MOLF")&(MLH1_data$quality < 5),]
unique(HetC.MOLF$mouse)

#non-passing mice
MOLF.remove.mice <- c("27oct18_MOLF_m3")
HetC.MOLF <- HetC.MOLF[ ! HetC.MOLF$mouse %in% MOLF.remove.mice, ]

#calq sex mean
MOLF.mean.MLH1.obs <- ddply(.data=HetC.MOLF, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                 mean.MLH1 = mean(adj_nMLH1.foci),
                 var.MLH1 = var(adj_nMLH1.foci)
)

MOLF.HETC.val <- MOLF.mean.MLH1.obs$mean.MLH1[1] / MOLF.mean.MLH1.obs$mean.MLH1[2]

```

```{r MLH1.HetC.plot, echo=FALSE, warning=FALSE}
#HETC plot
#change the main data? 

HetC_strain_table.M <- ddply(MLH1_data[(MLH1_data$quality < 5)&(MLH1_data$sex == "male"),], c("strain"), summarise,
                         Nmice = length(unique(mouse)),
                         Ncells  = length(adj_nMLH1.foci),
                         mean_co = format(round(  mean(adj_nMLH1.foci), 3 ), nsmall=3),
                         var = format(round(   var(nMLH1.foci),3), nsmall=3),
                         sd   = round(sd(adj_nMLH1.foci), 3),
                         se   = round(sd / sqrt(Ncells), 3),
                         cV = round( (as.numeric(sd) / as.numeric(mean_co) ),3)
)
#
#HetC_strain_table.M <- HetC_strain_table.M[ -c(7, 4, 10,11,12, 14, 17, 18,19), ]

HetC_strain_table.F <- ddply(MLH1_data[(MLH1_data$quality < 5)&(MLH1_data$sex == "female"),], c("strain"), summarise,
                         Nmice = length(unique(mouse)),
                         Ncells  = length(adj_nMLH1.foci),
                         mean_co = format(round(  mean(adj_nMLH1.foci), 3 ), nsmall=3),
                         var = format(round(   var(nMLH1.foci),3), nsmall=3),
                         sd   = round(sd(adj_nMLH1.foci), 3),
                         se   = round(sd / sqrt(Ncells), 3),
                         cV = round( (as.numeric(sd) / as.numeric(mean_co) ),3)
)
#HetC_strain_table.F <- HetC_strain_table.F[ -c(11), ]

#remove the 'other'

HetC_strain_table.M$sex <- "male"
HetC_strain_table.F$sex <- "female"

HetC_strain_table <- merge(HetC_strain_table.M, HetC_strain_table.F, by= "strain")

HetC_strain_table <- add_subsp(HetC_strain_table)

#HetC_strain_table$subsp <- c("Dom", "Dom", "Dom", "Musc", "Musc","Musc","Musc", "Cast", "Outgroup","Outgroup")

HetC_strain_table$mean_co.x <- as.numeric(HetC_strain_table$mean_co.x)
HetC_strain_table$mean_co.y <- as.numeric(HetC_strain_table$mean_co.y)
#str(HetC_strain_table)

Het.C.plot <- ggplot(data= HetC_strain_table, aes(x= mean_co.x, y=mean_co.y, color = subsp, label=strain ))+ geom_point(aes(size=2)) + xlim(c(20,35)) + ylim(c(20, 35)) + geom_abline(slope = 1, intercept = 0)+labs(x="male MLH1 mean",y="female MLH1 mean")+ggtitle("Heterochiasmy Plot")+
  scale_color_manual(values=c("royalblue", "limegreen",  "tomato", "slategray2", "plum", "grey")) + geom_text(data = HetC_strain_table, aes(label=HetC_strain_table$strain, x=mean_co.x, y= mean_co.y,  hjust = 0, nudge_x = 0.09, nudge_y = 0.09) )


```

```{r juvimice.list, echo=FALSE}
juvi_mice <- c(
  "30dec14_WSB_m2","30dec14_WSB_m4","30dec14_WSB_m1","30dec14_WSB_m3","30dec14_WSB_m5",
  "30sep14_G_m1","30sep14_G_m2","30sep14_G_m3","30sep14_G_m4","30sep14_G_m5",
  "12Feb17_LEW_m1","12Feb17_LEW_m2", "13sep17_LEW_m2", "13sep17_LEW_m1","13sep17_LEW_m3",
 
  "12sep17_PWD_m1","12sep17_PWD_m2","12sep17_PWD_m3",
   "30Dec16_MSM_m2","30Dec16_MSM_m3","30Dec16_MSM_m1"
  )

#REMOVE JUVI mice
#juvi.mouse_CurBiv_table <- mouse.avs_4MM[mouse.avs_4MM$mouse  %in% juvi_mice, ]  
#mouse.avs_4MM <- mouse.avs_4MM[! mouse.avs_4MM$mouse  %in% juvi_mice, ] 

```

```{r CurBiv.Mouse.av.TABLE, echo=FALSE}

#this block should be in the data set up
#the mouse averages will be used

#18jul19_31may19_SKIVE_m2_sp1_13.2_rev_8 is missing the 1CO position -- remove for now
cleanCurated_BivData <- cleanCurated_BivData %>% filter(fileName != "18jul19_31may19_SKIVE_m2_sp1_13.2_rev_8")

#cleanCurated_BivData
#removed the no hand foci count

mouse.avs_4MM <- ddply(cleanCurated_BivData, c("mouse"), summarise,
                        Nmice = length(unique(mouse)),
                        Ncells  = length(Obj.ID),
                       
                      # mean_CO = mean()
                       
                        mean_SC = as.numeric(format(round(  mean(chromosomeLength), 3 ), nsmall=3) ),
                        cV = cv(chromosomeLength),
                        var = as.numeric(format(round(   var(chromosomeLength),3), nsmall=3) ),
                        sd   = round(sd(chromosomeLength), 3),
                        se   = round(sd / sqrt(Ncells), 3),
                       
                       #calq mean.sc by chrm class
                       mean.SC_1CO = mean(chromosomeLength[which(hand.foci.count == 1)], na.rm = TRUE),
                       mean.SC_2CO = mean(chromosomeLength[which(hand.foci.count == 2)], na.rm = TRUE),
                       mean.SC_3CO = mean(chromosomeLength[which(hand.foci.count == 3)], na.rm = TRUE),
                       mean.SC_4CO = mean(chromosomeLength[which(hand.foci.count == 4)], na.rm = TRUE),
                       
                # IFD things
                #  mean_IFD_ABS = as.numeric(format(round(  mean(IFD1_ABS, na.rm = TRUE), 3 ), nsmall=3) ),
                # mean_IFD_PER = as.numeric(format(round(  mean(IFD1_PER, na.rm = TRUE), 3 ), nsmall=3) ),
                 
                 mean_IFD.2CO_ABS = mean(IFD1_ABS[which(hand.foci.count == 2)], na.rm = TRUE),
                 mean_IFD.2CO_PER = mean(IFD1_PER[which(hand.foci.count == 2)], na.rm = TRUE),
                  
                       #CO position things, requires the melt?
              norm.first.foci.pos =  mean(PER_Foci_1[which(hand.foci.count == 1)]),
                  
              #sis.co.ten!!
              mean.siscoten = mean(SisCoTen, na.rm=TRUE),
              
              mean.siscoten_1CO = mean(SisCoTen[which(hand.foci.count == 1)], na.rm = TRUE),
              mean.siscoten_2CO = mean(SisCoTen[which(hand.foci.count == 2)], na.rm = TRUE),
              mean.siscoten_3CO = mean(SisCoTen[which(hand.foci.count == 3)], na.rm = TRUE),
              
              mean.telo.dist = mean(telo_dist, na.rm=TRUE),
              mean.cent.dist = mean(dis.cent, na.rm=TRUE)
)

#vari
mouse.avs_4MM <- add_sex(mouse.avs_4MM)
mouse.avs_4MM <- add_strain(mouse.avs_4MM)
mouse.avs_4MM <- add_subsp(mouse.avs_4MM)
mouse.avs_4MM <- add_category(mouse.avs_4MM)

#unordered for the mixed models (move this to the MM blocks...?)
#keep order for the plots?
mouse.avs_4MM$subsp <- factor(mouse.avs_4MM$subsp, ordered = FALSE)
mouse.avs_4MM$sex <- factor(mouse.avs_4MM$sex, ordered = FALSE)
mouse.avs_4MM$strain <- factor(mouse.avs_4MM$strain, ordered = FALSE)

#MAKE the same version of the TABLE but for the longest_biv_Dataset (when I get around to making it)
#code if males are high or low?

#REMOVE JUVI mice
juvi.mouse_CurBiv_table <- mouse.avs_4MM[mouse.avs_4MM$mouse  %in% juvi_mice, ]  
mouse.avs_4MM <- mouse.avs_4MM[! mouse.avs_4MM$mouse  %in% juvi_mice, ]  

#assign high and low
#divide musc males into groups
mouse.avs_4MM$Rec.group <- ifelse(grepl("PWD male", mouse.avs_4MM$category), 1, 
                                ifelse(grepl("MSM male", mouse.avs_4MM$category), 1,
                                 ifelse(grepl("SKIVE male", mouse.avs_4MM$category), 1, 0)))


mouse.avs_4MM$Rec.group.exclu <- ifelse(grepl("PWD male", mouse.avs_4MM$category), 1, 
                                ifelse(grepl("MSM male", mouse.avs_4MM$category), 1, 0))

```

```{r Q1.biv.data_table.4MM.Male, echo=FALSE, eval=FALSE}

#add this to setup

#remove female, 
#add back in the BIVDATA from strains which were  removed for the sex-specific comparisons

#Curated_BivData_male

male.av_4MM <- ddply(cleanCurated_BivData, c("mouse"), summarise,
                        Nmice = length(unique(mouse)),
                        Ncells  = length(Obj.ID),
                       
                      # mean_CO = mean()
                       
                        mean_SC = as.numeric(format(round(  mean(chromosomeLength), 3 ), nsmall=3) ),
                        cV = cv(chromosomeLength),
                        var = as.numeric(format(round(   var(chromosomeLength),3), nsmall=3) ),
                        sd   = round(sd(chromosomeLength), 3),
                        se   = round(sd / sqrt(Ncells), 3),
                       
                       #calq mean.sc by chrm class
                       mean.SC_1CO = mean(chromosomeLength[which(hand.foci.count == 1)], na.rm = TRUE),
                       mean.SC_2CO = mean(chromosomeLength[which(hand.foci.count == 2)], na.rm = TRUE),
                       mean.SC_3CO = mean(chromosomeLength[which(hand.foci.count == 3)], na.rm = TRUE),
                       mean.SC_4CO = mean(chromosomeLength[which(hand.foci.count == 4)], na.rm = TRUE),
                       
                # IFD things
                #  mean_IFD_ABS = as.numeric(format(round(  mean(IFD1_ABS, na.rm = TRUE), 3 ), nsmall=3) ),
                # mean_IFD_PER = as.numeric(format(round(  mean(IFD1_PER, na.rm = TRUE), 3 ), nsmall=3) ),
                 
                 mean_IFD.2CO_ABS = mean(IFD1_ABS[which(hand.foci.count == 2)], na.rm = TRUE),
                 mean_IFD.2CO_PER = mean(IFD1_PER[which(hand.foci.count == 2)], na.rm = TRUE),
                  
                       #CO position things, requires the melt?
              norm.first.foci.pos =  mean(PER_Foci_1[which(hand.foci.count == 1)]),
                  
              #sis.co.ten!!
              mean.siscoten = mean(SisCoTen, na.rm=TRUE),
              
              mean.siscoten_1CO = mean(SisCoTen[which(hand.foci.count == 1)], na.rm = TRUE),
              mean.siscoten_2CO = mean(SisCoTen[which(hand.foci.count == 2)], na.rm = TRUE),
              mean.siscoten_3CO = mean(SisCoTen[which(hand.foci.count == 3)], na.rm = TRUE),
              
              mean.telo.dist = mean(telo_dist, na.rm=TRUE),
              mean.cent.dist = mean(dis.cent, na.rm=TRUE)
)

#vari
male.av_4MM <- add_sex(male.av_4MM)
male.av_4MM <- add_strain(male.av_4MM)
male.av_4MM <- add_subsp(male.av_4MM)
male.av_4MM <- add_category(male.av_4MM)

#unordered for the mixed models (move this to the MM blocks...?)
#keep order for the plots?
male.av_4MM$subsp <- factor(male.av_4MM$subsp, ordered = FALSE)
male.av_4MM$sex <- factor(male.av_4MM$sex, ordered = FALSE)
male.av_4MM$strain <- factor(male.av_4MM$strain, ordered = FALSE)

#REMOVE JUVI mice
juvi.from.male.biv_table <- male.av_4MM[male.av_4MM$mouse  %in% juvi_mice, ]  
mouse.avs_4MM <- male.av_4MM[! male.av_4MM$mouse  %in% juvi_mice, ] 

```

```{r totSC_setup, echo=FALSE}
#SC.skel
#"anon.batch" "rand_name"  "bin_size"   "skel_size" , all batches included

SC.skel.OG <- SC.skel #5683

#MERGEING MLH1 with SKEL
MLH1.merge.Skel <- merge(original_DF, SC.skel, by.x = "Random.Name", by.y = "rand_name", all= F)

MLH1.merge.Skel <- add_mouse(MLH1.merge.Skel)
MLH1.merge.Skel <- add_strain(MLH1.merge.Skel)
MLH1.merge.Skel <- add_sex(MLH1.merge.Skel)
MLH1.merge.Skel <- add_subsp(MLH1.merge.Skel)
MLH1.merge.Skel <- add_category(MLH1.merge.Skel)
MLH1.merge.Skel <- add_subsp(MLH1.merge.Skel)
#5110, 

OG.MLH1.merge.Skel <- MLH1.merge.Skel #raw version of DF

#check for DUPs, #check for duplicatations of the random name
MLH1.merge.Skel$ fileName  <- as.character(MLH1.merge.Skel$fileName)
tot.SC.dupies <- MLH1.merge.Skel[duplicated(MLH1.merge.Skel$ fileName ),]

MLH1.merge.Skel <- MLH1.merge.Skel %>% filter( X == "") %>% filter(REDO.crop == "no")

MLH1.merge.Skel$bin_size <- as.numeric(MLH1.merge.Skel$bin_size)
MLH1.merge.Skel$skel_size <- as.numeric(MLH1.merge.Skel$skel_size)
#table(MLH1.merge.Skel$category)

#remove outliers
MLH1.merge.Skel <- MLH1.merge.Skel  %>% filter(bin_size < 60000)

```


```{r totSC.viz4outliers, echo=FALSE}

# -- EXAMINE the ggplot scatter plots for outliers
tot.sc <- ggplot(data= MLH1.merge.Skel, aes(y = bin_size, x=strain))+ geom_jitter()+facet_wrap(~category, scales = "free")

#round 1
SC.skel.outliers_round1 <- c(
  #MLH1.merge.Skel  %>% filter(category == "WSB male") %>% filter(skel_size > 2000)
  "10apr15_31mar15_WSB_m3_sp1_4_rev.tif",
  
  #MLH1.merge.Skel  %>% filter(category == "WSB female") %>% filter(skel_size > 2700)       
            "17nov16_3nov15_WSB_f1_sp3_4_rev.tif","1may16_3nov15_WSB_f1_sp2_6_rev.tif",
                      
    #MLH1.merge.Skel  %>% filter(category == "G male") %>% filter(skel_size > 20000) 
                      "19oct15_12oct15_G_m1_sp1_17.1_rev.tif",
  
  #MLH1.merge.Skel  %>% filter(category == "G female") %>% filter(skel_size > 20000) 
              "1mar16_20feb16_G_f3_sp1_8_rev.tif","21nov16_16jan16_G_f2_sp1_15_rev.tif",
  
  #MLH1.merge.Skel  %>% filter(category == "LEW male") %>% filter(skel_size < 1000)          
          "6mar18_1feb18_LEW_m1_sp1_21.5_rev",
  
  #MLH1.merge.Skel  %>% filter(category == "LEW female") %>% filter(skel_size > 20000)   
          "12jan17_4jan17_LEW_f1_sp1_32_rev.tif","20mar17_13jan17_LEW_f2_sp1_12_rev.tif","9apr18_1feb18_LEW_f1_sp1_23_rev.tif","9apr18_1feb18_LEW_f1_sp1_12_rev.tif", 
                      
       #MLH1.merge.Skel  %>% filter(category == "PWD male") %>% filter(skel_size > 20000)  
                      "6mar15_1mar15_PWD_m1_sp4_6.1_rev.tif","7mar15_1mar15_PWD_m1_sp4_2_rev.tif","20jul15_18may15_PWD_m2_sp1_10.2_rev.tif",
           
   #MLH1.merge.Skel  %>% filter(category == "PWD female") %>% filter(skel_size > 20000)
   "6may15_17apr15_PWD_f1_sp1_7_rev",  
       
      #MLH1.merge.Skel  %>% filter(category == "KAZ male") %>% filter(skel_size < 1000)
            "6aug19_10jul19_KAZ_m1_sp1_7.1_rev.tif",
            
    # MLH1.merge.Skel  %>% filter(category == "KAZ female") %>% filter(skel_size > 30000)     
    "16feb18_1feb18_KAZ_f4_sp1_20_rev.tif",  "8feb18_1feb18_KAZ_f1_sp1_24_rev.tif",  
  "16feb18_1feb18_KAZ_f4_sp1_1_rev.tif",   "8feb18_1feb18_KAZ_f1_sp1_5_rev.tif",   
 "16feb18_1feb18_KAZ_f4_sp1_17_rev.tif",  "1mar18_1feb18_KAZ_f3_sp1_8_rev.tif",   
 "7feb18_11jan18_KAZ_f3_sp1_27_rev.tif",  "20apr18_11jan18_KAZ_f1_sp1_6_rev.tif", 
 "16feb18_1feb18_KAZ_f2_sp1_5.2_rev.tif", "16feb18_1feb18_KAZ_f2_sp1_15_rev.tif", 
"7feb18_11jan18_KAZ_f3_sp1_13_rev.tif",  "13apr18_1feb18_KAZ_f6_sp1_25_rev.tif" ,
            
  # MLH1.merge.Skel  %>% filter(category == "SKIVE female") %>% filter(skel_size > 10000)
    "10jun19_31may19_SKIVE_f6_sp1_47_rev.tif","10jun19_31may19_SKIVE_f6_sp1_7_rev.tif",

# MLH1.merge.Skel  %>% filter(category == "MOLF male") %>% filter(skel_size < 1200)
"4jun19_28sep18_MOLF_m1_sp1_10.1_rev.tif",

#MLH1.merge.Skel  %>% filter(category == "AST male") %>% filter(skel_size > 2100)
"16aug19_27aug18_AST_m1_sp1_23_rev.tif",

#MLH1.merge.Skel  %>% filter(category == "MSM female") %>% filter(skel_size > 20000)
"21nov17_16nov17_MSM_f3_sp1_24_rev.tif","21nov17_16nov17_MSM_f3_sp1_3_rev.tif", "21nov17_16nov17_MSM_f3_sp1_27_rev.tif","4dec17_16nov17_MSM_f2_sp1_10_rev.tif",
"5aug17_14jul17_MSM_f1_sp1_25_rev.tif","24jan18_31dec17_MSM_f3_sp1_3_rev.tif", 
"24jan18_31dec17_MSM_f3_sp1_22_rev.tif","21nov17_16nov17_MSM_f3_sp1_17_rev.tif",

#tt <- MLH1.merge.Skel  %>% filter(category == "SPIC female") %>% filter(skel_size > 20000)
"27dec17_4dec17_SPIC_f1_sp1_12_rev.tif",  "5apr18_31dec17_SPIC_f2_sp1_1_rev.tif",
 "26sep18_30may18_SPIC_f4_sp1_14_rev.tif", "4feb18_31dec17_SPIC_f1_sp1_21_rev.tif", 
 "5apr18_31dec17_SPIC_f2_sp1_22_rev.tif",  "27dec17_4dec17_SPIC_f1_sp1_40_rev.tif", 
 "4jul18_30may18_SPIC_f1_sp1_6_rev.tif",   "26sep18_30may18_SPIC_f4_sp1_6_rev.tif", 
 "29dec17_4dec17_SPIC_f2_sp1_10_rev.tif",  "5sep17_21aug17_SPI_f2_sp1_37_rev.tif" , 
 "5sep17_21aug17_SPI_f2_sp1_15_rev.tif",   "26sep18_30may18_SPIC_f4_sp1_24_rev.tif",
"29dec17_4dec17_SPIC_f2_sp1_16_rev.tif",  "5sep17_21aug17_SPI_f2_sp1_30_rev.tif",  
"4feb18_31dec17_SPIC_f1_sp1_5_rev.tif",   "5apr18_31dec17_SPIC_f2_sp1_25_rev.tif", 
 "26sep18_30may18_SPIC_f4_sp1_4_rev.tif",  "5sep17_21aug17_SPI_f2_sp1_22_rev.tif",  
 "5sep17_21aug17_SPI_f2_sp1_19_rev.tif",   "5sep17_21aug17_SPI_f2_sp1_29_rev.tif",  
 "26sep18_30may18_SPIC_f4_sp1_3_rev.tif",  "5sep17_21aug17_SPI_f2_sp1_5.2_rev.tif", 
 "29dec17_4dec17_SPIC_f2_sp1_6_rev.tif",   "13oct17_21aug17_SPI_f1_sp1_13_rev.tif"  )


MLH1.merge.Skel_2 <- MLH1.merge.Skel[!MLH1.merge.Skel$fileName %in% SC.skel.outliers_round1, ]#62 removed


remove.mice_round2 <- c(
    #MLH1.merge.Skel  %>% filter(category == "G male") %>% filter(skel_size > 2500)
    "28jul15_22jun15_G_m2_sp1_5_rev.tif","12may15_30sep14_G_m5_sp1_26_rev.tif",  
    
    #MLH1.merge.Skel  %>% filter(category == "G female") %>% filter(skel_size > 20000)
  "19oct15_12oct15_G_m1_sp1_17.1_rev.tif",
  
#MLH1.merge.Skel_2  %>% filter(category == "PWD male") %>% filter(skel_size > 2100)
"20jul15_18may15_PWD_m1_sp1_10_rev.tif","22mar15_10mar15_PWD_m2_sp1_11.2_rev.tif","22mar15_10mar15_PWD_m1_sp1_4.1_rev.tif","20jul15_18may15_PWD_m1_sp1_7.2_rev.tif",  

#MLH1.merge.Skel_2  %>% filter(category == "PWD female") %>% filter(skel_size > 20000)
  "6may15_17apr15_PWD_f1_sp1_7_rev.tif",
  
  #MLH1.merge.Skel_2  %>% filter(category == "KAZ female") %>% filter(skel_size < 1000)
  "8feb18_1feb18_KAZ_f1_sp1_19_rev.tif",

#MLH1.merge.Skel_2  %>% filter(category == "LEW male") %>% filter(skel_size < 1000 | skel_size > 1750)
"6mar18_1feb18_LEW_m1_sp1_10_rev.tif","3mar15_13jan17_LEW_m1_sp1_15_rev.tif", "4feb17_4jan17_LEW_m1_sp1_13_rev.tif","6mar18_1feb18_LEW_m1_sp1_21.5_rev.tif",

#MLH1.merge.Skel_2  %>% filter(category == "LEW female") %>% filter(skel_size > 2750 ) 
"20mar17_13jan17_LEW_f2_sp1_10_rev.tif",

# MLH1.merge.Skel_2  %>% filter(category == "CAROLI male") %>% filter(skel_size < 1400 )
"31aug18_1mar18_CAROLI_m1_sp3_29.2_rev.tif",

#MLH1.merge.Skel_2  %>% filter(category == "MSM male") %>% filter(skel_size < 1200 )
"16aug17_5jul17_MSM_m1_sp1_6.1_rev.tif",

#tt <- MLH1.merge.Skel_2  %>% filter(category == "MSM female") %>% filter(skel_size > 2850 ) 
"29nov17_16nov17_MSM_f1_sp1_10_rev.tif", "4dec17_16nov17_MSM_f2_sp1_24_rev.tif", "21nov17_16nov17_MSM_f3_sp1_26_rev.tif",
#SPIC male  x
#SPIC female  x
#TOM x
#AST x
#CZECH male x 

#MLH1.merge.Skel_2  %>% filter(category == "SPRET male") %>% filter(skel_size < 1900 ) 
 "24may17_8may17_SPRET_m1_sp1_21.1_rev.tif",
#SPRET female X
#MLH1.merge.Skel_2  %>% filter(category == "HMI male") %>% filter(skel_size < 1000 ) 
"29dec17_4dec17_HMI_m1_sp1_4.1_rev.tif"
)
#3remove.mice_round2


MLH1.merge.Skel <- MLH1.merge.Skel[! (MLH1.merge.Skel$fileName %in% SC.skel.outliers_round1),]
MLH1.merge.Skel <- MLH1.merge.Skel[! (MLH1.merge.Skel$fileName %in% remove.mice_round2),]

all.remove <- paste( c(SC.skel.outliers_round1, remove.mice_round2 ) )
```

```{r LONG.SHORT.biv_setup, echo=FALSE}

#FROM q1
#Q1.long.biv_1CO <-  real.long.bivData %>% filter(species =="M.musculus") %>% filter(hand.foci.count == 1) %>% filter(strain != "MOLF") %>% filter(strain != "CZECH")
#Q1.short.biv_1CO <-  real.short.bivData %>% filter(species =="M.musculus") %>% filter(hand.foci.count == 1)%>% filter(strain != "MOLF") %>% filter(strain != "CZECH")
#from Q2 IFD
#male.long.biv.2CO_IFD <- real.long.bivData %>% filter(sex== "male") %>% filter(species =="M.musculus") %>% filter(hand.foci.count == 2)
##long biv
#male.long.biv <- real.long.bivData %>% filter(sex== "male") %>% filter(species == "M.musculus") 
#table( droplevels( male.long.biv$strain)  )

#FROM q1 
REAL.short_mouse_table <- ddply(real.short.bivData, c("mouse"), summarise,
                        n_obs = length(Obj.ID),
                        ncells = length(unique(fileName)),
                         mean_SC = mean(chromosomeLength),
                        
                      mean.SC_1CO = mean(chromosomeLength[which(hand.foci.count == 1)], na.rm = TRUE),
                       mean.SC_2CO = mean(chromosomeLength[which(hand.foci.count == 2)], na.rm = TRUE),
                       mean.SC_3CO = mean(chromosomeLength[which(hand.foci.count == 3)], na.rm = TRUE),
                       mean.SC_4CO = mean(chromosomeLength[which(hand.foci.count == 4)], na.rm = TRUE),
                        
                      #make sure 1CO  
                       #IFD1_ABS[which(hand.foci.count == 2)], na.rm = TRUE
                    mean_F1pos_ABS =  mean(Foci1[which(hand.foci.count == 1)]),
                    mean_F1pos_PER =  mean(PER_Foci_1[which(hand.foci.count == 1)]),
                           
                      meanIFD2CO_raw = mean(IFD1_ABS[which(hand.foci.count == 2)], na.rm = TRUE),
                     meanIFD2CO_norm = mean(IFD1_PER[which(hand.foci.count == 2)], na.rm = TRUE)
                                            
                         )

REAL.short_mouse_table <- add_sex(REAL.short_mouse_table)
REAL.short_mouse_table <- add_strain(REAL.short_mouse_table)
REAL.short_mouse_table <- add_subsp(REAL.short_mouse_table)
REAL.short_mouse_table <- add_category(REAL.short_mouse_table)

REAL.short_mouse_table$sex <- factor(REAL.short_mouse_table$sex, ordered = FALSE)
REAL.short_mouse_table$strain <- factor(REAL.short_mouse_table$strain, ordered = FALSE)
REAL.short_mouse_table$subsp <- factor(REAL.short_mouse_table$subsp, ordered = FALSE)

#divide musc males into groups
REAL.short_mouse_table$Rec.group <- ifelse(grepl("PWD male", REAL.short_mouse_table$category), 1, 
                                  ifelse(grepl("MSM male", REAL.short_mouse_table$category), 1,
                                  ifelse(grepl("SKIVE male", REAL.short_mouse_table$category), 1,
                                           0)))

REAL.short_mouse_table$Rec.group.exclu <- ifelse(grepl("PWD male", REAL.short_mouse_table$category), 1, 
                                ifelse(grepl("MSM male", REAL.short_mouse_table$category), 1, 0))

###LONG BIV

REAL.long_mouse_table <- ddply(real.long.bivData, c("mouse"), summarise,
                       n_obs = length(Obj.ID),
                        ncells = length(unique(fileName)),
                         mean_SC = mean(chromosomeLength),
                        
                      mean.SC_1CO = mean(chromosomeLength[which(hand.foci.count == 1)], na.rm = TRUE),
                       mean.SC_2CO = mean(chromosomeLength[which(hand.foci.count == 2)], na.rm = TRUE),
                       mean.SC_3CO = mean(chromosomeLength[which(hand.foci.count == 3)], na.rm = TRUE),
                       mean.SC_4CO = mean(chromosomeLength[which(hand.foci.count == 4)], na.rm = TRUE),
                        
                      #make sure 1CO,  #IFD1_ABS[which(hand.foci.count == 2)], na.rm = TRUE
                    mean_F1pos_ABS =  mean(Foci1[which(hand.foci.count == 1)]),
                    mean_F1pos_PER =  mean(PER_Foci_1[which(hand.foci.count == 1)]),
                           
                      meanIFD2CO_raw = mean(IFD1_ABS[which(hand.foci.count == 2)], na.rm = TRUE),
                     meanIFD2CO_norm = mean(IFD1_PER[which(hand.foci.count == 2)], na.rm = TRUE)
                         )

REAL.long_mouse_table <- add_sex(REAL.long_mouse_table)
REAL.long_mouse_table <- add_strain(REAL.long_mouse_table)
REAL.long_mouse_table <- add_subsp(REAL.long_mouse_table)
REAL.long_mouse_table <- add_category(REAL.long_mouse_table)

REAL.long_mouse_table$sex <- factor(REAL.long_mouse_table$sex, ordered = FALSE)
REAL.long_mouse_table$strain <- factor(REAL.long_mouse_table$strain, ordered = FALSE)
REAL.long_mouse_table$subsp <- factor(REAL.long_mouse_table$subsp, ordered = FALSE)

#divide musc males into groups
REAL.long_mouse_table$Rec.group <- ifelse(grepl("PWD male", REAL.long_mouse_table$category), 1, 
                                  ifelse(grepl("MSM male", REAL.long_mouse_table$category), 1,
                                  ifelse(grepl("SKIVE male", REAL.long_mouse_table$category), 1,
                                           0)))

REAL.long_mouse_table$Rec.group.exclu <- ifelse(grepl("PWD male", REAL.long_mouse_table$category), 1, 
                                ifelse(grepl("MSM male", REAL.long_mouse_table$category), 1, 0))


#find which one 
```


<!-- 
two round of removing outliers were done. in total 84 skel_bin potential outlier observations were removed 
-->

```{r tot.skel.table, echo=FALSE}

totSC.mouse.av.table <- ddply(.data=MLH1.merge.Skel, 
                 .(mouse),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                # mean.MLH1 = mean(adj_nMLH1.foci, na.rm = TRUE),
                # var.MLH1 = var(adj_nMLH1.foci, na.rm = TRUE),
                 
                 mean.skel = mean(skel_size, na.rm = TRUE),
                #se for skel
                
                 mean.bin =  mean(bin_size, na.rm = TRUE)
                 
)

totSC.mouse.av.table <- add_strain(totSC.mouse.av.table)
totSC.mouse.av.table <- add_sex(totSC.mouse.av.table)
totSC.mouse.av.table <- add_category(totSC.mouse.av.table)
totSC.mouse.av.table <- add_subsp(totSC.mouse.av.table)
totSC.mouse.av.table <- add_species(totSC.mouse.av.table)

#unorder the factors
totSC.mouse.av.table$subsp <- factor(totSC.mouse.av.table$subsp, ordered = FALSE)
totSC.mouse.av.table$sex <- factor(totSC.mouse.av.table$sex, ordered = FALSE)
totSC.mouse.av.table$strain <- factor(totSC.mouse.av.table$strain, ordered = FALSE)
totSC.mouse.av.table$species <- factor(totSC.mouse.av.table$species, ordered = FALSE)

#add rec group
totSC.mouse.av.table$Rec.group <- ifelse(grepl("PWD male", totSC.mouse.av.table$category), 1, 
                              ifelse(grepl("MSM male", totSC.mouse.av.table$category), 1,
                                ifelse(grepl("SKIVE male", totSC.mouse.av.table$category), 1, 0)))

OG.totSC.mouse.av.table <- totSC.mouse.av.table

#REMOVE JUVI mice
tot.SC_juvi.male <- totSC.mouse.av.table[totSC.mouse.av.table$mouse  %in% juvi_mice, ]

totSC.mouse.av.table <- totSC.mouse.av.table[! totSC.mouse.av.table$mouse  %in% juvi_mice, ] 

totSC.mouse.av.table <- totSC.mouse.av.table %>% filter(strain!="CAST" & strain !="F1" & strain !="SPIC" & strain !=  "CAROLI" & strain !="SPRET"& strain !="HMI"& strain !="PERC")

mouse.av_sc_female <- ggplot(data= OG.totSC.mouse.av.table[OG.totSC.mouse.av.table$sex == "female",], aes(y = mean.skel, x=mouse, color=strain))+ geom_point()+scale_color_manual(values=colors_of_strains)+
  facet_wrap(Rec.group~subsp, scales="free")+ theme(axis.title.x=element_blank(),        axis.text.x=element_blank(), axis.ticks.x=element_blank()) + ylim(c(1500,3000 ))+ggtitle("Female Mouse av. total SC area")

mouse.av_sc_male <- ggplot(data= OG.totSC.mouse.av.table[OG.totSC.mouse.av.table$sex == "male",], aes(y = mean.skel, x=mouse, color=strain))+ geom_point()+scale_color_manual(values=colors_of_strains)+
  facet_wrap(Rec.group~subsp, scales="free")+ theme(axis.title.x=element_blank(),        axis.text.x=element_blank(), axis.ticks.x=element_blank()) + ylim(c(1200,1850 ))+ggtitle("Mouse av. total SC area")

```

```{r, tot.SC.setup, echo = FALSE}

totSC.strain.av.table <- ddply(.data=totSC.mouse.av.table, 
                 .(sex,strain),
                 summarize, 
                 ncells = sum(ncells),
                 nmice = length(unique(mouse)),
                # mean.MLH1 = mean(adj_nMLH1.foci, na.rm = TRUE),
                # var.MLH1 = var(adj_nMLH1.foci, na.rm = TRUE),
                 
                 mean.skel = mean(mean.skel, na.rm = TRUE),
                 mean.bin =  mean(mean.bin, na.rm = TRUE)
                 
)

totSC.strain.av.table <- add_subsp(totSC.strain.av.table)

```

<!--
From this figure of the male mouse average of total sc measures, Confirm the general pattern of more SC area for the higher rec strains, compared to the low. But there are some exceptions.


the low rec musc strains which are higher are czech (1) and ast (2) -- the PWD mouse which is lower 24sep14_PWD_m1

INSERT TABLE FOR THE TOTAL SC means ect

The difference between 

bin is short for binary, it reports the binarized area of the SC

skel -- takes the skeletonized midline of the bin objects
-->
# Main Figure

```{r table.for.plot, echo=FALSE, warning=FALSE}
#why do I add in a different table?

#this move this to the setup file
#MLH1_data  %>% filter(quality < 5)  -- low quality cells removed

#pooled
strain.table.HQ_Fig.og <- ddply(MLH1_data  %>% filter(quality < 5), c("subsp","strain", "sex"), summarise,
                        Nmice =length(unique(mouse)),
                        Ncells  =length(nMLH1.foci),
                        mean_co =as.numeric(format(round(mean(nMLH1.foci), 3 ), nsmall=3) ),
                        cV = cv(nMLH1.foci, na.rm = TRUE),
                        var = as.numeric(format(round(   var(nMLH1.foci, na.rm = TRUE),3), nsmall=3)), 
                        sd   = round(sd(nMLH1.foci, na.rm = TRUE), 3),
                       #Standard error (mice vs cells)
                        se  = round(sd / sqrt(Ncells), 3)
                       
                       #range_mouse_means = range()
                   #consider adding crhm proportions?
    )


strain.table.HQ_Fig <-  strain.table.HQ_Fig.og %>% filter(strain != "F1" & strain != "PERC" &
                       strain != "TOM" &  strain != "AST" & strain != "CZECH" & strain != "CAROLI" & strain != "HMI")

#adjust order of plot here
strain.table.HQ_Fig$strain <- factor(strain.table.HQ_Fig$strain,levels =c(
     "MSM", "MOLF",   
  "PWD", "SKIVE",  "KAZ",
  "WSB","G","LEW","PERC",
                     "CAST",
                         "SPRET","SPIC" ), order=T )

```


```{r Fig1, echo=FALSE, warning=FALSE}

#might need to nudge the X values

Fig1_v1_strain <- ggplot(strain.table.HQ_Fig, aes(y=mean_co, x=strain, color=strain, shape=sex))+ 
  ylab("Average MLH1 Foci per Cell")+

  geom_point(aes(shape = sex), size=3, position=position_dodge(.5))+

    geom_errorbar(aes(ymin=mean_co-(se*2), ymax=mean_co+(se*2) ), width=.3, position = position_dodge(.5), size=1.001)+
  
  ylim(c(18, 35))+
 scale_color_manual(values=colors_of_strains)+ 

    theme(legend.position="none",
  panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
  panel.background = element_blank(), axis.line=element_line(colour = "black"),
  axis.title.x=element_blank(),axis.text.x=element_blank(),axis.ticks.x=element_blank())
#remove y labfor male 
 
   #segments
#  annotate("segment", x=.75, xend=4.5, y=19.5, yend=19.5, colour="black", size=1)+
  #Musc
#  annotate("segment", x= 5.5, xend=10.2, y=19.5, yend=19.5, colour="black", size=1)+
  #mol
#  annotate("segment", x=11, xend=12.25, y=19.5, yend=19.5, colour="black", size=1)+
  #cast
#  annotate("segment", x=13, xend=14, y=19.5, yend=19.5, colour="black", size=1)+
  #outgroups
 # annotate("segment", x=14.5, xend=16, y=19.5, yend=19.5, colour="black", size=1)+
  #label

 #   annotate("text", x = c(2,4.4,6,8, 10), #list of all x points 
   #        y = c(18.5,18.5,33,18.5, 18.5),
           
  # label = c("M. m. domesticus","M. m. musculus", "M. m. molossinus", "M. m. castaneus", "Outgroups"),
    #       fontface = 'italic', hjust = 0.5, size=3.5)



```


```{r female.show.plots, echo=FALSE, fig.width = 4, fig.height=2.5, warning=FALSE}

female_mouse.av <- subset(AP_mouse_table_w.Ages, select= -c(sex.y,sex.x,
                            subsp.y,subsp.x,
                    category.y,category.x, strain.x,
                    strain.y, mouse.1  )  )

female_mouse.av <- add_sex(female_mouse.av)
female_mouse.av <- add_strain(female_mouse.av)
female_mouse.av <- add_subsp(female_mouse.av)
female_mouse.av <- add_species(female_mouse.av)

female_mouse.avs_for.analysis <-  female_mouse.av %>% filter(sex== "female") %>% filter(species == "M.musculus") 

female_mouse.avs_for.analysis$subsp <- factor(female_mouse.avs_for.analysis$subsp, ordered = FALSE)
female_mouse.avs_for.analysis$sex <- factor(female_mouse.avs_for.analysis$sex, ordered = FALSE)
female_mouse.avs_for.analysis$strain <- factor(female_mouse.avs_for.analysis$strain, ordered = FALSE)
female_mouse.avs_for.analysis$species <- factor(female_mouse.avs_for.analysis$species, ordered = FALSE)

#table( droplevels(female_mouse.avs_for.analysis$strain) )
#there are cast strains too


Musc.female_mouse.av <-  female_mouse.avs_for.analysis %>% filter(sex== "female") %>% filter(subsp == "Musc")

female.specific.plot <- ggplot(female_mouse.avs_for.analysis, aes(x = as.factor(strain), y = mean_co, color= strain))+ geom_boxplot()+  geom_jitter(aes( alpha=0.5), width = .2 )+ ggtitle("Female mouse av")+scale_color_manual(values=colors_of_strains)+ylim(c(20,34))+
  theme(legend.position="none",
  panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
  panel.background = element_blank(), axis.line=element_line(colour = "black"))+theme(axis.text.x = element_text(angle = 45, hjust = 1))

#female.specific.plot


```


```{r male.show.plots, echo=FALSE, fig.width = 4, fig.height=2.5, warning=FALSE, include=FALSE}
#try customize the plot size

male_mouse.av <- subset(AP_mouse_table_w.Ages, select= -c(sex.y,sex.x,
                            subsp.y,subsp.x,
                    category.y,category.x, strain.x,
                    strain.y, mouse.1  )  )

male_mouse.av <- add_sex(male_mouse.av)
male_mouse.av <- add_strain(male_mouse.av)
male_mouse.av <- add_subsp(male_mouse.av)
male_mouse.av <- add_species(male_mouse.av)

#just.male_mouse.av <-  male_mouse.av %>% filter(sex== "male") %>% #filter(species == "M.musculus") 


just.male_mouse.av <-  fomo.male_mouse.av %>% filter(sex== "male") %>% filter(species == "M.musculus") 

just.male_mouse.av$sex <- factor(just.male_mouse.av$sex, ordered = FALSE)
just.male_mouse.av$strain <- factor(just.male_mouse.av$strain, ordered = FALSE)
just.male_mouse.av$subsp <- factor(just.male_mouse.av$subsp, ordered = FALSE)
just.male_mouse.av$species <- factor(just.male_mouse.av$species, ordered = FALSE)

#table( droplevels(just.male_mouse.av$strain) )
#there are cast strains too

Musc.male_mouse.av <-  just.male_mouse.av %>% filter(sex== "male") %>% filter(subsp == "Musc")

#fomo.male_mouse.av

male.plot <- ggplot(just.male_mouse.av, aes(x = as.factor(strain), y = mean_co, color= strain))+geom_jitter( alpha=0.5, width = .2) +geom_boxplot()+ ggtitle("Male mouse mean")+scale_color_manual(values=colors_of_strains)+ylim(c(20,34))+theme(legend.position="none",
  panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
  panel.background = element_blank(), axis.line=element_line(colour = "black"))+theme(axis.text.x = element_text(angle = 45, hjust = 1))
male.plot

musc.male.plot <- ggplot(Musc.male_mouse.av, aes(color= strain))+geom_jitter(aes(x = as.factor(strain), y = mean_co, color= strain, alpha=0.5), width = .2 )+ ggtitle("Musc male of MLH1 distributions")+scale_color_manual(values=colors_of_strains)+ylim(c(20,34))

```


```{r three.plots, echo=FALSE, warning=FALSE}
#merge the 3 plots
#try patchwork

#Fig1_v1_strain + female.specific.plot + male.plot

#Fig1_v1_strain + female.specific.plot - male.plot + plot_layout(ncol=1)
Fig1_v1_strain / (female.specific.plot + male.plot )


#(p1 | p2 | p3)/p4

```

This is a draft of the main figure. Outgroups are not included in the analysis below. (add horizontal lines for denoting subspecies, group means and expected minimum)

<!--  insert Details of data set cleaning etc is in SetupData.rmd
Effects from different factors are reviewed in Effects.Rmd -->


```{r main.fig1s, echo=FALSE, include=FALSE}
strain.table.HQ_Fig_F <-  strain.table.HQ_Fig %>% filter(sex == "female") %>% filter(strain != "CAST" & strain != "SPRET" & strain != "SPIC" )

Fig1_Female <- ggplot(strain.table.HQ_Fig_F, aes(y=mean_co, x=strain, color=strain, shape=sex))+ 
  ylab("Average MLH1 Foci per Cell")+

  geom_point( size=3, position=position_dodge(.5))+

    geom_errorbar(aes(ymin=mean_co-(se*2), ymax=mean_co+(se*2) ), width=.3, position = position_dodge(.5), size=1.001)+
  
  ylim(c(18, 35))+
 scale_color_manual(values=colors_of_strains)+ 

    theme(legend.position="none",
  panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
  panel.background = element_blank(), axis.line=element_line(colour = "black"),
  axis.title.x=element_blank(),axis.text.x=element_blank(),axis.ticks.x=element_blank())


#strain.table.HQ_Fig.og

strain.table.HQ_Fig_M <-  strain.table.HQ_Fig.og %>% filter(sex == "male") %>% filter(strain != "CAST" & strain != "SPRET" & strain != "SPIC" & strain != "PERC" & strain != "F1" & strain != "HMI" & strain != "CAROLI")

strain.table.HQ_Fig_M$strain <- factor(strain.table.HQ_Fig_M$strain,levels =c(
     "MSM", "MOLF",   
  "PWD", "SKIVE",  "KAZ", "TOM", "AST","CZECH",
  "WSB","G","LEW"), order=T )


Fig1_Male <- ggplot( strain.table.HQ_Fig_M, aes(y=mean_co, x=strain, color=strain, shape=sex))+ 
  ylab("Average MLH1 Foci per Cell")+

  geom_point(size=3, position=position_dodge(.5))+

    geom_errorbar(aes(ymin=mean_co-(se*2), ymax=mean_co+(se*2) ), width=.3, position = position_dodge(.5), size=1.001)+
  
  ylim(c(18, 35))+
 scale_color_manual(values=colors_of_strains)+ 

    theme(legend.position="none",
  panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
  panel.background = element_blank(), axis.line=element_line(colour = "black"),
  axis.title.x=element_blank(),axis.text.x=element_blank(),axis.ticks.x=element_blank())

```

```{r basic.numbers, echo=FALSE, eval=FALSE}

#kable(table(AP_mouse_table.HQ$category), col.names = c("Category", "mice"), label = "Number of mice by category" )


#kable(table(MLH1_data$category), col.names = c("Category","Cells"), label = "Number of Cells by category" )

```

```{r rand.misc, echo=FALSE, include=FALSE, eval=FALSE}
#str(AP_mouse_table_w.Ages)

AP_mouse_table.HQ$var <- as.numeric(AP_mouse_table.HQ$var)

#remove sex.x == na, and strain.x=na or F1
AP_mouse_table.HQ <- AP_mouse_table.HQ[!(is.na(AP_mouse_table.HQ$sex) | AP_mouse_table.HQ$sex==""), ]

AP_mouse_table.HQ <- AP_mouse_table.HQ[!(is.na(AP_mouse_table.HQ$var) | AP_mouse_table.HQ$var==""), ]
#plot variance and mean (facet sex)

AP_mouse_table_w.Ages <- AP_mouse_table_w.Ages[!( AP_mouse_table_w.Ages$strain.x =="F1"), ]

VandMean.mouse <- ggplot(aes(x=mean_co, y = var), data=AP_mouse_table.HQ )+geom_point(aes(color=strain))+facet_wrap(~sex)+ggtitle("mouse MLH1 averages; variance by mean")

cVandMean.mouse <- ggplot(aes(x=mean_co, y = cV), data=AP_mouse_table.HQ )+geom_point(aes(color=strain))+facet_wrap(~sex)+ggtitle("mouse MLH1 averages; cV by means")

str(AP_mouse_table.HQ)

cor(AP_mouse_table.HQ$mean_co[AP_mouse_table.HQ$sex == "female"], AP_mouse_table.HQ$var[AP_mouse_table.HQ$sex == "female"])
#[1] 0.5750399
cor(AP_mouse_table.HQ$mean_co[AP_mouse_table.HQ$sex == "male"], AP_mouse_table.HQ$var[AP_mouse_table.HQ$sex == "male"])
#0.321601

cor(AP_mouse_table.HQ$mean_co[AP_mouse_table.HQ$sex == "female"], AP_mouse_table.HQ$cV[AP_mouse_table.HQ$sex == "female"])
# 0.3515544
cor(AP_mouse_table.HQ$mean_co[AP_mouse_table.HQ$sex == "male"], AP_mouse_table.HQ$cV[AP_mouse_table.HQ$sex == "male"])
#-0.03713536

cor.test(AP_mouse_table.HQ$mean_co[AP_mouse_table.HQ$sex == "male"], 
    AP_mouse_table.HQ$var[AP_mouse_table.HQ$sex == "male"] )
# p-value = 8.623e-09

cor.test(AP_mouse_table.HQ$mean_co[AP_mouse_table.HQ$sex == "male"], 
    AP_mouse_table.HQ$cV[AP_mouse_table.HQ$sex == "male"] )


#remove mice with fewer than 10 cells

AP_mouse_table.HQ.10plus <- AP_mouse_table.HQ[AP_mouse_table.HQ$Ncells >= 10,]

VandMean.mouse.10plus <- ggplot(aes(x=mean_co, y = var), data=AP_mouse_table.HQ.10plus )+geom_point(aes(color=strain))+facet_wrap(~sex)+ggtitle("10 plus mouse MLH1 averages; variance by mean")



cor(AP_mouse_table.HQ.10plus$mean_co[AP_mouse_table.HQ.10plus$sex == "female"], AP_mouse_table.HQ.10plus$cV[AP_mouse_table.HQ.10plus$sex == "female"])


cor.test(AP_mouse_table.HQ.10plus$mean_co[AP_mouse_table.HQ.10plus$sex == "female"], 
    AP_mouse_table.HQ.10plus$var[AP_mouse_table.HQ.10plus$sex == "female"] )



#why did I male just a female table?
female.musc.cells <- MLH1_data[( MLH1_data$sex == "female") & (MLH1_data$subsp == "Musc" ), ]

female.musc.mice.table <- ddply(female.musc.cells, c("sex"), summarise,
                        Nmice = length(unique(mouse)),
                        Ncells  = length(adj_nMLH1.foci),
                        mean_co = as.numeric(format(round(  mean(adj_nMLH1.foci), 3 ), nsmall=3) ),
                        cV = cv(adj_nMLH1.foci),
                        var = format(round(   var(adj_nMLH1.foci),3), nsmall=3),
                        sd   = round(sd(adj_nMLH1.foci), 3),
                        se   = round(sd / sqrt(Ncells), 3)
                        #quality?
)
```

# Genome wide recombiantion rate estimates for both sexes

```{r MLH1.summ.stats, echo=FALSE}

#strain.table.HQ_Fig.og  all strains
#strain.table.HQ_Fig  extra strains removed

main.table <- strain.table.HQ_Fig.og  %>% filter(strain != "F1")

main.table <- add_subsp(main.table)
main.table <- add_species(main.table)

#main.table %>% filter(sex == "female")
#main.table %>% filter(sex == "male")

#G.HETC.val #1.2
#LEW.HETC.val  #1
#WSB.HETC.val  #1
#KAZ.HETC.val  #1.1

#SKIVE.HETC.val # 0.96
#PWD.HETC.val  #0.88
#MSM.HETC.val #0.9

#kable(head(main.table),  caption = "Table 1. MLH1 counts per cell summary statistics by strain")

#strain.table.HQ_Fig

#sum( (main.table  %>% filter(sex == "male"))$Ncells)
#the kable version is bulkier than the other versaion

```


We used counts of MLH1 foci per cell to estimate the genome wide recombination rates across our panel of `r length(unique(  (main.table  %>% filter(species == "M.musculus") )$strain) )` wild derived inbred strains sampled from three subspecies within the house mouse species complex;  _M. m. domesticus,_ _M. m. musculus_. and _M. m. molossinus_. Mean MLH1 foci counts were quantified from a total of `r sum( (main.table %>% filter(species == "M.musculus") %>% filter(sex == "male"))$Ncells)` spermatocytes and `r sum( (main.table %>% filter(species == "M.musculus") %>% filter(sex == "female"))$Ncells)` oocytes. <!-- state the average number of mice
The sampling of prophase oocytes is diffecult and was the limiting factor thus there are some strains with only MLH1 counts from spermatocytes, which are included for sex specific analysises. -->

<!--
1. Fig1A  magnitude and direction of heterochiasmy;
-general patterns i) female higher and ii) low degree of hetC  (range of 1 to 1.2) (human is 1.6)
-All strain means are above the expected minimum of 20 (1CO per chromosome)
-there are exceptions they are followed up in the sex-specific plots  (male higher)

2. Fig1B and C mouse means and sex specific patterns; 
-female mouse mean distributions are have more spread across the range of means
-male distributions are more clustered, low and high (~30% evolution largest-rapid evolution of gwRR)
-sex-specific evolution, both sexes don't show equivalent elevation of gwRR
-Dom vs Molossinus and Musculus -->

The general patterns of heterochiasmy in house mouse are displayed in Figure 1A. Taking note of the direction and the magnitude of the sex differences, our results confirm two general patterns: i) that the genome wide recombination rates averages are greater females compared to males (female biaded heterochiasmy) and ii) the degree of heterochiasmy (F:M rates) is generally low in these house mouse strains, ranging from `r G.HETC.val` in _domesticus^G^_ to `r LEW.HETC.val` in _domesticus^LEW^_. Three notable exceptions of male biased heterochiasmy are from the strains _musculus^PWD^_, _molossinus^MSM^_ and _musculus^SKIVE^_, with heterochiasmy values of `r PWD.HETC.val`, `r MSM.HETC.val` and `r SKIVE.HETC.val` respectively.

Examining the mouse means of MLH1 foci per cell seperately for each sex reveal disctinct patterns of variation (Figure1 B and C). The female means can be discribed as more evenly distributed around the sex-wide mean of approximately 25 MLH1 foci per cell (Figure1 B). While in males, the strain specific means seperate more clearly into two groups of high genome wide recombination rates near 30 MLH1 foci per cell and low genome wide recombiantion rates, near 23 MLH1 foci per cell (Figure1 C).

<illustrate sex differences female mean `r mean((main.table %>% filter(sex=="female") )$mean_co )`>

<This striking differences across sexes suggest that the genome wide recombination rates have distinct evolutionary trajectories in males and females of the same species. The pattern of strain averages distributed around the total sample mean in females is supportive of stabilizing selection or neutral evolution. In contrast to the pattern from male means which have rapid evolution in a subset of genetic backgrounds this fits a model of directional selection. (We attempt to make better evolutionary models by applying an evolutionary framework)>

```{r female.strain.table, echo=FALSE}
female_mouse.avs_for.analysis <- female_mouse.avs_for.analysis %>% filter(strain != "CAST")
#female strain table
#female_mouse.avs_for.analysis
F.strain.table <- ddply(female_mouse.avs_for.analysis, c("strain"), summarise,
                       
                        Nmice = length(unique(mouse)),
                        
                        total.cells  = sum(Ncells, na.rm = TRUE),
                        
                        mouse.mean_co = mean(mean_co, na.rm=TRUE),
                       
                        mean.mouse_cV = mean(cV, na.rm = TRUE),
                        
                        mean.mouse_var = mean(var, na.rm = TRUE ), 
                        
                        mean.mouse_sd   = mean(sd, na.rm = TRUE ),
                       
                        #Standard error (mice vs cells)
                        mean.mouse_se  = mean(se, na.rm=TRUE)
    )
#F.strain.table

#F.strain.table

# I don't know why most of the table is blank
```

```{r female.dif, echo=FALSE}

#how much greater are the high strains?
G_female <-  female_mouse.avs_for.analysis %>% filter(strain == "G")
mean.G.female <- mean(G_female$mean_co, na.rm=TRUE)

LEW_female <-  female_mouse.avs_for.analysis %>% filter(strain == "LEW")
mean.LEW.female <- mean(LEW_female$mean_co, na.rm=TRUE)

#how much greater is MSM female
MSM_female <-  female_mouse.avs_for.analysis %>% filter(strain == "MSM")
mean.MSM.female <- mean(MSM_female$mean_co, na.rm=TRUE)


sans_high_female <- female_mouse.avs_for.analysis %>% filter(strain != "G") %>%  filter(strain != "MSM") %>%  filter(strain != "LEW")
mean.all.other <- mean(sans_high_female$mean_co, na.rm=TRUE)


G.female.diff <- (mean.G.female / mean.all.other)
LEW.female.diff <- (mean.LEW.female / mean.all.other)
MSM.female.diff <- (mean.MSM.female / mean.all.other)
```

```{r mouse.var.plots.female, echo=FALSE, fig.width = 4, fig.height=2.5, include=FALSE, eval=FALSE}
Musc.female_mouse.av <-  female_mouse.avs_for.analysis %>% filter(sex== "female") %>% filter(subsp == "Musc")

#variance plots
musc.female.VAR.plot <- ggplot(Musc.female_mouse.av, aes(color= strain))+geom_jitter(aes(x = as.factor(strain), y =var, color= strain, alpha=0.5), width = .2 )+ ggtitle("Musc female of within mouse MLH1 distributions")+scale_color_manual(values=colors_of_strains)
musc.female.VAR.plot

#Mol.female.VAR.plot <- ggplot(Mol.female_mouse.av, aes(color= strain))+geom_jitter(aes(x = as.factor(strain), y = var, color= strain, alpha=0.5), width = .2 )+ ggtitle("Mol female of within mouse MLH1 distributions")+scale_color_manual(values=colors_of_strains)
#Mol.female.VAR.plot

Dom.female.VAR.plot <- ggplot(Dom.female_mouse.av, aes(color= strain))+geom_jitter(aes(x = as.factor(strain), y = var, color= strain, alpha=0.5), width = .2 )+ ggtitle("Dom feale ofwithin mouse  MLH1 distributions")+scale_color_manual(values=colors_of_strains)
Dom.female.VAR.plot

```

```{r Female.specific.glm, echo=FALSE}

glm.female.strain.subsp <- glm(mean_co ~ subsp * strain, data= female_mouse.avs_for.analysis)

summry.female.strain.effects <- summary(glm.female.strain.subsp)
#round(summry.female.strain.effects$coefficients[,c(1,4)],5)
#write.table(summry.female.strain.effects$coefficients, "~./MLH1repo/summry.female.strain.effects.csv", sep=",",row.names = TRUE)

MLH1.female.spc.glm.M2 <- glm(mean_co ~ strain, data= female_mouse.avs_for.analysis)
sum_MLH1.female.spc.glm.M2 <- summary(MLH1.female.spc.glm.M2)

#round(lm.female.strain_sum$coefficients[,c(1,4)],5)

#write.table(lm.female.strain_sum$coefficients, "~./MLH1repo/lm.female.strain.csv", sep=",",row.names = TRUE)
```

<!--
The variance across mouse means for each strain is also variable across strains. With _domesticus^LEW^_ having the most significant strain effect in the second model (glm; p= ).
 -->
 
```{r FEMALE.VAR.MODELS, echo=FALSE}
## within mouse variance models 
lmVAR.female.strain.subsp <- glm(var ~ subsp * strain, data= female_mouse.avs_for.analysis)
summry.female.strain.VAReffects <- summary(lmVAR.female.strain.subsp)

lmVAR.female.strain <- glm(var ~ strain, data= female_mouse.avs_for.analysis)
lmVAR.female.strain_sum <- summary(lmVAR.female.strain)

#round( #round isn't working for this sumary
#lmVAR.female.strain_sum$coefficients[,c(1,4)]
#Q12
Q12_female_table <-  Q12_mouse_table_w.Ages %>% filter(sex == "female")

lm.m1.Q12_female <- glm(var ~ subsp * strain, data= Q12_female_table)
lm.m1.Q12_female_sum <- summary(lm.m1.Q12_female)


lm.m2.Q12_female <- glm(var ~ strain, data= Q12_female_table)
lm.m2.Q12VAR.female_sum <- summary(lm.m2.Q12_female)#LEW changes to PWD

lm.m2.Q12_female.cv <- glm(cV ~ strain, data= Q12_female_table)
lm.m2.Q12_female.cv_sum <- summary(lm.m2.Q12_female.cv)

#lm.m2.Q12VAR.female_sum$coefficients[,c(1,4)]
#LEW sig

```
<!-- male section -->

```{r male.strain.table, echo=FALSE}
M.strain.table <- ddply(fomo.male_mouse.av, c("strain"), summarise,
                        Nmice =length(unique(mouse)),
                        
                        totNcells  = sum(Ncells, na.rm=TRUE),
                        
                        mean.mouse_mean_co = mean(mean_co, na.rm=TRUE),
                       
                        mean.mouse_cV = mean(cV, na.rm = TRUE),
                        
                        mean.mouse_var = mean(var, na.rm = TRUE ), 
                        
                        mean.mouse_sd   = mean(sd, na.rm=TRUE),
                       
                        #Standard error (mice . cells)
                        mean.mouse_se  = mean(se, na.rm=TRUE)
    )
#M.strain.table
#same issue as the female table .. ugh
```

```{r male.diff, echo=FALSE}
just.male_mouse.av <-  fomo.male_mouse.av %>% filter(sex== "male") %>% filter(species == "M.musculus") 

just.male_mouse.av$sex <- factor(just.male_mouse.av$sex, ordered = FALSE)
just.male_mouse.av$strain <- factor(just.male_mouse.av$strain, ordered = FALSE)
just.male_mouse.av$subsp <- factor(just.male_mouse.av$subsp, ordered = FALSE)
just.male_mouse.av$species <- factor(just.male_mouse.av$species, ordered = FALSE)


#how much greater are the High Rec gwRRs
just.male_mouse.av$Rec.group <- ifelse(grepl("PWD", just.male_mouse.av$strain), 1, 
                              ifelse(grepl("MSM", just.male_mouse.av$strain), 1,
                                ifelse(grepl("SKIVE", just.male_mouse.av$strain), 1, 0)))


#how much bigger are the high rec male means?
sans_LewRec_male <- just.male_mouse.av %>% filter(Rec.group == 0)
#mean(sans_LewRec_male$mean_co)

PWD_male <-  just.male_mouse.av %>% filter(strain == "PWD")
#mean(PWD_male$mean_co) / mean(sans_LewRec_male$mean_co)

MSM_male <- just.male_mouse.av %>% filter(strain == "MSM")
#mean(MSM_male$mean_co) / mean(sans_LewRec_male$mean_co)

SKIVE_male <-   just.male_mouse.av %>% filter(strain == "SKIVE")
#mean(SKIVE_male$mean_co) /  
```

```{r mouse.var.plots.male, echo=FALSE, fig.width = 4, fig.height=2.5, include=FALSE, include=FALSE}

#variance plots
musc.male.VAR.plot <- ggplot(Musc.male_mouse.av, aes(color= strain))+geom_jitter(aes(x = as.factor(strain), y =var, color= strain, alpha=0.5), width = .2 )+ ggtitle("VAR Musc male of within mouse MLH1 distributions")+scale_color_manual(values=colors_of_strains)
musc.male.VAR.plot

Mol.male.VAR.plot <- ggplot(Mol.male_mouse.av, aes(color= strain))+geom_jitter(aes(x = as.factor(strain), y = var, color= strain, alpha=0.5), width = .2 )+ ggtitle("VAR Mol male of within mouse MLH1 distributions")+scale_color_manual(values=colors_of_strains)
Mol.male.VAR.plot

Dom.male.VAR.plot <- ggplot(Dom.male_mouse.av, aes(color= strain))+geom_jitter(aes(x = as.factor(strain), y = var, color= strain, alpha=0.5), width = .2 )+ ggtitle("VAR Dom male ofwithin mouse  MLH1 distributions")+scale_color_manual(values=colors_of_strains)
Dom.male.VAR.plot
```

```{r Male.specific.glm, echo=FALSE}
# 1 way anova, is at least 1 strain different from the rest?
# given a lm, how sig is the strain effect
glm.male.strain.subsp <- glm(mean_co ~ subsp * strain, data= just.male_mouse.av)

#summry.male.strain.effects <- summary(glm.male.strain.subsp)
#round(summry.male.strain.effects$coefficients[,c(1,4)], 5)

#write.table(summry.male.strain.effects$coefficients , "~./MLH1repo/summry.male.strain.effects.csv", sep=",",row.names = TRUE)

#1. determine which strain effects are sig

#M2
MLH1.male.specific.glm.M2 <- glm(mean_co ~ strain, data= just.male_mouse.av)
sum_MLH1.male.specific.glm.M2 <- summary(MLH1.male.specific.glm.M2)

#round(lm.male.strain_sum$coefficients[,c(1,4)], 5)

#write.table(lm.male.strain_sum$coefficients , "~./MLH1repo/lm.male.strain_sum.csv", sep=",",row.names = TRUE)

#even when all the other male strains are added -- the same result; PWD, MSM and SKIVE strain effects are significant.
```

```{r MALE.SPECIFIC.variance.model, echo=FALSE}
## within mouse variance models 
lmVAR.male.strain.subsp <- lm(var ~ subsp * strain, data= just.male_mouse.av)
summry.male.strain.VAReffects <- summary(lmVAR.male.strain.subsp)
#round(summry.male.strain.VAReffects$coefficients[,c(1,4)], 5)

#all p values kinda high, MSM and TOM are slightly sig
lmVAR.male.strain <- lm(var ~ strain, data= just.male_mouse.av)
lmVAR.male.strain_sum <- summary(lmVAR.male.strain)
#round(lmVAR.male.strain_sum$coefficients[,c(1,4)], 5)
#non sig

lmcv.male.strain <- lm(cV ~ strain, data= just.male_mouse.av)
lmcv.male.strain_sum <- summary(lmcv.male.strain)
#round(lmcv.male.strain_sum$coefficients[,c(1,4)], 5)
#HMI sig for cv model -- in the other model, CAST subsp and TOM strain are slightly significant

#Q12
Q12_male_table <-  Q12_mouse_table_w.Ages %>% filter(sex == "male")

lm.m1.Q12_male <- lm(var ~ subsp * strain, data= Q12_male_table)
lm.m1.Q12_male_sum <- summary(lm.m1.Q12_male)
#round(lm.m1.Q12_male_sum$coefficients[,c(1,4)], 5)
#NS

lm.m2.Q12_male <- lm(var ~ strain, data= Q12_male_table)
lm.m2.Q12VAR.male_sum <- summary(lm.m2.Q12_male)
#round(lm.m2.Q12VAR.male_sum$coefficients[,c(1,4)], 5)
#NS
```



<!--(the figure is made from pooling MLH1 cells)
the male 
- the ranges of mouse means are displayed in Sup table X
- How the means were calculated
-The male gwRR is adjusted for the PAR
-The means for each strains are report in Table 1
-Normality was assessed for the MLH1 counts (include in Supplement histograms)
-->

# Evolutionary Framework

<to better understand this varation in RR within an evolutionary framework. Make table of models>


**M1. Mixed Model** $$mouse \ av.\ MLH1\ foci\ per\ cell ~=~ subsp * sex + rand(strain) + \varepsilon $$

**M2. Linear Model** $$mouse \ av.\ MLH1\ foci\ per\ cell ~=~ subsp * sex * strain + \varepsilon$$

**M3. LInear Model** $$mouse \ av.\ MLH1\ foci\ per\ cell ~=~ sex * strain + \varepsilon$$

**M4. Linear Model** $$sex\ specific\ mouse \ av.\ MLH1\ foci\ per\ cell ~=~ subsp * strain + \varepsilon$$


```{r MLH1co.lmer.MM, echo=FALSE, warning = FALSE, digits = 3, message=FALSE}
#THIS IS THE NEW MIXED MODEL
#should be able to use the loaded 
#DF.HetC.MixedModel.HQ

MLH1.MM.M1 <- lmer(mean_co ~ subsp * sex + (1|strain), data=DF.HetC.MixedModel.HQ) 
MLH1.MM.M1.reduce.sex <- lmer( mean_co ~ subsp  +(1|strain), data=DF.HetC.MixedModel.HQ )
MLH1.MM.M1.reduce.subp <- lmer( mean_co ~ sex  +(1|strain), data=DF.HetC.MixedModel.HQ )

#summary(MLH1.MM.M1)$coefficients

MLH1.MM_interaction <- drop1(MLH1.MM.M1, test="Chisq")$`Pr(Chi)`[2]
MLH1.MM_sex <- anova(MLH1.MM.M1, MLH1.MM.M1.reduce.sex)$`Pr(>Chisq)`[2]
MLH1.MM_subp <- anova(MLH1.MM.M1, MLH1.MM.M1.reduce.subp)$`Pr(>Chisq)`[2]
MLH1.MM_strain<- MM.rand.pval <- exactRLRT(MLH1.MM.M1)$p.value 

MLH1.MMlmer_results <- data.frame(sub.sp = MLH1.MM_subp, 
sex.results =  MLH1.MM_sex, 
               interaction.pval = MLH1.MM_interaction,  rand.pval = 
MLH1.MM_strain)
```

We applied a series of models to our dataset of mean MLH1 foci counts from `r length(DF.HetC.MixedModel.HQ$mean_co )` mice to fit the patterns of variation across sex, subspecies and strain into a evolutionary framework. All of the effects within the mixed model M1. were highly significant (LRT: subsp p= `r MLH1.MMlmer_results$sub.sp`, sex p = `r MLH1.MMlmer_results$sex.results`, subspecies * sex p = `r MLH1.MMlmer_results$interaction.pval`) and random LTR; strain p= `r MLH1.MMlmer_results$rand.pval`). 

<These results allow use to reject the hypothesis of genome wide recombination rates (varying consistantly by sex across subepcies -- and having variance within subspecies due to strains)>

 <!-- (random strain effect was meant to reflect sampling of gentic backgrounds/standing genetic variation for a subspecies) -- (significant pvalue indicates that the variance is not 0)>
 indicating, fixed effects -- random effect variance due to each strain nested within subspecies -->

```{r MLH1.glmM2, echo=FALSE, warning=FALSE,  digits = 3}

remove_strains <- c("TOM","AST","CZECH","CAST","HMI","SPRET","CAROLI", "F1", "PERC", "SPIC")
#AP_mouse_table.HQ_target <- AP_mouse_table.HQ_target[ ! AP_mouse_table.HQ_target$strain %in% remove_strains, ]
#kkll <- DF.HetC.MixedModel.HQ[! DF.HetC.MixedModel.HQ$strain %in% remove_strains,]

#strain = fixed
MLH1.M2.glm.strain.fixed.full <- glm(mean_co ~ subsp * sex * strain, data=DF.HetC.MixedModel.HQ[  ! DF.HetC.MixedModel.HQ$strain %in% remove_strains,]   )

summ.MLH1.M2.glm.strain.fixed.full <- summary(MLH1.M2.glm.strain.fixed.full)

#table of just coeff anf pvalue
#round(summ$coefficients[,c(1,4)],5 )

#write.table(summ$coefficients , "~./MLH1repo/glm.strain.fixed.full.csv", sep=",",row.names = TRUE)
```

```{r MLH1.glm.M3, echo=FALSE, digits = 3}

remove_strains <- c("TOM","AST","CZECH","CAST","HMI","SPRET","CAROLI", "F1", "PERC", "SPIC")

MLH1.M3.strain.fixed <- lm(mean_co ~ sex * strain, data=DF.HetC.MixedModel.HQ[  ! DF.HetC.MixedModel.HQ$strain %in% remove_strains,])

summ.MLH1.M3.strain.fixed <- summary(MLH1.M3.strain.fixed)

#round(summ.strain.fixed$coefficients[,c(1,4)],5)

#write.table(summ.strain.fixed$coefficients , "~./MLH1repo/summ.strain.fixed.csv", sep=",",row.names = TRUE)
```

<The linear models, allow us to quantifying the significance in sepecific fixed effects and focus on how specific levels and interactions create these differences.>

The linear model results confirm the qualitative observations of sex-specific evolution and significant difference in the degree of sexual dimorphism in genome wide recombiantion rates. The subspecies fixed effects tested in M2, had weak to no significance (glm; p=`r summ.MLH1.M2.glm.strain.fixed.full$coefficients[,4][2]` and p= `r summ.MLH1.M2.glm.strain.fixed.full$coefficients[,4][3]` for _M. m. musculus_ and _M. m. molossinus_ respectively), indicating that most of the variance is due to strain effects. As such, we focus on results from M3. The two most significant strain effects were _domesticus^G^_ and _musculus^MSM^_ (p= `r summ.MLH1.M3.strain.fixed$coefficients[,4][3]` and p= `r summ.MLH1.M3.strain.fixed$coefficients[,4][6]` for _domesticus^G^_ and _molossinus^MSM^_ respectively).

In addition to significant strain effects, ( p = `r summ.MLH1.M3.strain.fixed$coefficients[,4][5]` and _molossinus^MSM^_ strain p = `r summ.MLH1.M3.strain.fixed$coefficients[,4][6]`)  both _musculus^PWD^_ and _musculus^MSM^_ had significant strain by sex effects (glm; p = `r summ.MLH1.M3.strain.fixed$coefficients[,4][12]`
and p = `r summ.MLH1.M3.strain.fixed$coefficients[,4][13]`) _musculus^PWD^_ by male and _musculus^MSM^_ by male.

Combined these results demonstrate that i) the significant sex-specific evolution in MSM and PWD and ii) the larger magnitude of sexual dimorphism /heterochiasmy in _domesticus^G^_. No significant sex effect showing that the sex effect on mean CO number is not uniform across the sample but has synergy with speciefic strains.

<domesticus^G^ had significant strain (p= `r summ.MLH1.M3.strain.fixed$coefficients[4,][3]`)
and sex by strain effects (p =`r  summ.MLH1.M3.strain.fixed$coefficients[,4][10]`>

A linear model was run on the sex specific data sets (M4), with `r length(fomo.male_mouse.av$mouse)` number of male and `r length(female_mouse.avs_for.analysis$mouse)` female mean MLH1 foci counts were used in each of these models. For the male specific data, 

_musculus^PWD^_ (glm; p=`r sum_MLH1.male.specific.glm.M2$coefficients[5,][4]`), _musculus^SKIVE^_ (glm; p=`r sum_MLH1.male.specific.glm.M2$coefficients[8,][4]`) and _molossinus^MSM^_ (glm; p=`r sum_MLH1.male.specific.glm.M2$coefficients[6,][4]` ). (with effect sizes ranging from 5, 7, and 2 foci respectively). Given these results, theses strains are classified as 'high rec' strains and point to rapid evolution in the recombiantion rate for spermatocytes.

three of the strain effects of 13 strains had significant effects on mean MLH1 per cell, _musculus^PWD^_  (glm; p=`r sum_MLH1.male.specific.glm.M2$coefficients[5,][4]`), _musculus^SKIVE^_ (glm; p=`r sum_MLH1.male.specific.glm.M2$coefficients[8,][4]`) and _molossinus^MSM^_ (glm; p=`r sum_MLH1.male.specific.glm.M2$coefficients[6,][4]` ). (with effect sizes ranging from 5, 7, and 2 foci respectively).

While the female specific linear model has many more had significant strain effects; _domesticus^G^_ (glm; p=`r sum_MLH1.female.spc.glm.M2$coefficients[2,][4]`), _domesticus^LEW^_ (glm; p=`r sum_MLH1.female.spc.glm.M2$coefficients[3,][4]`), _domesticus^PWD^_ (glm; p=`r sum_MLH1.female.spc.glm.M2$coefficients[4,][4]`), _domesticus^MSM^_ (glm; p=`r sum_MLH1.female.spc.glm.M2$coefficients[5,][4]`), _domesticus^MOLF^_(glm; p=`r sum_MLH1.female.spc.glm.M2$coefficients[6,][4]`), and _domesticus^KAZ^_(glm; p=`r sum_MLH1.female.spc.glm.M2$coefficients[8,][4]`) the effect sizes have a smaller range of four to one, indicateing specific, but small strain effects for the mean number of MLH1 foci per cell.

<!--
M2(glm; PWD*male p= `r summ.MLH1.M2.glm.strain.fixed.full$coefficients[,4][14]` and MSM-male p = `r summ.MLH1.M2.glm.strain.fixed.full$coefficients[,4][15]`)>
<Additionally, _domesticus^G^_ had significant strain and sex by strain effects (p =`r  summ.MLH1.M3.strain.fixed$coefficients[,4][3]` and `r summ.MLH1.M3.strain.fixed$coefficients[,4][10]` for strain and strain by sex effect). These results confirm the qualtative pattern of the _domesticus^G^_ strain having the largest degree of sexual dimorphism (heterochiasmy) in the dataset.

In order to test the effects of subspecies, sex and strain on the mean of MLH1 counts per cell, we fit `r length(DF.HetC.MixedModel.HQ$mean_co )` mouse averages of MLH1 foci per cell to a series of mixed and linear models. 
(*note on sampling /sample numbers, the numbers of mice across strains, is shown in supplemental table )
Check the DF of the number cited above
This is the break down of mice used within the Mixed model
`r kable(table( DF.HetC.MixedModel.HQ$sex, DF.HetC.MixedModel.HQ$strain))`
add number of cells
`r kable(table( all.cell.MM.HQ$sex, all.cell.MM.HQ$strain))`
 $$mouse \ av \ meteric ~=~ subsp * sex  + rand(1|strain) + \varepsilon$$

- only Mus musculus strains with sex matched observations  
- Male gwRR have been adjusted for the PAR
- Q5 excluded
- All the effects are un-ordered factors  
- male ages within 5 - 20 weeks
- Molossinus is coded as a subsp level
also g strain effect/ dom^G^ strain effect 
`r summ.MLH1.M2.glm.strain.fixed.full$coefficients[12,][4]` 
In both models strain by sex interaction effects were significant for (p = `r summ.MLH1.M3.strain.fixed$coefficients[,4][12]` and `r summ.MLH1.M3.strain.fixed$coefficients[,4][13]` for the second mixed model) for _musculus^PWD^_ and _molossinus^MSM^_  respectively. 
-->

<!-- transition gwRR to within mouse variance -->

## Within Mouse Variance in CO Count per Cell

Variance of a trait can contain information pertinant to it's mode of evolution and mechanistically could reflect the strength of constraints acting on the trait. We examine the within animal variance in MLH1 foci count per cell within the same models applied to mean MLH1 foci count (replacing mean MLH1 foci count with variance of MLH1 foci across cells as the dependant variable). <note about reduced power>
<in some traits, like gene expression patterns, evolution of the mean -- also has an effect on the variance>

<!-- 
**M5. Mixed Model** $$mouse\ variance\ in \ CO \ per cell ~=~ subsp * sex + rand(strain) + \varepsilon $$
 
 **M6. Linear Model** $$mouse\ variance\ in \ CO \ per cell ~=~ subsp * sex * strain + \varepsilon$$
 
 **M7. Linear Model** $$mouse\ variance\ in \ CO \ per cell  ~=~ sex * strain + \varepsilon$$>
-->

```{r M1.MLH1.var_cv, echo=FALSE, warning=FALSE, message=FALSE}

MLH1var_M1 <- lmer( var ~ subsp * sex +(1|strain), data=DF.HetC.MixedModel.HQ )
MLH1var_M1.reduce.sex <- lmer( var ~ subsp  +(1|strain), data=DF.HetC.MixedModel.HQ )
MLH1var_M1.reduce.subp <- lmer( var ~ sex  +(1|strain), data=DF.HetC.MixedModel.HQ )

#summary(MLH1var_M1)
LmerMLH1_VAR_int <- drop1(MLH1var_M1, test="Chisq")$`Pr(Chi)`[2]
LmerMLH1_VAR_sex <- anova(MLH1var_M1, MLH1var_M1.reduce.sex)$`Pr(>Chisq)`[2]
LmerMLH1_VAR_subsp <- anova(MLH1var_M1, MLH1var_M1.reduce.subp)$`Pr(>Chisq)`[2]
LmerMLH1_VAR_strain <- exactRLRT(MLH1var_M1)$p.value

LmerMLH1_VAR_results <- data.frame(sub.sp = LmerMLH1_VAR_subsp,
sex.results =  LmerMLH1_VAR_sex, interaction.pval = LmerMLH1_VAR_int,  rand.pval = LmerMLH1_VAR_strain)


#CV
MLH1cV_M1 <- lmer( cV ~ subsp * sex +(1|strain), data=DF.HetC.MixedModel.HQ )
MLH1cV_M1.reduce.sex <- lmer( cV ~ subsp  +(1|strain), data=DF.HetC.MixedModel.HQ )
MLH1cV_M1.reduce.subp <- lmer( cV ~ sex  +(1|strain), data=DF.HetC.MixedModel.HQ )

LmerMLH1_CV_int <- drop1(MLH1cV_M1, test="Chisq")$`Pr(Chi)`[2]
LmerMLH1_CV_sex <- anova(MLH1cV_M1, MLH1cV_M1.reduce.sex)$`Pr(>Chisq)`[2]
LmerMLH1_CV_subsp <- anova(MLH1cV_M1, MLH1cV_M1.reduce.subp)$`Pr(>Chisq)`[2]
LmerMLH1_CV_strain <- exactRLRT(MLH1cV_M1)$p.value

LmerMLH1_CV_results <- data.frame(sub.sp = LmerMLH1_CV_subsp,
sex.results =  LmerMLH1_CV_sex, interaction.pval = LmerMLH1_CV_int,  rand.pval = LmerMLH1_CV_strain)

#USE Q12 (Q123) table
#Q12_mouse_table_w.Ages
#Q12 var
MLH1var_M1_Q12 <- lmer( var ~ subsp * sex +(1|strain), data=Q12_mouse_table_w.Ages )
MLH1var_M1_Q12reduce.sex <- lmer( var ~ subsp  +(1|strain), data=Q12_mouse_table_w.Ages )
MLH1var_M1_Q12reduce.subp <- lmer( var ~ sex  +(1|strain), data=Q12_mouse_table_w.Ages )

LmerMLH1_M1_Q12VAR_int <- drop1(MLH1var_M1_Q12, test="Chisq")$`Pr(Chi)`[2]
LmerMLH1_M1_Q12_VAR_sex <- anova(MLH1var_M1_Q12, MLH1var_M1_Q12reduce.sex)$`Pr(>Chisq)`[2]
LmerMLH1_M1_Q12_VAR_subsp <- anova(MLH1var_M1_Q12, MLH1var_M1_Q12reduce.subp)$`Pr(>Chisq)`[2]
LmerMLH1_M1_Q12_VAR_strain <- exactRLRT(MLH1var_M1_Q12)$p.value

LmerMLH1_M1_Q12_VAR_results <- data.frame(sub.sp = LmerMLH1_M1_Q12_VAR_subsp,
sex.results =  LmerMLH1_M1_Q12_VAR_sex, interaction.pval = LmerMLH1_M1_Q12VAR_int,  rand.pval = LmerMLH1_M1_Q12_VAR_strain)

#Q12 cv
MLH1cv_M1_Q12 <- lmer( cV ~ subsp * sex +(1|strain), data=Q12_mouse_table_w.Ages )
MLH1cv_M1_Q12reduce.sex <- lmer( cV ~ subsp  +(1|strain), data=Q12_mouse_table_w.Ages )
MLH1cv_M1_Q12reduce.subp <- lmer( cV ~ sex  +(1|strain), data=Q12_mouse_table_w.Ages )

LmerMLH1_M1_Q12_CV_int <- drop1(MLH1cv_M1_Q12, test="Chisq")$`Pr(Chi)`[2]
LmerMLH1_M1_Q12_CV_sex <- anova(MLH1cv_M1_Q12, MLH1cv_M1_Q12reduce.sex)$`Pr(>Chisq)`[2]
LmerMLH1_M1_Q12_CV_subsp <- anova(MLH1cv_M1_Q12, MLH1cv_M1_Q12reduce.subp)$`Pr(>Chisq)`[2]
LmerMLH1_M1_Q12_CV_strain <- exactRLRT(MLH1cv_M1_Q12)$p.value

LmerMLH1_M1_Q12_VAR_results <- data.frame(sub.sp = LmerMLH1_M1_Q12_CV_subsp,
sex.results =  LmerMLH1_M1_Q12_CV_sex, interaction.pval = LmerMLH1_M1_Q12_CV_int,  rand.pval = LmerMLH1_M1_Q12_CV_strain)
```

```{r M2.MLH1.var, echo=FALSE, include=FALSE, eval=FALSE}
#not including this model for now / brivity
#$$mouse CO metric  ~=~ subsp * sex * strain + \varepsilon$$

MLH1var_M3 <- lm( var ~ subsp * sex * strain,  data=DF.HetC.MixedModel.HQ )

#summary(MLH1var_M3)
anova(MLH1var_M3)  
as.data.frame( round(summary(MLH1var_M3)$coefficients, 6))


MLH1cv_M3 <- lm(cV ~ subsp*sex*strain, data=DF.HetC.MixedModel.HQ)

#summary(MLH1cv_M3)

anova(MLH1cv_M3)
as.data.frame( round(summary(MLH1cv_M3)$coefficients, 6))


#Q12 var
MLH1var_M3_Q12 <- lm( var ~ subsp*sex*strain,  data=Q12_mouse_table_w.Ages )

#summary(MLH1var_M3_Q12)
anova(MLH1var_M3_Q12)  
as.data.frame( round(summary(MLH1var_M3_Q12)$coefficients, 6))

#Q12 cv
MLH1cv_M3_Q12 <- lm( cV ~ subsp*sex*strain, data=Q12_mouse_table_w.Ages )

#summary(MLH1cv_M3_Q12)
anova(MLH1cv_M3_Q12)#sex large F

as.data.frame( round(summary(MLH1cv_M3_Q12)$coefficients, 6))

```

```{r M3.MLH1.var, echo=FALSE}

glmMLH1Var_M3 <- glm( var ~ sex * strain,  data=DF.HetC.MixedModel.HQ )
glmMLH1Var_M3_sum <- summary(glmMLH1Var_M3)
#round(glmMLH1Var_M3_sum$coefficients[,c(1,4)], 5)

MLH1cv_M3 <- glm(cV ~ sex * strain, data=DF.HetC.MixedModel.HQ)
#summary(MLH1cv_M4)
#round(summary(MLH1cv_M3)$coefficients[,c(1,4)], 6)

### Q12 var
MLH1var_M3_Q12 <- glm( var ~ sex*strain, data=Q12_mouse_table_w.Ages )
MLH1var_M3_Q12_sum <- summary(MLH1var_M3_Q12)

#as.data.frame( round(summary(MLH1var_M3_Q12)$coefficients[,c(1,4)], 6))
#round(summary(MLH1var_M3_Q12)$coefficients[,c(1,4)], 6)

#Q12 cv
MLH1cv_M4_Q12 <- glm( cV ~ sex*strain, data=Q12_mouse_table_w.Ages )
#round(summary(MLH1cv_M4_Q12)$coefficients[,c(1,4)], 6)
```
<!--
These are the p values for the effects of the mixed models of the foloowing mixed models: VAR.FULL, CV.FULL, VAR.Q12 and cv.Q12
var, full.mixed, q12,mixed, M3.glm.full, M3.glm.q12
1.618e-10 ***, 2.092e-11, 0.044181 * (male),  0.000228 ***
cv, full.mixed, q12,mixed, M3.glm.full, M3.glm.q12
5.033e-12 ***, 1.315e-12,  3.195e-12 ***, 3.16e-05 ***
-->

**the general pattern is that females have more variance (alsmot twice as much (Figure 1))**

For both the mixed and linear models, sex was the only significant effect (LRT; p= `r round(LmerMLH1_VAR_results$sex.results, 5)`, glm; p= `r round(MLH1var_M3_Q12_sum$coefficients[,4][2], 5)`). Since the measures for within mouse variance may be more susceptible to technical error effects from the staining protocol (i.e. background noise), we replicated the model analysis using a subset of cells with higher quality scores. These models replicated the results of the full data set with sex being the most significant effect (LTR; p = `r round(LmerMLH1_M1_Q12_VAR_results$sex.results, 5)`, and glm; p = `r  round(MLH1var_M3_Q12_sum$coefficients[2,4], 5)` ). 

The sex specific effect linear models indicate that there is no significant difference in the amount of within mouse variance in males. While in females, one strain, _domesticus^LEW^_ has significantly higher within mouse variance (glm; p = `r lmVAR.female.strain_sum$coefficients[3,][4]`). However, in the model using the high quality cells, _musculus^PWD^_ has the only signicant strain effect (glm; p =`r lm.m2.Q12VAR.female_sum$coefficients[4,][4]`) suggesting a lack of a consistant signal for differences in the amount of within mouse variance in oocytes. 

**the general pattern is that females have more variance (alsmot twice as much (Figure 1))**

<there are some caveats, but the general pattern is that females have more variance ... For the estimates of variance, the replicates are at the level of mouse, unlike MLH1 counts, in which the replicates are at the level of the cell. This limits the power for detecting significance differences across groups,

the main/general pattern, is that females have significantly higher within mouse variance compared to males. This general pattern is replicated in higher quality dataset and using the coefficient of variance (cv) as a dependant variable.>

<!--
$$mouse\ variance\ for\ MLH1\ foci\ per\ cell   ~=~ subsp * sex + rand(strain)+ \varepsilon$$ -->
<!--
RW's main qtl paper used the within animal variance, 

@article{wang2017genetics,
  title={Genetics of genome-wide recombination rate evolution in mice from an isolated Island},
  author={Wang, Richard J and Payseur, Bret A},
  journal={Genetics},
  volume={206},
  number={4},
  pages={1841--1852},
  year={2017},
  publisher={Genetics Soc America}
}
references for phenotypic variance connected to robustness to environment, increase in developmental consistency

Waddington 1942, Rutherford and Lindquist 2998, Mackay and Lyman 2005, Levy and Siegal 2008)
phenotypic variance leading to adaptation - 
(Losick and desplan 2008, Beaumont 2009, Eldar and Eloqitz 2010)
Previous vQTLs (Shen etal 2012, Hulse and Cai 2013, Mulder et al 2016)
meiotic Checkpoints for ensuring consistency (Hochwagen and Amon 2006, Li and Schimenti 2007, Jaramillo-Lambert 2010 and Kauppi 2013)
-->


<SUMMERIZE BOTH THE MLH1 evolution and within mouse variance HERE>

(combined the summary of the MLH1 distributions evolution in the mean and consistant pattern of sex differences)

<The greater amount of within mouse variance in females (almost and, (males have no detecable difference across strains for within mouse variance)
-- supports the distinct evolutionary pressures acting on the genome wide recombination rate in oocytes and spermatocytes.>

3. -- the striking differences / patterns of variation across sexes suggest that the genome wide recombination rates have distinct evolutionary trajectories for male and female rates. 
For female the pattern of Strain averages distributed around a species wide average - fits a model of stabilizing / relaxed / neutral evolution in contrast to the male pattern where there is rapid evolution in a subset of genetic backgrounds / strains which fits a model of directional selection on genome wide recombination rate. 



# Evolution of Genome wide recombination rate associated with evolution of mean precursor number

<transition, >

```{r DMC1.setup, echo=FALSE}
#load DMC1 csv
#DMC1.DF <- read.csv("~./DMC1/newJuviQuant.csv", header = TRUE)
DMC1.DF <- read.csv("~./MLH1repo/data/DMC1newJuviQuant.csv", header = TRUE)


DMC1.DF$fileName <- DMC1.DF$Original.Name
DMC1.DF <- add_mouse(DMC1.DF)
DMC1.DF <- add_strain(DMC1.DF)
DMC1.DF <- add_category(DMC1.DF)
DMC1.DF <- add_subsp(DMC1.DF)
DMC1.DF <- DMC1.DF[DMC1.DF$stage != "P",]

#make DMC1 table
DMC1.mean.obs.L <- ddply(.data=DMC1.DF[DMC1.DF$stage == "L",],
                 .(category),
                 summarize, 
                 
                 ncells = length(fileName),
                 mean.count = mean( total.count)
)
DMC1.mean.obs.L$stage <- "L"

DMC1.mean.obs.Z <- ddply(.data=DMC1.DF[DMC1.DF$stage == "Z",],
                 .(category),
                 summarize, 
                 ncells = length(fileName),
                 mean.count = mean( total.count)
)
DMC1.mean.obs.Z$stage <- "Z"

DMC1.mean.table <- rbind(DMC1.mean.obs.L,DMC1.mean.obs.Z)

#write.table(DMC1.mean.table, "~./MLH1repo/DMC1.mean.table.csv", sep=",",
#            row.names = FALSE)

DMC1.DF$rec_group <- ifelse(grepl("WSB", DMC1.DF$strain), "Low",
                          ifelse(grepl("G",DMC1.DF$strain), "Low",
                                 
              ifelse(grepl("KAZ", DMC1.DF$strain), "Low",
            ifelse(grepl("PWD", DMC1.DF$strain), "High",
               ifelse(grepl("MSM", DMC1.DF$strain), "High","")))))
```

```{r MAIN.DMC1.plot, echo=FALSE, warning=FALSE}
#think of better color scheme
#make the jitter col thinner or make the stacked dots.
#remember that box-plots add outlier dots.
#change y labels

#experimenting with ggboplot
everyday <- ggboxplot(DMC1.DF[DMC1.DF$stage != "P",], x = "strain", y = "total.count",
          color = "stage", palette = "jco",
          add = "jitter", short.panel.labs = FALSE)
  #library gpubr has functions to add in sig/test bars.

#change order

DMC1.DF$strain<- factor(DMC1.DF$strain, levels =c("LEW",
                      "WSB","G", "KAZ", "SKIVE","PWD","MSM"), order=T )


DMC1.boxplot <- ggplot(aes(x=strain, y=total.count, color=stage), data=DMC1.DF[DMC1.DF$stage != "P",])+
  geom_boxplot(alpha=.5) +geom_jitter()+ggtitle("DMC1 averages in juvenile mice")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(), axis.line = element_line(colour = "black"))

DMC1.boxplot
```

```{r plot.stage.ttest, echo=FALSE, warning=FALSE}

anno_df_stage = compare_means(total.count ~ stage, group.by = "rec_group", data = DMC1.DF[DMC1.DF$stage != "P",],method = "t.test") %>% mutate(y_pos = 300)
#

DMC1.bxlt_Z_to_L <- ggplot(DMC1.DF[DMC1.DF$stage != "P",], aes(x=stage, y=total.count)) + 
  geom_boxplot(position=position_dodge()) + 
  geom_point(aes(color=rec_group), position=position_jitterdodge()) + 
  
  #facet_wrap(~stage, nrow=1) + 
  
  ggsignif::geom_signif(
    data=anno_df_stage, 
    aes(xmin=group1, xmax=group2, annotations=p.adj, y_position=y_pos), 
    manual=TRUE
  )+ggtitle("Comparison of DMC1 means across high and low recombing groups")+theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(), axis.line = element_line(colour = "black"))



#Musc.mouse.av <-  Mouse.Table_BivData_4MM %>%  filter(subsp == "Musc") 

ttest.ZtoL <- t.test(DMC1.DF$total.count[(DMC1.DF$stage == "L")],
               DMC1.DF$total.count[(DMC1.DF$stage == "Z")]  )


DMC1.High.goup <-  DMC1.DF %>%  filter(rec_group == "High") 

ttest.ZtoL.high <- t.test(DMC1.High.goup$total.count[(DMC1.High.goup$stage == "L")],
               DMC1.High.goup$total.count[(DMC1.High.goup$stage == "Z")]  )


DMC1.Low.goup <-  DMC1.DF %>%  filter(rec_group == "Low") 

ttest.ZtoL.Low <- t.test(DMC1.Low.goup$total.count[(DMC1.Low.goup$stage == "L")],
               DMC1.Low.goup$total.count[(DMC1.Low.goup$stage == "Z")]  )
```

```{r plot.t.test, echo=FALSE, include=FALSE}

  #this will add neat bars, but still need to figure out what groups ect it's actually using
#THE stat_compare function adds a arm thing and pvalue for 
#  stat_compare_means(comparisons = my_comparisons)

#status - legend, expression y-axis, facet, facet_wrap SYMBOL
#legend - stage. y=total count, FACet strain


#add column to distinguish high vs low rec strains
DMC1.DF$rec_group <- ifelse(grepl("WSB", DMC1.DF$strain), "Low",
                          ifelse(grepl("G",DMC1.DF$strain), "Low",
                                 
              ifelse(grepl("KAZ", DMC1.DF$strain), "Low",
            ifelse(grepl("PWD", DMC1.DF$strain), "High",
               ifelse(grepl("MSM", DMC1.DF$strain), "High","")))))

DMC1.DF$rec_group <- as.factor(DMC1.DF$rec_group)

                   
# annotation table with adjusted pvals and y-position of the labels

anno_df = compare_means(total.count ~ rec_group, group.by = "stage", data = DMC1.DF[DMC1.DF$stage != "P",],method = "t.test") %>% mutate(y_pos = 300)
#

DMC1.bxlt_High.to.Low <- ggplot(DMC1.DF[DMC1.DF$stage != "P",], aes(x=rec_group, y=total.count)) + 
  geom_boxplot(position=position_dodge()) + 
  geom_point(aes(color=stage), position=position_jitterdodge()) + 
  facet_wrap(~stage, nrow=1) + 
  ggsignif::geom_signif(
    data=anno_df, 
    aes(xmin=group1, xmax=group2, annotations=p.adj, y_position=y_pos), 
    manual=TRUE
  )+ggtitle("Comparison of DMC1 means across high and low recombing groups")+theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(), axis.line = element_line(colour = "black"))

#example for ggplot+ test from here
#https://github.com/kassambara/ggpubr/issues/65

#https://www.r-bloggers.com/add-p-values-and-significance-levels-to-ggplots/

```

```{r DMC1_t.tests, echo=FALSE, warning=FALSE}
#2. t-tests for the DMC1 counts 
#need to account for sample sizes when comparing t.tests (KAZ and G don't have equal across )
#best case scnario -- more microscoping to get ~20 cells for each stage by strain combo


#high vs low group (L)
ttest.HighLow.L.pval <- t.test(DMC1.DF$total.count[( (DMC1.DF$stage == "L") & ( ( DMC1.DF$strain == "MSM") | (DMC1.DF$strain == "PWD") ) )  ],
               DMC1.DF$total.count[( (DMC1.DF$stage == "L") & ( ( DMC1.DF$strain == "WSB") | (DMC1.DF$strain == "G") | (DMC1.DF$strain == "KAZ") ) )]    )$p.value
# p-value =  0.002145


#high vs low group (Z)
ttest.HighLow.Z.pval <- t.test(DMC1.DF$total.count[( (DMC1.DF$stage == "Z") & ( ( DMC1.DF$strain == "MSM") | (DMC1.DF$strain == "PWD") ) )  ],
               DMC1.DF$total.count[( (DMC1.DF$stage == "Z") & ( ( DMC1.DF$strain == "WSB") | (DMC1.DF$strain == "G") | (DMC1.DF$strain == "KAZ") ) )]    )$p.value
# p-value = 0.6628


#dom vs musc (L) (2 subsp vs)
ttest.subsp.L <- t.test(DMC1.DF$total.count[(DMC1.DF$subsp == "Dom") & (DMC1.DF$stage == "L")], DMC1.DF$total.count[(DMC1.DF$subsp == "Musc") & (DMC1.DF$stage == "L")] )
# p-value = 0.003016

#dom vs musc (Z)
ttest.subsp.Z <- t.test(DMC1.DF$total.count[(DMC1.DF$subsp == "Dom") & (DMC1.DF$stage == "Z")], DMC1.DF$total.count[(DMC1.DF$subsp == "Musc") & (DMC1.DF$stage == "Z")] )
#p-value = 0.09588

#KAZ vs Dom (Z) (only 1 L cell)
ttest.KAZ.Dom.z <- t.test(DMC1.DF$total.count[(DMC1.DF$subsp == "Dom") & (DMC1.DF$stage == "Z")], 
      DMC1.DF$total.count[(DMC1.DF$strain == "KAZ") & (DMC1.DF$stage == "Z")] )

#p-value = 0.004148
#Kaz Z are higher than Dom
```

```{r DMC1_anova, echo=FALSE, warning=FALSE}
#1. ANOVA, is there a difference in DMC1 counts across strains?

#unorder strain
DMC1.DF$strain <- factor(DMC1.DF$strain, levels = c("G","WSB","KAZ","PWD","MSM"), ordered = FALSE )

mod.L <- lm(total.count ~ strain, data = DMC1.DF[DMC1.DF$stage == "L",])
mod.Z <- lm(total.count ~ strain, data = DMC1.DF[DMC1.DF$stage == "Z",])

l.anova.pval <- anova(mod.L)$`Pr(>F)`[1] #0.0002701 ***
z.anova.pval <- anova(mod.Z)$`Pr(>F)`[1] #NS
```

```{r DMC1_MLH1.MAIN_TABLE, echo=FALSE, warning = FALSE}
#make category averages from AP_mouse_table.HQ
mean.MLH1.obs <- ddply(.data=AP_mouse_table.HQ,
                 .(category),
                 summarize, 
                 mean.MLH1 = mean(mean_co),
                 var.MLH1 = var(mean_co)
)

DMC1.target.strains <- subset(mean.MLH1.obs, category == "G male" |
                   category == "WSB male" |
                     category == "KAZ male" |
                  category == "MSM male" |
                    category == "PWD male")
#attach this mini DF to the DMC1 data


DMC1.mean.obs.L$mean.count.L <- DMC1.mean.obs.L$mean.count
DMC1.mean.obs.Z$mean.count.Z <- DMC1.mean.obs.Z$mean.count


DMC1table <- merge(DMC1.mean.obs.L, DMC1.mean.obs.Z, by="category")

DMC1table.main <- merge(DMC1.target.strains, DMC1table, by="category")

MLH1.DMC1.L_cor <- cor(DMC1table.main$mean.MLH1, DMC1table.main$mean.count.L, method = c("pearson"))#MLH1.DMC1.Z_cor

MLH1.DMC1.Z_cor <- cor(DMC1table.main$mean.MLH1, DMC1table.main$mean.count.Z, method = c("pearson"))#0.2793237

##RATIOS
DMC1table.main$MLH1.L_ratio <- DMC1table.main$mean.MLH1/ DMC1table.main$mean.count.L 
DMC1table.main$MLH1.Z_ratio <- DMC1table.main$mean.MLH1 / DMC1table.main$mean.count.Z
  
#colnames(DMC1table.main) <- c("category", "mean.MLH1", "", "", "")

#DMC1.mean.table
DMC1.mean.table.99 <- subset(DMC1table.main, select= -c(var.MLH1,  mean.count.x,  mean.count.y, stage.x, stage.y))

colnames(DMC1.mean.table.99) <- c("category","mean.MLH1","ncells.L","mean.L",
                  "ncells.Z", "mean.Z", "MLH1.L_ratio", "MLH1.Z_ratio")


WSB_DMC1.Z_CO_ratio <-   ( DMC1table.main %>% filter(category == "WSB male") )$mean.count.L / (DMC1table.main %>% filter(category == "WSB male"))$mean.MLH1
  
G_DMC1.Z_CO_ratio <-   ( DMC1table.main %>% filter(category == "G male") )$mean.count.L / (DMC1table.main %>% filter(category == "G male"))$mean.MLH1

MSM_DMC1.Z_CO_ratio <- ( DMC1table.main %>% filter(category == "MSM male") )$mean.count.L / (DMC1table.main %>% filter(category == "MSM male"))$mean.MLH1

PWD_DMC1.Z_CO_ratio <-( DMC1table.main %>% filter(category == "PWD male") )$mean.count.L / (DMC1table.main %>% filter(category == "PWD male"))$mean.MLH1



DMC1table.main$category <- factor(DMC1table.main$category, levels = c("WSB male","G male","KAZ male",               "PWD male","MSM male"), ordered = TRUE )

DMC1table.main <- DMC1table.main[order(DMC1table.main$category),]

kable(head(DMC1table.main),  caption = "Table X. DMC1 foci counts per cell summary")


DMC1.Lratio_t.test <- t.test(  (DMC1table.main %>% filter(  ( category == "G male" | category == "WSB male" | category == "KAZ male" ) ) )$MLH1.L_ratio, 
                (DMC1table.main %>% filter(  ( category == "PWD male" | category == "MSM male") ) )$MLH1.L_ratio        )
                               

DMC1.Zratio_t.test <- t.test(  (DMC1table.main %>% filter(  ( category == "G male" | category == "WSB male" | category == "KAZ male" ) ) )$MLH1.Z_ratio, 
                (DMC1table.main %>% filter(  ( category == "PWD male" | category == "MSM male") ) )$MLH1.Z_ratio        )

```

<!--
explain why male oer female measures were used, -- mention some measures were made -- but the number of cells was so too small.
-->

In an attempt to localize  the above male specific crossover number evolution within the meiotic pathway we quantified a marker for DSBs, DMC1, in early prophase spermtaocytes. DMC1 foci were scored from a total of `r sum( (DMC1.mean.table %>%  filter(stage == "L") )$ncells)` leptotene and `r sum( (DMC1.mean.table %>%  filter(stage == "Z") )$ncells)` zygotene staged spermatocytes form juvenille mice (12 to 18 days). A subset of strains were choosen to test for differences between the high and low recombination groups; _musculus^PWD^_, _molossinus^MSM^_, _musculus^KAZ^_ ,  _domesticus^WSB^_ , and _domesticus^G^_ (Figure2). One mouse was sampled for each strain.

For all strains there were significantly more foci from leptotene cells compared to zygotene (t.test; p = `r ttest.ZtoL$p.value`), which reflects the expected patterns across the timing of meiosis. Confirming the prediction that evolution of the mean number of crossovers per cell is associated with evolution of the number of precursors (DSBs), the two high recombining strains tested have significantly more DMC1 foci compared to low recombining strains in leptoene cells (t.test; p=`r ttest.HighLow.L.pval`, one-way-anova; p=`r format(l.anova.pval, scientific=F)`).

However the the differences in DMC1 foci were not significant for (later prophase) zygotene cells between the two groups (t.test; p=`r ttest.HighLow.Z.pval`, one-way-anova; p=`r z.anova.pval`). Indicating that this marker at early prophase is more precitive of the downstream crossover number differences.

<explain the biological significance of the ratios!>

The ratios, calculated for DMC1 means from both stages of prophase, are not significantly differet between the strain groups (t.test; p = `r DMC1.Lratio_t.test$p.value` and p=`r DMC1.Zratio_t.test$p.value` for leptotene and zygotene based ratios respectively). These results indicate that the targets of evolution which lead to differences in crossover numbers may be established before DSB formation (cite Baier et al 2014).


<!-- which maybe be attributed to noise effects 
The ratios for MLH1:DMC1^Leptotene^ and MLH1:DMC1^zygotene^ are shown in Table X, and can be used as a proxy for the CO:NCO ratio (instead CO:(CO+NCO) ratio). -->

<!--
The correlation with MLH1 and leptotene cells is `r MLH1.DMC1.L_cor`. The correlation with MLH1 and zygotene cells is `r MLH1.DMC1.Z_cor`. `r c(WSB_DMC1.Z_CO_ratio, G_DMC1.Z_CO_ratio, MSM_DMC1.Z_CO_ratio, PWD_DMC1.Z_CO_ratio )` for  _M. m. domesticus^WSB^_ , _M. m. domesticus^G^_, _M. m. molossinus^MSM^_, and  _M. m. musculus^PWD^_ respecitively. 
-->

```{r DMC1.subsampling, echo=FALSE, eval = FALSE, include = FALSE}

#Subsampling code -- don't report
 
#The number of observations are not quite equal across strains and cell stages, so I ran permutations sampling 25 cells from the high and low recombining groups of cells. Below are the distributions of p values for permuted t-tests from sampling 25 cells from the pooled high and low recombining groups. Permutations were done for both cell stages. The actual p value is in red. 

#The observed p value does not fall outside of the permuted distribution with assumes equal cell sampling.

#recreate the DMC1 data by subsampling ~9 or lowest cell number from 
# the 2 groups

#make the 2 pools
High.L.pool <- DMC1.DF$total.count[( (DMC1.DF$stage == "L") & ( ( DMC1.DF$strain == "MSM") | (DMC1.DF$strain == "PWD") ) )  ]
High.Z.pool <- DMC1.DF$total.count[( (DMC1.DF$stage == "Z") & ( ( DMC1.DF$strain == "MSM") | (DMC1.DF$strain == "PWD") ) )  ]

Low.L.pool <- DMC1.DF$total.count[( (DMC1.DF$stage == "L") & ( ( DMC1.DF$strain == "WSB") | (DMC1.DF$strain == "G") | (DMC1.DF$strain == "KAZ") ) )]
Low.Z.pool <- DMC1.DF$total.count[( (DMC1.DF$stage == "Z") & ( ( DMC1.DF$strain == "WSB") | (DMC1.DF$strain == "G") | (DMC1.DF$strain == "KAZ") ) )]


#permute drawing / subsampling and performing the t-test
#put the p.value results into a DF!


#permut.SC.cv.diff  <- as.data.frame(
#sample(THE.DF, SAMPLE.SIZE, REPLACE?)

#permuting samples -- and re-licating t-tests

#nrep <- 1000

#permuting samples to replicate the t.test across cell stages from high and low with equal sample sizes.
bunchOpvals.L <- replicate(1000, 
    # t.test
    t.test( sample(High.L.pool, 25, replace = FALSE),
            sample(Low.L.pool, 25, replace = FALSE), var.equal = FALSE )$p.value
          )
bunchOpvals.L <- as.data.frame(bunchOpvals.L)
bunchOpvals.L$stage <- "L"

bunchOpvals.Z <- replicate(1000, 
    # t.test
    t.test( sample(High.Z.pool, 25, replace = FALSE),
            sample(Low.Z.pool, 25, replace = FALSE), var.equal = FALSE)$p.value
          )

bunchOpvals.Z <- as.data.frame(bunchOpvals.Z)
bunchOpvals.Z$stage <- "Z"

#could merge the lists of p-values
replicated.Ttests <- cbind(bunchOpvals.L, bunchOpvals.Z)
#have to melt, the rbinded dataframe to get the cols 'stage' and 'pvale'


#ASSUMPTIONS; normality, equal variance

#plot the lists of pvals in histogram? compare to the observed pvalue to get a sense as of how much sample size shifts the pvalues.

#plot.pvals.L <- ggplot(bunchOpvals.L, aes(bunchOpvals.L) )+geom_histogram(bins = 50)+ ggtitle("1000 rep t-tests sample size 25, High vs Low Leptotene")+geom_vline(xintercept=ttest.HighLow.L$p.value, linetype="dotted", color="red")


#plot.pvals.Z <- ggplot(bunchOpvals.Z, aes(bunchOpvals.Z) )+geom_histogram(bins = 50)+ ggtitle("1000 rep t-tests sample size 25, High vs Low Zygotene")+geom_vline(xintercept=ttest.HighLow.Z$p.value, linetype="dotted", color="red")
#plot.pvals.L
#plot.pvals.Z


#The t.test( ) function produces a variety of t-tests. Unlike most statistical packages, the default assumes #unequal variance and applies the Welsh df modification.
#n use the var.equal = TRUE
#https://www.statmethods.net/stats/ttest.html
```

```{r normality, echo=FALSE, warning=FALSE, eval=FALSE}

#these are histogram plots for testing Normality shape -- move to supplemenatry section


#- check that distributions are close to normal
#- check for equal variance.

DMC1.histogram <- ggplot(DMC1.DF, aes(total.count, fill = stage)) + geom_histogram(alpha = 0.4)+facet_wrap(~strain)+ ggtitle("DMC1 histogram")

DMC1.histogram
#checking equal variance, https://www.statmethods.net/stats/anovaAssumptions.html
#bartlet's test, Levine's test

#The counts look close to normal.
#Potentially figure 2. Using a function to #display the p-vales for mean comparison

```


<!--
interpertation notes, 
the number of DMC1 across stages (L and Z stages) are driven by different equations
Leptoene the number / rate of DMC1 is primarily driven by the substrate -- number of DNA loops -- empty sites are being filled.
Zygotene the number / rate of DMC1 foci are more variable across cells because it is driven by the rate / success of homolog pairing  / if this varies across cells (via the timing) - the there might be a different total number across cells

*the repair decision can happen at both stages and remove DMC1. formation of new DSB -> DMC1 foci happens at both stages, but the are triggered by different events (L: cell wide signal to start the homolog pairing and Z: unpaired bivalent signal)

back to CO number -- variation in CO number is best predicted by metrics/indicators placed BEFORE DSB formation starts.

expectations, predictions if this is the case -- more within mouse/group variation for DMC1 number in Z compared to L?
OR 

variance in L stage for DMC1 per cell would be proportional to the variation in the 'packing ratio' across pachtyene cells
-Baier et al 2014 supports this model (Chmr axis / DNA loop model)
-->


# Evolution of genome wide recombination rate is reflected at the single chromosome level 

<come up with a better section title>

```{r chrm_prop_data, echo=FALSE, warning=FALSE}
#shut off eval for now because of wd / path error
#chrm.class.DF = read.csv("~./MLH1repo/data/ChrmClassProportions.csv", header = TRUE)
#chrm.class.DF = read.csv("C:/Users/alpeterson7/Documents/MLH1repo/data/ChrmClassProportions.csv", header = TRUE)

chrm.class.DF = read.csv("~./MLH1repo/data/ChrmClassProportions2.csv", header = TRUE)

#try adding the BivData version of the proportions


#chrm.class.DF = read.csv("C:/Users/alpeterson7/Documents/MLH1repo/data/ChrmClassProportions2.csv", header = TRUE)

#this excel file is ~100 cells (10 per category) where I manually verified the number of chrom classes. (this file can also be used to quantify error for CO counts)

chrm.class.DF$fileName <- (chrm.class.DF$Original.Name)
chrm.class.DF <- add_mouse(chrm.class.DF)
chrm.class.DF <- add_strain(chrm.class.DF)
chrm.class.DF <- add_sex(chrm.class.DF)
chrm.class.DF <- add_category(chrm.class.DF)
#chrm.class.DF <- add_euth_date(meta.data.file2)
#chrm.class.DF <- add_age(meta.data.file2)

#table(chrm.class.DF$category)
#str(chrm.class.DF)

#summarize by category
proportion_stats_category  <- ddply(.data=chrm.class.DF, 
                 .(category),
                 summarize, 
                 ncells = length(unique(Original.Name)),
                 nmice = length(unique(mouse)),
                  tot_biv  = sum(biv.sum.check,  na.rm = TRUE),

                 n0CO = sum(n0CO, na.rm = TRUE),
                  n1CO = sum(n1CO,na.rm = TRUE),
                  n2CO = sum(n2CO,na.rm = TRUE),
                  n3CO = sum(n3CO,na.rm = TRUE),
                  n4CO = sum(n4CO,na.rm = TRUE),
                 
                 per0CO = sum(n0CO) / tot_biv,                 
                 per1CO= sum(n1CO)/ tot_biv,
                 per2CO= sum(n2CO)/ tot_biv,
                 per3CO= sum(n3CO)/ tot_biv,
                 per4CO= sum(n4CO)/ tot_biv,
                 
                 mean.MLH1.foci = mean(MLH1.sum.check),
                 mean.MLH1.foci2 = mean(nMLH1.foci),
                 meanQ = mean(quality)

)
#this is the table that needs melting
#but is also easy to redo by hand

#write.table(proportion_stats_category, "~./MLH1repo/prop_category.csv", sep=",", row.names = FALSE)
```

```{r plot.chrm.prop.melt, echo=FALSE, warning=FALSE}

#add subspecies to these values
melt.Chrm.Prop.table = read.csv("~./MLH1repo/data/ChrmClassProportions2_melt.csv", header = TRUE)

#melt.Chrm.Prop.table = read.csv("MLH1repo/data/ChrmClassProportions2_melt.csv", header = TRUE)

melt.Chrm.Prop.table <- melt.Chrm.Prop.table[c(1:5)]

#add the subsps

melt.Chrm.Prop.table$strain<- factor(melt.Chrm.Prop.table$strain, levels =c("LEW",
                      "WSB","G", "TOM","AST","KAZ", "SKIVE","PWD","MSM", "MOLF"), order=T )



melt.Chrm.Prop.table <- with(melt.Chrm.Prop.table, melt.Chrm.Prop.table[order(strain),])

chrm.class.barplot <- ggplot(data = melt.Chrm.Prop.table, aes(y = value, x = strain, fill=as.factor(type)) ) + geom_col(position = "dodge")+facet_grid(sex~strain, scales = "free")+ggtitle("Chrm Proportions from Hand Measured Data")+ylim(c(0,1))

#need a second DF / the first df
#+ annotate() text 
#annotate()text

#proportion_stats_category,  tot_biv, ncells
#add number of cells and bivalents in the corner

#chrm.class.barplot
```

```{r test.Chrm.Prop, echo=FALSE, eval=TRUE, warning=FALSE}

#prop.test(x, y)  x number of successes, y total  
#strain sums of n
#look up continuity correction

#mouse.level.BivData_table <-- has more male strains
#Mouse.Table_BivData_4MM <- only sex matched strains

#perform test across strain 
#removed all lines with no hand.foci.count entry
Chrm.prop_strain_table <- ddply(mouse.level.BivData_table, c("strain","sex"), summarise,
                  
                  nbiv = sum(tot.biv ),              
                 n0CO = sum(n0CO ),
                  n1CO = sum(n1CO ),
                 n2CO = sum(n2CO ),
                  n3CO = sum(n3CO ),
                 n4CO = sum(n4CO )
                          )

Chrm.prop_strain_table <- add_subsp(Chrm.prop_strain_table)

for(g in 1:length(Chrm.prop_strain_table$nbiv)){
  
  Chrm.prop_strain_table$sum2[g] <- sum(c(Chrm.prop_strain_table$n0CO[g],
                                     Chrm.prop_strain_table$n1CO[g],
                                     Chrm.prop_strain_table$n2CO[g],
                                     Chrm.prop_strain_table$n3CO[g] ))  
}

Chrm.prop_strain_table$dcheck <- Chrm.prop_strain_table$nbiv - Chrm.prop_strain_table$sum2
#there are still some wacky entries -- but much fewer than before

chrm.prop.strain.female <-  Chrm.prop_strain_table %>%  filter(sex == "female")

female.0CO.prop.test <- prop.test(chrm.prop.strain.female$n0CO, chrm.prop.strain.female$sum2 )
female.1CO.prop.test <- prop.test(chrm.prop.strain.female$n1CO, chrm.prop.strain.female$sum2 )
female.2CO.prop.test <- prop.test(chrm.prop.strain.female$n2CO, chrm.prop.strain.female$sum2 )
#the pvalues for 0 are not sig dif
#the p values for 1 and 2CO are sig diff

#seperate by subsp
chrm.prop.female.dom <-  chrm.prop.strain.female %>%  filter(subsp == "Dom")
chrm.prop.female.musc <-  chrm.prop.strain.female %>%  filter(subsp == "Musc")
chrm.prop.female.mol <-  chrm.prop.strain.female %>%  filter(subsp == "Mol")

Dom_female.0CO.prop.test <- prop.test(chrm.prop.female.dom$n0CO, chrm.prop.female.dom$sum2 )
Dom_female.1CO.prop.test <- prop.test(chrm.prop.female.dom$n1CO, chrm.prop.female.dom$sum2 )
Dom_female.2CO.prop.test <- prop.test(chrm.prop.female.dom$n2CO, chrm.prop.female.dom$sum2 )
Dom_female.3CO.prop.test <- prop.test(chrm.prop.female.dom$n3CO, chrm.prop.female.dom$sum2 )
#WSB has fewer 2COs than other 3 strains

Musc_female.0CO.prop.test <- prop.test(chrm.prop.female.musc$n0CO, chrm.prop.female.musc$sum2 )
Musc_female.1CO.prop.test <- prop.test(chrm.prop.female.musc$n1CO, chrm.prop.female.musc$sum2 )
Musc_female.2CO.prop.test <- prop.test(chrm.prop.female.musc$n2CO, chrm.prop.female.musc$sum2 )
#KAZ has slightly more 1CO than the other two


#high.female_G.LEW.
chrm.prop.female.high <-  chrm.prop.strain.female %>%  filter(strain == "G" | strain == "LEW")

high_female.0CO.prop.test <- prop.test(chrm.prop.female.high$n0CO, chrm.prop.female.high$sum2 )
high_female.1CO.prop.test <- prop.test(chrm.prop.female.high$n1CO, chrm.prop.female.high$sum2 )
high_female.2CO.prop.test <- prop.test(chrm.prop.female.high$n2CO, chrm.prop.female.high$sum2 )


#Low.female.G.LEW.
chrm.prop.female.low <-  chrm.prop.strain.female %>%  filter(strain != "G" | strain != "LEW")

low_female.0CO.prop.test <- prop.test(chrm.prop.female.low$n0CO, chrm.prop.female.low$sum2 )
low_female.1CO.prop.test <- prop.test(chrm.prop.female.low$n1CO, chrm.prop.female.low$sum2 )
low_female.2CO.prop.test <- prop.test(chrm.prop.female.low$n2CO, chrm.prop.female.low$sum2 )


#all females

chrm.prop.strain.male <-  Chrm.prop_strain_table %>%  filter(sex == "male")
male.0CO.prop.test <- prop.test(chrm.prop.strain.male$n0CO, chrm.prop.strain.male$sum2 )

male.1CO.prop.test <- prop.test(chrm.prop.strain.male$n1CO, chrm.prop.strain.male$sum2 )
male.2CO.prop.test <- prop.test(chrm.prop.strain.male$n2CO, chrm.prop.strain.male$sum2 )
#the pvals for these 3 class are sig dif

#seperate by subsp
chrm.prop.male.dom <-  chrm.prop.strain.male %>%  filter(subsp == "Dom")
chrm.prop.male.musc <-  chrm.prop.strain.male %>%  filter(subsp == "Musc")
chrm.prop.male.mol <-  chrm.prop.strain.male %>%  filter(subsp == "Mol")

Dom_male.0CO.prop.test <- prop.test(chrm.prop.male.dom$n0CO, chrm.prop.male.dom$nbiv )
Dom_male.1CO.prop.test <- prop.test(chrm.prop.male.dom$n1CO, chrm.prop.male.dom$nbiv )
Dom_male.2CO.prop.test <- prop.test(chrm.prop.male.dom$n2CO, chrm.prop.male.dom$nbiv )
#non of the Dom proportions are different

Musc_male.0CO.prop.test <- prop.test(chrm.prop.male.musc$n0CO, chrm.prop.male.musc$nbiv )
Musc_male.1CO.prop.test <- prop.test(chrm.prop.male.musc$n1CO, chrm.prop.male.musc$nbiv )
Musc_male.2CO.prop.test <- prop.test(chrm.prop.male.musc$n2CO, chrm.prop.male.musc$nbiv )
#all v different

Mol_male.0CO.prop.test <- prop.test(chrm.prop.male.mol$n0CO, chrm.prop.male.mol$nbiv )
Mol_male.1CO.prop.test <- prop.test(chrm.prop.male.mol$n1CO, chrm.prop.male.mol$nbiv )
Mol_male.2CO.prop.test <- prop.test(chrm.prop.male.mol$n2CO, chrm.prop.male.mol$nbiv )


#high vs low
chrm.prop.strain.male$Rec.group <- ifelse(grepl("PWD", chrm.prop.strain.male$strain), 1, 
                              ifelse(grepl("MSM", chrm.prop.strain.male$strain), 1,
                                 ifelse(grepl("SKIVE", chrm.prop.strain.male$strain), 1, 0)))

chrm.prop.male.low <-  chrm.prop.strain.male %>%  filter(Rec.group == 0)
chrm.prop.male.low2 <-  chrm.prop.strain.male  %>%  filter(Rec.group == 0 | strain == "SKIVE") #cool this worked


chrm.prop.male.high1 <-  chrm.prop.strain.male %>%  filter(Rec.group == 1) %>%  filter(strain != "SKIVE")
chrm.prop.male.high2 <-  chrm.prop.strain.male %>%  filter(Rec.group == 1)


Low_male.0CO.prop.test <- prop.test(chrm.prop.male.low$n0CO, chrm.prop.male.low$nbiv )
Low_male.1CO.prop.test <- prop.test(chrm.prop.male.low$n1CO, chrm.prop.male.low$nbiv )
Low_male.2CO.prop.test <- prop.test(chrm.prop.male.low$n2CO, chrm.prop.male.low$nbiv )
#the low strains do vary slightly in their 1CO and 2CO proportions

#includes SKIVE
Low2_male.0CO.prop.test <- prop.test(chrm.prop.male.low2$n0CO, chrm.prop.male.low2$nbiv )
Low2_male.1CO.prop.test <- prop.test(chrm.prop.male.low2$n1CO, chrm.prop.male.low2$nbiv )
Low2_male.2CO.prop.test <- prop.test(chrm.prop.male.low2$n2CO, chrm.prop.male.low2$nbiv )
#all very sig p values


High2_male.0CO.prop.test <-prop.test(chrm.prop.male.high2$n0CO,chrm.prop.male.high2$nbiv )
High2_male.1CO.prop.test <- prop.test(chrm.prop.male.high2$n1CO,chrm.prop.male.high2$nbiv )
High2_male.2CO.prop.test <- prop.test(chrm.prop.male.high2$n2CO,chrm.prop.male.high2$nbiv )
#these proportions are v different

#no SKIVE
High1_male.0CO.prop.test <- prop.test(chrm.prop.male.high1$n0CO, chrm.prop.male.high1$nbiv )
High1_male.1CO.prop.test <- prop.test(chrm.prop.male.high1$n1CO, chrm.prop.male.high1$nbiv )
High1_male.2CO.prop.test <- prop.test(chrm.prop.male.high1$n2CO, chrm.prop.male.high1$nbiv )
#MSM and PWD only slightly sig dif 2CO

#MSM-MOLF
MSM.MOLF.0CO.prop.test <- prop.test( (chrm.prop.strain.male %>% filter(subsp == "Mol") )$n0CO, (chrm.prop.strain.male %>% filter(subsp == "Mol"))$nbiv  )

MSM.MOLF.1CO.prop.test <- prop.test( (chrm.prop.strain.male %>% filter(subsp == "Mol") )$n1CO, (chrm.prop.strain.male %>% filter(subsp == "Mol"))$nbiv  )

MSM.MOLF.2CO.prop.test <- prop.test( (chrm.prop.strain.male %>% filter(subsp == "Mol") )$n2CO, (chrm.prop.strain.male %>% filter(subsp == "Mol"))$nbiv  ) 
 

#PWD-KAZ
PWD.KAZ.0CO.prop.test <- prop.test( (chrm.prop.strain.male %>% filter( (strain == "PWD") | strain == "KAZ") )$n0CO, (chrm.prop.strain.male %>% filter(strain == "PWD" | strain == "KAZ") )$nbiv )


PWD.KAZ.1CO.prop.test <- prop.test( (chrm.prop.strain.male %>% filter( (strain == "PWD") | strain == "KAZ") )$n1CO, (chrm.prop.strain.male %>% filter(strain == "PWD" | strain == "KAZ") )$nbiv )


PWD.KAZ.2CO.prop.test <- prop.test( (chrm.prop.strain.male %>% filter( (strain == "PWD") | strain == "KAZ") )$n2CO, (chrm.prop.strain.male %>% filter(strain == "PWD" | strain == "KAZ") )$nbiv )


```

```{r Chrm.props.plot2, echo=FALSE}

#barplots for proportions of chrm classes

#consider adding this the the setup block (could have done this to category level table)
mouse.level.BivData_table$per.0CO <- mouse.level.BivData_table$n0CO / mouse.level.BivData_table$tot.biv

mouse.level.BivData_table$per.1CO <- mouse.level.BivData_table$n1CO / mouse.level.BivData_table$tot.biv

mouse.level.BivData_table$per.2CO <- mouse.level.BivData_table$n2CO / mouse.level.BivData_table$tot.biv

mouse.level.BivData_table$per.3CO <- mouse.level.BivData_table$n3CO / mouse.level.BivData_table$tot.biv

mouse.level.BivData_table$per.4CO <- mouse.level.BivData_table$n4CO / mouse.level.BivData_table$tot.biv

#DF needs to be melted -- so they all in 1 col
#per code, I've done this by hand

#subset DF for ease
skinny.mouse.level <- mouse.level.BivData_table[-c(12:21) ]
col.names.list.99 <- colnames(skinny.mouse.level)

#all of the cols to use
MELT_Chrm.Proportion <- melt(skinny.mouse.level[,c(  "mouse", "ncells","boxs.IDd", "nboxes_passed","Nbiv","tot.biv","n0CO","n1CO","n2CO","n3CO","n4CO",
             
               #leave out the Foci
                "per.0CO", "per.1CO", "per.2CO", "per.3CO", "per.4CO")], 
               
  id=c("mouse", "ncells","boxs.IDd",    "nboxes_passed","Nbiv","tot.biv","n0CO","n1CO","n2CO","n3CO","n4CO"))

MELT_Chrm.Proportion <- MELT_Chrm.Proportion[(!is.na(MELT_Chrm.Proportion$value)),]
MELT_Chrm.Proportion <- MELT_Chrm.Proportion[!(is.na(MELT_Chrm.Proportion$value) | MELT_Chrm.Proportion$value==0), ]

#no filename for add mouse, strain subsp
MELT_Chrm.Proportion <- add_category(MELT_Chrm.Proportion)
MELT_Chrm.Proportion <- add_sex(MELT_Chrm.Proportion)
MELT_Chrm.Proportion <- add_strain(MELT_Chrm.Proportion)


#then this DF has to be summarized into categories
chrm.class.category_MELT <- ddply(MELT_Chrm.Proportion, c("strain","sex", "variable"), summarise,
                             mean.percent.type = mean(value)
                             #sum is the wrong!
                             #used average percentage, might consider a different summary
)

#can't apply function use this
chrm.class.category_MELT$subsp <- ifelse(grepl("WSB", chrm.class.category_MELT$strain), "Dom", 
             ifelse(grepl("G", chrm.class.category_MELT$strain), "Dom",
                 ifelse(grepl("LEW", chrm.class.category_MELT$strain), "Dom", 
                ifelse(grepl("PERC", chrm.class.category_MELT$strain), "Dom",
                                                            
          ifelse(grepl("CAST", chrm.class.category_MELT$strain), "Cast",
              ifelse(grepl("HMI", chrm.class.category_MELT$strain), "Cast",
                                                                                 
                                       ifelse(grepl("MSM", chrm.class.category_MELT$strain), "Mol", #added Mol as distinct subsp class
                                      ifelse(grepl("MOLF", chrm.class.category_MELT$strain), "Mol",    
            ifelse(grepl("PWD", chrm.class.category_MELT$strain), "Musc", 
                     ifelse(grepl("CZECH", chrm.class.category_MELT$strain), "Musc", 
                 ifelse(grepl("AST", chrm.class.category_MELT$strain), "Musc",     
                      ifelse(grepl("TOM", chrm.class.category_MELT$strain), "Musc",  
                   ifelse(grepl("SKIVE", chrm.class.category_MELT$strain), "Musc",
                        ifelse(grepl("KAZ", chrm.class.category_MELT$strain), "Musc",
                                                                                                                                              
             ifelse(grepl("SPRET", chrm.class.category_MELT$strain), "Spretus",
               ifelse(grepl("SPIC", chrm.class.category_MELT$strain), "Spic",  "other"))))))))))))))))
  

chrm.class.category_MELT$subsp <- factor(chrm.class.category_MELT$subsp, ordered = TRUE,levels =c( "Dom",                    "Cast",
                                                                 "Musc",
                                                                 "Mol",
                                                                 "Spretus","Spic",
                                                                 "other") )
#ggplot

chrm.class.barplot <- ggplot(data = chrm.class.category_MELT, aes(y = mean.percent.type, x = strain, fill=as.factor(variable)) ) + geom_col(position = "dodge")+ggtitle("Chrm Proportions from BivData Data")+ylim(c(0,1)) + facet_wrap(sex~subsp, scales = "free")
#chrm.class.barplot
chrm.class.barplot

```

```{r Chrm.Prop.misc.calulations, echo=FALSE}

total.biv.all <- sum( MELT_Chrm.Proportion$n0CO, MELT_Chrm.Proportion$n1CO, MELT_Chrm.Proportion$n2CO, MELT_Chrm.Proportion$n3CO, MELT_Chrm.Proportion$n4CO)

#sum( MELT_Chrm.Proportion$n1CO) / total.biv.all
#sum( MELT_Chrm.Proportion$n2CO) / total.biv.all

```

<!--
- This is motivation to investigate more traits at the chromosome level
1.	Basic stats - bivalent data numbers 
2.	Rang of co per chrm, mostly 1co 2co
3.	High rec strains have evolved a higher proportion of 2CO
For males Chi squared test, p = low group,  p = MSM and PWD,  SKIVE is intermediate)
MSM 60% PWD 50% SKIVE 30% 20 - 10% remaining (low) strains
4.	explain why this complicates further comparisons (chromosome size effect * chromosome cl.ass)
-->

<from the means the expected number of COs per chroosomes ranges from X to X we confirm these numbers Can't have fractions of COs>

We distill the variance observed in the mean MLH1 foci counts per cell to number of MLH1 foci per chromosomes (bivalent). 

Because our observations of MLH1 foci are made in pachtyene cells, our chromosome opbservations are fully synapsesed paired homologous chromosomes (i.e. 4n copies of each chromsomes) -- additionally we can distinguish the idenetiy of indiviual chromsomes (Chromsome 1 or Chromsome 2)
for these reasons we use the term bivalent.
 
Of the pooled dataset of `r total.biv.all` single bivalents, 96% of bivalents had either one or two crossovers. <these results fit with previous observations on the number of crossovers per chromosome (Stapley 2019)>

The proportion of 1CO:2CO (are displayed in figure X) --

-- distinguishes -- the high and low rec males



Most of the bivalents are 1CO and 2CO class bivalents -- and the proportion of these two classes distinguish high and low rec male strains. (distinguish the high rec males from minimal rec groups)

(A previously reported for house mouse, the most prevelent class of chromosomes is the 1CO class. The high rec group of males are the exception indicating, which fits with the conclusion that higher cell wide CO counts are due to more chromosomes(bivalents) have 2 instead of 1 CO.  *(much smaller proportions and variation in the fringe classes*


**2. most of the male proportions are 2/3's 1CO and 1/3 2COs. but The high rec strains are distinguishable by having more 2COs at the cost of 1COs, with closer to 1:1**

**MSM 60%  PWD 50%  SKIVE 30%   20 - 10% remaining (low) strains**

PWD and MSM have slightly different proportions of 1CO and 2CO, (with MSM having more 2COs) (chi-square test; 1CO p =`r High1_male.1CO.prop.test$p.value` and 2CO p = `r High1_male.2CO.prop.test$p.value` ). 

While the proportions are significantly different, PWD male and MSM male proportions are more similar to each other than strains from the same subspcies (PWD and KAZ chi-square test; 1CO p =`r PWD.KAZ.1CO.prop.test$p.value`, 2CO p=`r PWD.KAZ.2CO.prop.test$p.value`)
 and MSM and MOLF chi-square test; 1CO p = `r MSM.MOLF.1CO.prop.test$p.value`, 2CO p=`r MSM.MOLF.2CO.prop.test$p.value`).


**3. these results confirm the intuitive conclusion/result that the rapid male specific evolution in gwRR proceeds though the chromosome level/though increasing the number of 2CO bivalents at the expense of 1CO bivalents**

The general pattern is that most of the chromosom classes are from 1CO (`r sum( MELT_Chrm.Proportion$n1CO) / total.biv.all`) and 2CO (`r sum( MELT_Chrm.Proportion$n2CO) / total.biv.all`). <This relpicates what has been notice in the literature (cite stapley).> 

Six strain have been designated low rec groups, WSB, LEW, G, KAZ, CZECH, and MOLF.
They have significantly different proportions for 1CO and 2CO (chi-square; 1CO p = `r Low_male.1CO.prop.test$p.value` and 2CO p= `r Low_male.2CO.prop.test$p.value`). This significant p value indicates that at least one of the six strains had significant differences).>

Breif conclusion: the evolution of genome wide recombiantion rates proceeds through the chromosome level.

<something about the female results>

<SUMMARY the / TRANSITION to single bivalen measures>


# Single Bivalent Level Results

In order to better deconstruct the genome wide recombiation rate, we decided to deconstruct cell wide average by examining features of the recombiantion landscape at the single bivalent level. recombination landscape refers to the i) total length of the SC-AE (potential area for COs), ii) the position of single crossovers along bivalents and iii) the spacing between multiple COs on the same bivalent.

From our total set of cell images `r length(unique(new.Curate.FULL.org %>% filter(SC.pass == "1"))$fileName )` chromosome objects were isolated by the image analysis software. After the human curation step, `r length(unique(new.Curate.FULL.org %>% filter(hand.foci.count != ""))$Obj.ID )` single bivalent measures were left. (make sure these are the hand.foci bivalents).


```{r algorithm_performance_stats, echo=FALSE}

#sort by category, 
#compare total boxes and rate passed
#Curated_BivData  -- pass rate?

#number of mice, number of cells
#mean number of isolate bivs per cell,
#use these two metrics
#cell.level.BivData_table$boxs.IDd 
#cell.level.BivData_table$nboxes_passed

#calq the mean rate by category

cell.level.BivData_table$pass.rate <- cell.level.BivData_table$nboxes_passed / cell.level.BivData_table$boxs.IDd 

algorithm.stat.mouse <- ddply(cell.level.BivData_table, c("mouse"), summarise,
                        n_obs = length( unique(fileName) ),
                         mean_pass.rate = mean(pass.rate)
                         )

cell.level.BivData_table <- add_sex(cell.level.BivData_table)

algorithm.stat.category <- ddply(cell.level.BivData_table, c("sex", "strain"), summarise,
                        n_obs = length(unique(fileName)),
                        mean_pass.rate = mean(pass.rate)
                         )

#showing the table..
#data.frame(table( droplevels( Curated_BivData  %>%  filter(hand.foci.count  == 0 | hand.foci.count  == 1 |                                       hand.foci.count  == 2 | hand.foci.count  == 3 | hand.foci.count  == 4 ) )$category ) )
```

<Table X illustrates the break down of single bivalent observations by category. Error compared to human measures and other details have been discribed in (Peterson_2019).>

The ranges of of rates of isolating chromosomes per cell are `r min(algorithm.stat.category$mean_pass.rate)` in _molossinus^MSM^_ male to 
`r max(algorithm.stat.category$mean_pass.rate)` ) _musculus^KAZ^_ female, we assume that the isolation process is not biased. Because there are hundreds of observations per category, we assume that each of the 19 or 20 chromosomes are equally represented in male and female single bivalent data datasets. We assume the bivalent datasets across all categories are equally eqivilent for representing general patterns.

<curation, x from curation, XY excluded, replication to hand measures>

Using this data set, We address two main questions 1) Which traits are sexually dimorphic? and 2) which traits fit distinsutinguish the high and low recombining strains for males?

<These questions use different datasets: Q1 uses a dataset with only sex matched strains  and Q2 uses the data set with all male observations with including strains including those not in the Q1 set.>
<The same basic models from the MLH1 counts per cell will be used. In addition to basic t.tests and logistic regression models for Q2 to distinguish betwee high and low recombining strains.>


# Q1 Sex Differences in SC-AE Lengths

DO THE HIGH QUALITY dataset tests

<!-- breif background on longer female SC  (mouse, human) Gruhn-->

Our data set provides an opportunity to test if the canonical pattern of females having higher gwRR and more SC area) have been uncoupled in the instance of rapid male specific evolution in the gwRR of _musculus^PWD^_  and _molossinus^MSM^_  A simple model might predict if SC area is the strongest predictor for gwRR / CO number, the high recombining males would have greater SC area proportional to the difference in CO number per cell.

<Comparing the means of single bivalents is complicated by the fact we can't identify the chromosome identitied the physical size (in terms of SC-AE) is the closet approximation we can make.>


```{r Q1.Short.Long_biv.setup, echo=FALSE}

#REAL.long_mouse_table
#REAL.short_mouse_table

#make the Q1 strain list
Q1.real.short_mouse_table <- REAL.short_mouse_table %>% filter( (strain != "CZECH") & (strain != "MOLF") )

Q1.real.long_mouse_table <- REAL.long_mouse_table %>% filter( (strain != "CZECH") & (strain != "MOLF") )

```

We use a reduced bivalent data set made up of chromsomes (below the 2nd quartile for SC length of single cells) to make single bivalent groups which exclude the XX homologs from the female data. A total of `r sum(Q1.real.short_mouse_table$n_obs)` 'short' bivalents were isolated from `r sum(Q1.real.short_mouse_table$ncells)` oocytes and spermatocytes.

<Even though this smaller data set has decreased power due to redcued sample size, it offers a mors comparable set of SC-AE meterics across the sexes>

This 'short biv' data set has advantages in that it is closer to sampling across a similar set of chromosomes (ideally Chromosomes 15 to 19) and it excludes the XX in females, which is predicted to be the 3rd longest based on the physical (Mb) length.

```{r Q1.show.SC.length, echo=FALSE, warning=FALSE}

Curated_BivData99 <- Curated_BivData[Curated_BivData$hand.foci.count < 5,]
Curated_BivData99 <- Curated_BivData99[(!is.na(Curated_BivData99$sex)),]

#CLEAN UP THE FULL DATA SET

#adjust order of plot here
Curated_BivData99$strain <- factor(Curated_BivData99$strain,levels =c(
  "WSB","G","LEW","PERC",  
"PWD", "SKIVE",  "KAZ",
     "MSM", "MOLF",   
                     "CAST",
                         "SPRET","SPIC" ), order=T )


#nested boxplots, pooled Bivalents

Q1.SC.length_Nested_boxplots_Dom <- ggplot(Curated_BivData99[Curated_BivData99$subsp == "Dom",], aes(y = chromosomeLength, x=strain, fill=as.factor(hand.foci.count) ))+geom_boxplot(alpha=0.5)+ggtitle("Bivalent SC Lengths by Chrm Class \n Dom")+facet_wrap(subsp~sex, scales="free")+ylim(c(25,255))+theme(legend.position="none") 

Q1.SC.length_Nested_boxplots_Musc <- ggplot(Curated_BivData99 %>% filter(subsp == "Musc") %>% filter(strain != "CZECH"), aes(y = chromosomeLength, x=strain, fill=as.factor(hand.foci.count) ))+geom_boxplot(alpha=0.5)+ggtitle("Bivalent SC Lengths by Chrm Class \nMusc")+facet_wrap(subsp~sex, scales="free")+ylim(c(25,255))+theme(legend.position="none") 

Q1.SC.length_Nested_boxplots_Mol <- ggplot(Curated_BivData99 %>% filter(strain == "MSM"), aes(y = chromosomeLength, x=as.factor(hand.foci.count), fill=as.factor(hand.foci.count) ))+geom_boxplot(alpha=0.5)+geom_jitter(alpha=.3)+
  ggtitle("Bivalent SC Lengths by Chrm Class\n Mol")+facet_wrap(strain~sex, scales="free")+ylim(c(25,255))+theme(legend.position="none") 

```

```{r Q1.SC.show.short.biv, echo=FALSE}
Q1.SC.short.mouse_scatter_boxplot <- ggplot( Q1.real.short_mouse_table, aes(y = mean_SC, x = sex, color=strain)) + geom_jitter()+ geom_boxplot(alpha=.2) +scale_color_manual(values=colors_of_strains)+facet_wrap(~subsp, scales="free")+ylim(c(20,115) )+ggtitle("Mouse average SC Length \n short bivalent dataset")

Q1.SC.short.mouse_scatter_boxplot

#ploting SKIVE alone, confirms the ranges overlap, KAZ also have close ranges
skiveQ1.SC.short.mouse_scatter_boxplot <- ggplot( Q1.real.short_mouse_table %>% filter(strain == "SKIVE"), aes(y = mean_SC, x = sex, color=strain)) + geom_jitter()+ geom_boxplot(alpha=.2) +scale_color_manual(values=colors_of_strains)+facet_wrap(~subsp, scales="free")+ylim(c(20,115) )+ggtitle("Mouse average SC Length \n short bivalent dataset")
#skiveQ1.SC.short.mouse_scatter_boxplot
#there are 2 lowish female means, 

```

```{r Q1_SC.biv.short_ttest, echo=FALSE}

#run t.tests for the mouse av. for reduced single bivalent datasets
#WSB
Q1_mus.av_shortSC_t.test.WSB <- t.test((Q1.real.short_mouse_table %>% filter(sex == "male") %>% filter(strain == "WSB") )$mean_SC,
                           (Q1.real.short_mouse_table %>% filter(sex == "female") %>% filter(strain == "WSB") )$mean_SC)

#G
Q1_mus.av_shortSC_t.test.G <- t.test((Q1.real.short_mouse_table %>% filter(sex == "male") %>% filter(strain == "G") )$mean_SC,
                           (Q1.real.short_mouse_table %>% filter(sex == "female") %>% filter(strain == "G") )$mean_SC)

#LEW
Q1_mus.av_shortSC_t.test.LEW <- t.test((Q1.real.short_mouse_table %>% filter(sex == "male") %>% filter(strain == "LEW") )$mean_SC,
                           (Q1.real.short_mouse_table %>% filter(sex == "female") %>% filter(strain == "LEW") )$mean_SC)

#PWD
Q1_mus.av_shortSC_t.test.PWD <- t.test((Q1.real.short_mouse_table %>% filter(sex == "male") %>% filter(strain == "PWD") )$mean_SC,
                           (Q1.real.short_mouse_table %>% filter(sex == "female") %>% filter(strain == "PWD") )$mean_SC)

#KAZ
Q1_mus.av_shortSC_t.test.KAZ <- t.test((Q1.real.short_mouse_table %>% filter(sex == "male") %>% filter(strain == "KAZ") )$mean_SC,
                           (Q1.real.short_mouse_table %>% filter(sex == "female") %>% filter(strain == "KAZ") )$mean_SC)

#SKIVE
Q1_mus.av_shortSC_t.test.SKIVE <- t.test((Q1.real.short_mouse_table %>% filter(sex == "male") %>% filter(strain == "SKIVE") )$mean_SC,
                           (Q1.real.short_mouse_table %>% filter(sex == "female") %>% filter(strain == "SKIVE") )$mean_SC)

#MSM
#Q1_mus.av_shortSC_t.test.MSM <-t.test( (Q1.real.short_mouse_table %>% filter(sex == "male") %>% filter(strain == "MSM") )$mean_SC, 
#          (Q1.real.short_mouse_table %>% filter(sex == "female") %>% filter(strain == #"MSM"))$mean_SC )
#1 male, 3 females measures
#try to merge the manual - whole cell observations (at least 2 males)

```

<in the above boxplots of mouse means for shortbivalents>

For all but one strain, _musculus^SKIVE^_, the mouse mean for the short bivalents are significantly longer <how much longer, the largest difference across sexes by strain are>  (t.test; p = `r Q1_mus.av_shortSC_t.test.WSB$p.value`, p =`r format(Q1_mus.av_shortSC_t.test.G$p.value, scientific = F)`, and p=`r format(Q1_mus.av_shortSC_t.test.LEW$p.value, scientific = F)`
for _domesticus^WSB^_, _domesticus^G^_ and _domesticus^LEW^_ respectively.
This pattern is also true for the musculus strins (t.test; p= `r format(Q1_mus.av_shortSC_t.test.PWD$p.value, scientific = F)`, `r Q1_mus.av_shortSC_t.test.KAZ$p.value`, and `r  format(Q1_mus.av_shortSC_t.test.SKIVE$p.value, scientific = F)` for _musculus^PWD^_ , _musculus^KAZ^_ , _musculus^SKIVE^_  respectively. 


<EXPLAIN WHY USING TOTAL SC AREA>

An additional measure of the SC-AE we tested is the total SC area per cell (the whole genome).**to get a comparison of the SC-AE area for the whole genome (not just a subset of chromosomes)** We apply a skeletonizing image analysis algorithm to all MLH1 meiocyte spreads. A total of X images were run trhought the image analysis pipeline, and after outliers removed, X remained. / cell based measures were left. Mouse means were calculated for their total SC-AE area and are displayed in figure X with the strain means shown as horizontal lines (below).

```{r Q1.tot.SC_show, echo=FALSE, warning=FALSE}

Q1.totSC.mouse.av.table <-  totSC.mouse.av.table ##%>% filter(species == "") 
  
Q1.totSC.mouse.av.table$strain <- factor(Q1.totSC.mouse.av.table$strain,levels =c(
   "WSB","G","LEW","PERC", 
  "PWD", "SKIVE",  "KAZ",
  "MSM", "MOLF"), order=T )

Q1.totSC.mouse.av.table <- with(Q1.totSC.mouse.av.table, Q1.totSC.mouse.av.table[order(sex, strain),])

#order the strains
Q1.totSC.mouse.av.table <- Q1.totSC.mouse.av.table %>%
  arrange(strain) %>%
  mutate(mouse = factor(mouse))

Q1.totSC.mouse.av.table <- with(Q1.totSC.mouse.av.table, Q1.totSC.mouse.av.table[order(sex, strain),])

#add category mean
#totSC.strain.av.table %>% filter
totsc.dom_tot.sc <- Q1.totSC.mouse.av.table %>% filter( subsp == "Dom" )

totsc_strain.av.dom <- totSC.strain.av.table %>% filter( subsp == "Dom" )

Tot.SC_mouse.av_Dom <- ggplot(data= totsc.dom_tot.sc, aes(y = mean.skel, x=mouse, color=strain))+ geom_point()+scale_color_manual(values=colors_of_strains)+
  facet_wrap(~sex, scales="free", ncol = 3)+ 
  
  geom_hline(data =totsc_strain.av.dom, aes(yintercept=mean.skel) )+
  
  theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank())+  ylim(c(1000,2600 ))+ggtitle("Mouse av. total SC area")

Tot.SC_mouse.av_Dom



totsc.musc <- Q1.totSC.mouse.av.table %>% filter( subsp == "Musc" )
totsc_strain.av.msc <- totSC.strain.av.table %>% filter( subsp == "Musc" )

Tot.SC_mouse.av_Musc <- ggplot(data= totsc.musc, aes(y = mean.skel, x=mouse, color=strain))+ geom_point()+scale_color_manual(values=colors_of_strains)+
  facet_wrap(~sex, scales="free", ncol = 3)+ 
  
  geom_hline(data =totsc_strain.av.msc, aes(yintercept=mean.skel) )+
  
  theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank())+  ggtitle("Mouse av. total SC area")+  ylim(c(1000,3000 ))

Tot.SC_mouse.av_Musc


totsc.mol <- Q1.totSC.mouse.av.table %>% filter( subsp == "Mol" )
totsc_strain.av.mol <- totSC.strain.av.table %>% filter( subsp == "Mol" )

Tot.SC_mouse.av_Mol <- ggplot(data= totsc.mol, aes(y = mean.skel, x=mouse, color=strain))+ geom_point()+scale_color_manual(values=colors_of_strains)+
  facet_wrap(~sex, scales="free", ncol = 3)+ 
  
  geom_hline(data =totsc_strain.av.mol, aes(yintercept=mean.skel) )+
  
  theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank())+  ylim(c(1000,3000 ))+ggtitle("Mouse av. total SC area")

Tot.SC_mouse.av_Mol
```

<these are the number of mice for each of the categories r as.data.frame( table(droplevels( totSC.mouse.av.table$category) ) )

```{r Q1.total.SC_t.test, echo=FALSE}

#run t-test for male female avergage for each strain

#WSB
Q1_tot.sc_t.test.WSB <- t.test((totSC.mouse.av.table %>% filter(sex == "male") %>% filter(strain == "WSB") )$mean.skel,
                           (totSC.mouse.av.table %>% filter(sex == "female") %>% filter(strain == "WSB") )$mean.skel)
#G
Q1_tot.sc_t.test.G <- t.test((totSC.mouse.av.table %>% filter(sex == "male") %>% filter(strain == "G") )$mean.skel,
                           (totSC.mouse.av.table %>% filter(sex == "female") %>% filter(strain == "G") )$mean.skel)
#LEW
Q1_tot.sc_t.test.LEW <- t.test((totSC.mouse.av.table %>% filter(sex == "male") %>% filter(strain == "LEW") )$mean.skel,
                           (totSC.mouse.av.table %>% filter(sex == "female") %>% filter(strain == "LEW") )$mean.skel)

#PWD
Q1_tot.sc_t.test.PWD <- t.test((totSC.mouse.av.table %>% filter(sex == "male") %>% filter(strain == "PWD") )$mean.skel,
                           (totSC.mouse.av.table %>% filter(sex == "female") %>% filter(strain == "PWD") )$mean.skel)

#KAZ
Q1_tot.sc_t.test.KAZ <- t.test((totSC.mouse.av.table %>% filter(sex == "male") %>% filter(strain == "KAZ") )$mean.skel,
                           (totSC.mouse.av.table %>% filter(sex == "female") %>% filter(strain == "KAZ") )$mean.skel)

#SKIVE
#Q1_tot.sc_t.test.SKIVE <- t.test((totSC.mouse.av.table %>% filter(sex == "male") #%>% filter(strain == "SKIVE") )$mean.skel,
#           (totSC.mouse.av.table %>% filter(sex == "female") %>% filter(strain == #"SKIVE") )$mean.skel, paired = FALSE, alternative = "greater")

#error not enought female observations

#MSM
Q1_tot.sc_t.test.MSM <- t.test((totSC.mouse.av.table %>% filter(sex == "male") %>% filter(strain == "MSM") )$mean.skel,
                           (totSC.mouse.av.table %>% filter(sex == "female") %>% filter(strain == "MSM") )$mean.skel)

```

```{r Q1.totSC_male.first.comp, echo=FALSE}

Q1.male.Dom.Musc.t.test <- t.test( ( totSC.mouse.av.table %>% filter(sex == "male") %>% filter(subsp == "Dom") )$mean.skel,
(totSC.mouse.av.table %>% filter(sex == "male") %>% filter(subsp == "Musc") )$mean.skel)

Q1.male.Dom.Mol.t.test <- t.test( ( totSC.mouse.av.table %>% filter(sex == "male") %>% filter(subsp == "Dom") )$mean.skel,
(totSC.mouse.av.table %>% filter(sex == "male") %>% filter(subsp == "Mol") )$mean.skel)
```


The most averages for total SC-AE area (skeletonizes) are significantly longer in females for all strains (p=
`r format(Q1_tot.sc_t.test.WSB$p.value, scientific = F )`, `r format(Q1_tot.sc_t.test.G$p.value, scientific= F )`, and `r format(Q1_tot.sc_t.test.LEW$p.value, scientific= F )`
for the _domesticues^WSB^ _, _domesticues^G^ _, and _domesticues^LEW^ _ respectively).

Some of the p values for PWD and MSM are much closer (less significant), ( p = `r format(Q1_tot.sc_t.test.PWD$p.value,scientific = F )`, `r format(Q1_tot.sc_t.test.KAZ$p.value, scientific = F )`, and `r format(Q1_tot.sc_t.test.MSM$p.value, scientific = F )` for the _musculus^PWD^ _, _musculus^KAZ^ _, and _molossinus^MSM^_ respectively)

The comparison of this metric is somewhat complicated across sexes due to the sex chromosomes. (simply put, the oocytes have an additional autosome, and the spermatocytes have mroe mostly non-recombinationly active SC-AE area, the XY).  <But we use this metric anyways>

<We estimate the effect of the sex chromosomes from hand measured data...the XX (3rd longest chromsomes) in females is ~10% of total the summed SC area, the average propotion of the XY in spermatocytes is XX%?  however the morphology of the XY is highly variable across cells>


The mouse means for total SC in domesticus strains are significant shorter compared to Musc (t.test; p= `r Q1.male.Dom.Musc.t.test$p.value`) and Molossinus (t.test; p = `r Q1.male.Dom.Mol.t.test$p.value`, indecating evolution (divergence) of SC length and (chromatin compaction) <this will be expanded on n later sections>
-- we follow up with models (which focus on sex differences). 

<explain why applying models.  We then use models to examine/ think about the evolution>

The above t.tests were done to establish the sex differences for SC-AE (chromatin condensation), we now apply models (for understanding within an evolutionary framework) -- we apply the same models to both mean mouse total sc and mouse mean short bivalent.

**M3. Mixed Model** $$mouse \ average \ SC \ metric ~=~ subsp * sex + rand(strain) + \varepsilon $$

```{r Q1.lmer_totalSC, echo=FALSE, warning=FALSE, message=FALSE}

Q1.lmer_totSC.mouse.av.table <- totSC.mouse.av.table %>% filter(strain != "AST" & strain != "TOM" & strain != "CZECH" & strain != "MOLF")

#str(Q1.lmer_totSC.mouse.av.table)

#unorder factors
Q1.lmer_totSC.mouse.av.table$subsp <- factor(Q1.lmer_totSC.mouse.av.table$subsp, ordered = FALSE)
Q1.lmer_totSC.mouse.av.table$sex <- factor(Q1.lmer_totSC.mouse.av.table$sex, ordered = FALSE)
Q1.lmer_totSC.mouse.av.table$strain <- factor(Q1.lmer_totSC.mouse.av.table$strain, ordered = FALSE)
Q1.lmer_totSC.mouse.av.table$species <- factor(Q1.lmer_totSC.mouse.av.table$species, ordered = FALSE)


#TOTAL SC AREA!!
lmer.Q1_mus.av_totSC <- lmer(mean.skel ~ subsp * sex + (1|strain), data=Q1.lmer_totSC.mouse.av.table, REML = TRUE) 

lmer.Q1_mus.av_totSC_R.sex <- lmer(mean.skel ~ subsp + (1|strain), data=Q1.lmer_totSC.mouse.av.table, REML = TRUE) 

lmer.Q1_mus.av_totSC_R.subsp <- lmer(mean.skel ~ sex + (1|strain), data=Q1.lmer_totSC.mouse.av.table, REML = TRUE) 


lmer.Q1_mus.av_totSC_interaction  <-drop1(lmer.Q1_mus.av_totSC, test="Chisq" )$`Pr(Chi)`[2]

lmer.Q1_mus.av_totSC_sex  <-anova(lmer.Q1_mus.av_totSC, lmer.Q1_mus.av_totSC_R.sex)$`Pr(>Chisq)`[2]

lmer.Q1_mus.av_totSC_subp  <- anova(lmer.Q1_mus.av_totSC, lmer.Q1_mus.av_totSC_R.subsp)$`Pr(>Chisq)`[2]

lmer.Q1_mus.av_totSC_strain  <- exactRLRT(lmer.Q1_mus.av_totSC)$p.value 

#as.data.frame(round(summary(lmer.SCLength_all_MM.M1)$coefficients, 6))#these Dont return p values for the fixed effects

Q1.lmer.totSC_results <- data.frame(sub.sp = lmer.Q1_mus.av_totSC_subp, 
sex.results =  lmer.Q1_mus.av_totSC_sex, 
               interaction.pval = lmer.Q1_mus.av_totSC_interaction,  rand.pval = 
lmer.Q1_mus.av_totSC_strain)

#Q1.lmer.totSC_results

#G
```

For the mixed model run on the total sc metric, there are more significant effects for strain and subsp for the mixed model of mouse.av total sc.

The three fixed effects and one random effect were significant (LTR; subsp p= `r Q1.lmer.totSC_results$sub.sp`, 
sex p = `r Q1.lmer.totSC_results$sex.results`, and interaction p = `r Q1.lmer.totSC_results$interaction.pval`, (random LRT; strain p = `r Q1.lmer.totSC_results$rand.pval`).


```{r Q1_short_biv_lmer, echo=FALSE, warning=FALSE, message=FALSE}
lmer.Q1_SCLength_short_M1 <- lmer(mean_SC ~ subsp * sex + (1|strain), data=Q1.real.short_mouse_table, REML = TRUE) 

lmer.Q1_SCLength_short_M1_R.sex <- lmer(mean_SC ~ subsp + (1|strain), data=Q1.real.short_mouse_table, REML = TRUE) 

lmer.Q1_SCLength_short_M1_R.subsp <- lmer(mean_SC ~ sex + (1|strain), data=Q1.real.short_mouse_table, REML = TRUE) 

Q1lmer.MM_SC_short_interaction  <-drop1(lmer.Q1_SCLength_short_M1, test="Chisq" )$`Pr(Chi)`[2]
Q1lmer.MM_SC_short_sex  <-anova(lmer.Q1_SCLength_short_M1, lmer.Q1_SCLength_short_M1_R.sex)$`Pr(>Chisq)`[2]
Q1lmer.MM_SC_short_subp  <- anova(lmer.Q1_SCLength_short_M1, lmer.Q1_SCLength_short_M1_R.subsp)$`Pr(>Chisq)`[2]
Q1lmer.MM_SC_short_strain  <- exactRLRT(lmer.Q1_SCLength_short_M1)$p.value 

#as.data.frame(round(summary(lmer.SCLength_all_MM.M1)$coefficients, 6))#these Dont return p values for the fixed effects

Q1lmer.MM_SC_short_results <- data.frame(sub.sp = Q1lmer.MM_SC_short_subp, 
sex.results =  Q1lmer.MM_SC_short_sex, 
               interaction.pval = Q1lmer.MM_SC_short_interaction,  rand.pval = 
Q1lmer.MM_SC_short_strain)


#Q1lmer.MM_SC_short_results

```


Mixed models for the sex differences in mouse averages for short bivalents Sex is the most significant factor for the mixed model of mean short bivalents (LTR; p =`r Q1lmer.MM_SC_short_results$sex.results`). The interaction (subspecies by sex) effect also significant (p = `r Q1lmer.MM_SC_short_results$interaction.pval`).

These effects were followed up the glms, again applied to both mouse means for short biv SC lengths and total sc area per cell.

**M2.Linear Model** $$mouse \ average \ SC \ metric ~=~ sex * strain + \varepsilon $$

**M3. Linear Model** $$mouse \ average \ SC \ metric ~=~ sex * strain + \varepsilon $$

```{r Q1.glm_totalSC_M2.M3, echo=FALSE, message=FALSE}
#total-SC
#Q1.lmer_totSC.mouse.av.table

Q1.glm_totSC_M2 <- glm(mean.skel ~ subsp*strain * sex, data=Q1.lmer_totSC.mouse.av.table)
sum.Q1.glm_totSC_M2 <- summary(Q1.glm_totSC_M2)

#subsp effect, many strain, #male
#G * male v.sig and male*LEW

Q1.glm_totSC_M3 <- glm(mean.skel ~ strain * sex, data=Q1.lmer_totSC.mouse.av.table)
sum.Q1.glm_totSC_M3 <- summary(Q1.glm_totSC_M3)
#LOT OF SIGNIFICant strain effects in this model
```


```{r Q1.SC_short_M2, echo=FALSE, warning=FALSE, message=FALSE}

Q1.real.short_mouse_table$subsp <- factor(Q1.real.short_mouse_table$subsp, ordered = FALSE)
Q1.real.short_mouse_table$sex <- factor(Q1.real.short_mouse_table$sex, ordered = FALSE)
Q1.real.short_mouse_table$strain <- factor(Q1.real.short_mouse_table$strain, ordered = FALSE)

Q1.glm_short.SC_M2 <- glm(mean_SC ~ subsp * strain * sex, data=Q1.real.short_mouse_table)

#summary(Q1.glm_short.SC_M2)

summ.Q1.glm_short.SC_M2 <- summary(Q1.glm_short.SC_M2)

#round(summ.Q1.glm_short.SC_M2$coefficients[,c(1,4)],5)

```


```{r Q1.glm_short.SC_M3, echo=FALSE, message=FALSE}

Q1.glm_short.SC_M3 <- glm(mean_SC ~ strain * sex, data=Q1.real.short_mouse_table)

#summary(Q1.glm_short.SC_M3)

summ.Q1.glm_short.SC_M3 <- summary(Q1.glm_short.SC_M3)

#round(summ.Q1.glm_short.SC_M3$coefficients[,c(1,4)],5)
#sex male very sig -- only sig
```

The following effects had significant p values for the **total SC** from the above model

subspecies musc p = `r sum.Q1.glm_totSC_M2$coefficients[,4][2]`, subspecies mol p =`r sum.Q1.glm_totSC_M2$coefficients[,4][3]`, sex male  p =`r sum.Q1.glm_totSC_M2$coefficients[,4][8]`, strain lew p =`r sum.Q1.glm_totSC_M2$coefficients[,4][5]`, strain g  p =`r sum.Q1.glm_totSC_M2$coefficients[,4][4]`,
<this is the smallest p vale, elevated female total, driven by the high means in females>

(subsp*sex) mol.male  p =`r sum.Q1.glm_totSC_M2$coefficients[,4][10]`, sex.g  p =`r sum.Q1.glm_totSC_M2$coefficients[,4][11]`, lew.male p =`r sum.Q1.glm_totSC_M2$coefficients[,4][12]`

<!--
for the _mmdomesticues^WSB^ _, _mmdomesticues^G^ _, and _mmdomesticues^LEW^ _
-->

Fr the **short bivalent** sc the model results are listed below. From this (M2) model sex effect, male is most significant (glm; p =`r summ.Q1.glm_short.SC_M2$coefficients[,4][8]`).

while the skive strain effect (glm; p=`r summ.Q1.glm_short.SC_M2$coefficients[,4][7]`) and and SKIVE*male (glm; p =`r summ.Q1.glm_short.SC_M2$coefficients[,4][16]`) are slightly significant.

In the M3 model (strains and sex as fixed effects), the Sex effect (male) is the only significant p value from this model (glm; p= `r summ.Q1.glm_short.SC_M3$coefficients[,4][14]`).

<SUMMARY, ttests and models indicate female have strongest effect -- longer AC-A >

<In summary, females (oocytes) have loser different chromatin compaction patterns compared to males. There are suggestions of evolution for this trait across the domesticus and musculus subspecies.>

<1. the longer female sc's is consistant and seen across all strains,
even in the high rec strains, suggest a decoupling of sc and gwRR - when comparing across sexes -- or the sc area is used differently for CO in oocytes and spermatocytes.
2. when the total sc per cell is compared, domesticus males have significantly lower SC arae compared to molossinus and musculus strains.Except in Q2 I Say this isn't true.>

```{r show.long.biv, echo=FALSE, include=FALSE}
#mouse av
Q1.SC.long.mouse_scatter_boxplot <- ggplot( Q1.real.long_mouse_table, aes(y = mean_SC, x = sex, color=strain)) + geom_jitter()+ geom_boxplot(alpha=.2) +scale_color_manual(values=colors_of_strains)+facet_wrap(~subsp, scales="free")+ylim(c(75,150) )+ggtitle("Mouse average SC Length \n long bivalent dataset")

```

```{r Q1_SC.biv.long_ttest, echo=FALSE, include=FALSE}

#Q1.real.long_mouse_table

Q1_mus.av_longSC_t.test.WSB <- t.test((Q1.real.long_mouse_table %>% filter(sex == "male") %>% filter(strain == "WSB") )$mean_SC,
                           (Q1.real.long_mouse_table %>% filter(sex == "female") %>% filter(strain == "WSB") )$mean_SC)

#G
Q1_mus.av_longSC_t.test.G <- t.test((Q1.real.long_mouse_table %>% filter(sex == "male") %>% filter(strain == "G") )$mean_SC,
                           (Q1.real.long_mouse_table %>% filter(sex == "female") %>% filter(strain == "G") )$mean_SC)

#LEW
Q1_mus.av_longSC_t.test.LEW <- t.test((Q1.real.long_mouse_table %>% filter(sex == "male") %>% filter(strain == "LEW") )$mean_SC,
                           (Q1.real.long_mouse_table %>% filter(sex == "female") %>% filter(strain == "LEW") )$mean_SC)

#PWD
Q1_mus.av_longSC_t.test.PWD <- t.test((Q1.real.long_mouse_table %>% filter(sex == "male") %>% filter(strain == "PWD") )$mean_SC,
                           (Q1.real.long_mouse_table %>% filter(sex == "female") %>% filter(strain == "PWD") )$mean_SC)

#KAZ
Q1_mus.av_longSC_t.test.WSB <- t.test((Q1.real.long_mouse_table %>% filter(sex == "male") %>% filter(strain == "KAZ") )$mean_SC,
                           (Q1.real.long_mouse_table %>% filter(sex == "female") %>% filter(strain == "KAZ") )$mean_SC)

#SKIVE
Q1_mus.av_longSC_t.test.SKIVE <- t.test((Q1.real.long_mouse_table %>% filter(sex == "male") %>% filter(strain == "SKIVE") )$mean_SC,
                           (Q1.real.long_mouse_table %>% filter(sex == "female") %>% filter(strain == "SKIVE") )$mean_SC)

#MSM
#Q1_mus.av_longSC_t.test.MSM <- t.test((Q1.real.long_mouse_table %>% filter(sex #== "male") %>% filter(strain == "MSM") )$mean_SC,
#                           (Q1.real.long_mouse_table %>% filter(sex == "female") #%>% filter(strain == "MSM") )$mean_SC)


#run t.tests for the mouse av. for reduced single bivalent datasets
```

```{r Q1other_SC_lmer, echo=FALSE, include=FALSE}

#1CO  lmer.Q1_SCLength_M1
Q1.lmer_SCLength_1CO_M1 <- lmer(mean.SC_1CO ~ subsp * sex + (1|strain), data=Mouse.Table_BivData_4MM)
Q1.lmer_SCLength_1CO_M1_R.sex <- lmer(mean.SC_1CO ~ subsp + (1|strain), data=Mouse.Table_BivData_4MM, REML = TRUE)
Q1.lmer_SCLength_1CO_M1_R.subsp <- lmer(mean.SC_1CO ~ sex + (1|strain), data=Mouse.Table_BivData_4MM, REML = TRUE) 

#as.data.frame(round(summary(Q1.lmer_SCLength_1CO_M1)$coefficients, 6))

Q1.lmer_SCLength.1CO_sex <- anova(Q1.lmer_SCLength_1CO_M1, Q1.lmer_SCLength_1CO_M1_R.sex)$`Pr(>Chisq)`[2]
Q1.lmer_SCLength.1CO_subsp <- anova(Q1.lmer_SCLength_1CO_M1, Q1.lmer_SCLength_1CO_M1_R.subsp)$`Pr(>Chisq)`[2]
Q1.lmer_SCLength.1CO_int <- drop1(Q1.lmer_SCLength_1CO_M1, test="Chisq")$`Pr(Chi)`[2]
Q1.lmer_SCLength.1CO_strain <- exactRLRT(Q1.lmer_SCLength_1CO_M1)$p.value

Q1.lmer_SCLength.1CO_results <- data.frame(sub.sp = Q1.lmer_SCLength.1CO_subsp, 
sex.results =  Q1.lmer_SCLength.1CO_sex, 
               interaction.pval = Q1.lmer_SCLength.1CO_int,  rand.pval = 
Q1.lmer_SCLength.1CO_strain)


#2CO
Q1.lmer_SCLength_2CO_M1 <- lmer(mean.SC_2CO ~ subsp * sex + (1|strain), data=Mouse.Table_BivData_4MM)
Q1.lmer_SCLength_2CO_M1_R.sex <- lmer(mean.SC_2CO ~ subsp + (1|strain), data=Mouse.Table_BivData_4MM, REML = TRUE) 
Q1.lmer_SCLength_2CO_M1_R.subsp <- lmer(mean.SC_2CO ~ sex + (1|strain), data=Mouse.Table_BivData_4MM, REML = TRUE) 

#as.data.frame(round(summary(Q1.lmer_SCLength_2CO_M1)$coefficients, 6))

Q1.lmer_SCLength.2CO_sex <- anova(Q1.lmer_SCLength_2CO_M1, Q1.lmer_SCLength_2CO_M1_R.sex)$`Pr(>Chisq)`[2]
Q1.lmer_SCLength.2CO_subsp <- anova(Q1.lmer_SCLength_2CO_M1, Q1.lmer_SCLength_2CO_M1_R.subsp)$`Pr(>Chisq)`[2]
Q1.lmer_SCLength.2CO_int <- drop1(Q1.lmer_SCLength_2CO_M1, test="Chisq")$`Pr(Chi)`[2]
Q1.lmer_SCLength.2CO_strain <- exactRLRT(Q1.lmer_SCLength_2CO_M1)$p.value


Q1.lmer_SCLength.2CO_results <- data.frame(sub.sp = Q1.lmer_SCLength.2CO_subsp, 
sex.results =  Q1.lmer_SCLength.2CO_sex, 
               interaction.pval = Q1.lmer_SCLength.2CO_int,  rand.pval = 
Q1.lmer_SCLength.2CO_strain)
#as.data.frame(round(summary(SCLength_long.biv_MM.M1)$coefficients, 6))
```

```{r Q1.SC.M2plot, echo=FALSE, include=FALSE}

#removing these plots for space

Musc.mouse.av <-  Mouse.Table_BivData_4MM %>%  filter(subsp == "Musc") 
#Musc.all
Q1glm_SCLength_M2.Musc <- glm(mean_SC ~ strain * sex, data = Musc.mouse.av)

#BivData Level
Musc_BivData <-  Curated_BivData %>%  filter(subsp == "Musc") 

Q1.SClength_Musc.BivData_plot <- ggplot( Musc_BivData, aes(y = chromosomeLength, x = strain, color=strain)) + geom_jitter(alpha=.3)+ geom_boxplot(alpha=.3)+scale_color_manual(values=colors_of_strains)+ggtitle("Musc av. SC Length \nBivData Level")+facet_wrap(~sex, scales="free")+ylim(c(30,225))

#BivData Level
Dom_BivData <-  Curated_BivData %>%  filter(subsp == "Dom") 

Q1.SClength_Dom.BivData_plot <- ggplot( Dom_BivData, aes(y = chromosomeLength, x = strain, color=strain)) + geom_jitter(alpha=.3)+ geom_boxplot(alpha=.3)+scale_color_manual(values=colors_of_strains)+ggtitle("Dom SC Lengths \nBivData Level")+facet_wrap(~sex, scales="free")+ylim(c(30,225))

```

## Adjusting for XX

(rethink where this section should go)
new outline

1. illustrate problem(affects mostly SC length)  

2. Expected impact on sex comparisons, estimated effect size of the X  
3. (prove general pattern that ALL bivalents are longer), chrms sorted by bin comparisons

4. 19 female, 19male, 20female 20 male

The female mouse averages should have adjustments for the XX. working on code to estimate the SC length from 3rd largest bivalent from female whole cell data across strains. Subtract this amount from the female mouse averages ... This isn't the best solution -- since I can't determine what proportion of cells for female mouse averages include the XX, (most cells are missing at least 3 bivalents)


- Of all female single bivalents observations, 5% are XX (1 of 20). 

- The XX is large, likely within the top 25% longest bivalents of the cell (3rd largest by Mb).

- The average % of XX for whole cell SC (sum(all bivalents)) can be calculated from the whole.cell data set. Lets guess 12% of a cell's total SC area is XX.

I think a formula something like this can be applied to adjust for XX

- rate of bivalent segmentation /* rate of XX, 5% /* mean SC length for 3rd longest bivalent / total SC area (by bivalent) = proportion of SC area due to XX /* 




# short transition

<transition>  -- following up on SC area - with how COs are place --- placement and number of COs on single bivalents
1. Transition, --- the SC-AE area can be thought of as the 'area / available real-estate for COs
-- these results suggest that this area is 'utilized' in different ways across sexes and strains - so we next investigate the recombination landscape - (ie the relationship between the placement and number of COs along chromosomes).

2. We focus on two metrics/ aspects i) the placement of single foci along a bivalent (just 1CO) and ii) the placement of two foci on the same bivalent / the interfocal distance of 2CO bivalents (reflects crossover interference)



# Q1. 1CO Position Sex Differences


```{r Q1.F1pos_long.short_setup, echo=FALSE}

#REAL.short_mouse_table
#REAL.long_mouse_table

Q1.long.biv_1CO <-  real.long.bivData %>% filter(species =="M.musculus") %>% filter(hand.foci.count == 1) %>% filter(strain != "MOLF") %>% filter(strain != "CZECH")

```

<!-- motivating

To quantify one of the major -- patterns that is different between male and female recombiantion landscapes
In order to test for evolution -- and examine sex differences in the recombination landscape (crossover position bias)
-->

We focus on the foci positions from one crossover (1CO n= 
) bivalents since the landscape patterns for multi crossover bivalents will be highly influences by crossover interference.

We anchor the position measures at the centromere (and normalize by deviding the position by the length of the chromosome, so the value range from 0 to 1, with smaller values representing positions closer to the centromere end of the bivalent, and larger values meaning foci positions closer to the opposite end (refered here as terminal or telomeric position).

We plot the mouse averages across all of the single bivalent measures, with the strain averages shown as boxplots...

in order to get a better picture of the sex differences in this foci-position metric

```{r CO.positions.figs, echo=FALSE, include=FALSE, warning=FALSE}
#main figure of density plots,

MELT_Curated_BivData <- add_sex(MELT_Curated_BivData)
MELT_Curated_BivData <- add_strain(MELT_Curated_BivData)
MELT_Curated_BivData <- add_subsp(MELT_Curated_BivData)#subsp, needs strain

#clean up all the non 1 to 4 hand.foci counts
MELT_Curated_BivData.1CO <- MELT_Curated_BivData[MELT_Curated_BivData$hand.foci.count < 2, ]

MELT_Curated_BivData.1CO <- MELT_Curated_BivData.1CO[MELT_Curated_BivData.1CO$PER.Foci.Pos < 1.01, ]

# remove the na's
MELT_Curated_BivData.1CO <- MELT_Curated_BivData.1CO[!(is.na(MELT_Curated_BivData.1CO$hand.foci.count) | MELT_Curated_BivData.1CO$hand.foci.count==""), ]

CO.Pos.density_1CO <- ggplot(MELT_Curated_BivData.1CO, aes(x=PER.Foci.Pos, fill=strain) ) + geom_density(alpha=.2) + facet_wrap(sex~subsp, ncol=2)+ggtitle("Normalized 1CO Foci Positions")+scale_fill_manual(values=colors_of_strains)
CO.Pos.density_1CO


#category specific plots, don't run for now.
Dom.female_MELT_curBivData <- MELT_Curated_BivData[ ((MELT_Curated_BivData$sex == "female") & (MELT_Curated_BivData$subsp == "Dom") ), ]
CO.Pos.density_Dom_female <- ggplot(Dom.female_MELT_curBivData, aes(x=PER.Foci.Pos, fill=strain) ) + geom_density(alpha=.2) + facet_wrap(~hand.foci.count, ncol=1)+ggtitle("Dom female \nNormalized Foci Positions")
#Musc male
Musc.male_MELT_curBivData <- MELT_Curated_BivData[ ((MELT_Curated_BivData$sex == "male") & (MELT_Curated_BivData$subsp == "Musc") ), ]

CO.Pos.density_Musc_male <- ggplot(Musc.male_MELT_curBivData, aes(x=PER.Foci.Pos, fill=strain) ) + geom_density(alpha=.2) + facet_wrap(~hand.foci.count, ncol=1)+ggtitle("Musc male \nNormalized Foci Positions")
#Dom male
Dom.male_MELT_curBivData <- MELT_Curated_BivData[ ((MELT_Curated_BivData$sex == "male") & (MELT_Curated_BivData$subsp == "Dom") ), ]

CO.Pos.density_Dom_male <- ggplot(Dom.male_MELT_curBivData, aes(x=PER.Foci.Pos, fill=strain) ) + geom_density(alpha=.2) + facet_wrap(~hand.foci.count, ncol=1)+ggtitle("Dom male \nNormalized Foci Positions")
```

```{r Q1show.F1pos_Full, echo=FALSE, warning=FALSE}

Q1.mouse.av.biv_scatr_box   <- ggplot(mouse.avs_4MM, aes(x=sex, y=norm.first.foci.pos, color=strain) )+geom_jitter( )+ geom_boxplot(aes(y= norm.first.foci.pos ),alpha=.3 ) + scale_color_manual(values=colors_of_strains)+facet_wrap(~subsp, scales = "free")+ylim(c(0,1.1))+ggtitle("Mouse av. \nNormalized 1CO Foci pos \n all bivalents")
Q1.mouse.av.biv_scatr_box

```

```{r Q1_1COpos.t.test, echo=FALSE}
#testing,
#norm.first.foci.pos  - mean nrm.F1.pos for just 1CO bivalents

#WSB
Q1_1CO.pos_t.test.WSB <- t.test((mouse.avs_4MM %>% filter(sex == "male") %>% filter(strain == "WSB") )$norm.first.foci.pos,
                           (mouse.avs_4MM %>% filter(sex == "female") %>% filter(strain == "WSB") )$norm.first.foci.pos)
#G
Q1_1CO.pos_t.test.G <- t.test((mouse.avs_4MM %>% filter(sex == "male") %>% filter(strain == "G") )$norm.first.foci.pos,
                           (mouse.avs_4MM %>% filter(sex == "female") %>% filter(strain == "G") )$norm.first.foci.pos)
#LEW
Q1_1CO.pos_t.test.LEW <- t.test((mouse.avs_4MM %>% filter(sex == "male") %>% filter(strain == "LEW") )$norm.first.foci.pos,
                           (mouse.avs_4MM %>% filter(sex == "female") %>% filter(strain == "LEW") )$norm.first.foci.pos)

#PWD
Q1_1CO.pos_t.test.PWD <- t.test((mouse.avs_4MM %>% filter(sex == "male") %>% filter(strain == "PWD") )$norm.first.foci.pos,
                           (mouse.avs_4MM %>% filter(sex == "female") %>% filter(strain == "PWD") )$norm.first.foci.pos)

#KAZ
Q1_1CO.pos_t.test.KAZ <- t.test((mouse.avs_4MM %>% filter(sex == "male") %>% filter(strain == "KAZ") )$norm.first.foci.pos,
                           (mouse.avs_4MM %>% filter(sex == "female") %>% filter(strain == "KAZ") )$norm.first.foci.pos)

#SKIVE
Q1_1CO.pos_t.test.SKIVE <- t.test((mouse.avs_4MM %>% filter(sex == "male") %>% filter(strain == "SKIVE") )$norm.first.foci.pos,
           (mouse.avs_4MM %>% filter(sex == "female") %>% filter(strain == "SKIVE") )$norm.first.foci.pos )

#MSM
Q1_1CO.pos_t.test.MSM <- t.test((mouse.avs_4MM %>% filter(sex == "male") %>% filter(strain == "MSM") )$norm.first.foci.pos,
                           (mouse.avs_4MM %>% filter(sex == "female") %>% filter(strain == "MSM") )$norm.first.foci.pos)



#Single Biv Level
#single.biv.1CO_pos_t.test <- t.test( 
  
 # Curated_BivData$PER_Foci_1[(Curated_BivData$subsp=="Musc")&(Curated_BivData$sex=="male")&(Curated_BivData$hand.foci.count==1)&(Curated_BivData$Rec.group==1)], 
 
#   Curated_BivData$PER_Foci_1[(Curated_BivData$subsp=="Musc")&(Curated_BivData$sex=="male")&(Curated_BivData$hand.foci.count==1)&(Curated_BivData$Rec.group==0)]
#  )

#t.tests for mous.av_nrm.1CO.pos 
```

In the domesticus strains the mouse averages for normalized single foci positions were significantly sexually dimorphic (fitting the predictions/pattern of the 'typical recombination landscape' (Sardell)). The single CO positions are medially placed in females and terminall placed in males. 

The pavlues across the strains are; (t.test; p = `r Q1_1CO.pos_t.test.WSB$p.value`, `r Q1_1CO.pos_t.test.G$p.value`, and `r Q1_1CO.pos_t.test.LEW$p.value` for WSB, G and LEW respectively).

With the musculus and molossinus strains displaying a similar pattern (t.test p = `r Q1_1CO.pos_t.test.PWD$p.value` and `r Q1_1CO.pos_t.test.MSM$p.value`) PWD and MSM. Two musculus  strains had a weaker signal, but it was still in the same direction (t.test; p =  `r Q1_1CO.pos_t.test.KAZ$p.value`, and `r Q1_1CO.pos_t.test.SKIVE$p.value` for KAZ and SKIVE respectively). 

After confirming the significan sex differences for 1CO position, we apply the familiar models


**M1. Mixed Model**

$$mouse \ average \ F1 position ~=~ subsp * sex + rand(strain) + \varepsilon $$


```{r Q1.CO1pos_lmer.M1, echo=FALSE, message=FALSE}

#make sure this dataset has all the right strains
Q1.Foci.pos_DF <- Mouse.Table_BivData_4MM %>% filter(strain != "CZECH") %>% filter(strain != "MOLF")

lmer.Q1.nrm1CO_M1 <- lmer(norm.first.foci.pos ~ subsp * sex + (1|strain), data=Q1.Foci.pos_DF, REML = TRUE) 

lmer.Q1.nrm1CO_M1reduce.sex <- lmer( norm.first.foci.pos ~ subsp  +(1|strain), data=Q1.Foci.pos_DF )

lmer.Q1.nrm1CO_M1reduce.subp <- lmer( norm.first.foci.pos ~ sex  +(1|strain), data=Q1.Foci.pos_DF )


lmer.Q1.nrm1CO_M1.sex <-  anova(lmer.Q1.nrm1CO_M1, lmer.Q1.nrm1CO_M1reduce.sex)$`Pr(>Chisq)`[2]
lmer.Q1.nrm1CO_M1.subp <-  anova(lmer.Q1.nrm1CO_M1, lmer.Q1.nrm1CO_M1reduce.subp)$`Pr(>Chisq)`[2]

lmer.Q1.nrm1CO_M1_interaction <- drop1(lmer.Q1.nrm1CO_M1, test="Chisq" )$`Pr(Chi)`[2]
lmer.Q1.nrm1CO_M1_strain <- exactRLRT(lmer.Q1.nrm1CO_M1)$p.value 

lmer.Q1.nrm1CO_M1_results <- data.frame(sub.sp = lmer.Q1.nrm1CO_M1.subp, 
sex.results = lmer.Q1.nrm1CO_M1.sex , 
               interaction.pval = lmer.Q1.nrm1CO_M1_interaction,  rand.pval = lmer.Q1.nrm1CO_M1_strain)

#lmer.Q1.nrm1CO_M1_results

#write.table(round(summry.glm.Q1_IFD.PER.M2$coefficients, 6), "~./MLH1repo/summry.glm.Q1_IFD.PER.M2.csv", sep=",",row.names = TRUE)

```


For the Mixed model, sex is the most significant effect (LTR; p =`r lmer.Q1.nrm1CO_M1_results$sex.results`).The random effect of strain is also significant
(LRT; p =`r lmer.Q1.nrm1CO_M1_results$rand.pval`). These results confirm the sex difference establish in the t.tests above -- and suggesting variation across the strains (in the sexual dimorphism for this pattern).


**M2. Linear model**

$$mouse \ average \ F1 position ~=~ subsp * sex * strain + \varepsilon $$

```{r Q1.CO1pos_glm.M2, echo=FALSE}

#in case factors aren't ordered
#Q1.real.short_mouse_table$subsp <- factor(Q1.real.short_mouse_table$subsp, ordered = FALSE)
#Q1.real.short_mouse_table$sex <- factor(Q1.real.short_mouse_table$sex, ordered = FALSE)
#Q1.real.short_mouse_table$strain <- factor(Q1.real.short_mouse_table$strain, ordered = FALSE)

Q1.glm_F1pos_M2 <- glm(norm.first.foci.pos ~ subsp * strain * sex, data=Q1.Foci.pos_DF)

#summary(Q1.glm_F1pos_M2)

summry.Q1.glm_F1pos_M2 <- summary(Q1.glm_F1pos_M2)
#round(summry.Q1.glm_F1pos_M2$coefficients[,c(1,4)],5)

#write.table(round(summry.Q1.glm_nrmF1pos_M2$coefficients,5), "~./MLH1repo/summry.Q1.glm_nrmF1pos_M2.csv", #sep=",",row.names = TRUE)


#DOM
Q1.glm_nrmF1pos_M2_Dom <- glm(norm.first.foci.pos ~ strain * sex, data=Mouse.Table_BivData_4MM[Mouse.Table_BivData_4MM$subsp == "Dom",])
#summary(Q1.glm_nrmF1pos_M2_Dom)

#MUSC
Q1.glm_nrmF1pos_M2_Musc <- glm(norm.first.foci.pos ~ strain * sex, data=Mouse.Table_BivData_4MM[Mouse.Table_BivData_4MM$subsp == "Musc",])
#summary(Q1.glm_nrmF1pos_M2_Musc)

#MOL
Q1.glm_nrmF1pos_M2_Mol <- glm(norm.first.foci.pos ~ strain * sex, data=Mouse.Table_BivData_4MM[Mouse.Table_BivData_4MM$subsp == "Mol",])
#summary(Q1.glm_nrmF1pos_M2_Mol)

```

**M3. Linear model**

$$mouse \ average \ F1 position ~=~ sex * strain + \varepsilon $$

```{r Q1.CO1pos_glm.M3, echo=FALSE, eval=TRUE}

Q1.glm_1CO.pos_M3 <- glm(norm.first.foci.pos ~ strain * sex, data=Q1.Foci.pos_DF)

#summary(Q1.glm_1CO.pos_M3)

summry.Q1.glm_1CO.pos_M3 <- summary(Q1.glm_1CO.pos_M3)
#round(summry.Q1.glm_1CO.pos_M3$coefficients[,c(1,4)],5)
#write.table(round(summry.Q1.glm_nrmF1pos_M3$coefficients,5), "~./MLH1repo/summry.Q1.glm_nrmF1pos_M3.csv", sep=",",row.names = TRUE)

```


The male effects were the most significant (glm M2; p =`r summry.Q1.glm_F1pos_M2$coefficients[,4][8]`, glm M3; p = `r summry.Q1.glm_1CO.pos_M3$coefficients[,4][8]`). While the musc subspecies, LEW and KAZ strains had slightly signifcant effects (glm M2; Musc subsp p = `r summry.Q1.glm_F1pos_M2$coefficients[,4][2]`, LEW strain p = `r summry.Q1.glm_F1pos_M2$coefficients[,4][5]`) and (glm M3; LEW strain p = `r summry.Q1.glm_1CO.pos_M3$coefficients[,4][3]`, KAZ strain p = `r summry.Q1.glm_1CO.pos_M3$coefficients[,4][7]`).

(these results will be explored more fully in the Q2 section). The general pattern that emerges across these models is the sex effect being the most significant, (with the male single foci position being more telomeric compared to females). 

<we don't explore the effects on the redcued bivalent data sets! The main expected pattern is confirmed>

Chromosome size effects are a confounding factor for CO position, with the expectation that shorter chromosomes will have a more uniform landscape compared to larger/longer chromosomes (cite). To account for this we again use the reduced bivalent sets (long.biv and short.biv)

```{r Q1.F1.pos.LONG.SHORT.BIV_scatter, echo=FALSE}

#real.long.bivData


Q1.short.biv_scatr_box   <- ggplot( (real.short.bivData %>% filter(hand.foci.count == 1) ), aes(x=sex, y=PER_Foci_1, color=strain) )+geom_jitter( )+ geom_boxplot(aes(y= PER_Foci_1 ),alpha=.3 ) + scale_color_manual(values=colors_of_strains)+facet_wrap(~subsp, scales = "free")+ylim(c(0,1.1))+ggtitle("Mouse av. \nNormalized 1CO Foci pos \nShort Biv")
Q1.short.biv_scatr_box

Q1.long.biv_scatr_box   <- ggplot((real.long.bivData %>% filter(hand.foci.count == 1) ), aes(x=sex, y=PER_Foci_1, color=strain) )+geom_jitter( )+ geom_boxplot(aes(y= PER_Foci_1 ),alpha=.3 ) + scale_color_manual(values=colors_of_strains)+facet_wrap(~subsp, scales = "free")+ylim(c(0,1.1))+ggtitle("Mouse av. \nNormalized 1CO Foci pos \nLong Biv")
Q1.long.biv_scatr_box
```


```{r Q1.mouse.av.F1.pos.LONG.BIV_scatter, echo=FALSE}

#REAL.short_mouse_table
#REAL.long_mouse_table
#mouse.av.long.biv.table

Q1.av.long.biv_scatr_box   <- ggplot(REAL.long_mouse_table  , aes(x=sex, y=mean_F1pos_PER, color=strain) )+geom_jitter( )+ geom_boxplot(aes(y= mean_F1pos_PER ),alpha=.3 ) + scale_color_manual(values=colors_of_strains)+facet_wrap(~subsp, scales = "free")+ylim(c(0,1.1))+ggtitle("Mouse av. \nNormalized 1CO Foci pos \nLong Biv")

Q1.av.short.biv_scatr_box   <- ggplot(REAL.short_mouse_table, aes(x=sex, y=mean_F1pos_PER, color=strain) )+geom_jitter( )+ geom_boxplot(aes(y= mean_F1pos_PER ),alpha=.3 ) + scale_color_manual(values=colors_of_strains)+facet_wrap(~subsp, scales = "free")+ylim(c(0,1.1))+ggtitle("Mouse av. \nNormalized 1CO Foci pos \nShort Biv")
```

These box plot show that females have a much more medial position of single foci bivalents, (much closer to 50% compared to males). They also show that Musc males' Foci1 position is slightly more central / medial compared to the same type of positions in the Dom male strains. MOLF males have much more medial positions than other strains.

<!--
### Siscoten
The metric Sis-co-ten measures the amount of sister cohesion connected to the other pole.
The logic of how the sis-co-ten metric is outlined in the figure below. The goal is to use this metric to model different tension active cohesion amounts as a consequence of different numbers and placements of chiasmata/CO. This metric is calculated using SC area as a proxy to the amount of cohesion at metaphase.

![from (Lee, J. (2019). Is age-related increase of chromosome segregation errors in mammalian oocytes caused by cohesin deterioration?. Reproductive Medicine and Biology.)](figureFiles/Lee2019_Fig2.png)

```{r show.siscoten, echo=FALSE}
#think of a good way to display this

sis.co.ten.plot_WSB_female <- ggplot(data = Curated_BivData[Curated_BivData$category == "WSB female",], aes(y=SisCoTen, x=chromosomeLength, color = as.factor(hand.foci.count) ))+
  geom_point()+facet_wrap(sex~strain)
sis.co.ten.plot_WSB_female

sis.co.ten.plot_WSB_male <- ggplot(data = Curated_BivData[Curated_BivData$category == "WSB male",], aes(y=SisCoTen, x=chromosomeLength, color = as.factor(hand.foci.count) ))+
  geom_point()+facet_wrap(sex~strain)
sis.co.ten.plot_WSB_male

sis.co.ten.plot_LEW_female <- ggplot(data = Curated_BivData[Curated_BivData$category == "LEW female",], aes(y=SisCoTen, x=chromosomeLength, color = as.factor(hand.foci.count) ))+
  geom_point()+facet_wrap(sex~strain)
#sis.co.ten.plot_LEW_female

sis.co.ten.plot_LEW_male <- ggplot(data = Curated_BivData[Curated_BivData$category == "LEW male",], aes(y=SisCoTen, x=chromosomeLength, color = as.factor(hand.foci.count) ))+
  geom_point()+facet_wrap(sex~strain)
#sis.co.ten.plot_LEW_male


sis.co.ten.plot_PWD_female <- ggplot(data = Curated_BivData[Curated_BivData$category == "PWD female",], aes(y=SisCoTen, x=chromosomeLength, color = as.factor(hand.foci.count) ))+
  geom_point()+facet_wrap(sex~strain)
#sis.co.ten.plot_PWD_female


sis.co.ten.plot_PWD_male <- ggplot(data = Curated_BivData[Curated_BivData$category == "PWD male",], aes(y=SisCoTen, x=chromosomeLength, color = as.factor(hand.foci.count) ))+
  geom_point()+facet_wrap(sex~strain)
#sis.co.ten.plot_PWD_male


Curated_BivData <- Curated_BivData[Curated_BivData$hand.foci.count < 5,]
Curated_BivData <- Curated_BivData[!(is.na(Curated_BivData$hand.foci.count) | Curated_BivData$hand.foci.count==""), ]


jitter_siscoten_PWDf <- ggplot(data = Curated_BivData[Curated_BivData$category == "PWD female",], aes(y=SisCoTen, x=hand.foci.count, color = as.factor(hand.foci.count) ))+ ggtitle("PWD female Sis.co.ten")+
  geom_jitter()+facet_wrap(sex~strain)
#jitter_siscoten_PWDf

jitter_siscoten_PWDm <- ggplot(data = Curated_BivData[Curated_BivData$category == "PWD male",], aes(y=SisCoTen, x=hand.foci.count, color = as.factor(hand.foci.count) ))+ggtitle("PWD male sis.co.ten")+
  geom_jitter()+facet_wrap(sex~strain)
#jitter_siscoten_PWDm

jitter_siscoten_WSBf <- ggplot(data = Curated_BivData[Curated_BivData$category == "WSB female",], aes(y=SisCoTen, x=hand.foci.count, color = as.factor(hand.foci.count) ))+ggtitle("WSB female sis.co.ten")+
  geom_jitter()+facet_wrap(sex~strain)
#jitter_siscoten_WSBf

jitter_siscoten_WSBm <- ggplot(data = Curated_BivData[Curated_BivData$category == "WSB male",], aes(y=SisCoTen, x=hand.foci.count, color = as.factor(hand.foci.count) ))+ggtitle("WSB male sis.co.ten")+
  geom_jitter()+facet_wrap(sex~strain)
#jitter_siscoten_WSBm
```

Males have much clearer separation of siscoten across chrm classes. This is emphasized when SC length is also plotted. It seems like musc males have higher amounts of this metric compared to Dom males.
To formally test the differences in sis-co-ten I plan to write a sub sampling / permutation loop to compare the mean(sis.co.ten) of the same numbers of bivalents of the same class.
BUT females have a greater range -- so maybe it's just a scale issue.  

```{r show.norm.siscoten, echo=FALSE}
Curated_BivData$nrm.siscoten <- Curated_BivData$SisCoTen/Curated_BivData$chromosomeLength

Nrm.sis.co.ten.plot_WSB <- ggplot(data = Curated_BivData[Curated_BivData$strain == "WSB",], aes(y=SisCoTen/chromosomeLength, x=chromosomeLength, color = as.factor(hand.foci.count) ))+
  geom_point()+facet_wrap(~sex)+ggtitle("Normalized siscoten \n WSB")+theme(legend.position="none")
Nrm.sis.co.ten.plot_WSB

Nrm.sis.co.ten.plot_G <- ggplot(data = Curated_BivData[Curated_BivData$strain == "G",], aes(y=SisCoTen/chromosomeLength, x=chromosomeLength, color = as.factor(hand.foci.count) ))+
  geom_point()+facet_wrap(~sex)+ggtitle("Normalized siscoten \n G")+theme(legend.position="none")
Nrm.sis.co.ten.plot_G


Nrm.sis.co.ten.plot_LEW <- ggplot(data = Curated_BivData[Curated_BivData$strain == "LEW",], aes(y=SisCoTen/chromosomeLength, x=chromosomeLength, color = as.factor(hand.foci.count) ))+
  geom_point()+facet_wrap(~sex)+ggtitle("Normalized siscoten\nLEW")+theme(legend.position="none")
Nrm.sis.co.ten.plot_LEW



Nrm.sis.co.ten.plot_PWD <- ggplot(data = Curated_BivData[Curated_BivData$strain == "PWD",], aes(y=SisCoTen/chromosomeLength, x=chromosomeLength, color = as.factor(hand.foci.count) ))+
  geom_point()+facet_wrap(~sex)+ggtitle("Normalized siscoten \n PWD")+theme(legend.position="none")
Nrm.sis.co.ten.plot_PWD


Nrm.sis.co.ten.plot_KAZ <- ggplot(data = Curated_BivData[Curated_BivData$strain == "KAZ",], aes(y=SisCoTen/chromosomeLength, x=chromosomeLength, color = as.factor(hand.foci.count) ))+
  geom_point()+facet_wrap(~sex)+ggtitle("Normalized siscoten \n KAZ")+theme(legend.position="none")
Nrm.sis.co.ten.plot_KAZ


Nrm.sis.co.ten.plot_SKIVE <- ggplot(data = Curated_BivData[Curated_BivData$strain == "PWD",], aes(y=SisCoTen/chromosomeLength, x=chromosomeLength, color = as.factor(hand.foci.count) ))+
  geom_point()+facet_wrap(~sex)+ggtitle("Normalized siscoten \n SKIVE")+theme(legend.position="none")
Nrm.sis.co.ten.plot_SKIVE

Nrm.sis.co.ten.plot_MSM <- ggplot(data = Curated_BivData[Curated_BivData$strain == "MSM",], aes(y=SisCoTen/chromosomeLength, x=chromosomeLength, color = as.factor(hand.foci.count) ))+
  geom_point()+facet_wrap(~sex)+ggtitle("Normalized siscoten \n MSM")+theme(legend.position="none")
Nrm.sis.co.ten.plot_MSM

```

I think the the normalized sis.co.ten plots also show that the there is more clustering of the sis.co.ten for the males.

```{r sis.co.ten.test_HetC, echo=FALSE, eval=FALSE}

#glm for siscoten
#mouse.avs_4MM  - mouse averages with mean sis.co.ten
#MM
lme_MM_siscoten <- lme(mean.siscoten ~ subsp * sex, data= mouse.avs_4MM, random=list(strain=pdDiag(~sex) ) )
as.data.frame( round(  summary(lme_MM_siscoten)$tTable, 6 ) )

#sex and sex*subsp interaction effect are significant for mean sis.co.ten
lmer_MM_siscoten <- lmer(mean.siscoten ~ subsp * sex + (1|strain), data=mouse.avs_4MM, REML = TRUE)
#anova(lmer_MM_siscoten) 

#thing1  <- summary(lmer_MM_siscoten)
#drop1(lmer_MM_siscoten, test="Chisq") 

rand_siscoten_MM <- exactRLRT(lmer_MM_siscoten)
rand_siscoten_MM
#slightly sig
```


```{r siscoten.test.poly, echo=FALSE, eval=FALSE}

#move this to the set up blocks

mouse.avs_4MM$Rec.group <- ifelse(grepl("PWD", mouse.avs_4MM$strain), 1, 
                            ifelse(grepl("MSM", mouse.avs_4MM$strain), 1,
                             ifelse(grepl("SKIVE", mouse.avs_4MM$strain), 1, 0)) )

#test high vs low Musc strains
#mouse.avs_4MM
```


```{r log.reg_siscoten, echo=FALSE}

#replace the table

male_mouse.av$Rec.group <- ifelse(grepl("PWD", male_mouse.av$strain), 1, 
                            ifelse(grepl("MSM", male_mouse.av$strain), 1,
                             ifelse(grepl("SKIVE", male_mouse.av$strain), 1, 0)) )

Male.poly.Mouse.Table_BivData_4MM$Rec.group <- ifelse(grepl("PWD", Male.poly.Mouse.Table_BivData_4MM$strain), 1, 
                            ifelse(grepl("MSM", Male.poly.Mouse.Table_BivData_4MM$strain), 1,
                             ifelse(grepl("SKIVE", Male.poly.Mouse.Table_BivData_4MM$strain), 1, 0)) )

#SISCOTEN
#mouse av, norm.first foci is from 1CO bivalents

log.reg_SIS.co.ten_mouse.av <- glm(Rec.group ~ mean.siscoten, 
                           
                                 data=Male.poly.Mouse.Table_BivData_4MM[(Male.poly.Mouse.Table_BivData_4MM$subsp == "Musc"),], family=binomial(link="logit"))
summary(log.reg_SIS.co.ten_mouse.av)


#single biv level
log.reg_Sis.co.ten_biv <- glm(Rec.group ~ SisCoTen, 
                            data=Curated_BivData[(Curated_BivData$subsp == "Musc")&(Curated_BivData$sex == "male"),], family=binomial(link="logit"))

summary(log.reg_Sis.co.ten_biv)

```


```{r t.test_sis.co.ten,echo=FALSE, eval=FALSE}

#bivData -- 

t.test_mouse.av_siscoten <- t.test(
  
  Male.poly.Mouse.Table_BivData_4MM$mean.siscoten[(mouse.avs_4MM$Rec.group == 1)],
  
  Male.poly.Mouse.Table_BivData_4MM$mean.siscoten[(Male.poly.Mouse.Table_BivData_4MM$subsp == "Musc") &(Male.poly.Mouse.Table_BivData_4MM$Rec.group == 0)] 
  )

#1CO
t.test_mouse.av_siscoten_1CO <- t.test( 
  Male.poly.Mouse.Table_BivData_4MM$mean.siscoten_1CO[(Male.poly.Mouse.Table_BivData_4MM$subsp == "Musc")& (Male.poly.Mouse.Table_BivData_4MM$Rec.group == 1)], 
 
   Male.poly.Mouse.Table_BivData_4MM$mean.siscoten_1CO[(Male.poly.Mouse.Table_BivData_4MM$subsp == "Musc")& (Male.poly.Mouse.Table_BivData_4MM$Rec.group == 0)]
  )

#2CO
t.test_mouse.av_siscoten_2CO <- t.test( 
  Male.poly.Mouse.Table_BivData_4MM$mean.siscoten_2CO[(Male.poly.Mouse.Table_BivData_4MM$subsp == "Musc")&(Male.poly.Mouse.Table_BivData_4MM$Rec.group == 1)], 
 
   Male.poly.Mouse.Table_BivData_4MM$mean.siscoten_2CO[(Male.poly.Mouse.Table_BivData_4MM$subsp == "Musc")& (Male.poly.Mouse.Table_BivData_4MM$Rec.group == 0)]
  )

#3CO
t.test_mouse.av_siscoten_3CO <- t.test( 
  Male.poly.Mouse.Table_BivData_4MM$mean.siscoten_3CO[(Male.poly.Mouse.Table_BivData_4MM$subsp == "Musc")& (Male.poly.Mouse.Table_BivData_4MM$Rec.group == 1)], 
 
   Male.poly.Mouse.Table_BivData_4MM$mean.siscoten_3CO[(Male.poly.Mouse.Table_BivData_4MM$subsp == "Musc")&(Male.poly.Mouse.Table_BivData_4MM$Rec.group == 0)]
  )



####
#Single Biv Level -- from the main 
#pull the single BivData -- MUSC

#check that these have the extra male strains

t.test_single.biv_siscoten <- t.test( 
  
  Curated_BivData$SisCoTen[(Curated_BivData$subsp == "Musc")&(Curated_BivData$sex == "male")& (Curated_BivData$Rec.group == 1)], 
 
   Curated_BivData$SisCoTen[(Curated_BivData$subsp == "Musc")&(Curated_BivData$sex == "male")& (Curated_BivData$Rec.group == 0)]
  )

#1CO
mouse.avs_4MM$mean.siscoten_2CO

```

All the sis.co.ten tests are highly significant. Maybe I should consider running a normalized sis.co.ten? I think nrm_siscoten would still reflect the differing cohesion structure/outcome.

### Telomere and centromere Distance

My metric for telomere and centromere distance measure the distance of the nearest foci to the ends of the bivalent (SC). In the plots below each point is a single bivalent. I choose not to use the mark for centromere because it seems noisy and inconsistent... 

```{r telo.dist, echo=FALSE}

sis.co.ten.plot_WSB_female <- ggplot(data = Curated_BivData[Curated_BivData$category == "WSB female",], aes(y=SisCoTen, x=chromosomeLength, color = as.factor(hand.foci.count) ))+
  geom_point()+facet_wrap(sex~strain)

Curated_BivData <- Curated_BivData[Curated_BivData$hand.foci.count < 5,]
Curated_BivData <- Curated_BivData[!(is.na(Curated_BivData$hand.foci.count) | Curated_BivData$hand.foci.count==""), ]


telo.dist_boxplots <- ggplot(Curated_BivData, aes(x= hand.foci.count, y=telo_dist ) )+ geom_boxplot(aes( fill=hand.foci.count, group=strain), alpha=.2) + facet_wrap(sex~subsp, ncol=2, scales="free")+ggtitle("boxy plots")+ ylim(c(0, 150))


telo_jitter <- ggplot(Curated_BivData, aes(x= strain, y=telo_dist, color=as.factor(hand.foci.count) ) )+ geom_jitter(alpha=.2) + facet_wrap(sex~subsp, ncol=2, scales="free")+ggtitle("raw telomere distance")+ ylim(c(0, 150))
telo_jitter


nrm.telo_jitter <- ggplot(Curated_BivData, aes(x= strain, y=telo_dist_PER, color=as.factor(hand.foci.count) ) )+ geom_jitter(alpha=.2) + facet_wrap(sex~subsp, ncol=2, scales="free")+ggtitle("Normalized telomere distance")+ylim(c(0,1.2))
nrm.telo_jitter

#why do a different number of rows removed each time??
```

Males on average have much lower raw telomere distance (reflects the telomere bias) compared to females. In Males, 2CO bivalents have very low telomere distances, while the 1CO bivalents have a greater range. In females the ranges of telomere distances have much more overlap.   

```{r MM.telo, echo=FALSE}
#HetC

lme_MM_telo.dist <- lmer(mean.telo.dist ~ subsp * sex + (1|strain), data= Mouse.Table_BivData_4MM )
drop1(lme_MM_telo.dist, test="Chisq")
as.data.frame( round(  summary(lme_MM_telo.dist)$coefficients, 6 ) )

#sex and sex*subsp interaction effect are significant for mean sis.co.ten

lmer_MM_telo.dist <- lmer(mean.telo.dist ~ subsp * sex + (1|strain), data=Mouse.Table_BivData_4MM, REML = TRUE)
#anova(lmer_MM_siscoten) 

#thing1  <- summary(lmer_MM_siscoten)
drop1(lmer_MM_telo.dist, test="Chisq") 
rand_telo.dist_MM <- exactRLRT(lmer_MM_telo.dist)
rand_telo.dist_MM
```

Mixed model result summary:

```{r log.reg_telo.dist, echo=FALSE}

Mouse.Table_BivData_4MM$Rec.group <- ifelse(grepl("PWD male", Mouse.Table_BivData_4MM$category), 1, 
                            ifelse(grepl("MSM male", Mouse.Table_BivData_4MM$category), 1,
                             ifelse(grepl("SKIVE male", Mouse.Table_BivData_4MM$category), 1, 0)) )


#TELO.DIST
#mouse av, norm.first foci is from 1CO bivalents
log.reg_Telo.Dist_mouse.av <- glm(Rec.group ~ mean.telo.dist, 
                           
                                 data=Mouse.Table_BivData_4MM[(Mouse.Table_BivData_4MM$subsp == "Musc")&(Mouse.Table_BivData_4MM$sex == "male"),], family=binomial(link="logit"))
summary(log.reg_Telo.Dist_mouse.av)


#single biv level
log.reg_Telo.Dist_biv <- glm(Rec.group ~ telo_dist, 
                            data=Curated_BivData[(Curated_BivData$subsp == "Musc")&(Curated_BivData$sex == "male"),], family=binomial(link="logit"))

summary(log.reg_Telo.Dist_biv)


#PER versions
log.reg_Telo.Dist.PER_biv <- glm(Rec.group ~ telo_dist_PER, 
                            data=Curated_BivData[(Curated_BivData$subsp == "Musc")&(Curated_BivData$sex == "male"),], family=binomial(link="logit"))

summary(log.reg_Telo.Dist.PER_biv)
```

```{r t.test_Telo.Dist, echo=FALSE}

#mouse av
t.test_mouse.av_telo.dist <- t.test(
  
  Mouse.Table_BivData_4MM$mean.telo.dist[(Mouse.Table_BivData_4MM$Rec.group == 1)],
  
  Mouse.Table_BivData_4MM$mean.telo.dist[(Mouse.Table_BivData_4MM$sex == "male") & (Mouse.Table_BivData_4MM$subsp == "Musc") &(Mouse.Table_BivData_4MM$Rec.group == 0)] 
  )


#Single Biv Level
single.biv_t.test_Telo.Dist <- t.test( 
  
  Curated_BivData$telo_dist[(Curated_BivData$subsp == "Musc")&(Curated_BivData$sex == "male") & (Curated_BivData$Rec.group == 1)], 
 
   Curated_BivData$telo_dist[(Curated_BivData$subsp == "Musc")&(Curated_BivData$sex == "male") & (Curated_BivData$Rec.group == 0)]
  )
   
#Telo Dist PER
single.biv_t.test_Telo.Dist_PER <- t.test( 
  
  Curated_BivData$telo_dist_PER[(Curated_BivData$subsp == "Musc")&(Curated_BivData$sex == "male")& (Curated_BivData$Rec.group == 1)], 
 
   Curated_BivData$telo_dist_PER[(Curated_BivData$subsp == "Musc")&(Curated_BivData$sex == "male") & (Curated_BivData$Rec.group == 0)]
  )
   
```

```{r cent.dist_show, echo=FALSE}
cent.dist_boxplots <- ggplot(Curated_BivData, aes(x= strain, y=dis.cent ) )+ geom_boxplot(aes( fill=hand.foci.count, group=strain), alpha=.2) + facet_wrap(sex~subsp, ncol=2, scales="free")+ggtitle("raw centromere distance")+ ylim(c(0, 150))


cent.dist_jitter <- ggplot(Curated_BivData, aes(x= strain, y=dis.cent, color=as.factor(hand.foci.count) ) )+ geom_jitter(alpha=.2) + facet_wrap(sex~subsp, ncol=2, scales="free")+ggtitle("raw centromere dist")+ ylim(c(0, 200))
cent.dist_jitter

norm.cent.dist_jitter <- ggplot(Curated_BivData, aes(x= strain, y=dis.cent.PER, color=as.factor(hand.foci.count) ) )+ geom_jitter(alpha=.2) + facet_wrap(sex~subsp, ncol=2, scales="free")+ggtitle("Normalized centromere distance")+ylim(c(0,1.2))
norm.cent.dist_jitter

```

```{r MM.cent.dist, echo=FALSE, eval=FALSE}

lme_MM_cent.dist <- lme(mean.cent.dist ~ subsp * sex, data= Mouse.Table_BivData_4MM, random=list(strain=pdDiag(~sex) ) )
as.data.frame( round(  summary(lme_MM_cent.dist)$tTable, 6 ) )

lmer_MM_cent.dist <- lmer(mean.cent.dist ~ subsp * sex + (1|strain), data=Mouse.Table_BivData_4MM, REML = TRUE)

exactRLRT(lmer_MM_cent.dist)
```

```{r log.reg_CENT.dist, echo=FALSE}

#CENT.DIST
#mouse av, norm.first foci is from 1CO bivalents
log.reg_CENT.Dist_mouse.av <- glm(Rec.group ~ mean.cent.dist, 
                           
                                 data=Mouse.Table_BivData_4MM[(Mouse.Table_BivData_4MM$subsp == "Musc")&(Mouse.Table_BivData_4MM$sex == "male"),], family=binomial(link="logit"))
summary(log.reg_CENT.Dist_mouse.av)


#single biv level
log.reg_CENT.Dist_biv <- glm(Rec.group ~ dis.cent, 
                            data=Curated_BivData[(Curated_BivData$subsp == "Musc")&(Curated_BivData$sex == "male"),], family=binomial(link="logit"))

summary(log.reg_CENT.Dist_biv)


#PER versions
log.reg_CENT.Dist.PER_biv <- glm(Rec.group ~ dis.cent.PER, 
                            data=Curated_BivData[(Curated_BivData$subsp == "Musc")&(Curated_BivData$sex == "male"),], family=binomial(link="logit"))

summary(log.reg_CENT.Dist.PER_biv)
```

```{r t.test_CENT.Dist, echo=FALSE}
#CENT.DIST
#mouse av
t.test_mouse.av_Cent.dist <- t.test(
  
  Mouse.Table_BivData_4MM$mean.cent.dist[(Mouse.Table_BivData_4MM$Rec.group == 1)],
  
  Mouse.Table_BivData_4MM$mean.cent.dist[(Mouse.Table_BivData_4MM$sex == "male") & (Mouse.Table_BivData_4MM$subsp == "Musc") &(Mouse.Table_BivData_4MM$Rec.group == 0)] 
  )


#Single Biv Level
single.biv_t.test_Cent.Dist <- t.test( 
  
  Curated_BivData$dis.cent[(Curated_BivData$subsp == "Musc")&(Curated_BivData$sex == "male") & (Curated_BivData$Rec.group == 1)], 
 
   Curated_BivData$dis.cent[(Curated_BivData$subsp == "Musc")&(Curated_BivData$sex == "male") & (Curated_BivData$Rec.group == 0)]
  )
   
#Cent Dist PER
single.biv_t.test_Cent.Dist_PER <- t.test( 
  
  Curated_BivData$dis.cent.PER[(Curated_BivData$subsp == "Musc")&(Curated_BivData$sex == "male") & (Curated_BivData$Rec.group == 1)], 
 
   Curated_BivData$dis.cent.PER[(Curated_BivData$subsp == "Musc")&(Curated_BivData$sex == "male")& (Curated_BivData$Rec.group == 0)]
  )
   
```

The normalized centromere plots show that in Musc males, on 2CO bivalents the 1st CO is closer to the centromere end than in Dom males. Females have more overlap in the distributions of centromere distances across chromosome class compared to males.-->


```{r Q1.CO1_foci_pos_boxplot, echo=FALSE, eval=FALSE}

MELT_long_biv <- MELT_Curated_BivData[ MELT_Curated_BivData$Obj.ID %in% real.long.bivData$Obj.ID, ]

CO.Pos.boxplots_1CO <- ggplot(MELT_Curated_BivData.1CO, aes(x= strain, y=PER.Foci.Pos, fill=strain) ) + geom_boxplot(alpha=.2) + facet_wrap(sex~subsp, ncol=2, scales="free")+ggtitle("Percent Foci Position")+geom_jitter(alpha=.1)+scale_fill_manual(values=colors_of_strains)
CO.Pos.boxplots_1CO

```

```{r CO1_foci_pos_boxplot, echo=FALSE, eval=FALSE}

MELT_long_biv <- MELT_Curated_BivData[ MELT_Curated_BivData$Obj.ID %in% real.long.bivData$Obj.ID, ]

CO.Pos.boxplots_1CO <- ggplot(MELT_Curated_BivData.1CO, aes(x= strain, y=PER.Foci.Pos, fill=strain) ) + geom_boxplot(alpha=.2) + facet_wrap(sex~subsp, ncol=2, scales="free")+ggtitle("Percent Foci Position")+geom_jitter(alpha=.1)+scale_fill_manual(values=colors_of_strains)
CO.Pos.boxplots_1CO

```

```{r Q1.1CO.pos_Sardell.eq, echo=FALSE, include=FALSE}
#IGNORE THIS FOR KNOW, THIS 

#filter all Curated biv data for 1 strain
#WSB.Bivalents.DF , use single bivalent level observations, with the strain specific DFs


# the stardell paper sums the number of COs for bins (10)

#in my data it would be like adding -- 10 more colums, and marking 1 if there is a bin within it
#but then the model would be close to logistic

#summing the number of COs per bin, losses the chrom specific data....
#!! need to use a melted Data frame to capture all nrm.foci obs
#wsb.ttest<- mouse.avs_4MM %>% filter()
#start with same chrm classes
#the telomere dist and centromere dis were celulcate for the nearest foci ... but really these shhould be calq for each foci

WSB.Biv.1CO <- WSB.Bivalents.DF %>% filter(hand.foci.count == 1)

Sardell.mo.WSB <- lmer(PER_Foci_1 ~ dis.cent + telo_dist + sex+ (1 | chromosomeLength), data = WSB.Biv.1CO)


PWD.Biv.1CO <- PWD.Bivalents.DF %>% filter(hand.foci.count == 1)

Sardell.mo.PWD <- lmer(PER_Foci_1 ~ dis.cent + telo_dist + sex + (1 | chromosomeLength), data = PWD.Biv.1CO)
Sardell.mo.PWD_R.dis.cent <- lmer(PER_Foci_1 ~  telo_dist + sex + (1 | chromosomeLength), data = PWD.Biv.1CO)
Sardell.mo.PWD_R.dis.telo <- lmer(PER_Foci_1 ~  dis.cent + sex + (1 | chromosomeLength), data = PWD.Biv.1CO)

anova(Sardell.mo.PWD,Sardell.mo.PWD_R.dis.cent) #p= highly sig

anova(Sardell.mo.PWD,Sardell.mo.PWD_R.dis.telo) #p= v low

anova(Sardell.mo.PWD_R.dis.cent, Sardell.mo.PWD_R.dis.telo) #p = 1

#or AIC(model1,model2)

```

```{r Q1.mean.nrmF1_table, echo=FALSE,eval=FALSE}

#make a table for the sex specific values of the normalized F1 positions

#for strain, sex -- Curated_BivData
#or (Mouse.Table_BivData_4MM)
#norm.first.foci.pos
Q1.normF1.pos_table <- ddply(Mouse.Table_BivData_4MM, c("strain","sex"), summarise,
                  
                    mean_1CO_nrm.pos = mean(norm.first.foci.pos, na.rm = TRUE), 
                  n1CO = sum(n1CO )

)

```


```{r norm.CO.pos_longBiv, echo=FALSE, eval=FALSE}

#real.long.bivData
#MELT_Curated_BivData

#these isn't long.biv data
MELT_long_biv <- MELT_Curated_BivData[ MELT_Curated_BivData$Obj.ID %in% real.long.bivData$Obj.ID, ]

real.long.bivData
MELT_Curated_BivData.1CO <- MELT_Curated_BivData[MELT_Curated_BivData$hand.foci.count < 2, ]
MELT_Curated_BivData.1CO <- MELT_Curated_BivData.1CO[MELT_Curated_BivData.1CO$PER.Foci.Pos < 1.01, ]

MELT_long_biv <- MELT_Curated_BivData[ MELT_Curated_BivData$Obj.ID %in% real.long.bivData$Obj.ID, ]


MELT_long_biv <- add_sex(MELT_long_biv)
MELT_long_biv <- add_strain(MELT_long_biv)
MELT_long_biv <- add_subsp(MELT_long_biv)
MELT_long_biv <- add_category(MELT_long_biv)


long.biv_Dom.male <- MELT_long_biv[ (MELT_long_biv$subsp == "Dom") & (MELT_long_biv$sex == "male"), ]

Dom.male_MELT_long <- ggplot( long.biv_Dom.male, aes(x=PER.Foci.Pos, fill=strain) ) + geom_density(alpha=.2) + facet_wrap(~hand.foci.count, ncol=1)+ggtitle("Dom male long biv\nNormalized Foci Positions")+scale_fill_manual(values=colors_of_strains)
Dom.male_MELT_long


long.biv_Musc.male <- MELT_long_biv[ (MELT_long_biv$subsp == "Musc") & (MELT_long_biv$sex == "male"), ]

Musc_male_MELT_long <- ggplot( long.biv_Musc.male, aes(x=PER.Foci.Pos, fill=strain) ) + geom_density(alpha=.2) + facet_wrap(~hand.foci.count, ncol=1)+ggtitle("Musc male long biv\nNormalized Foci Positions")+scale_fill_manual(values=colors_of_strains)
Musc_male_MELT_long

#there are v few observations in this dataset
```

```{r Q1.log.reg.CO1_pos, echo=FALSE, eval=FALSE}

#LOGISTIC REGRESSION for FOCI POSITION

#mouse av, norm.first foci is from 1CO bivalents
log.reg_1CO_pos_mouse.av <- glm(Rec.group ~ norm.first.foci.pos, 
                            data=mouse.avs_4MM[(mouse.avs_4MM$subsp == "Musc")&(mouse.avs_4MM$sex == "male"),], family=binomial(link="logit"))
summary(log.reg_1CO_pos_mouse.av)


#single biv level
log.reg_1CO_pos_biv <- glm(Rec.group ~ PER_Foci_1, 
                            data=Curated_BivData[(Curated_BivData$subsp == "Musc")&(Curated_BivData$sex == "male")&(Curated_BivData$hand.foci.count == 1),], family=binomial(link="logit"))
summary(log.reg_1CO_pos_biv)

```




#Q1. Sex Differences in CO Interference (IFD)

In order to test if there are sex differences between crossover interference, a major determinant of the positioning of crossovers along chromosomes, we examined ... ran models with the interfocal distance (IFD) of two foci on the same bivalent. 

-We focus on observations from two-crossover bivalents for more comparable observations.
-comparisons of all IFDs of mulit crossover bivalents is also included
Interference is a major determinant of the positioning of chromosomes

<!-- 
Brief lit review for IFD / interference expectations  
**Petkov et al 2007,  CO interference underlies sex differences in RR**  

"Here we show that in mice, this is because of a shorter genomic interference distance in females than in males, measured in Mb. However, the interference distance is the same in terms of bivalent length. We propose a model in which the interference distance in the two sexes reflects the compaction of chromosomes at the pachytene stage of meiosis."

Chrm1 genetic map, 
(other human and mice refs 
Tease n Hulten 2004: -- no difference between MLH1 foci in males and females

DeBoer 2006, 13 -- measured chrm 1 in males and females, both were 2.8 microns
<!--
ToDo: make a melt.DF with the IFD1, IFD2, IFD3 in a single cols

explain the difference between the raw and PER measures. Why should I use /examine the IFD.PER?
-->


<!--  set up -->

<!--  plots-->
 
```{r Q1.IFD_show.IFD_99, echo=FALSE, warning=FALSE}

#first draft of the IFD comparisons
#make the mouse boxplots -- instead of geom_points

#Only use 2CO bivalents
Curated_BivData.22 <- Curated_BivData[Curated_BivData$hand.foci.count == 2,]
Curated_BivData.22 <- Curated_BivData.22[(!is.na(Curated_BivData.22$hand.foci.count)),]

#boxplots
boxplots_first.IFD.plot_ABS <- ggplot(Curated_BivData.22, aes(y = IFD1_ABS, x= mouse, color=strain))+geom_boxplot(alpha=.2)+
  geom_jitter(alpha=.2)+
  ggtitle("Interfocal Distances, ABS")+
  scale_color_manual(values=colors_of_strains)+  facet_wrap(sex~subsp, scales = "free")+ylim(c(20,160))+theme(legend.position="none")
#order is out of wack, but -- good for now

boxplots_first.IFD.plot_PER <- ggplot(Curated_BivData.22, aes(y = IFD1_PER, x= mouse, color=strain))+geom_boxplot(alpha=.2)+
  geom_jitter(alpha=.2)+
  ggtitle("Interfocal Distances, PER")+
  scale_color_manual(values=colors_of_strains)+  facet_wrap(sex~subsp, scales = "free")+theme(legend.position="none")+ylim(c(0,1))


first.IFD.plot_ABS <- ggplot(Curated_BivData.22, aes(y = IFD1_ABS, x= strain, color=mouse))+geom_jitter()+ggtitle("Interfocal Distances, ABS")+facet_wrap(sex~subsp, scales = "free")+ylim(c(20,160))+theme(legend.position="none") #+scale_color_manual(values=colors_of_strains)
first.IFD.plot_ABS

first.IFD.plot_PER <- ggplot(Curated_BivData.22 %>% filter(chromosomeLength < 150), aes(y = IFD1_PER, x= strain, color=chromosomeLength))+geom_jitter()+ggtitle("Interfocal Distances, PER \ncut off of chrm lengths, 150")+facet_wrap(sex~subsp, scales = "free")+ylim(c(0,1)) 
first.IFD.plot_PER
#+theme(legend.position="none")
```


```{r Q1.IFD_IFD.old.plots, echo=FALSE, eval=FALSE}
#long SC Long_biv_mouse.avs_4MM
#real.long.bivData
#Only use 2CO bivalents

real.long.bivData.22 <- real.long.bivData[real.long.bivData$hand.foci.count == 2,]
real.long.bivData.22 <- real.long.bivData.22[(!is.na(real.long.bivData.22$hand.foci.count)),]


long.biv_IFD_look_ABS <- ggplot(real.long.bivData.22, aes(y = IFD1_ABS, x= mouse, color=strain))+geom_boxplot(alpha=.2)+
  geom_jitter(alpha=.2)+
  ggtitle("Interfocal Distances, ABS")+
  scale_color_manual(values=colors_of_strains)+  facet_wrap(sex~subsp, scales = "free")+ylim(c(20,160))+theme(legend.position="none")+scale_color_manual(values=colors_of_strains)


long.biv_IFD_look_PER <- ggplot(real.long.bivData.22, aes(y = IFD1_PER, x= mouse, color=strain))+geom_boxplot(alpha=.2)+
  geom_jitter(alpha=.2)+
  ggtitle("Interfocal Distances, ABS")+
  scale_color_manual(values=colors_of_strains)+  facet_wrap(sex~subsp, scales = "free")+ylim(c(0,1))+theme(legend.position="none")+scale_color_manual(values=colors_of_strains)
```

<!-- EXPLAIN WHY WE SHOULD CARE ABOUT RAW AND PER IFD values-->

The raw IFD measures -- are usuful for thinking of crossover interference as a mechnical - force mediated through the SC -- In some models this metric should be thought of as the best metric for charaecterizing crossover interference -- bc it reflect the physcial -- 


The above plots display the raw and normalized area which seperate two foci on the same chromosome. In the normalized plot, 9 observations fall above the length threshold of 150.  The female measures fall more uniformly across the range of (0 to 1).

In the abosolute plots, you can start to see clustering/ enrichment of IFD values between 40 and 80 pixels for PWD and SKIVE, 
(and maybe slight clustering for MSM).

<!--t.tests -->

t.tests are performed on the normalized metrics for establish if there are sex differences in the IFD meterics (raw and normalized).


```{r Q1.IFD_Ttests, echo=FALSE}
#below are calq just from 2COs, have no fear
#mean_IFD.2CO_ABS
#mean_IFD.2CO_PER

#FULL all strains
Q1.IFD.raw_FULL_t.test <- t.test( (mouse.avs_4MM %>% filter(sex == "male") %>% filter(strain !=  "CZECH" ) %>% filter(strain !=  "MOLF" )    )$mean_IFD.2CO_ABS,
                           
                                ( mouse.avs_4MM %>% filter(sex == "female") )$mean_IFD.2CO_ABS)
#p-value = 0.07


Q1.IFD.PER_FULL_t.test <- t.test( (mouse.avs_4MM %>% filter(sex == "male") %>% filter(strain !=  "CZECH" ) %>% filter(strain !=  "MOLF" )    )$mean_IFD.2CO_PER,
                                ( mouse.avs_4MM %>% filter(sex == "female") )$mean_IFD.2CO_PER)
#p-value = 1e-12


#WSB
Q1.IFD.raw_t.test_WSB <- t.test( (mouse.avs_4MM %>% filter(sex == "male") %>% filter(strain =="WSB" ))$mean_IFD.2CO_ABS,
                                (mouse.avs_4MM %>% filter(sex == "female") %>% filter(strain =="WSB"))$mean_IFD.2CO_ABS)

Q1.IFD.PER_t.test_WSB <- t.test( (mouse.avs_4MM %>% filter(sex == "male") %>% filter(strain =="WSB" ))$mean_IFD.2CO_PER,
                                ( mouse.avs_4MM %>% filter(sex == "female") %>% filter(strain =="WSB"))$mean_IFD.2CO_PER)

#LEW
Q1.IFD.raw_t.test_LEW <- t.test( (mouse.avs_4MM %>% filter(sex == "male") %>% filter(strain =="LEW" ))$mean_IFD.2CO_ABS,
                                ( mouse.avs_4MM %>% filter(sex == "female") %>% filter(strain =="LEW"))$mean_IFD.2CO_ABS)

Q1.IFD.PER_t.test_LEW <- t.test( (mouse.avs_4MM %>% filter(sex == "male") %>% filter(strain =="LEW"))$mean_IFD.2CO_PER,
                                ( mouse.avs_4MM %>% filter(sex == "female") %>% filter(strain =="LEW"))$mean_IFD.2CO_PER)

#G
Q1.IFD.raw_t.test_G <- t.test( (mouse.avs_4MM %>% filter(sex == "male") %>% filter(strain =="G" ))$mean_IFD.2CO_ABS,
                                ( mouse.avs_4MM %>% filter(sex == "female") %>% filter(strain =="G"))$mean_IFD.2CO_ABS)

Q1.IFD.PER_t.test_G <- t.test( (mouse.avs_4MM %>% filter(sex == "male") %>% filter(strain =="G" ))$mean_IFD.2CO_PER,
                                ( mouse.avs_4MM %>% filter(sex == "female") %>% filter(strain =="G"))$mean_IFD.2CO_PER)


#PWD
Q1.IFD.raw_t.test_PWD <- t.test( (mouse.avs_4MM %>% filter(sex == "male") %>% filter(strain =="PWD" ))$mean_IFD.2CO_ABS,
                                ( mouse.avs_4MM %>% filter(sex == "female") %>% filter(strain =="PWD"))$mean_IFD.2CO_ABS)

Q1.IFD.PER_t.test_PWD <- t.test( (mouse.avs_4MM %>% filter(sex == "male") %>% filter(strain =="PWD" ))$mean_IFD.2CO_PER,
                                ( mouse.avs_4MM %>% filter(sex == "female") %>% filter(strain =="PWD"))$mean_IFD.2CO_PER)


#KAZ
Q1.IFD.raw_t.test_KAZ <- t.test( (mouse.avs_4MM %>% filter(sex == "male") %>% filter(strain =="KAZ" ))$mean_IFD.2CO_ABS,
                                ( mouse.avs_4MM %>% filter(sex == "female") %>% filter(strain =="KAZ"))$mean_IFD.2CO_ABS)

Q1.IFD.PER_t.test_KAZ <- t.test( (mouse.avs_4MM %>% filter(sex == "male") %>% filter(strain =="KAZ" ))$mean_IFD.2CO_PER,
                                ( mouse.avs_4MM %>% filter(sex == "female") %>% filter(strain =="KAZ"))$mean_IFD.2CO_PER)


#SKIVE
Q1.IFD.raw_t.test_SKIVE <- t.test( (mouse.avs_4MM %>% filter(sex == "male") %>% filter(strain =="SKIVE" ))$mean_IFD.2CO_ABS,
                                ( mouse.avs_4MM %>% filter(sex == "female") %>% filter(strain =="SKIVE"))$mean_IFD.2CO_ABS)

Q1.IFD.PER_t.test_SKIVE <- t.test( (mouse.avs_4MM %>% filter(sex == "male") %>% filter(strain =="SKIVE" ))$mean_IFD.2CO_PER,
                                ( mouse.avs_4MM %>% filter(sex == "female") %>% filter(strain =="SKIVE"))$mean_IFD.2CO_PER)

#MSM
Q1.IFD.raw_t.test_MSM <- t.test( (mouse.avs_4MM %>% filter(sex == "male") %>% filter(strain =="MSM" ))$mean_IFD.2CO_ABS,
                                ( mouse.avs_4MM %>% filter(sex == "female") %>% filter(strain =="MSM"))$mean_IFD.2CO_ABS)

Q1.IFD.PER_t.test_MSM <- t.test( (mouse.avs_4MM %>% filter(sex == "male") %>% filter(strain =="MSM" ))$mean_IFD.2CO_PER,
                                ( mouse.avs_4MM %>% filter(sex == "female") %>% filter(strain =="MSM"))$mean_IFD.2CO_PER)


#FULL all strains
Q1.IFD.raw_sansG_t.test <- t.test( (mouse.avs_4MM %>% filter(sex == "male") %>% filter(strain !=  "CZECH" ) %>% filter(strain !=  "MOLF" )  %>%  filter(strain !="G") )$mean_IFD.2CO_ABS,
                           
                                ( mouse.avs_4MM %>% filter(sex == "female") %>%  filter(strain !="G") )$mean_IFD.2CO_ABS)
#p-value = 0.3


```


The raw IFD measures for 2CO bivalents are slightly significant sexes (t.test; p = `r Q1.IFD.raw_FULL_t.test$p.value`). BUT WHEN G IS REMOVE, the pvalue is no longer significant (t.test; p = `r Q1.IFD.raw_sansG_t.test$p.value`). <look closer at G>

The normalized IFD measures are highly significant (t.test; p = `r Q1.IFD.PER_FULL_t.test$p.value`) between sexes, indicating there a

When the t.tests were performed within each strain, all but KAZ show (p-value = 0.3), significant sex differences in the normalized IFD measures
MSM   `r `
SKIVE  `r `
PWD  `r `
MSM  `r `
LEW  `r `
G  `r `
WSB  `r `

This indicates a general pattern of two foci are seperated by more area (hance have stronger interference) in males, this pattern is also for genetic maps.


<!-- short and long -->

```{r Q1.IFD.long.short.setup, echo=FALSE}

#real.long.bivData
#real.short.bivData

Q1.IFD.long <- ddply(real.long.bivData, c("mouse"), summarise,
                             
                        n_obs = length(Obj.ID),
                         mean.SC = mean(chromosomeLength),
            
                      meanIFD2CO_raw = mean(IFD1_ABS[which(hand.foci.count == 2)], na.rm = TRUE),
                     meanIFD2CO_norm = mean(IFD1_PER[which(hand.foci.count == 2)], na.rm = TRUE)
                         
                     )

Q1.IFD.long <- add_sex(Q1.IFD.long)
Q1.IFD.long <- add_strain(Q1.IFD.long)
Q1.IFD.long <- add_subsp(Q1.IFD.long)

Q1.IFD.long$strain <- factor(Q1.IFD.long$strain, ordered = FALSE)
Q1.IFD.long$subsp <- factor(Q1.IFD.long$subsp, ordered = FALSE)



Q1.IFD.short <- ddply(real.short.bivData, c("mouse"), summarise,
                             
                        n_obs = length(Obj.ID),
                         mean.SC = mean(chromosomeLength),
            
                      meanIFD2CO_raw = mean(IFD1_ABS[which(hand.foci.count == 2)], na.rm = TRUE),
                     meanIFD2CO_norm = mean(IFD1_PER[which(hand.foci.count == 2)], na.rm = TRUE)
                         
                     )

Q1.IFD.short <- add_sex(Q1.IFD.short)
Q1.IFD.short <- add_strain(Q1.IFD.short)
Q1.IFD.short <- add_subsp(Q1.IFD.short)

Q1.IFD.short$strain <- factor(Q1.IFD.short$strain, ordered = FALSE)
Q1.IFD.short$subsp <- factor(Q1.IFD.short$subsp, ordered = FALSE)

#t.test. mouse av
Q1.IFD.raw_short.ttest_KAZ <-  t.test( (Q1.IFD.short %>% filter(sex == "male") %>% filter(strain =="KAZ" ))$meanIFD2CO_norm,
                                ( Q1.IFD.short %>% filter(sex == "female") %>% filter(strain =="KAZ"))$meanIFD2CO_norm)

  
Q1.IFD.raw_long.ttest_KAZ <- t.test( (Q1.IFD.long %>% filter(sex == "male") %>% filter(strain =="KAZ" ))$meanIFD2CO_norm,
                                ( Q1.IFD.long %>% filter(sex == "female") %>% filter(strain =="KAZ"))$meanIFD2CO_norm)
#both the short and long norm IFD are NS
  

#t.test Biv
Q1.IFD.raw_BIV.long.ttest_KAZ <-  t.test( (real.long.bivData %>% filter(sex == "male") %>% filter(strain =="KAZ" )  %>% filter(hand.foci.count == "2") )$IFD1_PER,
                ( real.long.bivData %>% filter(sex == "female") %>% filter(strain =="KAZ") %>%  filter(hand.foci.count == "2"))$IFD1_PER)

Q1.IFD.raw_BIV.short.ttest_KAZ <-  t.test( (real.short.bivData %>% filter(sex == "male") %>% filter(strain =="KAZ" )  %>% filter(hand.foci.count == "2") )$IFD1_PER,
                ( real.short.bivData %>% filter(sex == "female") %>% filter(strain =="KAZ") %>%  filter(hand.foci.count == "2"))$IFD1_PER)
```


```{r Q1.show.IFD.things, echo=FALSE}
nrm.IFD_KAZ_long <- ggplot( (Q1.IFD.long %>% filter(strain =="KAZ" ) ) , aes(y = meanIFD2CO_norm, x= sex,color=mouse))+geom_point()+ggtitle("Interfocal Distances, ABS KAZ")

nrm.IFD_KAZ_BIV.long <- ggplot( (real.long.bivData %>% filter(strain =="KAZ" ) %>%  filter(hand.foci.count == "2")  ) , aes(y = IFD1_PER, x= sex, color=mouse))+geom_point()+ggtitle("Interfocal Distances, ABS KAZ")
#11jan18_KAZ_m3, 26jan18_11jan18_KAZ_m3_sp1_8_rev_19
#looked up image, measures confirmed 

####
nrm.IFD_KAZ_short <- ggplot( (Q1.IFD.short %>% filter(strain =="KAZ" ) ) , aes(y = meanIFD2CO_norm, x= sex,color=mouse))+geom_point()+ggtitle("Interfocal Distances, ABS KAZ")

nrm.IFD_KAZ_BIV.short <- ggplot( (real.short.bivData %>% filter(strain =="KAZ" ) %>%  filter(hand.foci.count == "2")  ) , aes(y = IFD1_PER, x= sex, color=mouse))+geom_point()+ggtitle("Interfocal Distances, ABS KAZ")
#21may18_KAZ_m2 -- seems like it might be an outlier
#5jun18_21may18_KAZ_m2_sp1_19.2_rev_16
#second foci slightly smaller, kinda fainter -- but also much brighter than background

#check all KAZ males

nrm.IFD_KAZ_BIV.data <- ggplot( (Curated_BivData %>% filter(strain =="KAZ" ) %>%  filter(hand.foci.count == "2") %>%  filter(sex == "male")  ) , aes(y = IFD1_PER, x= fileName, color=chromosomeLength))+geom_point()+ggtitle("Interfocal Distances, ABS KAZ")+facet_wrap(~mouse, scales = "free")+ylim(c(0,1))

nrm.IFD_KAZ.female_BIV.data <- ggplot( (Curated_BivData %>% filter(strain =="KAZ" ) %>%  filter(hand.foci.count == "2") %>%  filter(sex == "female")  ) , aes(y = IFD1_PER, x= fileName, color=chromosomeLength))+geom_point()+ggtitle("Normalized Interfocal Distances, female KAZ")+facet_wrap(~mouse, scales = "free")+ylim(c(0,1))
nrm.IFD_KAZ.female_BIV.data


nrm.IFD_PWD.female_BIV.data <- ggplot( (Curated_BivData %>% filter(strain =="PWD" ) %>%  filter(hand.foci.count == "2") %>%  filter(sex == "female")  ) , aes(y = IFD1_PER, x= fileName, color=chromosomeLength))+geom_point()+ggtitle("Normalized Interfocal Distances, female PWD")+facet_wrap(~mouse, scales = "free")+ylim(c(0,1))
nrm.IFD_PWD.female_BIV.data

nrm.IFD_LEW.female_BIV.data <- ggplot( (Curated_BivData %>% filter(strain =="LEW" ) %>%  filter(hand.foci.count == "2") %>%  filter(sex == "female")  ) , aes(y = IFD1_PER, x= fileName, color=chromosomeLength))+geom_point()+ggtitle("Normalized Interfocal Distances, female LEW")+facet_wrap(~mouse, scales = "free")+ylim(c(0,1))
nrm.IFD_LEW.female_BIV.data

nrm.IFD_G.female_BIV.data <- ggplot( (Curated_BivData %>% filter(strain =="G" ) %>%  filter(hand.foci.count == "2") %>%  filter(sex == "female")  ) , aes(y = IFD1_PER, x= fileName, color=chromosomeLength))+geom_point()+ggtitle("Normalized Interfocal Distances, female G")+facet_wrap(~mouse, scales = "free")+ylim(c(0,1))
nrm.IFD_G.female_BIV.data

nrm.IFD_WSB.female_BIV.data <- ggplot( (Curated_BivData %>% filter(strain =="WSB" ) %>%  filter(hand.foci.count == "2") %>%  filter(sex == "female")  ) , aes(y = IFD1_PER, x= fileName, color=chromosomeLength))+geom_point()+ggtitle("Normalized Interfocal Distances, female WSB")+facet_wrap(~mouse, scales = "free")+ylim(c(0,1))
nrm.IFD_WSB.female_BIV.data    #18nov17_WSB_f2 has continuous ifd range



nrm.IFD_LEW_BIV.data <- ggplot( (Curated_BivData %>% filter(strain =="LEW" ) %>%  filter(hand.foci.count == "2") %>%  filter(sex == "male")  ) , aes(y = IFD1_PER, x= fileName, color=chromosomeLength))+geom_point()+ggtitle("Interfocal Distances, ABS LEW")+facet_wrap(~mouse, scales = "free")+ylim(c(0,1))


###look at why KAZ males are weird



nrm.IFD_KAZ_BIV.data <- ggplot( (Curated_BivData %>% filter(strain =="KAZ" ) %>%  filter(hand.foci.count == "2") %>%  filter(sex == "male")  ) , aes(y = IFD1_PER, x= fileName, color=chromosomeLength))+geom_point()+ggtitle("Normalized Interfocal Distances, male KAZ")+facet_wrap(~mouse, scales = "free")+ylim(c(0,1))


nrm.IFD_LEW_BIV.data <- ggplot( (Curated_BivData %>% filter(strain =="LEW" ) %>%  filter(hand.foci.count == "2") %>%  filter(sex == "male")  ) , aes(y = IFD1_PER, x= fileName, color=chromosomeLength))+geom_point()+ggtitle("Normalized Interfocal Distances, male LEW")+facet_wrap(~mouse, scales = "free")+ylim(c(0,1))

#the difference the the Nrm IFD cut off is striking with these plots


nrm.IFD_PWD_BIV.data <- ggplot( (Curated_BivData %>% filter(strain =="PWD" ) %>%  filter(hand.foci.count == "2") %>%  filter(sex == "male") ) , aes(y = IFD1_PER, x= fileName, color=chromosomeLength))+geom_point()+ggtitle("Normalized Interfocal Distances, male PWD")+facet_wrap(~mouse, scales = "free")+ylim(c(0,1))

#this might 


nrm.IFD_MSM_BIV.data <- ggplot( (Curated_BivData %>% filter(strain =="MSM" ) %>%  filter(hand.foci.count == "2") %>%  filter(sex == "male")  ) , aes(y = IFD1_PER, x= fileName, color=chromosomeLength))+geom_point()+ggtitle("Normalized Interfocal Distances, male GMSM")+facet_wrap(~mouse, scales = "free")+ylim(c(0,1))


nrm.IFD_SKIVE_BIV.data <- ggplot( (Curated_BivData %>% filter(strain =="SKIVE" ) %>%  filter(hand.foci.count == "2") %>%  filter(sex == "male")  ) , aes(y = IFD1_PER, x= fileName, color=chromosomeLength))+geom_point()+ggtitle("Normalized Interfocal Distances, male SKIVE")+facet_wrap(~mouse, scales = "free")+ylim(c(0,1))

nrm.IFD_G_BIV.data <- ggplot( (Curated_BivData %>% filter(strain =="G" ) %>%  filter(hand.foci.count == "2") %>%  filter(sex == "male")  ) , aes(y = IFD1_PER, x= fileName, color=chromosomeLength))+geom_point()+ggtitle("Normalized Interfocal Distances, male G")+facet_wrap(~mouse, scales = "free")+ylim(c(0,1))

```

see the reference plots within this dir, --

this section -- all strains but KAZ, males have a higher mean normalized IFDs

the female norm IFD are centered at ~50% -- but they also seem to have a higher rate of short (<25%) normalized IFDs

The KAZ strain is the only strain without significant p value for the normalized IFDs, due to KAZ males having a higher rate (enrichment of short normalized IFDs.

There seems to be a 25% norm.IFD cutoff (in females) -- but some mice seem to have less of this cut
off.
Is there a default expectation for why .25 would be the cut off...(given female uniform CO positioning?

<The plots above are focusing on the male pattern -- first meant to look at the pattern in KAZ -- WHICH DOESN"T have significant sex difference in NRM IFD -- INSTEAD it has a low rate of 2CO bivalents with very short IFD(nrm)  --- what if these were mistakes? they were supposed to be 1CO -- but somehow a second CO formed anyways .. and very close by

Where as in PWD, SKIVE and MSM there ARE NO 2CO bivalents which have such short IFD
WHAT espicially werid about this -- 

whats the expectations following the confonding proportions? 
high rec strains have more 2COs so they there is potential for more variance // chromsome specific effects   where as for the low rec strains -- which have fewer 2COs, there might (big might) be less chromosome specific effects.>


**THE male female pattern** female close to 50% -- which fits with uniform rec landscape / weak interference

to rule out chromosome size effects, (espically in KAZ) I re-run the ttests using the short and long biv data sets. Our predictions for these are that longer chromsomes will show more of the 'typical male pattern' while short chromosomes often have a more uniform positing (hence tighter)

-- run these comparisons to see if the small NRM IFD are due to shorter chromosomes, which are predicted to have weaker interference.

Looked at the KAZ pattern

Plotting the single bivalent data for the short and the long data set show in KAZ -- there are two observations near the lowerest IFD1.PER, (even lower than female). These come from 2 different males and I looked up the images to confirm that the measures are accurate (by the bivalent).

Considering that the short-long bivData is much smaller than the full data set

<!-- lmer/ glms-->


### Mixed Model Tests, Fixed Effects

Mixed model analysis for IFD (interference), the first set of models are made with the lme() functions.

**M1. Mixed Model** $$mouse \ average \ IFD ~=~ subsp * sex + rand(strain) + \varepsilon $$

```{r Q1.IFD.M1, echo=FALSE, message=FALSE}
#get a list of the old mice to double check
#old.mice <- DF.HetC.MixedModel.HQ$mouse[DF.HetC.MixedModel.HQ$age.weeks < 25,]
#MIXED MODEL

#https://rdrr.io/cran/lme4/man/pvalues.html notes of fixed effect p values

#Mouse.Table_BivData_4MM #this one is the HetC table (all the Curated BivData)
#Long.biv_mouse_table_HetC  (long biv data set for HetC models)

#M1_IFD_2CO
lmer.IFD.ABS_MM.M1 <- lmer(mean_IFD.2CO_ABS ~ subsp * sex + (1|strain), data=Mouse.Table_BivData_4MM, REML = TRUE) 
lmer.IFD.ABS_MM.M1.reduce.sex <- lmer( mean_IFD.2CO_ABS ~ subsp  +(1|strain), data=Mouse.Table_BivData_4MM )
lmer.IFD.ABS_MM.M1.reduce.subp <- lmer( mean_IFD.2CO_ABS ~ sex  +(1|strain), data=Mouse.Table_BivData_4MM )


ABS.IFD_Q1.M1_sex  <- anova(lmer.IFD.ABS_MM.M1, lmer.IFD.ABS_MM.M1.reduce.sex)$`Pr(>Chisq)`[2]
ABS.IFD_Q1.M1_subsp  <- anova(lmer.IFD.ABS_MM.M1, lmer.IFD.ABS_MM.M1.reduce.subp)$`Pr(>Chisq)`[2]
ABS.IFD_Q1.M1_intr  <- drop1(lmer.IFD.ABS_MM.M1, test="Chisq" )$`Pr(Chi)`[2]
ABS.IFD_Q1.M1_strain  <- exactRLRT(lmer.IFD.ABS_MM.M1)$p.value

Q1.IFD.ABS_M1_lmer.results <- data.frame(sub.sp = ABS.IFD_Q1.M1_subsp, sex.results =  ABS.IFD_Q1.M1_sex, 
               interaction.pval = ABS.IFD_Q1.M1_intr,  rand.pval = ABS.IFD_Q1.M1_strain)

#M1_IFD_2CO
lmer.IFD.PER_MM.M1 <- lmer(mean_IFD.2CO_PER ~ subsp * sex + (1|strain), data=Mouse.Table_BivData_4MM, REML = TRUE) 
lmer.IFD.PER_MM.M1.reduce.sex <- lmer( mean_IFD.2CO_PER ~ subsp  +(1|strain), data=Mouse.Table_BivData_4MM )
lmer.IFD.PER_MM.M1.reduce.subp <- lmer( mean_IFD.2CO_PER ~ sex  +(1|strain), data=Mouse.Table_BivData_4MM )

sex.pval <- anova(lmer.IFD.ABS_MM.M1, lmer.IFD.ABS_MM.M1.reduce.sex)$`Pr(>Chisq)`[2]
subsp.pval <- anova(lmer.IFD.ABS_MM.M1, lmer.IFD.ABS_MM.M1.reduce.subp)$`Pr(>Chisq)`[2]
subsp.sex.pval <- drop1(lmer.IFD.ABS_MM.M1, test="Chisq" )$`Pr(Chi)`[2]
strain.pval <- exactRLRT(lmer.IFD.ABS_MM.M1)$p.value

Q1.IFD.PER_M1_lmer.results <- data.frame(sub.sp = subsp.pval, sex.results =  sex.pval, 
               interaction.pval = subsp.sex.pval,  rand.pval = strain.pval)
#don't understand why 2 rows?

#M1_IFD_2CO
lmer.IFD.PER_MM.M1 <- lmer(mean_IFD.2CO_PER ~ subsp * sex + (1|strain), data=Mouse.Table_BivData_4MM, REML = TRUE) 
lmer.IFD.PER_MM.M1.reduce.sex <- lmer( mean_IFD.2CO_PER ~ subsp  +(1|strain), data=Mouse.Table_BivData_4MM )
lmer.IFD.PER_MM.M1.reduce.subp <- lmer( mean_IFD.2CO_PER ~ sex  +(1|strain), data=Mouse.Table_BivData_4MM )

#anova(lmer.IFD.PER_MM.M1, lmer.IFD.PER_MM.M1.reduce.sex)
#anova(lmer.IFD.PER_MM.M1, lmer.IFD.PER_MM.M1.reduce.subp)

#these 2 below are LR tests
#drop1(lmer.IFD.PER_MM.M1, test="Chisq" )
#exactRLRT(lmer.IFD.PER_MM.M1)

```

The table above should display the slightly unusual pattern, where the coefficients for the significant sex fixed effect are positive and negative in the raw and normalized values respectively. That is for the raw IFD values, females are significantly longer but the normalized IFD values, males are significantly longer.

I tested 2 versions of the mixed model for this flavor of trait, raw IFD and normalized IFD measure. The tables below are from anova( for the lmer model ). Random effect of strain is not significant for ABS IFD, and only slightly significant for the IFD.PER

**M2. Linear Model** $$mouse \ average \ IFD ~=~ subsp * sex * strain + \varepsilon $$

```{r IFD_HetC_M2, echo=FALSE}

glm.Q1_IFD.ABS.M2  <- glm(mean_IFD.2CO_ABS ~ subsp * sex * strain, data=Mouse.Table_BivData_4MM)

summry.glm.Q1_IFD.ABS.M2 <- summary(glm.Q1_IFD.ABS.M2)
#round(summry.glm.Q1_IFD.ABS.M2$coefficients[,c(1,4)],5)

#write.table(round(summry.glm.Q1_IFD.ABS.M2$coefficients,5), "~./MLH1repo/summry.glm.Q1_IFD.ABS.M2.csv", sep=",",row.names = TRUE)

#PER
glm.Q1_IFD.PER.M2 <- glm(mean_IFD.2CO_PER ~ subsp * sex * strain, data=Mouse.Table_BivData_4MM)

summry.glm.Q1_IFD.PER.M2 <- summary(glm.Q1_IFD.PER.M2)
#round(summry.glm.Q1_IFD.PER.M2$coefficients[,c(1,4)],5)

#write.table(round(summry.glm.Q1_IFD.PER.M2$coefficients, 6), "~./MLH1repo/summry.glm.Q1_IFD.PER.M2.csv", sep=",",row.names = TRUE)

```

**M3. Linear model**

$$mouse \ average \ IFD ~=~  sex * strain + \varepsilon $$

```{r IFD_HetC_M3, echo=FALSE}

glm.Q1_IFD.ABS.M3  <- glm(mean_IFD.2CO_ABS ~  sex * strain, data=Mouse.Table_BivData_4MM)

summry.glm.Q1_IFD.ABS.M3 <- summary(glm.Q1_IFD.ABS.M3)
#round(summry.glm.Q1_IFD.ABS.M3$coefficients[,c(1,4)],5)

#write.table(round(summry.glm.Q1_IFD.ABS.M3$coefficients,6), #"~./MLH1repo/summry.glm.Q1_IFD.ABS.M3.csv", sep="\t",row.names = TRUE)

#PER
glm.Q1_IFD.PER.M3  <- glm(mean_IFD.2CO_PER ~ sex * strain, data=Mouse.Table_BivData_4MM)
summry.glm.Q1_IFD.PER.M3 <- summary(glm.Q1_IFD.PER.M3)
round(summry.glm.Q1_IFD.PER.M3$coefficients[,c(1,4)],5)

#write.table(round(summry.glm.Q1_IFD.PER.M3$coefficients,6), #"~./MLH1repo/summry.glm.Q1_IFD.PER.M3.csv", sep=",",row.names = TRUE)

```

For the Mixed models of IFDs, sex is a significant effect for both raw and nrmIFD. for the nrm.IFD, subspecies.

the interaction effects were slightly significant for both raw and nrm.IFD.

The most significant value was the sex effect for nrm.IFDs. 

The the random strain effect was not significant for either model.

For the raw measures, in M2, I think the 2 SKIVE effects mean that the female raw IFD is shorter than the male IFD.raw. The other effects are for PWD and SKIVE (larger raw IFD from intercept.)  I think the MSM and MOLF interaction effects were too far down the list to sop up any variance. For the M3, only SKIVE*male effect is close to significant

For the normalized values in both M2 and M3, sex is a significant effect, increasing nrm.IFD in males. SKIVE*male is the only other consistently significant effect, which also increases the nrm.IFD measure.

Overall There's a low amount of significant effects across the 2CO IFD measures. This might be an indication that interference is conserved across these samples and/or that there is too much noise across from chromosome specific effects.


Dive deeper into the sex specific pattern for each strain.  Below are code chunks which show the unusual sex specific results for IFD measures. The general pattern is that, female raw IFD > male IFD and  female PER IFD < male PER IFD. The scatter plots show that female raw measures are longer than male and for the PER values, the female mean is brought down by an enrichment of short IFDs.   

For some strains, PWD, MSM and SKIVE there's a 30% threshold in the male PER IFD distributions. (What does that mean?). How do I test / quantify this pattern? Cluster metric?


```{r Comp_Ratios, echo=FALSE}

#calq ratios of belo 30

top.male <-  G.Bivalents.DF %>% filter(sex== "male") %>%  filter(hand.foci.count== 2) %>%filter(IFD1_PER <= .3) 

bottom.male <-  G.Bivalents.DF %>% filter(sex== "male") %>%  filter(hand.foci.count== 2) 

ratio.male <- length(top.male$Obj.ID)/ length(bottom.male$Obj.ID)


top.female <-  G.Bivalents.DF %>% filter(sex== "female") %>%  filter(hand.foci.count== 2) %>% filter(IFD1_PER <= .3)

bottom.female <-  G.Bivalents.DF %>% filter(sex== "female")%>%  filter(hand.foci.count== 2)

ratio.female <- length(top.female$Obj.ID)/ length(bottom.female$Obj.ID)

#what is the fastest way to calq all these ratios?  also what's the fastst way to get them into a DF / table?


#put all the DFs in a list, for each DF in the list -- apply the filters for the nom and denominator
MOLF.Bivalents.DF <- Curated_BivData %>% filter(strain == "MOLF")


DF.strain.list <- list(
  WSB.Bivalents.DF, G.Bivalents.DF, LEW.Bivalents.DF, 
  PWD.Bivalents.DF, KAZ.Bivalents.DF, SKIVE.Bivalents.DF,
  MSM.Bivalents.DF, MOLF.Bivalents.DF
  )

entr_DF <- data.frame(strain=c("WSB", "G", "LEW", "PWD", "KAZ", "SKIVE", "MSM", "MOLF"), 
        total.2CO_female=c(0,0,0,0,0,0,0,0), total.2CO_male=c(0,0,0,0,0,0,0,0), below30_male = c(0,0,0,0,0,0,0,0), below30_female = c(0,0,0,0,0,0,0,0),     
          female.ratio = c(0,0,0,0,0,0,0,0), male.ratio = c(0,0,0,0,0,0,0,0))
  
i = 1
for(x in (DF.strain.list) ){
  
  #print(head())
  
  #these filters won't apply to the indexed DF
  top.male =  x %>% filter(sex== "male") %>%  filter(hand.foci.count== 2) %>%filter(IFD1_PER <= .3)

  bottom.male <-  x %>% filter(sex== "male") %>%  filter(hand.foci.count== 2)

  entr_DF$total.2CO_male[i] <- length(bottom.male$Obj.ID)
  entr_DF$below30_male[i] <- length(top.male$Obj.ID)
  entr_DF$male.ratio[i] <- length(top.male$Obj.ID)/ length(bottom.male$Obj.ID)
  
  top.female <-  x %>% filter(sex== "female") %>%  filter(hand.foci.count== 2) %>% filter(IFD1_PER <= .3)
  
  bottom.female <-  x %>% filter(sex== "female")%>%  filter(hand.foci.count== 2)

  
  entr_DF$total.2CO_female[i] <- length(bottom.female$Obj.ID)
  entr_DF$below30_female[i] <- length(top.female$Obj.ID)
  entr_DF$female.ratio[i] <- length(top.female$Obj.ID)/ length(bottom.female$Obj.ID)
  
  i = i+1
  
  }
#entr_DF

entr_DF$sex.dif <- entr_DF$female.ratio - entr_DF$male.ratio

```

Above is a table of the proportion of 2Co bivalents which have a norm IFD below 30%, For all strains but KAZ, the females have a greater proportion of these shorted IFD values.



# transition Q1 to Q2

we tested the sex differences -- now we focus on the largest aspect of variation, the high rec males vs the low rec males.


# Q2 SC Length

<put in plot for total SC and mean gwRR count>

<motivation: the positive correlation, Our prediction for this metric is that the high rec strain will have longer sc lengths (axis lengths)>

<pattern for the single bivalent, pos correlation, The predicted pattern for SC length becomes more nuanced when the data are split up by Chromosome class. Mainly that 1CO are shorter in the high rec strains then low rec strains. This is likely due to the fact that more physically longer chromosomes have 1CO in the low rec strains, which pushes the mean SC length up. Where as in the High rec strains, physically longer chromosomes are more likely to be in the 2CO group.  This supports a general pattern of tighter clustering of SC lengths across chromosome classes in the high rec group. They have a lower probability of 2COs below a certain SC length threshold.>

while testing the sex differences, there were results (the significant subspecies effect) which suggest that the musculus and molossinus strains had longer SC compared to the domesticus strains - these two strains also contain the high rec strains - motivating a model/hypothesis that sc length / chromatin compaction evolution may proceed - facilitate the rapid male specific evolution in gwRR

<!-- basic stats
`r length(OG.MLH1.merge.Skel$fileName)` total skelentonize measures were made from all of the MLH1 pachtyene cell images. X observations removed for having extra signal from other cells. `r length(unique( all.remove ) )` measures removed as potential outliers (by examining the category specific plots)

 place set ups for total SC, short biv and long biv here -->

```{r Q2.SC.short.long.setup, echo=FALSE}

#table(male.long.biv$strain)
#the Long biv dataset is quite unbalance (esp MSM)
male.long.biv <- real.long.bivData %>% filter(sex== "male")

#mouse av for long biv
cell.av.long.biv.table <- ddply(male.long.biv, c("fileName"), summarise,
                             
                        n_obs = length(Obj.ID),
                         mean.SC = mean(chromosomeLength)
                         )
cell.av.long.biv.table <- add_mouse(cell.av.long.biv.table)
cell.av.long.biv.table <- add_strain(cell.av.long.biv.table)
cell.av.long.biv.table <- add_subsp(cell.av.long.biv.table)

cell.av.long.biv.table$strain <- factor(cell.av.long.biv.table$strain, ordered = FALSE)
cell.av.long.biv.table$strain <- factor(cell.av.long.biv.table$strain, ordered = FALSE)
cell.av.long.biv.table$strain <- factor(cell.av.long.biv.table$strain, ordered = FALSE)


Q2mouse.av.long.biv.table <- ddply(male.long.biv, c("mouse"), summarise,
                        n_obs = length(Obj.ID),
                         mean.SC = mean(chromosomeLength),
                        
                         #mean.first.1CO
                        norm.first.foci.pos =  mean(PER_Foci_1[which(hand.foci.count == 1)])
                         )

Q2mouse.av.long.biv.table <- add_strain(Q2mouse.av.long.biv.table)
Q2mouse.av.long.biv.table <- add_subsp(Q2mouse.av.long.biv.table)

Q2mouse.av.long.biv.table$strain <- factor(Q2mouse.av.long.biv.table$strain, ordered = FALSE)
Q2mouse.av.long.biv.table$subsp <- factor(Q2mouse.av.long.biv.table$subsp, ordered = FALSE)

#divide musc males into groups
Q2mouse.av.long.biv.table$Rec.group <- ifelse(grepl("PWD", Q2mouse.av.long.biv.table$strain), 1, 
                                  ifelse(grepl("MSM", Q2mouse.av.long.biv.table$strain), 1,
                                  ifelse(grepl("SKIVE", Q2mouse.av.long.biv.table$strain), 1,
                                           0)))

Q2mouse.av.long.biv.table$Rec.group.exclu <- ifelse(grepl("PWD", Q2mouse.av.long.biv.table$strain), 1, 
                                ifelse(grepl("MSM", Q2mouse.av.long.biv.table$strain), 1, 0))


male.short.biv <- real.short.bivData %>% filter(sex== "male")

Q2mouse.av.short.biv.table <- ddply(male.short.biv, c("fileName"), summarise,
                             
                        n_obs = length(Obj.ID),
                         mean.SC = mean(chromosomeLength),
                        
                        #mean.first.1CO
                        norm.first.foci.pos =  mean(PER_Foci_1[which(hand.foci.count == 1)]
                         ) ) 

Q2mouse.av.short.biv.table <- add_mouse(Q2mouse.av.short.biv.table)
Q2mouse.av.short.biv.table <- add_strain(Q2mouse.av.short.biv.table)
Q2mouse.av.short.biv.table <- add_subsp(Q2mouse.av.short.biv.table)

Q2mouse.av.short.biv.table$strain <- factor(Q2mouse.av.short.biv.table$strain, ordered = FALSE)
Q2mouse.av.short.biv.table$subsp <- factor(Q2mouse.av.short.biv.table$subsp, ordered = FALSE)

Q2mouse.av.short.biv.table$Rec.group <- ifelse(grepl("PWD", Q2mouse.av.short.biv.table$strain), 1, 
ifelse(grepl("MSM", Q2mouse.av.short.biv.table$strain), 1,
                                  ifelse(grepl("SKIVE", Q2mouse.av.short.biv.table$strain), 1,
                                           0)))

Q2mouse.av.short.biv.table$Rec.group.exclu <- ifelse(grepl("PWD", Q2mouse.av.short.biv.table$strain), 1, 
                                ifelse(grepl("MSM", Q2mouse.av.short.biv.table$strain), 1, 0))
```

<!-- plots -->

```{r Q2.show.SC.length, echo=FALSE, warning=FALSE}

Curated_BivData99 <- Curated_BivData[Curated_BivData$hand.foci.count < 5,]
Curated_BivData99 <- Curated_BivData99[(!is.na(Curated_BivData99$sex)),]

#adjust order of plot here
Curated_BivData99$strain <- factor(Curated_BivData99$strain,levels =c(
  "WSB","G","LEW","PERC",  
"PWD", "SKIVE",  "KAZ",
     "MSM", "MOLF",   
                     "CAST",
                         "SPRET","SPIC" ), order=T )

Q2.SC_BivData <- Curated_BivData %>% filter(sex=="male") %>% filter( hand.foci.count < 5)

#nested boxplots, pooled Bivalents
#x axis for hand.focicount -- good for seperating the points --- but it pools 

#prediction for Dom strains: low variation 
Q2.SC.length_Nested_boxplots_Dom <- ggplot(Q2.SC_BivData[Q2.SC_BivData$subsp == "Dom",], aes(y = chromosomeLength, x=strain, fill=as.factor(hand.foci.count) ))+geom_boxplot(alpha=0.5)+ geom_jitter(alpha = .1)+
  ggtitle("Bivalent SC Lengths by Chrm Class \n Dom")+facet_wrap(subsp~sex, scales="free")+ylim(c(25,160))+theme(legend.position="none") + scale_color_manual(values=color_chrm_class)

Q2.SC.length_Nested_boxplots_Dom2 <- ggplot(Q2.SC_BivData[Q2.SC_BivData$subsp == "Dom",], aes(y = chromosomeLength, x=as.factor(hand.foci.count), fill=as.factor(hand.foci.count) ))+geom_boxplot(aes(x=strain),alpha=0.5)+ geom_jitter(alpha = .1)+
  ggtitle("Bivalent SC Lengths by Chrm Class \nDom")+facet_wrap(subsp~sex, scales="free")+ylim(c(25,160))+theme(legend.position="none") +scale_color_manual(values=color_chrm_class)

Q2.SC.length_Nested_boxplots_Musc2 <- ggplot(Q2.SC_BivData[Q2.SC_BivData$subsp == "Musc",], aes(y = chromosomeLength, x=as.factor(hand.foci.count), fill=as.factor(hand.foci.count) ))+geom_boxplot( aes(x=strain), alpha=0.5)+geom_jitter(alpha=.2)+
  ggtitle("Bivalent SC Lengths by Chrm Class \nMusc")+facet_wrap(subsp~sex, scales="free")+ylim(c(25,160))+theme(legend.position="none") +scale_color_manual(values=color_chrm_class)


Q2.SC.length_Nested_boxplots_Mol <- ggplot(Q2.SC_BivData[Q2.SC_BivData$subsp == "Mol",], aes(y = chromosomeLength, x=as.factor(hand.foci.count), fill=as.factor(hand.foci.count) ))+geom_boxplot(aes(x=strain),alpha=0.5)+geom_jitter(alpha=.3)+  ggtitle("Bivalent SC Lengths by Chrm Class\n Mol")+facet_wrap(subsp~sex, scales="free")+ylim(c(25,160))+theme(legend.position="none")+scale_color_manual(values=color_chrm_class)

#+theme(legend.position="none") 
```

```{r Q2.SC.long.biv_plots, echo=FALSE}

#REAL.long_mouse_table
#REAL.short_mouse_table

#real.long.bivData
#real.short.bivData

shoo.long.biv <- ggplot( (real.long.bivData %>% filter(sex == "male") ), aes(y=chromosomeLength, x=strain, color=strain))+geom_jitter()+facet_wrap(~subsp, scales = "free")+scale_color_manual(values=colors_of_strains)+ylim(c(75,140))+geom_boxplot(alpha=.2) + ggtitle("Single Biv \nMale Long Biv SC Lengths")


#cell.avlong.biv <- ggplot(cell.av.long.biv.table, aes(y=mean.sc, x=strain, color=strain))+geom_jitter()+facet_wrap(~subsp, scales = "free")+scale_color_manual(values=colors_of_strains)+ylim(c(75,140))+geom_boxplot(alpha=.2) + ggtitle("Cell av. \nMale Long Biv SC Lengths")


mouse.shoo.long.biv <- ggplot( REAL.long_mouse_table, aes(y=mean.sc, x=strain, color=strain))+geom_jitter()+facet_wrap(~subsp, scales = "free")+scale_color_manual(values=colors_of_strains)+ylim(c(75,140))+geom_boxplot(alpha=.2) + ggtitle("Mouse av. \nMale Long Biv SC Lengths")

                      
#cell.avlong.biv <- ggplot( cell.av.long.biv.table, aes(y=mean.sc, x=strain, color=strain))+geom_jitter()+facet_wrap(~subsp, scales = "free")+scale_color_manual(values=colors_of_strains)+ylim(c(75,140))+geom_boxplot(alpha=.2) + #ggtitle("Cell av. \nMale Long Biv SC Lengths")

```

```{r Q2_tot.sc_plot, echo=FALSE, warning=FALSE}

#totSC.mouse.av.table <- totSC.mouse.av.table %>% filter(strain!="CAST" & strain !="F1"& strain !="SPIC" & strain !=  "CAROLI" & strain !="SPRET"& strain !="HMI"& strain !="PERC")

#MLH1_data <- MLH1_data %>%
#  arrange(strain, sex, mouse) %>%
#  mutate(file.name = factor(file.name))

mouse.av_totsc_male <- ggplot(data= totSC.mouse.av.table %>% filter(sex == "male"), aes(y = mean.skel, x=mouse, color=strain))+ geom_point()+scale_color_manual(values=colors_of_strains)+
  facet_wrap(Rec.group~subsp, scales="free")+ theme(axis.title.x=element_blank(),        axis.text.x=element_blank(), axis.ticks.x=element_blank()) + ylim(c(1200,1850 ))+ggtitle("Mouse av. total SC area")

mouse.av_totsc_male
```


```{r Q2.SC_MLH1.by.totalSC, echo=FALSE}

#make plot of mouse averages for MLH1 and total SC

bep <- merge(  fomo.male_mouse.av,   totSC.mouse.av.table)

X.by.texas <- ggplot(bep, aes(x=mean.skel, y=mean_co, color=strain))+geom_point()+
  scale_color_manual(values=colors_of_strains)+ggtitle("Total SC by mean CO\nMouse Means")

X.by.texas
```

```{r Q2totalSC_checkdates, echo=FALSE, eval=FALSE}

#investigating potential skew / technical effects in the algorithm

Neg.control <- ggplot(bep, aes(x=mean.skel, y=mean.bin, color=strain))+geom_point()+
  scale_color_manual(values=colors_of_strains)+ggtitle("Total SC by mean CO\nMouse Means")
#not sure what my expectations for this plot are, 

median(bep$oldest.image.date) #"2017-08-03"
#there might be an issue with the algorithm performance 
#given the parrelle diagonal lines -- 
#this might be due to the dates imaged  2017-02-07

bep$date.cut.off <- ifelse( bep$oldest.image.date < "2017-02-07", 0, 1 )

Neg.control.find.date <- ggplot(bep, aes(x=mean.skel, y=mean.bin, color=strain))+geom_point()+
  scale_color_manual(values=colors_of_strains)+ggtitle("Total SC by mean CO\nMouse Means")+facet_wrap(~date.cut.off)

#there's evidence that the algorithm might have some biased effects -- 
#that will be investigated later.

```


<!-- tests below;  t.test(high low) OR logistic regression, mixed modlels, glm 
remember the pairwise t.test
Pwise.ttest_Dom.male <- pairwise.t.test(male.Dom.mouse.av$mean_SC, male.Dom.mouse.av$strain)
G-WSB is sig, WSB-LEW are slightly sig
-->

```{r Q2.totSC.ttest, echo=FALSE}
#totalSC high and low rec group

Q2tot.sc_t.test_High.Low <- t.test((totSC.mouse.av.table %>% filter(sex == "male") %>% filter(Rec.group == "0") )$mean.skel,
                (totSC.mouse.av.table %>% filter(sex == "male") %>% filter(Rec.group == "1") )$mean.skel)
#p-value = 0.007


#mol
Q2tot.sc_t.test_High.Low_Mol <- t.test((totSC.mouse.av.table %>% filter(sex == "male") %>% filter(Rec.group == "0") %>% filter(subsp == "Mol") )$mean.skel,
         (totSC.mouse.av.table %>% filter(sex == "male") %>% filter(Rec.group == "1") %>% filter(subsp == "Mol") )$mean.skel)


#musc
Q2tot.sc_t.test_High.Low_Musc <- t.test((totSC.mouse.av.table %>% filter(sex == "male") %>% filter(Rec.group == "0") %>% filter(subsp == "Musc"))$mean.skel,
                (totSC.mouse.av.table %>% filter(sex == "male") %>% filter(subsp == "Musc") %>% filter(Rec.group == "1") )$mean.skel)

```

```{r Q2.short.long.ttest, echo=FALSE}

#long and short 
#mouse.av.short.biv.table
#Q2mouse.av.short.biv.table
Q2short.biv_t.test_High.Low <- t.test((Q2mouse.av.short.biv.table %>% filter(Rec.group == "0") )$mean.SC,
                                      (Q2mouse.av.short.biv.table %>% filter(Rec.group == "1") )$mean.SC)

#p-value = 0.9

#mouse.av.long.biv.table
Q2long.biv_t.test_High.Low <- t.test((Q2mouse.av.long.biv.table  %>% filter(Rec.group == "0") )$mean.SC,
                          (Q2mouse.av.long.biv.table  %>% filter(Rec.group == "1") )$mean.SC)
#p-value = 0.2


#mUSC-short
Q2.short_t.test_High.Low.musc <- t.test((Q2mouse.av.short.biv.table  %>% filter(Rec.group == "0") %>% filter(subsp == "Musc") )$mean.SC,
                          (Q2mouse.av.short.biv.table  %>% filter(Rec.group == "1") %>% filter(subsp == "Musc"))$mean.SC)

#mol-short
#Q2.short_t.test_High.Low.mol <- t.test((Q2mouse.av.short.biv.table  %>% filter(Rec.group == "0") %>% filter(subsp == "Mol") )$mean.SC,
#                          (Q2mouse.av.short.biv.table  %>% filter(Rec.group == "1") %>% filter(subsp == "Mol"))$mean.SC)


#musc-long
Q2.long_t.test_High.Low.musc <- t.test((Q2mouse.av.long.biv.table  %>% filter(Rec.group == "0") %>% filter(subsp == "Musc") )$mean.SC,
                          (Q2mouse.av.long.biv.table  %>% filter(Rec.group == "1") %>% filter(subsp == "Musc"))$mean.SC)

#mol-long
#Q2.long_t.test_High.Low.mol <- t.test((Q2mouse.av.long.biv.table  %>% filter(Rec.group == "0") %>% filter(subsp == "Mol") )$mean.SC,
#                          (Q2mouse.av.long.biv.table  %>% filter(Rec.group == "1") %>% filter(subsp =="Mol"))$mean.SC)

#not enought observations 

```

```{r Q2.SC_HvL_allBiv, echo=FALSE}

#High-Low_All-BivData

Q2.all.BIV_t.test_HL <- t.test((Curated_BivData %>%filter(sex == "male")%>%filter(Rec.group == "0") )$chromosomeLength,
              (Curated_BivData %>%filter(sex == "male")%>%filter(Rec.group == "1"))$chromosomeLength)


#short
Q2.BIV_short_t.test_HL <- t.test((real.short.bivData %>%filter(sex == "male")%>%filter(Rec.group == "0") )$chromosomeLength,
              (real.short.bivData %>%filter(sex == "male")%>%filter(Rec.group == "1"))$chromosomeLength)

#long
Q2.BIV_long_t.test_HL <- t.test((real.long.bivData %>%filter(sex == "male")%>%filter(Rec.group == "0") )$chromosomeLength,
              (real.long.bivData %>%filter(sex == "male")%>%filter(Rec.group == "1"))$chromosomeLength)

```

**mouse means of total SC area for Higher rec strains are significantly higher compared to low rec strains (ttest; p = 
`r Q2tot.sc_t.test_High.Low$p.value`).**

Molossinus have significant difference between mouse means for total sc, between the high and low strains (mol p = `r Q2tot.sc_t.test_High.Low_Mol$p.value`). The total sc mouse means are not significant for the high and low strains in musc (musc p= `r Q2tot.sc_t.test_High.Low_Musc$p.value`).

```{r Q2.SC_across.subsp_tot.SC, echo=FALSE}

#Dom-Musc
Q2.totSC_t.test_DomMusc <- t.test((totSC.mouse.av.table  %>% filter(subsp == "Dom") )$mean.skel,
                          (totSC.mouse.av.table  %>%  filter(subsp == "Musc"))$mean.skel)
#p-value = 0.5

#Dom-Mol
Q2.totSC_t.test_DomMol <- t.test((totSC.mouse.av.table  %>% filter(subsp == "Dom") )$mean.skel,
                          (totSC.mouse.av.table  %>%  filter(subsp == "Mol"))$mean.skel)
#p-value = 0.1
```

```{r Q2.SC_across.subsp_short.long.mus.av, echo=FALSE}

#DOM-musc-long
Q2.long_t.test_DomMusc <- t.test((Q2mouse.av.long.biv.table  %>% filter(subsp == "Dom") )$mean.SC,
                          (Q2mouse.av.long.biv.table %>% filter(subsp == "Musc"))$mean.SC)
#p-value = 0.08

#DOM-musc-short
Q2.short_t.test_DomMusc <- t.test((Q2mouse.av.short.biv.table  %>% filter(subsp == "Dom") )$mean.SC,
                          (Q2mouse.av.short.biv.table %>% filter(subsp == "Musc"))$mean.SC)
#p-value = 0.8

#DOM-mol-long
Q2.long_t.test_DomMol <- t.test((Q2mouse.av.long.biv.table  %>% filter(subsp == "Dom") )$mean.SC,
                          (Q2mouse.av.long.biv.table %>% filter(subsp == "Mol"))$mean.SC)
#p-value = 0.4

#DOM-mol-short
Q2.short_t.test_DomMol <- t.test((Q2mouse.av.short.biv.table  %>% filter(subsp == "Dom") )$mean.SC,
                          (Q2mouse.av.short.biv.table %>% filter(subsp == "Mol"))$mean.SC)
#p-value = 0.09

```

```{r Q2.SC_across.subsp_short.long_biv, echo=FALSE}

#DOM-MUSC-ALL
Q2.all.BIV_t.test_DomMusc <- t.test((Curated_BivData %>%filter(subsp == "Dom")%>%filter(sex == "male") )$chromosomeLength,
              (Curated_BivData %>% filter(subsp == "Musc")%>%filter(sex == "male"))$chromosomeLength)
#p-value = 4e-13

#DOM-MOL-ALL
Q2.all.BIV_t.test_DomMol <- t.test((Curated_BivData%>%filter(subsp == "Dom")%>%filter(sex == "male") )$chromosomeLength,
              (Curated_BivData %>% filter(subsp == "Mol")%>%filter(sex == "male"))$chromosomeLength)
#p-value = 6e-10

#DOM-musc-long
Q2.longBIV_t.test_DomMusc <- t.test((real.long.bivData%>%filter(subsp == "Dom")%>%filter(sex == "male") )$chromosomeLength,
              (real.long.bivData %>% filter(subsp == "Musc")%>%filter(sex == "male"))$chromosomeLength)
#p-value = 0.2

#DOM-musc-short
Q2.shortBIV_t.test_DomMusc <- t.test((real.short.bivData  %>% filter(subsp == "Dom")%>%filter(sex == "male") )$chromosomeLength,
              (real.short.bivData %>% filter(subsp == "Musc")%>%filter(sex == "male"))$chromosomeLength)
#p-value = 0.8


#DOM-mol-long
Q2.longBIV_t.test_DomMol <- t.test((real.long.bivData  %>% filter(subsp == "Dom")%>%filter(sex == "male") )$chromosomeLength,
                          (real.long.bivData %>% filter(subsp == "Mol")%>%filter(sex == "male"))$chromosomeLength)
#p-value = 0.4

#DOM-mol-short
Q2.shortBIV_t.test_DomMol <- t.test((real.short.bivData  %>% filter(subsp == "Dom")%>%filter(sex == "male") )$chromosomeLength,
                          (real.short.bivData %>% filter(subsp == "Mol")%>%filter(sex == "male"))$chromosomeLength)
#p-value = 0.06

```

<across subspecies: only significant at the bivalent level, (Dom and Musc are sig different, Dom - Mol are sig difference) (also when high rec and low rec are compared).>

<!-- models -->

```{r Q2.totSC.glm, echo=FALSE}
#(totSC.mouse.av.table %>% filter(sex == "male") %>% filter(Rec.group == "0") )$mean.skel

Q2.glm_mus.av_totSC_M1 <- glm(mean.skel ~ subsp * strain, data= totSC.mouse.av.table %>% filter(sex == "male"))

sum.Q2.glm_mus.av_totSC_M1  <- summary(Q2.glm_mus.av_totSC_M1)$coefficients

#sum.Q2.glm_mus.av_totSC_M1$coefficients[,4]   pvalues of all effects
#sum.Q2.glm_mus.av_totSC_M1$coefficients[,1]  coefficients of all effects
#round(sum.Q2.glm_mus.av_totSC_M1$coefficients[,c(1,4)], 6)

Q2.glm_mus.av_totSC_M2 <- glm(mean.skel ~ strain, data= totSC.mouse.av.table %>% filter(sex == "male"))

sum.Q2.glm_mus.av_totSC_M2  <- summary(Q2.glm_mus.av_totSC_M2)
#round(summary(sum.Q2.glm_mus.av_totSC_M2)$coefficients[,c(1,4)], 6)
```

```{r Q2.glm.SC.long, echo=FALSE}

#Q2.glm.M2.SC.cell.av_long.biv <- glm(mean.SC ~ strain, data= cell.av.long.biv.table)
#round(summary(glm.Q2.M2.SC.cell.av_long.biv)$coefficients[,c(1,4)], 6)
#table(mouse.av.long.biv.table$mouse)
#Q2mouse.av.long.biv.table


glmQ2_long.biv_M1 <- glm(mean.SC ~ subsp * strain, data= Q2mouse.av.long.biv.table)

sum.glmQ2_long.biv_M1 <- summary(glmQ2_long.biv_M1)

glmQ2_long.biv_M2 <- glm(mean.SC ~ strain, data= Q2mouse.av.long.biv.table)

sum.glmQ2_long.biv_M2 <- summary(glmQ2_long.biv_M2)

#round(sum.glmQ2_long.biv_M2$coefficients[,c(1,4)], 6)
```

```{r Q2.glm.SC.short, echo=FALSE}
#Q2mouse.av.short.biv.table
#mouse.av.short.biv.table

glmQ2_short.biv_M1 <- glm(mean.SC ~ subsp*strain, data= Q2mouse.av.short.biv.table)
sum.glmQ2_short.biv_M1 <- summary(glmQ2_short.biv_M1)

#round(summary(sum.glmQ2_short.biv_M1)$coefficients[,c(1,4)], 6)

glmQ2_short.biv_M2 <- glm(mean.SC ~ strain, data= Q2mouse.av.short.biv.table)
glmQ2_short.biv_M2 <- summary(glmQ2_short.biv_M2)
```

```{r Q2.SC.long.glm.bivglm, echo=FALSE}

#Dom
glm_SC.male_Dom_long.biv <- glm(mean_SC ~ strain, data= Long.biv_mouse_table[Long.biv_mouse_table$subsp == "Dom",])
#round(summary(glm_SC.male_Dom_long.biv)$coefficients[,c(1,4)], 6)

#Musc
glm_SC.male_Musc_long.biv <- glm(mean_SC ~ strain, data= Long.biv_mouse_table[Long.biv_mouse_table$subsp == "Musc",])
#round(summary(glm_SC.male_Musc_long.biv)$coefficients[,c(1,4)], 6)
#summary(glm_SC.male_Musc_long.biv)

#Mol
glm_SC.male_Mol_long.biv <- glm(mean_SC ~ strain, data= Long.biv_mouse_table[Long.biv_mouse_table$subsp == "Mol",])
#round(summary(glm_SC.male_Mol_long.biv)$coefficients[,c(1,4)], 6)


## logistic regression for Rec group
#log.SC.male_long.SC <- glm(Rec.group ~ mean_SC, 
#              data=Long.biv_mouse_table,  family=binomial(link="logit"))

#summary(log.SC.male_long.SC)


log.SC.male.poly_musc <- glm(Rec.group ~ mean_SC, 
              data=Male.poly.Mouse.Table_BivData_4MM[Male.poly.Mouse.Table_BivData_4MM$subsp == "Musc",],  family=binomial(link="logit"))

```

**For the total SC glm**

<Thse are our prediction for the M3 glm (just straint) models based on the variation of the gwRR's
1. no difference in DOM strains (for SC lengths )
2. PWD > SKIVE > the rest
3. Msm > mol>


For total SC in M1 (glm including subsp), the two subspecies effects are significant (glm; p= Musculus `r sum.Q2.glm_mus.av_totSC_M1[,4][2]` and Molossinus p= `r sum.Q2.glm_mus.av_totSC_M1[,4][3]`)   

In the short bivalent data set, analyzed by M1, both subspecies are sig (glm; p= Musculus `r sum.glmQ2_short.biv_M1$coefficients[,4][2]` and Molossinus p= `r sum.glmQ2_short.biv_M1$coefficients[,4][3]`). 

<And the strain effect for G is significant (glm; G p =`r sum.glmQ2_short.biv_M1$coefficients[,4][4]`).>

<For the M2 model analyzing the mouse means short bivalents, all strains have significant effect but LEW, G,PWD, MSM, MOLF, SKIVE, KAZ, CZECH.>


<!--24sep14_PWD_m1 is lower than the others, 5jul17_MSM_m1 is the low MSM-->

<just like the sex differences in SC area -- this points to differential usage of the SC for placing CO -- investigate the recombination landscape>

<!--
blocks of other male Q2 - SC lengths
what are the pairwise t-test functions -- use pairwise.t.test(Ozone, Month, p.adj = "bonf")
The mouse average and single bivalents are significantly longer in the high rec group p = r T.test_SC.Len_mouse.av$p.value and p=  r T.test_SC.Len_single.biv$p.value.
-->

**The mouse means for the reduced bivalent datasets (short and long) are not significantly different between the high and low rec strains (for the total pooled data) (ttest; short p = `r Q2short.biv_t.test_High.Low$p.value` and long p = `r Q2long.biv_t.test_High.Low$p.value`).**

**For the long bivalent dataset, In M1 where the subsp are tested, only Musc is significant (glm; Musc p= `r sum.glmQ2_long.biv_M1$coefficients[,4][2]`). The significant strain effects in M1 are slight just MSM (glm; MSM p= `r sum.glmQ2_long.biv_M1$coefficients[,4][7]`).**

**For the long bivalent dataset, In M2 (limited to just the strain effects) a few strain effects are significant (glm; MSM p = `r sum.glmQ2_long.biv_M2$coefficients[,4][5]`, SKIVE `r sum.glmQ2_long.biv_M2$coefficients[,4][7]` and CZECH p = `r sum.glmQ2_long.biv_M2$coefficients[,4][9]`).**

> **Summary Points**

- confirm that the positive correlation of SC and CO number is maintained

- Predicted differences, between High and low strains OR between Dom and Musc/Mol, only have significant p values when the single bivalent levels are used.

- (the glms, indecate there are many strain effects), Using the mouse average SC lengths, in the full data set all strain effects are significant. This is an indication that there is more variation for the SC lengths than for gwRR / CO counts.

- For the M3 models of mouse av.s across SC meterics, for many of the SC metrics, most all of the strain effects are significant

- for the M2 models of the mouse av, musc - mol are significant fixed effects

- many strain effects, complications in the SC meterics; short/long single bivalents have lower sample size and the total SC might have noise due to the algorithm / differences in the imaging

- th predictions for M3 (strain) differencs (motivated by the gwrr variation are not met,  (correlation of SC means and CO means?))

<Each point is a mouse average SC. When all bivalents are pooled, the logistic regression SC lengths predict the groups. When the data is divided by CO number, 1CO's have a higher SC length in the low group - in the low group more chrms, including the longer ones have 1CO - whereas in the high group, only shorter chrms (SC) are in the 1CO class. For the 2CO class of bivalents - all the SC are longer, the high group averages were longer than the low.>

<!-- junk chunks below -->

```{r Q2.SC.logreg.chrmClass_BivData, echo=FALSE, include=FALSE, eval=FALSE}

log.SC_male_chrm.class_all <- glm(chrm.class ~ chromosomeLength,
              data=Bivalent_1o2,  family=binomial(link="logit"))

summary(log.SC_male_chrm.class_all)


#use the 1CO or 2CO DF
Bivalent_1o2_WSB <-  Bivalent_1o2 %>% filter(strain == "WSB")

log.SC_male_chrm.class.WSB <- glm(chrm.class ~ chromosomeLength,
              data=Bivalent_1o2_WSB,  family=binomial(link="logit"))

summary(log.SC_male_chrm.class.WSB)


Bivalent_1o2_G <-  Bivalent_1o2 %>% filter(strain == "G") 
log.SC_chrm.class.G <- glm(chrm.class ~ chromosomeLength,
              data=Bivalent_1o2_G,  family=binomial(link="logit"))
summary(log.SC_chrm.class.G)


Bivalent_1o2_LEW <-  Bivalent_1o2 %>% filter(strain == "LEW") 
log.SC_chrm.class.LEW <- glm(chrm.class ~ chromosomeLength,
              data=Bivalent_1o2_LEW,  family=binomial(link="logit"))
summary(log.SC_chrm.class.LEW)


Bivalent_1o2_PWD <-  Bivalent_1o2 %>% filter(strain == "PWD") 
log.SC_chrm.class.PWD <- glm(chrm.class ~ chromosomeLength,
              data=Bivalent_1o2_PWD,  family=binomial(link="logit"))
summary(log.SC_chrm.class.PWD)


Bivalent_1o2_SKIVE <-  Bivalent_1o2 %>% filter(strain == "SKIVE") 
log.SC_chrm.class.SKIVE <- glm(chrm.class ~ chromosomeLength,
              data=Bivalent_1o2_SKIVE,  family=binomial(link="logit"))
summary(log.SC_chrm.class.SKIVE)


Bivalent_1o2_KAZ <-  Bivalent_1o2 %>% filter(strain == "KAZ") 
log.SC_chrm.class.KAZ <- glm(chrm.class ~ chromosomeLength,
              data=Bivalent_1o2_KAZ,  family=binomial(link="logit"))
summary(log.SC_chrm.class.KAZ)


Bivalent_1o2_CZECH <-  Bivalent_1o2 %>% filter(strain == "CZECH") 
log.SC_chrm.class.CZECH <- glm(chrm.class ~ chromosomeLength,
              data=Bivalent_1o2_CZECH,  family=binomial(link="logit"))
summary(log.SC_chrm.class.CZECH)

Bivalent_1o2_MSM <-  Bivalent_1o2 %>% filter(strain == "MSM") 
log.SC_chrm.class.MSM <- glm(chrm.class ~ chromosomeLength,
              data=Bivalent_1o2_MSM,  family=binomial(link="logit"))
summary(log.SC_chrm.class.MSM)


Bivalent_1o2_MOLF <-  Bivalent_1o2 %>% filter(strain == "MOLF")
log.SC_chrm.class.MOLF <- glm(chrm.class ~ chromosomeLength,
              data=Bivalent_1o2_MOLF,  family=binomial(link="logit"))
summary(log.SC_chrm.class.MOLF)

```

```{r Q2.logreg.plots, echo=FALSE, include=FALSE, eval=FALSE}

log.reg.chrm.class_all.strains <- ggplot(Bivalent_1o2,
                          aes(x=chromosomeLength, y=chrm.class)) + geom_point(aes(color=strain)) +  scale_color_manual(values=colors_of_strains)+ stat_smooth(method="glm", method.args=list(family="binomial"), se=FALSE) + ggtitle("Chrm Class prediction (1CO or 2CO)")+facet_wrap(subsp~ strain)
log.reg.chrm.class_all.strains



WSB.log.reg.chrm.class <- ggplot(Bivalent_1o2_WSB,
                          aes(x=chromosomeLength, y=chrm.class)) + geom_point(aes(color=strain)) +  scale_color_manual(values=colors_of_strains)+ stat_smooth(method="glm", method.args=list(family="binomial"), se=FALSE) + ggtitle("Chrm Class prediction (1CO or 2CO, WSB)")+facet_wrap(~subsp)


MSMlog.reg.chrm.class <- ggplot(Bivalent_1o2_MSM,
                          aes(x=chromosomeLength, y=chrm.class)) + geom_point(aes(color=strain)) +  scale_color_manual(values=colors_of_strains)+ stat_smooth(method="glm", method.args=list(family="binomial"), se=FALSE) + ggtitle("MSM Chrm Class prediction (1CO or 2CO)")+facet_wrap(~subsp)


PWDlog.reg.chrm.class <- ggplot(Bivalent_1o2_PWD,
                          aes(x=chromosomeLength, y=chrm.class)) + geom_point(aes(color=strain)) +  scale_color_manual(values=colors_of_strains)+ stat_smooth(method="glm", method.args=list(family="binomial"), se=FALSE) + ggtitle("PWD Chrm Class prediction (1CO or 2CO)")+facet_wrap(~subsp)

```

```{r Q2.SC.Chrm.Class, echo=FALSE, include=FALSE, eval=FALSE}
#REMOVE COMPARES ACROSS FROM FULL BIVDAAT

#1CO
log.SC_male_1CO.poly <- glm(Rec.group ~ mean.SC_1CO,
              data=Male.poly.Mouse.Table_BivData_4MM,  family=binomial(link="logit"))
#summary(log.SC_male_1CO.poly)

log.SC_male_1CO.Musc <- glm(Rec.group ~ mean.SC_1CO,
              data=Male.poly.Mouse.Table_BivData_4MM[Male.poly.Mouse.Table_BivData_4MM$subsp == "Musc",],  family=binomial(link="logit"))
#summary(log.SC_male_1CO.Musc)
#double check these directions

log.SC_male_1CO.Mol <- glm(Rec.group ~ mean.SC_1CO,
              data=Male.poly.Mouse.Table_BivData_4MM[Male.poly.Mouse.Table_BivData_4MM$subsp == "Mol",],  family=binomial(link="logit"))

#summary(log.SC_male_1CO.Mol)


#2CO
log.SC_male_2CO.poly <- glm(Rec.group ~ mean.SC_2CO,
              data=Male.poly.Mouse.Table_BivData_4MM,  family=binomial(link="logit"))
#summary(log.SC_male_2CO.poly)


log.SC_male_2CO.Musc <- glm(Rec.group ~ mean.SC_2CO,
              data=Male.poly.Mouse.Table_BivData_4MM[Male.poly.Mouse.Table_BivData_4MM$subsp == "Musc",],  family=binomial(link="logit"))
#summary(log.SC_male_2CO.Musc)


log.SC_male_2CO.Mol <- glm(Rec.group ~ mean.SC_2CO,
              data=Male.poly.Mouse.Table_BivData_4MM[Male.poly.Mouse.Table_BivData_4MM$subsp == "Mol",],  family=binomial(link="logit"))
#summary(log.SC_male_2CO.Mol)


#3CO
log.SC_male_3CO.poly <- glm(Rec.group ~ mean.SC_3CO,
              data=Male.poly.Mouse.Table_BivData_4MM,  family=binomial(link="logit"))
#summary(log.SC_male_3CO.poly)


log.SC_male_3CO.Musc <- glm(Rec.group ~ mean.SC_3CO,
              data=Male.poly.Mouse.Table_BivData_4MM[Male.poly.Mouse.Table_BivData_4MM$subsp == "Musc",],  family=binomial(link="logit"))
#summary(log.SC_male_3CO.Musc)


#there's no rec group for this table
log.SC_male_3CO.Mol <- glm(Rec.group ~ mean.SC_3CO,
              data=Male.poly.Mouse.Table_BivData_4MM[Male.poly.Mouse.Table_BivData_4MM$subsp == "Mol",],  family=binomial(link="logit"))
#summary(log.SC_male_3CO.Mol)


#reverse logistic regression (single bivalent level)
#how well does SC length predict 1 or 2 COs?

Bivalent_1o2 <-  Curated_BivData_male %>% filter(species== "M.musculus") %>% filter(hand.foci.count > 0) %>% filter(hand.foci.count < 3) 

#need to switch the co counts to 1 - 0
Bivalent_1o2$chrm.class <- ifelse(grepl(1, Bivalent_1o2$hand.foci.count), 0, 
                                    ifelse(grepl(2, Bivalent_1o2$hand.foci.count), 1, ""))


Bivalent_1o2$chrm.class <- as.numeric(Bivalent_1o2$chrm.class)
Bivalent_1o2$strain <- factor(Bivalent_1o2$strain, ordered = FALSE)

log.SC_male_chrm.class <- glm(chrm.class ~ chromosomeLength*strain,
              data=Bivalent_1o2,  family=binomial(link="logit"))

summary(log.SC_male_chrm.class)
#not sure if this is the equivilent of running each strain seperatly
#the strain effect is probably also pulling in the different proportion of chrm class
#useing the chrm.length*strain gives fewer significant factors. 

#i'd be more convinced by running all those log regression plots for each strain
#come back later
```


<These are the plots of the logistic regression.>

```{r Q2.SC.log.regression.show.plots, echo=FALSE, include=FALSE}

plot.blah.mouse.av <- ggplot(Male.poly.Mouse.Table_BivData_4MM,
                          aes(x=mean_SC, y=Rec.group)) + geom_point(aes(color=strain)) +  scale_color_manual(values=colors_of_strains)+ stat_smooth(method="glm", method.args=list(family="binomial"), se=FALSE) + ggtitle("Mouse av. SC (pooled) \nHigh vs low gwRR strains")+facet_wrap(~subsp)


plot.logReg.mouse.av_1CO <- ggplot(Male.poly.Mouse.Table_BivData_4MM,
                          aes(x=mean.SC_1CO, y=Rec.group)) + geom_point(aes(color=strain)) +  scale_color_manual(values=colors_of_strains)+ stat_smooth(method="glm", method.args=list(family="binomial"), se=FALSE) + ggtitle("Mouse av. SC 1CO \nHigh vs low gwRR strains")+facet_wrap(~subsp)


plot.logReg.mouse.av_2CO <- ggplot(Male.poly.Mouse.Table_BivData_4MM,
                          aes(x=mean.SC_2CO, y=Rec.group)) + geom_point(aes(color=strain)) +  scale_color_manual(values=colors_of_strains)+ stat_smooth(method="glm", method.args=list(family="binomial"), se=FALSE) + ggtitle("Mouse av. SC 2CO \nHigh vs low gwRR strains")+facet_wrap(~subsp)


```

```{r Q2.SClength.ttest, echo=FALSE, include=FALSE}
#COMPARISONS ACROSS ALL CLASSES

#MOUSE AVE, all
Rec.group.mouse.SC_Ttest <- t.test(Male.poly.Mouse.Table_BivData_4MM$mean_SC[Male.poly.Mouse.Table_BivData_4MM$Rec.group == 1], 
                                 Male.poly.Mouse.Table_BivData_4MM$mean_SC[Male.poly.Mouse.Table_BivData_4MM$Rec.group == 0])
 
Rec.group.mouse.SC_Ttest$p.value  #p-value = 0.00238

#mouse.av 1CO
Rec.group.mouse.SC.1CO_Ttest <- t.test(Male.poly.Mouse.Table_BivData_4MM$mean.SC_1CO[Male.poly.Mouse.Table_BivData_4MM$Rec.group == 1], 
                                 Male.poly.Mouse.Table_BivData_4MM$mean.SC_1CO[Male.poly.Mouse.Table_BivData_4MM$Rec.group == 0])
 
Rec.group.mouse.SC.1CO_Ttest$p.value  #p-value = 0.9993


#MOUSE.av 2CO
Rec.group.mouse.SC.2CO_Ttest <- t.test(Male.poly.Mouse.Table_BivData_4MM$mean.SC_2CO[Male.poly.Mouse.Table_BivData_4MM$Rec.group == 1], 
                                 Male.poly.Mouse.Table_BivData_4MM$mean.SC_2CO[Male.poly.Mouse.Table_BivData_4MM$Rec.group == 0])
 
Rec.group.mouse.SC.2CO_Ttest$p.value  #p-value = 0.03428


#single biv level
#male.biv?

#Rec.group_BivData_F1post_Ttest <- t.test(male.bivdata.1CO$PER_Foci_1[male.bivdata.1CO$Rec.group == 1], 
#                                 male.bivdata.1CO$PER_Foci_1[male.bivdata.1CO$Rec.group == 0])
 
#Rec.group_BivData_F1post_Ttest$p.value 

```

```{r log.reg_SC.length, echo=FALSE, include=FALSE}
#REMOVE THIS COMPARES FROM FULL BIV DATA

#male_mouse.av
#Male.poly.Mouse.Table_BivData_4MM

Curated_BivData$Rec.group <- ifelse(grepl("PWD male", Curated_BivData$category), 1, 
                                ifelse(grepl("MSM male", Curated_BivData$category), 1,
                                ifelse(grepl("SKIVE male", Curated_BivData$category), 1, 0)))

#male poly shouldn't be here
Male.poly.Mouse.Table_BivData_4MM$Rec.group <- ifelse(grepl("PWD male", Male.poly.Mouse.Table_BivData_4MM$category), 1, 
                                ifelse(grepl("MSM male", Male.poly.Mouse.Table_BivData_4MM$category), 1,
                                ifelse(grepl("SKIVE male", Male.poly.Mouse.Table_BivData_4MM$category), 1, 0)))


#All chrm
log.reg_SC.length_mouse.av <- glm(Rec.group ~ mean_SC, 
                            data=Male.poly.Mouse.Table_BivData_4MM[(Male.poly.Mouse.Table_BivData_4MM$subsp == "Musc"),], family=binomial(link="logit"))
#summary(log.reg_SC.length_mouse.av)

#considering seperating out by hand foci counts

#1CO
log.reg_SC.length_mouse.av_1CO <- glm(Rec.group ~ mean.SC_1CO, 
                            data=Male.poly.Mouse.Table_BivData_4MM[(Male.poly.Mouse.Table_BivData_4MM$subsp == "Musc"),], family=binomial(link="logit"))
#summary(log.reg_SC.length_mouse.av_1CO)


#2CO
log.reg_SC.length_mouse.av_2CO <- glm(Rec.group ~ mean.SC_2CO, 
                            data=Male.poly.Mouse.Table_BivData_4MM[(Male.poly.Mouse.Table_BivData_4MM$subsp == "Musc"),], family=binomial(link="logit"))

#summary(log.reg_SC.length_mouse.av_2CO)


#3CO
log.reg_SC.length_mouse.av_3CO <- glm(Rec.group ~ mean.SC_3CO, 
                            data=Male.poly.Mouse.Table_BivData_4MM[(Male.poly.Mouse.Table_BivData_4MM$subsp == "Musc"),], family=binomial(link="logit"))
#summary(log.reg_SC.length_mouse.av_3CO)


# bivalent level log regression

log.reg_SC.length_BIV <- glm(Rec.group ~ chromosomeLength, 
                            data=Curated_BivData[(Curated_BivData$subsp == "Musc")&(Curated_BivData$sex == "male"),], family=binomial(link="logit"))

#summary(log.reg_SC.length_BIV)

#1CO  (BIVDATA)
log.reg_SC.length_BIV_1CO <- glm(Rec.group ~ chromosomeLength, 
                            
                data=Curated_BivData[(Curated_BivData$subsp == "Musc") & (Curated_BivData$hand.foci.count == 1) &(Curated_BivData$sex == "male"),], family=binomial(link="logit"))
#summary(log.reg_SC.length_BIV_1CO)


#2CO
log.reg_SC.length_BIV_2CO <- glm(Rec.group ~ chromosomeLength, 
                            data=Curated_BivData[(Curated_BivData$subsp == "Musc")&(Curated_BivData$hand.foci.count == 2) & (Curated_BivData$sex == "male"),], family=binomial(link="logit"))
#summary(log.reg_SC.length_BIV_2CO)


#3CO
log.reg_SC.length_BIV_3CO <- glm(Rec.group ~ chromosomeLength, 
                            data=Curated_BivData[(Curated_BivData$subsp == "Musc")&(Curated_BivData$hand.foci.count == 3)&(Curated_BivData$sex == "male"),], family=binomial(link="logit"))

#summary(log.reg_SC.length_BIV_3CO)
```

```{r Q2.SC_class.rec.group_setup, echo=FALSE, include=FALSE}
#REMOVE THIS, IT uses total bivalent data


#rerun the models for datasets -- deviding by chrm class and rec group
Curated_BivData$Rec.group <- ifelse(grepl("PWD male", Curated_BivData$category), 1, 
                            ifelse(grepl("MSM male", Curated_BivData$category), 1,
                             ifelse(grepl("SKIVE male", Curated_BivData$category), 1, 0)) )


Q2.SC_BivData_low.rec <- Curated_BivData %>% filter(sex=="male") %>% filter( hand.foci.count < 5)%>% filter(Rec.group == 0 )
Q2.SC_BivData_high.rec <- Curated_BivData %>% filter(sex=="male") %>% filter( hand.foci.count < 5) %>% filter(Rec.group == 1 )

#calq_mouse av
mouse.av.SC.low.rec <- ddply(Q2.SC_BivData_low.rec, c("mouse"), summarise,
                        n_obs = length(Obj.ID),
                        n_cells = length(unique(fileName) ),
                        mean.SC.total = mean(chromosomeLength),

                    mean.SC_0CO = mean(chromosomeLength[which(hand.foci.count == 0)], na.rm = TRUE),
                  n_0CO =  sum((hand.foci.count == 0), na.rm = TRUE ),
                  
                   mean.SC_1CO = mean(chromosomeLength[which(hand.foci.count == 1)], na.rm = TRUE),
                    n_1CO = sum((hand.foci.count == 1), na.rm = TRUE),
                  
                   mean.SC_2CO = mean(chromosomeLength[which(hand.foci.count == 2)], na.rm = TRUE),
                   n_2CO = sum((hand.foci.count == 2), na.rm = TRUE),
                  
                  mean.SC_3CO = mean(chromosomeLength[which(hand.foci.count == 3)], na.rm = TRUE),
                  n_3CO = sum((hand.foci.count == 3), na.rm = TRUE)
                                                
                         )

mouse.av.SC.low.rec <- add_strain(mouse.av.SC.low.rec)
mouse.av.SC.low.rec <- add_subsp(mouse.av.SC.low.rec)

mouse.av.SC.low.rec$strain <- factor(mouse.av.SC.low.rec$strain, ordered = FALSE)
mouse.av.SC.low.rec$subsp <- factor(mouse.av.SC.low.rec$subsp, ordered = FALSE)


#calq_mouse av
mouse.av.SC.high.rec <- ddply(Q2.SC_BivData_high.rec, c("mouse"), summarise,
                        n_obs = length(Obj.ID),
                         mean.SC = mean(chromosomeLength)
                         )
mouse.av.SC.high.rec <- add_mouse(mouse.av.SC.high.rec)
mouse.av.SC.high.rec <- add_strain(mouse.av.SC.high.rec)
mouse.av.SC.high.rec <- add_subsp(mouse.av.SC.high.rec)

mouse.av.SC.high.rec$strain <- factor(mouse.av.SC.high.rec$strain, ordered = FALSE)
mouse.av.SC.high.rec$add_subsp <- factor(mouse.av.SC.high.rec$subsp, ordered = FALSE)
```

```{r male.show.plotsEXample, echo=FALSE, eval=FALSE}
male_mouse.av <- subset(AP_mouse_table_w.Ages, select= -c(sex.y,sex.x,
                            subsp.y,subsp.x,
                    category.y,category.x, strain.x,
                    strain.y, mouse.1  )  )

male_mouse.av <- add_sex(male_mouse.av)
male_mouse.av <- add_strain(male_mouse.av)
male_mouse.av <- add_subsp(male_mouse.av)
male_mouse.av <- add_species(male_mouse.av)

fomo.male_mouse.av <-  male_mouse.av %>% filter(sex== "male") %>% filter(species == "M.musculus") 

fomo.male_mouse.av$subsp <- factor(fomo.male_mouse.av$subsp, ordered = FALSE)
fomo.male_mouse.av$sex <- factor(fomo.male_mouse.av$sex, ordered = FALSE)
fomo.male_mouse.av$strain <- factor(fomo.male_mouse.av$strain, ordered = FALSE)
fomo.male_mouse.av$species <- factor(fomo.male_mouse.av$species, ordered = FALSE)

table( droplevels(fomo.male_mouse.av$strain) )
#there are cast strains too


Musc.male_mouse.av <-  fomo.male_mouse.av %>% filter(sex== "male") %>% filter(subsp == "Musc")

musc.male.plot <- ggplot(Musc.male_mouse.av, aes(color= strain))+geom_jitter(aes(x = as.factor(strain), y = mean_co, color= strain, alpha=0.5), width = .2 )+ ggtitle("Musc male of MLH1 distributions")+scale_color_manual(values=colors_of_strains)+ylim(c(20,34))
musc.male.plot

# + theme(axis.title.x=element_blank(),
#        axis.text.x=element_blank(),
#        axis.ticks.x=element_blank()) +
#ylim( c((min(MLH1_data$adj_nMLH1.foci)-3) ), (max(MLH1_data$adj_nMLH1.foci)+3)) +  


#make the Mol and
Mol.male_mouse.av <-  fomo.male_mouse.av %>% filter(sex== "male") %>% filter(subsp == "Mol")

Mol.male.plot <- ggplot(Mol.male_mouse.av, aes(color= strain))+geom_jitter(aes(x = as.factor(strain), y = mean_co, color= strain, alpha=0.5), width = .2 )+ ggtitle("Mol male of MLH1 distributions")+scale_color_manual(values=colors_of_strains)+ylim(c(20,33))
Mol.male.plot


#Dom version 
Dom.male_mouse.av <-  fomo.male_mouse.av %>% filter(sex== "male") %>% filter(subsp == "Dom")

Dom.male.plot <- ggplot(Dom.male_mouse.av, aes(color= strain))+geom_jitter(aes(x = as.factor(strain), y = mean_co, color= strain, alpha=0.5), width = .2 )+ ggtitle("Dom male of MLH1 distributions")+scale_color_manual(values=colors_of_strains)+ylim(c(20,33))
Dom.male.plot
```

```{r SC.length.male_tests, echo=FALSE, include=FALSE, eval=FALSE}
## run the tests to check following expectations

#1.Dom strains not different from ech other
#2. PWD and SKIVE are different from the other strains
#3.MSM and MOLF are different from each other

#male.Musc.mouse.av <- Male.poly.Mouse.Table_BivData_4MM[(Male.poly.Mouse.Table_BivData_4MM$subsp == "Musc"), ]
  
male.Dom.mouse.av <- Male.poly.Mouse.Table_BivData_4MM[(Male.poly.Mouse.Table_BivData_4MM$subsp == "Dom"), ]
#the subsp labels get messed up, unless they are designated chr first
Dom.male_SC.length_lm <- lm(mean_SC~ strain,  data =male.Dom.mouse.av  )

Dom.male_SC.length_lm.1 <- lm(mean.SC_1CO~ strain/mouse,  data =male.Dom.mouse.av  ) #wierd sytax 
Dom.male_SC.length_lm.2 <- lm(mean.SC_2CO~ strain,  data =male.Dom.mouse.av  )
#Dom.male_SC.length_lm.3 <- lm(mean.SC_3CO~ strain,  data =male.Dom.mouse.av  ) #only WSB and LEW?
#total sc sig, 1CO sig, 2CO sig


Dom.male_SC.aov <- aov(mean_SC ~ strain, data = male.Dom.mouse.av)
#0.000369 ***
Dom.male_1CO_SC.aov <- aov(mean.SC_1CO ~ strain, data = male.Dom.mouse.av)
#0.000816 ***
Dom.male_2CO_SC.aov <- aov(mean.SC_2CO ~ strain, data = male.Dom.mouse.av)
#0.0148 *

show.Dom.male.SC <- ggplot(male.Dom.mouse.av, aes(y= mean_SC, x= strain, color=strain))+ geom_jitter()

show.Dom.male.SC <- ggplot(male.Dom.mouse.av, aes(y= mean.SC_2CO, x= strain, color=strain))+ geom_jitter()
#making plots above show that mouse average SC lengths are lower could there be microscope issue?


#make sure to remove some of the juvi mice!! from the AP_mouse table
show.Dom.male.CO <- ggplot(
male_mouse.av[(male_mouse.av$subsp.x == "Dom"),], aes(y= mean_co, x= strain.x, color=strain.x))+ geom_jitter()

Dom.male_CO.aov <- aov(mean_co ~ strain, data = male_mouse.av[ (male_mouse.av$subsp == "Dom"),])
#P not sig for CO mean

#seems like there are more mice in the MLH1 data set than bivData
#

#WSB all dissected and imaged in '15
#G imaged 15 - 16
#LEW imaged 15 to 17

ggboxplot(data = male.Dom.mouse.av, x = "strain", y = "mean_SC", 
          color = "strain", palette = c("#00AFBB", "#E7B800", "#FC4E07"),
          ylab = "", xlab = "")


show.Dom.male.SC <- ggplot(male.Dom.mouse.av, aes(y= mean_SC, x= strain, color=strain))+ geom_jitter()

ggboxplot(data = male.Dom.mouse.av, x = "strain", y = "mean_SC", 
          color = "strain", palette = c("#00AFBB", "#E7B800", "#FC4E07"),
          ylab = "", xlab = "")


ggboxplot(data = male.Dom.mouse.av, x = "strain", y = "mean.SC_1CO", 
          color = "strain", palette = c("#00AFBB", "#E7B800", "#FC4E07"),
          ylab = "1CO SC Length", xlab = "")


ggboxplot(data = male.Dom.mouse.av, x = "strain", y = "mean.SC_2CO", 
          color = "strain", palette = c("#00AFBB", "#E7B800", "#FC4E07"),
          ylab = "2CO SC Length", xlab = "")
```

```{r SC.length.pairwise.T, echo=FALSE, include=FALSE, eval=FALSE}

#REMOVE RUNS COMPARISONS ACROSS FULL DATASET
#male.Dom.mouse.av, male.Mol.mouse.av, male.Musc.mouse.av

male.Musc.mouse.av <- Male.poly.Mouse.Table_BivData_4MM[(Male.poly.Mouse.Table_BivData_4MM$subsp == "Musc"),]
male.Dom.mouse.av <- Male.poly.Mouse.Table_BivData_4MM[(Male.poly.Mouse.Table_BivData_4MM$subsp == "Dom"),]
male.Mol.mouse.av <- Male.poly.Mouse.Table_BivData_4MM[(Male.poly.Mouse.Table_BivData_4MM$subsp == "Mol"),]

Pwise.ttest_Dom.male <- pairwise.t.test(male.Dom.mouse.av$mean_SC, male.Dom.mouse.av$strain)
#G-WSB is sig, WSB-LEW are slightly sig

Pwise.ttest_Musc.male <- pairwise.t.test(male.Musc.mouse.av$mean_SC, male.Musc.mouse.av$strain)
#PWD and MSM are most diff from SKIVE?

Pwise.ttest_Mol.male <- pairwise.t.test(male.Mol.mouse.av$mean_SC, male.Mol.mouse.av$strain)
#the pooled SC value tests are sig

#the mol comparisons are different due to adj p values


Pwise.ttest_Dom.male.1CO <- pairwise.t.test(male.Dom.mouse.av$mean.SC_1CO, male.Dom.mouse.av$strain)
#G-WSB is sig, WSB-LEW are slightly sig

Pwise.ttest_Musc.male.1CO <- pairwise.t.test(male.Musc.mouse.av$mean.SC_1CO, male.Musc.mouse.av$strain)
#no sig (only 3) -- does include CZECCH

#non-sig, with mol pooled with musc

Pwise.ttest_Mol.male.1CO <- pairwise.t.test(male.Mol.mouse.av$mean.SC_1CO, male.Mol.mouse.av$strain)

Pwise.ttest_Dom.male.2CO <- pairwise.t.test(male.Dom.mouse.av$mean.SC_2CO, male.Dom.mouse.av$strain)

Pwise.ttest_Musc.male.2CO <- pairwise.t.test(male.Musc.mouse.av$mean.SC_2CO, male.Musc.mouse.av$strain)
#non of these sig with the mulitple adjust

Pwise.ttest_Mol.male.2CO <- pairwise.t.test(male.Mol.mouse.av$mean.SC_2CO, male.Mol.mouse.av$strain)

#pairwise.t.test(Ozone, Month, p.adj = "bonf")
```

# brief transition, rec landscape

the lack of predictive power of SC length to predict CO number indicates that there might be different patterns of 'SC usage' / different patterns in the layout of the recombination landscape /  these patterns could stem from placement or spacing of foci.

# Q2 Normalized Single CO positions

<!-- motivation / background of the sex differences, males more terminall in CO placement compared to females, now we ask if there are significant differences in the placement of single foci along bivalents  -->
<!-- set up chunks here -->
<!-- plots /figures below-->

```{r Q2.1CO.show_plots, echo=FALSE,  warning=FALSE}
#Curated_BivData_male

#boxplots of nrm F1
male.bivdata.1CO <- Curated_BivData_male %>% filter(species =="M.musculus") %>% filter(hand.foci.count == 1)
  
Q2.show.nrmF1 <- ggplot(male.bivdata.1CO, aes(x=strain, y=PER_Foci_1, color=strain) )+geom_jitter(alpha=.5 )+ geom_boxplot(aes(y= PER_Foci_1), alpha=.3 ) + scale_color_manual(values=colors_of_strains)+facet_wrap(~subsp, scales = "free")+ylim(c(0,1.1))+ggtitle("Normalized 1CO Foci pos All bivalents")
Q2.show.nrmF1

nrmF1_mouse_level <- ggplot(Male.poly.Mouse.Table_BivData_4MM, aes(x=strain, y=norm.first.foci.pos, color=strain) )+geom_jitter(alpha=.5 )+ geom_boxplot(aes(y= norm.first.foci.pos), alpha=.5 ) + scale_color_manual(values=colors_of_strains)+facet_wrap(~subsp, scales = "free")+ylim(c(0,1.1))+ggtitle("Mouse Level, \nNormalized 1CO Foci pos")

```

<not sure what the beenfit is to test for chrm effectss>

```{r Q2.1COpos.LONG.BIV_plot, echo=FALSE, warning=FALSE}
#make the same plots as above but with the long biv data set
male.long.biv.1CO <- real.long.bivData %>% filter(sex== "male") %>% filter(species =="M.musculus") %>% filter(hand.foci.count == 1)

#mouse.av.long.biv.table

Q2.show.nrmF1_long.biv <- ggplot(male.long.biv.1CO, aes(x=strain, y=PER_Foci_1, color=strain) )+geom_jitter( )+ geom_boxplot(aes(y= PER_Foci_1 ),alpha=.3 ) + scale_color_manual(values=colors_of_strains)+facet_wrap(~subsp, scales = "free")+ylim(c(0,1.1))+ggtitle("Normalized 1CO Foci pos \nLong Biv")

Mouse.av.Long.biv <- ddply(male.long.biv.1CO, c("mouse"), summarise,
                        Nmice = length(unique(mouse)),
                        Nbiv  = length(Obj.ID),
                       mean.1CO_PER = mean(PER_Foci_1)
                      # mean_CO = mean()
                       )

Mouse.av.Long.biv <- add_strain(Mouse.av.Long.biv)
Mouse.av.Long.biv <- add_subsp(Mouse.av.Long.biv)


Q2.shoooo   <- ggplot(Mouse.av.Long.biv, aes(x=strain, y=mean.1CO_PER, color=strain) )+geom_jitter( )+ geom_boxplot(aes(y= mean.1CO_PER ),alpha=.3 ) + scale_color_manual(values=colors_of_strains)+facet_wrap(~subsp, scales = "free")+ylim(c(0,1.1))+ggtitle("mouse ave \nNormalized 1CO Foci pos \nLong Biv")

```

The previous section found that the single crossover landscape is significantly different between males and females.
In this section we focus on differences between the high and the low rec male strains
In order to test if if there are distinguishing meterics for these two groups.

<!-- t tests mouse.avs_4MM, REAL.short_mouse_table-->

```{r Q2.1COpos_mouse.av.t.test_all, echo=FALSE}
#mouse.avs_4MM, #31may19_SKIVE_m2  has NA for norm.first.foci.pos

#High = MSM, PWD, SKIVE
Q2.1COpos_Ttest <- t.test( (mouse.avs_4MM %>% filter(sex == "male") %>% filter(Rec.group == "0"))$norm.first.foci.pos, 
            (mouse.avs_4MM %>% filter(sex == "male") %>% filter(Rec.group == "1"))$norm.first.foci.pos )
#p-value = 1

#high = PWD and MSM
Q2.1COpos_Ttest.HL.ex <- t.test( (mouse.avs_4MM %>% filter(sex == "male") %>% filter(Rec.group.exclu == "0"))$norm.first.foci.pos, 
            (mouse.avs_4MM %>% filter(sex == "male") %>% filter(Rec.group.exclu == "1"))$norm.first.foci.pos )
#p-value = 1

```

```{r Q2.1COpos_within.subsp_ttest, echo=FALSE}

#Dom (predicted no difference) -- can't DO a t.test for DOM across groups..

#Musc
Q2.1COpos_mus.av_Musc_ttest  <-  t.test( (mouse.avs_4MM %>%  filter(sex == "male") %>%  filter(subsp == "Musc") %>% filter(Rec.group == "0"))$norm.first.foci.pos, 
                                         
            (mouse.avs_4MM  %>%  filter(sex == "male") %>% filter(subsp == "Musc") %>% filter(Rec.group == "1"))$norm.first.foci.pos )
# p-value = 0.4


#Mol
Q2.1COpos_mus.av_Mol_ttest  <-  t.test( (mouse.avs_4MM %>%  filter(sex == "male") %>%  filter(subsp == "Mol") %>% filter(Rec.group == "0"))$norm.first.foci.pos, 
            
              (mouse.avs_4MM  %>%  filter(sex == "male") %>% filter(subsp == "Mol") %>% filter(Rec.group == "1"))$norm.first.foci.pos   )
# p-value = 0.07
# exclusive rec group not sig
```

```{r Q2.1CO_mouse.av.short.long_t.test, echo=FALSE}
#these don't have mean.first.foci

#short -- real.long.bivData -- REAL.short_mouse_table
Q2.1COposABS_short_ttest_HL <- t.test( (REAL.short_mouse_table %>% filter(sex == "male") %>%  filter(Rec.group == "0"))$mean_F1pos_ABS, 
      
      (REAL.short_mouse_table %>% filter(sex == "male")  %>% filter(Rec.group == "1"))$mean_F1pos_ABS   )
#p-value = 0.3


Q2.1COpos.PER_short_ttest_HL <- t.test( (REAL.short_mouse_table %>% filter(sex == "male") %>% filter(Rec.group == "0"))$mean_F1pos_PER, 
      (REAL.short_mouse_table %>% filter(sex == "male")  %>% filter(Rec.group == "1"))$mean_F1pos_PER   )
# p-value = 0.1

#remove SKIVE??
```

```{r Q2.1CO.pos_short.long.ttest, echo=FALSE}

#real.long.bivData
#single biv level
Rec.group_LONG.BivData_F1post_Ttest <- t.test(
  (real.long.bivData %>% filter(sex == "male") %>% filter(hand.foci.count == 1) %>% filter(Rec.group == "0") )$PER_Foci_1, 
(real.long.bivData %>% filter(sex == "male") %>% filter(hand.foci.count == 1) %>% filter(Rec.group == "1"))$PER_Foci_1  )
#p-value = 0.5 

Rec.group_BivData_F1post_Ttest <- t.test(
  (Curated_BivData %>% filter(sex == "male") %>% filter(hand.foci.count == 1) %>% filter(Rec.group == "0") )$PER_Foci_1, 
(Curated_BivData %>% filter(sex == "male") %>% filter(hand.foci.count == 1) %>% filter(Rec.group == "1"))$PER_Foci_1  )
#p-value = 0.2

#
#whoo.mama <-  t.test( (Curated_BivData %>% filter(sex == "male") %>% filter(hand.foci.count #== 1)  %>%  filter(Rec.group == "0"))$PER_Foci_1, 
#            (Curated_BivData  %>% filter(sex == "male") %>% filter(hand.foci.count == 1)   #%>% filter(Rec.group == "1"))$PER_Foci_1   )
#p-value = 0.2


#whoo.mama.rec.exc <-  t.test( (Curated_BivData  %>% filter(sex == "male") %>% #filter(hand.foci.count == 1)  %>%  filter(Rec.group.exclu == "0"))$PER_Foci_1, 
#            (Curated_BivData  %>% filter(sex == "male") %>% filter(hand.foci.count == 1)   #%>% filter(Rec.group.exclu == "1"))$PER_Foci_1   )
#p-value = 0.4

```


**The figure above shows, WSB has significantly more terminal positioning, while MOLF has significantly more medial position.**

T tests of the single crossover positions reveal that the normalized single foci positions are not significantly different between the total pooled high and low rec male strains (t.test; p = `r Rec.group_BivData_F1post_Ttest$p.value`). 

<further tests -- within the subspecies, musculus and molossinus?

The high rec musc strains, _musculus^PWD^_ and _musculus^SKVIE^_, do not have signifcantly different single foci landscapes from the other musc low rec strains (ttest; p = `r Q2.1COpos_mus.av_Musc_ttest$p.value`).

While _molossinus^MOLF^_ is significantly more centrally positioned compared to _molossinus^MSM^_ (t.test; p = `r Q2.1COpos_mus.av_Mol_ttest$p.value`). This result is somewhat puzzeling and it doesn't match up with the strains with high recombination rates.

 <BUT this might be due to sampling for MSM -- but it doesn't seem to be the case in MOLF, 
 (MSM and MOLF have different proportions of chromosome classes -- (MSM has the smallest propotion of 1COs, across the strains, BUT the mean and distributions for MSM of single foci positions are equivilent across all the strains.>

<long p = `r Rec.group_LONG.BivData_F1post_Ttest$p.value` short 1CO, male, p = `r Q2.1COpos.PER_short_ttest_HL$p.value`> 
<!-- GLMs because there's no difference 
toal data set, short and long biv. There's only a difference between MSM and MOKF
test ppredictions, Dom = NULL,  -->

```{r Q2.1COpos_mus.av.full_glm, echo=FALSE}

#(mouse.avs_4MM %>%  filter(sex == "male") %>%  filter(subsp == "Musc") %>% filter(Rec.group == "0"))$norm.first.foci.pos

Q2.glm_mus.av_1COpos_M1 <- glm(norm.first.foci.pos ~ subsp * strain, data= mouse.avs_4MM %>% filter(sex == "male"))
sum.Q2.glm_mus.av_1COpos_M1  <- summary(Q2.glm_mus.av_1COpos_M1)$coefficients
#summary(Q2.glm_mus.av_1COpos_M1)$coefficients[,4]

#with.chrm.legth  ?
Q2.glm_mus.av_1COpos_M1 <- glm(norm.first.foci.pos ~ subsp * strain, data= mouse.avs_4MM %>% filter(sex == "male"))


#M2
Q2.glm_mus.av_1COpos_M2 <- glm(norm.first.foci.pos ~ strain, data= mouse.avs_4MM %>% filter(sex == "male"))
sum.Q2.glm_mus.av_1COpos_M2 <- summary(Q2.glm_mus.av_1COpos_M2)$coefficients

```


in addition models which use mouse average of foci position as a dependat variable, we decided to run models for the single bivalent level, and which include chromosome length as a fixed effect.

in the above chunk I tried making another glm's which uses the single bivalent measures and include chromosome size for the switching the model to include interaction (instead of additive), all of the pvalues get lower. .. the chromosome length has the most significant effect, LEW by chrm effect are significant

In the reduced model, chrm length is the most significant, strain SKIVE, MOLF, G and LEW are significant strain effects


**For the M2 (subsp)**
There are differences for the dom strains: G strain(p = `r summary(Q2.glm_mus.av_1COpos_M1)$coefficients[,4][4]`).


LEW (slight) (p =`r summary(Q2.glm_mus.av_1COpos_M1)$coefficients[,4][5]`).  

But big effects from Mol (p = `r summary(Q2.glm_mus.av_1COpos_M1)$coefficients[,4][3]`).  

and MSM strain (p = `r summary(Q2.glm_mus.av_1COpos_M1)$coefficients[,4][7]`)  


**For M3 (all strains)**

strain G (p = `r sum.Q2.glm_mus.av_1COpos_M2[,4][2]`).

strain Lew (p = `r sum.Q2.glm_mus.av_1COpos_M2[,4][3]`).

strain PWD ( p = `r sum.Q2.glm_mus.av_1COpos_M2[,4][4]`).

slight strain MSM  (p = `r sum.Q2.glm_mus.av_1COpos_M2[,4][5]`).

very sig strain MOLF (p = `r sum.Q2.glm_mus.av_1COpos_M2[,4][6]`).

strain SKIVE (p = `r sum.Q2.glm_mus.av_1COpos_M2[,4][7]`).

strain KAZ (p = `r sum.Q2.glm_mus.av_1COpos_M2[,4][8]`).

from the main figure above, you can see that G and MOLF 1CO foci positions are the most different

<no prediction for the strains within subspecies>

For MOLF 326 1CO bivalents,  79, 126, 62, 59 1CO observations across the 4 MOLF mice

there are only 77 MSM 1COs across 4 mice in MSM (MSM is less central than MOLF)
(motivation for MOLF female observations and more male MSM biv measures)


```{r Q2.1COpos.show.MOL, echo=FALSE}

#(mouse.avs_4MM %>%  filter(sex == "male") %>%  filter(subsp == "Mol") %>% filter(Rec.group == #"0"))$norm.first.foci.pos

mus.av1COpos.plot <- ggplot(data= mouse.avs_4MM %>%  filter(sex == "male") %>%  filter(subsp == "Mol"),
                      aes(x= mouse, y = norm.first.foci.pos))+geom_point()


#+facet_wrap(~mouse, scales = "free")

single.biv_1COpos.plot_Mol <- ggplot(data= Curated_BivData %>%  filter(sex == "male") %>%  filter(subsp == "Mol") %>%  filter(hand.foci.count == "1") %>%  filter(chromosomeLength < 100),
                      aes(x= fileName, y = PER_Foci_1, color=chromosomeLength))+geom_point()+facet_wrap(~mouse, scales = "free") + ylim(c(0,1))

#+scale_color_manual(values=colors_of_strains)


single.biv_1COpos.plot_Dom <- ggplot(data= Curated_BivData %>% filter(sex == "male") %>%filter(subsp == "Dom") %>%  filter(hand.foci.count == "1"),
                      aes(x= fileName, y = PER_Foci_1, color=strain ))+geom_point()+facet_wrap(~mouse, scales = "free") + ylim(c(0,1))+scale_color_manual(values=colors_of_strains)

#CHRM.length is very significant-- some of the same subsp and strains are sig     
```


<I ran another model that used the single bivalents observations to predict the normalize foci position with fixed effects of chromosome length, subspecies, strian and the interaction effects.

**M.X. Linear Model** $$1CO \ Normalized \ Position ~=~ chromosomeLength * subsp * strain + \varepsilon $$>
 

```{r Q2.1CO.pos_glm.chrm.Length, echo=FALSE}
#remove this cell, 

Curated_BivData <- Curated_BivData %>% filter(Obj.ID != "13sep18_28jun18_MOLF_m1_sp1_18.1_rev_14")

Curated_BivData.Mol <- Curated_BivData %>% filter(Foci1 != "")

#making new models 

Curated_BivData$subsp <- factor(Curated_BivData$subsp , ordered = F)
Curated_BivData$strain <- factor(Curated_BivData$strain , ordered = F)

chrm.1CO.pos.chrmLength <- glm(PER_Foci_1 ~ chromosomeLength * subsp * strain, data= Curated_BivData.Mol %>% filter(sex == "male")  %>%  filter(hand.foci.count == "1") )

sum.chrm.1CO.pos.chrmLength <- summary(chrm.1CO.pos.chrmLength)$coefficients
#summary(Q2.glm_mus.av_1COpos_M1)$coefficients[,4]


chrm.1CO.pos.chrmLength_strain <- glm(PER_Foci_1 ~ chromosomeLength *  strain, data= Curated_BivData.Mol %>% filter(sex == "male")  %>%  filter(hand.foci.count == "1") )

#summary(chrm.1CO.pos.chrmLength_strain)

```

<for the first model, 

chromosome length `r summary(chrm.1CO.pos.chrmLength)$coefficients[,4][2]`  

subsp MOL `r summary(chrm.1CO.pos.chrmLength)$coefficients[,4][4]`  
the mol obj above isn't showing the reight value. 

strain G   `r summary(chrm.1CO.pos.chrmLength)$coefficients[,4][5]`  


strain LEW   `r summary(chrm.1CO.pos.chrmLength)$coefficients[,4][6]`  


Strain MSM `r summary(chrm.1CO.pos.chrmLength)$coefficients[,4][8]`  


chrm * LEW `r summary(chrm.1CO.pos.chrmLength)$coefficients[,4][14]`  


chrm *MSM  `r summary(chrm.1CO.pos.chrmLength)$coefficients[,4][16]`


M2

-chromosome length, `r summary(chrm.1CO.pos.chrmLength_strain)$coefficients[,4][2]`  


-strain G, `r summary(chrm.1CO.pos.chrmLength_strain)$coefficients[,4][3]`  


-strain LEW, `r summary(chrm.1CO.pos.chrmLength_strain)$coefficients[,4][4]`  


-strain MOLF, `r summary(chrm.1CO.pos.chrmLength_strain)$coefficients[,4][7]`  


-strain SKIVE, `r summary(chrm.1CO.pos.chrmLength_strain)$coefficients[,4][8]`  


-chrm length LEW, `r summary(chrm.1CO.pos.chrmLength_strain)$coefficients[,4][12]`>  


males from WSB and MOLF strains have 'strang'unique patterns for the single foci positions, which don't fit with predicted patterns from the genome wide recombination rate variation

There are significant difference across the 1CO positions, yet they don't match fit any of the predictions. there are also complications due to the chromsome class proportions that

```{r Q2.1COpos_strain, echo=FALSE, eval=FALSE}
#isolate all musc
#Curated_BivData_male <- Curated_BivData[Curated_BivData$sex == "male",]

##long biv
male.long.biv <- real.long.bivData %>% filter(sex== "male") %>% filter(species == "M.musculus") 
#table( droplevels( male.long.biv$strain)  )


mouse.av.long.biv.table <- ddply(male.long.biv, c("mouse"), summarise,
                             
                        n_obs = length(Obj.ID),
                         mean_F1 = mean(Foci1, na.rm=TRUE),
                        mean_F1.nrm = mean(PER_Foci_1,na.rm=TRUE),
                        
                        mean_SC = mean(chromosomeLength),

                      mean.SC_1CO = mean(chromosomeLength[which(hand.foci.count == 1)], na.rm = TRUE),
                       mean.SC_2CO = mean(chromosomeLength[which(hand.foci.count == 2)], na.rm = TRUE),
                       mean.SC_3CO = mean(chromosomeLength[which(hand.foci.count == 3)], na.rm = TRUE),
                       mean.SC_4CO = mean(chromosomeLength[which(hand.foci.count == 4)], na.rm = TRUE)

                                               )
#add subsp

mouse.av.long.biv.table <- add_strain(mouse.av.long.biv.table)
mouse.av.long.biv.table <- add_subsp(mouse.av.long.biv.table)

mouse.av.long.biv.table$strain <- factor(mouse.av.long.biv.table$strain, ordered = FALSE)
mouse.av.long.biv.table$subsp <- factor(mouse.av.long.biv.table$subsp, ordered = FALSE)

#add rec group
mouse.av.long.biv.table$Rec.group <- ifelse(grepl("PWD", mouse.av.long.biv.table$strain), 1, 
                            ifelse(grepl("MSM", mouse.av.long.biv.table$strain), 1,
                             ifelse(grepl("SKIVE", mouse.av.long.biv.table$strain), 1, 0)) )
```


```{r Q2.1COpos.glm, echo=FALSE, eval=FALSE}
male.bivdata.1CO$Rec.group <- ifelse(grepl("PWD", male.bivdata.1CO$strain), 1, 
                                    ifelse(grepl("MSM", male.bivdata.1CO$strain), 1, 
                                            ifelse(grepl("SKIVE", male.bivdata.1CO$strain), 1, 0)))

Log.Q2.1COnrm.F1.PER <- glm(Rec.group ~ PER_Foci_1,
              data=male.bivdata.1CO,  family=binomial(link="logit"))
#summary(Log.glmQ2_M2.1COnrm)

#Musc
Log.Q2.F1_PER_Musc <- glm(Rec.group ~ PER_Foci_1,
              data=male.bivdata.1CO[male.bivdata.1CO$subsp == "Musc",], 
              family=binomial(link="logit"))
#summary(Log.Q2.F1_PER_Musc)


#Mol
Log.Q2.F1_PER_Mol <- glm(Rec.group ~ PER_Foci_1,
              data=male.bivdata.1CO[male.bivdata.1CO$subsp == "Mol",],
              family=binomial(link="logit"))

#summary(Log.glmQ2_M2.1COnrm_Mol)

```



# Q2 Evolution of interference is associated with genome wide recombination rate evolution


```{r Q2.IFD.long.short.setup, echo=FALSE}
#long BivData
male.long.biv.2CO_IFD <- real.long.bivData %>% filter(sex== "male") %>% filter(species =="M.musculus") %>% filter(hand.foci.count == 2)

male.long.biv.2CO_IFD$Rec.group <- ifelse(grepl("PWD", male.long.biv.2CO_IFD$strain), 1, 
                                    ifelse(grepl("MSM", male.long.biv.2CO_IFD$strain), 1, 
                                           ifelse(grepl("SKIVE", male.long.biv.2CO_IFD$strain), 1, 0)))

#show
Long.biv.IFDs_PER_by.group <- ggplot(male.long.biv.2CO_IFD, aes(y = IFD1_PER, x= mouse, color=strain))+geom_boxplot(alpha=.2)+
  geom_jitter(alpha=.2)+
  ggtitle("Normalized Interfocal Distances, Long BivData\nLow Rec group =0, High Rec group =1")+
  scale_color_manual(values=colors_of_strains)+ facet_wrap(subsp~Rec.group, scales = "free")+theme(legend.position="none")+ylim(c(0,1.1))

```


<!-- set up chunks here-->


```{r Q2.show.IFD, echo=FALSE, warning = FALSE}

#subset
male.biv.2CO <- Curated_BivData %>% filter(sex== "male") %>% filter(species =="M.musculus") %>% filter(hand.foci.count == 2)

male.biv.2CO$Rec.group <- ifelse(grepl("PWD", male.biv.2CO$strain), 1, 
                                    ifelse(grepl("MSM", male.biv.2CO$strain), 1, 
                                           ifelse(grepl("SKIVE", male.biv.2CO$strain), 1, 0)))


biv.IFDs_by_group <- ggplot(male.biv.2CO, aes(y = IFD1_ABS, x= mouse, color=strain))+geom_boxplot(alpha=.2)+
  geom_jitter(alpha=.2)+
  ggtitle("Interfocal Distances, ABS \nSeperated by Rec Groups")+
  scale_color_manual(values=colors_of_strains)+ facet_wrap(subsp~Rec.group, scales = "free")+ylim(c(0,110))


biv.IFDs_PER_by.group <- ggplot(male.biv.2CO, aes(y = IFD1_PER, x= mouse, color=strain))+geom_boxplot(alpha=.2)+
  geom_jitter(alpha=.2)+
  ggtitle("Normalized Interfocal Distances,\n Low Rec group =0, High Rec group =1")+
  scale_color_manual(values=colors_of_strains)+ facet_wrap(subsp~Rec.group, scales = "free")+theme(legend.position="none", axis.text.x = element_blank())+ylim(c(0,1.1))

biv.IFDs_PER_by.group

```


<Previous prediction for differences in the long and short bivalent dataset. Examine the pattern in the Long bivalent data set to get a feel for if the chromosome size effect skews the pattern. In the long bivalent data set, the longest bivalents (top 25%) from each cell are isolated, so they are more likely to be the same chromosome identities (Chrm1 to Chrm5). The section on IFD only looks at the 2CO bivalents.>

```{r Q2.t.test_IFD_mouse.av, echo=FALSE}

#t.test for IFD
#mouse.av - the mouse average IFD is from just 2CO
t.test_mouse.av.IFD_ABS <- t.test(  (mouse.avs_4MM %>% filter(Rec.group == "1") %>% filter(sex == "male") )$mean_IFD.2CO_ABS,
   (mouse.avs_4MM %>% filter(Rec.group == "0") %>% filter(sex == "male") )$mean_IFD.2CO_ABS   )
  
t.test_mouse.av.IFD_PER <- t.test(  (mouse.avs_4MM %>% filter(Rec.group == "1") %>% filter(sex == "male") )$mean_IFD.2CO_PER,
   (mouse.avs_4MM %>% filter(Rec.group == "0") %>% filter(sex == "male") )$mean_IFD.2CO_PER   )

```


```{r Q2.t.test_IFD,echo=FALSE, eval=FALSE}

#already made
#subset
male.bivdata.2CO_IFD <- Curated_BivData %>% filter(sex== "male") %>% filter(species =="M.musculus") %>% filter(hand.foci.count == 2)
#t.tests for the Rec group
Curated_BivData$Rec.group <- ifelse(grepl("PWD male", Curated_BivData$category), 1, 
                                    ifelse(grepl("MSM male", Curated_BivData$category), 1,
                                           ifelse(grepl("SKIVE male",Curated_BivData$category),1, 0)))

#Curated_BivData <- add_IFD(Curated_BivData)
#THIS SHOULD ALREADY BE ADDED

#single biv level
Rec.group.BivData_IFD.PER_Ttest <- t.test(male.bivdata.2CO_IFD$IFD1_PER[male.bivdata.2CO_IFD$Rec.group == 1], 
                                 male.bivdata.2CO_IFD$IFD1_PER[male.bivdata.2CO_IFD$Rec.group == 0])
 
Rec.group.BivData_IFD.PER_Ttest$p.value
#p-value < 2.2e-16

#MOUSE AVE  -- replace with mouse    mouse.avs_4MM,   male.av_4MM
Rec.group_IFD.PER_Ttest <- t.test(  (mouse.avs_4MM %>% filter(sex == "male") %>% filter(Rec.group == "1"))$mean_IFD.2CO_PER, 
                                (mouse.avs_4MM %>% filter(sex == "male") %>% filter(Rec.group == "0"))$mean_IFD.2CO_PER )
 
Rec.group_IFD.PER_Ttest$p.value 
#p-value = 6.872e-07

```

The mouse averages for IFDs were significantly longer in the high rec groups (t.test; p = `r t.test_mouse.av.IFD_PER$p.value` for IFD^PER^ and p = `r t.test_mouse.av.IFD_ABS$p.value` for IFD^raw^). The pattern is confirmed with t.test comparing groups within the same subepcies.

<glms>

The following model framework was applied to further investigate these patterns using mouse averages of IFD^raw^ and IFD^PER^.

**Linear model M1 ** $$mouse \ average \ IFD ~=~ subsp * strain + \varepsilon $$

**Linear model M2 ** $$mouse \ average \ IFD ~=~ strain + \varepsilon $$


```{r Q2.IFD.glm.test, echo=FALSE}
#REPLACE Male.poly with ..
#(mouse.avs_4MM %>% filter(sex == "male") %>% filter(Rec.group == "1")

#M1.ABS
Q2.M1glm_IFD.raw <- glm(mean_IFD.2CO_ABS ~ subsp * strain, data= mouse.avs_4MM %>% filter(sex == "male")    )
#summary(Q2.M1glm_IFD.raw) 

#M1.PER
Q2.M1glm_IFD.PER <- glm(mean_IFD.2CO_PER ~ subsp * strain, data= mouse.avs_4MM %>% filter(sex == "male") )

sum.Q2.M1glm_IFD.PER <- summary(Q2.M1glm_IFD.PER)
#summary(Q2.M1glm_IFD.PER) 


#M2.raw
Q2.M2glm_IFD.raw <- glm(mean_IFD.2CO_ABS ~ strain, data= mouse.avs_4MM %>% filter(sex == "male"))
#summary(Q2.M2glm_IFD.raw)

Q2.M2glm_IFD.PER <- glm(mean_IFD.2CO_PER ~ strain, data= mouse.avs_4MM %>% filter(sex == "male"))
sum.Q2.M2glm_IFD.PER <- summary(Q2.M2glm_IFD.PER)
#  MSM strain effect for 2CO PER not significant



#within subspecies
#Dom
Q2.glm.M2_IFD_Dom <- glm(mean_IFD.2CO_ABS ~ strain, data= Male.poly.Mouse.Table_BivData_4MM[Male.poly.Mouse.Table_BivData_4MM$subsp == "Dom",])
#summary(Q2.glm.M2_IFD_Dom)

#Musc
Q2.glm.M2_IFD_Musc <- glm(mean_IFD.2CO_ABS ~ strain, data= Male.poly.Mouse.Table_BivData_4MM[Male.poly.Mouse.Table_BivData_4MM$subsp == "Musc",])
#summary(Q2.glm.M2_IFD_Musc)

#Mol
Q2.glm.M2_IFD_Mol <- glm(mean_IFD.2CO_ABS ~ strain, data= Male.poly.Mouse.Table_BivData_4MM[Male.poly.Mouse.Table_BivData_4MM$subsp == "Mol",])
#summary(Q2.glm.M2_IFD_Mol)


#Dom
Q2.glm.M2_IFD.PER_Dom <- glm(mean_IFD.2CO_PER ~ strain, data= Male.poly.Mouse.Table_BivData_4MM[Male.poly.Mouse.Table_BivData_4MM$subsp == "Dom",])
#summary(Q2.glm.M2_IFD.PER_Dom) 

#Musc
Q2.glm.M2_IFD.PER_Musc <- glm(mean_IFD.2CO_PER ~ strain, data= Male.poly.Mouse.Table_BivData_4MM[Male.poly.Mouse.Table_BivData_4MM$subsp == "Musc",])
#summary(Q2.glm.M2_IFD.PER_Musc) 

#Mol
Q2.glm.M2_IFD.PER_Mol <- glm(mean_IFD.2CO_PER ~ strain, data= Male.poly.Mouse.Table_BivData_4MM[Male.poly.Mouse.Table_BivData_4MM$subsp == "Mol",])
#summary(Q2.glm.M2_IFD.PER_Mol)
```

In M1, the model build on the mouse averages of IFD metrics with subspecies and strain as fixed effects, the high rec strains had significant strain effects (glm; p = `r sum.Q2.M1glm_IFD.PER$coefficients[,4][6]`, p = `r sum.Q2.M1glm_IFD.PER$coefficients[,4][7]` and p = `r sum.Q2.M1glm_IFD.PER$coefficients[,4][8]`  for PWD, MSM and SKIVE respectively). In M2, the mouse average for normalzied IFD is modeled with only strain effects. Only PWD and SKIVE IFD^PER^ had significant strain effects (glm; p = `r sum.Q2.M2glm_IFD.PER$coefficients[,4][4]` and p = `r sum.Q2.M2glm_IFD.PER$coefficients[,4][7]` for PWD and SKIVE respectively).

<when the M2 model was run within subspecies, the results generally confirmed our predictions; no sig values for Dom, PWD and SKIVE have longer IFDs than other musc strains and MSM is longer than MOLF>


<!-- we ran similar models using 2CO bivalent observations to examine what effects the SC length might have on IFD^PER^ length. 

**model BivLen.M1 ** $$Single bivalent\ IFD ~=~ subsp * strain * chromosomeLength + \varepsilon $$

**model BivLen.M2 ** $$Single bivalent\ IFD ~=~ strain * chromosomeLength + \varepsilon $$

```{r Q2.IFD.glm.chrmLength, echo=FALSE}

#repeating chrm models with chromsome length
#FOR BIVALENT LEVEL MODELS

Curated_BivData$subsp  <- factor(Curated_BivData$subsp, ordered = FALSE)

#M.Len.ABS
Q2.BIV.LEN.glm_IFD.raw <- glm(IFD1_ABS ~ subsp * strain * chromosomeLength, data= Curated_BivData %>% filter(sex == "male")%>% filter(hand.foci.count == "2")    )
#summary(Q2.BIV.LEN.glm_IFD.raw)

#M.Len.PER
Q2.BIV.LEN.glm_IFD.PER <- glm(IFD1_PER ~ subsp * strain * chromosomeLength, data= Curated_BivData %>% filter(sex == "male")%>% filter(hand.foci.count == "2")    )
#summary(Q2.BIV.LEN.glm_IFD.PER)


#M.Len.raw
Q2.BIV.LEN.M2glm_IFD.ABS <- glm(IFD1_ABS ~ strain* chromosomeLength, data= Curated_BivData %>% filter(sex == "male")%>% filter(hand.foci.count == "2"))
#summary(Q2.BIV.LEN.M2glm_IFD.ABS)

##M.Len PER
Q2.BIV.LEN.M2glm_IFD.PER <- glm(IFD1_PER ~ strain* chromosomeLength, data= Curated_BivData %>% filter(sex == "male")%>% filter(hand.foci.count == "2"))
#summary(Q2.BIV.LEN.M2glm_IFD.PER)
#  MSM strain effect for 2CO PER not significant

```

For the first version of the bivalent level model (which has subspecies effect), the most significant effect is chromosome length, is slight positive direction (both for raw and normalized IFDs). Molossinus subspecies and Mol\*chrm length effect for IFD^Normalized^ and IFD^raw^. The Mol\*chrmLength is negative 
<PWD, MSM, MOLF, SKIVE are the only significant strain effects.
- all of the strin\*chrm lengths effects are negative, PWD, MSM, MOLF, SKIVE significant effects (for IFD.ABS and IFD.PER)
With each unit increase in SC length, the IFD^PER^ also increases (for the high rec strains + MOLF). All of the high rec strains and MOLF have significantly longer IFD^PER^. There are also significant interaction effect of strain and chromsome length, in the high rec strains and MOLF, with each increase in bivalent length, the IFD^PER^ decreases slightly.
-->

```{r Q2.IFD_rate.shortIFD, echo=FALSE, warning = FALSE}

#calculate the rates of 'close foci' (<30%)
#examine the changes in the distribution closer.


#for the relevant data, by strain, calcu the number of observations of 2CO above and below 30%

close.IFD.rates.mus.rate <- ddply(Curated_BivData %>% filter(sex== "male") %>% filter(species =="M.musculus") %>% filter(hand.foci.count == 2), 
                         c("mouse"), summarise,
                        N.obs  = length(Obj.ID),

                      #  mean_SC = as.numeric(format(round(  mean(chromosomeLength, na.rm = TRUE), 3 ), nsmall=3) ),
                        mean.per.IFD = mean(IFD1_PER, na.rm=TRUE),
                        var.IFD = var(IFD1_PER, na.rm=TRUE),
                        
                      inverse.close.foci = length(which(IFD1_PER >= .31)),  
                      n.close.foci = length(which(IFD1_PER <= .3)),
                      
                      off.by = (N.obs - sum(c(inverse.close.foci,n.close.foci)) ),
                      
                      rate.close.foci = n.close.foci / N.obs
)

close.IFD.rates.mus.rate <- add_strain(close.IFD.rates.mus.rate)
close.IFD.rates.mus.rate <- add_category(close.IFD.rates.mus.rate)
close.IFD.rates.mus.rate <- add_subsp(close.IFD.rates.mus.rate)


close.IFD.rates_strain <- ddply(Curated_BivData %>% filter(sex== "male") %>% filter(species =="M.musculus") %>% filter(hand.foci.count == 2), 
                         c("strain"), summarise,
                        N.obs  = length(Obj.ID),

                        mean_SC = as.numeric(format(round(  mean(chromosomeLength, na.rm = TRUE), 3 ), nsmall=3) ),
                        mean.per.IFD = mean(IFD1_PER, na.rm=TRUE),
                        var.IFD = var(IFD1_PER),
                        
                      inverse.close.foci = length(which(IFD1_PER > .3)),  
                      n.close.foci = length(which(IFD1_PER < .3)),
                      rate.close.foci = n.close.foci / N.obs,
                     
                       off.by = (N.obs - sum(c(inverse.close.foci,n.close.foci)) )
                      
)
#the 
 

show.close.IFD.rate <- ggplot(close.IFD.rates_strain, aes(y=rate.close.foci, x=strain, color=strain))+geom_jitter()+scale_color_manual(values=colors_of_strains)

#add rec.group
close.IFD.rates_strain$Rec.group <- ifelse(grepl("PWD", close.IFD.rates_strain$strain), 1, 
                                ifelse(grepl("MSM", close.IFD.rates_strain$strain), 1,
                                 ifelse(grepl("SKIVE", close.IFD.rates_strain$strain), 1, 0)))

#POOL all short and 
#chi square test
high.rec.props <- prop.test( ( close.IFD.rates_strain %>% filter(Rec.group == 1) )$n.close.foci,
                              ( close.IFD.rates_strain %>% filter(Rec.group == 1) )$N.obs)     
#p-value = 0.03, slight dif tween 3 high rec strains

#don't need to run chi square for this pattern, since it's enough to 

```

We determined that the main difference in IFD disbribution across these groups is an enrichment of 'close foci', IFD^PER^s less than 30% of the total SC length in low rec strains. The rate of these close foci range from 8.2% (G) to 16% (KAZ) in the low rec strains, while the high strains all had rates under 5% (0%, 1.3%, and 3.3% for skive, MSM and PWD respectively). <no chi sqaure test see above>


```{r Q2.IFD.PER.logreg, echo=FALSE, eval=FALSE}

#Log regression all show significant prediction

#IFD.PER
log.reg.Q2.IFD.PER <- glm(Rec.group ~ mean_IFD.2CO_PER,
              data=Male.poly.Mouse.Table_BivData_4MM,  family=binomial(link="logit"))
#summary(log.reg.Q2.IFD.PER)

#MUSC
log.reg.Q2.IFD.PER.Musc <- glm(Rec.group ~ mean_IFD.2CO_PER,
              data=Male.poly.Mouse.Table_BivData_4MM[Male.poly.Mouse.Table_BivData_4MM$subsp == "Musc",],  family=binomial(link="logit"))
#summary(log.reg.Q2.IFD.PER.Musc)

#Mol
log.reg.Q2.IFD.PER.Mol <- glm(Rec.group ~ mean_IFD.2CO_PER,
              data=Male.poly.Mouse.Table_BivData_4MM[Male.poly.Mouse.Table_BivData_4MM$subsp == "Mol",],  family=binomial(link="logit"))
#summary(log.reg.Q2.IFD.PER.Mol)



#SINGLE BIVALENT LEVEL

Log.Q2.IFD.per_bivdata <- glm(Rec.group ~ IFD1_PER,
              data=male.bivdata.2CO_IFD,  family=binomial(link="logit"))
#summary(Log.Q2.IFD.per_bivdata)

#Musc
Log.Q2.IFD.per_bivdata.Musc <- glm(Rec.group ~ IFD1_PER,
              data=male.bivdata.2CO_IFD[male.bivdata.2CO_IFD$subsp == "Musc",],  family=binomial(link="logit"))

#summary(Log.Q2.IFD.per_bivdata.Musc)

#Mol
Log.Q2.IFD.per_bivdata.Mol <- glm(Rec.group ~ IFD1_PER,
              data=male.bivdata.2CO_IFD[male.bivdata.2CO_IFD$subsp == "Mol",],  family=binomial(link="logit"))

#summary(Log.Q2.IFD.per_bivdata.Mol)

```

```{r Q2IFD_logreg, echo=FALSE, eval=FALSE}

Male.poly.Mouse.Table_BivData_4MM$Rec.group <- ifelse(grepl("PWD", Male.poly.Mouse.Table_BivData_4MM$strain), 1, 
                                    ifelse(grepl("MSM", Male.poly.Mouse.Table_BivData_4MM$strain), 1, 
                                           ifelse(grepl("SKIVE", Male.poly.Mouse.Table_BivData_4MM$strain), 1, 0)))

#IFD.raw
log.reg.Q2.IFD.raw <- glm(Rec.group ~ mean_IFD.2CO_ABS,
              data=Male.poly.Mouse.Table_BivData_4MM,  family=binomial(link="logit"))
#summary(log.reg.Q2.IFD.raw)

#MUSC
log.reg.Q2.IFD.raw.Musc <- glm(Rec.group ~ mean_IFD.2CO_ABS,
              data=Male.poly.Mouse.Table_BivData_4MM[Male.poly.Mouse.Table_BivData_4MM$subsp == "Musc",],  family=binomial(link="logit"))
#summary(log.reg.Q2.IFD.raw.Musc)

#Mol
log.reg.Q2.IFD.raw.Mol <- glm(Rec.group ~ mean_IFD.2CO_ABS,
              data=Male.poly.Mouse.Table_BivData_4MM[Male.poly.Mouse.Table_BivData_4MM$subsp == "Mol",],  family=binomial(link="logit"))
#summary(log.reg.Q2.IFD.raw.Mol)


#SINGLE BIV LEVELs
male.bivdata.2CO_IFD <- Curated_BivData_male %>% filter(species =="M.musculus") %>% filter(hand.foci.count == 2)

male.bivdata.2CO_IFD$Rec.group <- ifelse(grepl("PWD", male.bivdata.2CO_IFD$strain), 1, 
                                    ifelse(grepl("MSM", male.bivdata.2CO_IFD$strain), 1, 
                                            ifelse(grepl("SKIVE", male.bivdata.2CO_IFD$strain), 1, 0)))


Log.Q2.IFD.raw_bivdata <- glm(Rec.group ~ IFD1_ABS,
              data=male.bivdata.2CO_IFD,  family=binomial(link="logit"))
#summary(Log.Q2.IFD.raw.PER_bivdata)

#Musc
Log.Q2.IFD.raw_bivdata.Musc <- glm(Rec.group ~ IFD1_ABS,
              data=male.bivdata.2CO_IFD[male.bivdata.2CO_IFD$subsp == "Musc",],  family=binomial(link="logit"))

#summary(Log.Q2.IFD.raw.PER_bivdata.Musc)

#Mol
Log.Q2.IFD.raw_bivdata.Mol <- glm(Rec.group ~ IFD1_ABS,
              data=male.bivdata.2CO_IFD[male.bivdata.2CO_IFD$subsp == "Mol",],  family=binomial(link="logit"))

#summary(Log.Q2.IFD.raw.PER_bivdata.Mol)
```

```{r Q2.IFD.longbiv_glm, echo=FALSE, eval=FALSE}

#Full dataset

male.long.biv.2CO_IFD$strain <- factor(male.long.biv.2CO_IFD$strain, ordered = FALSE)

#single bivalent
Q2.glm_long.biv_M2.IFD.PER <- glm(IFD1_PER ~ strain, data= male.long.biv.2CO_IFD)
#summary(Q2.glm_long.biv_M2.IFD.PER)
#The PER.IFDs don't have significant strain effects
```


 **(unlike total SC length and 1CO pattern, IFD metreics are a significant predictors of rapid genome wide recombination rate evolution. However, they go agaisnt a standard perdiction for the relationship between the chromosome level recombination landscape and genome wide recombiantion rate.** 

<!-- 
Since the majority of cells from the Automated BivData set do not have the complete complement of bivalents this subset is much smaller and might have 'smaller' bivalents from cells missing the longer bivalents. But assuming the algorithm's identification of bivalents in random, with a large enough sample size, hopefully this error would be minimized.
--> 

<Below are examples of plots of SC length distributions across cells. The top figure shows whole cell hand measured data and the bottom shows the Automated bivData from cells with at least 15 bivalents measured. Most plots excluded for space.  

Each point is a bivalent plotted by cell on the x axis. X's are the 4th quartile, big point is the mean and smaller black point is the median. I'm using these to compare the patterns of these statistics in the automated data set which is missing some bivalent data. (the extra stats are not correctly mapped)>


<HAND measure METRICS, below chunk goes with the above chunk>
```{r long.biv.plots, echo=FALSE, eval=FALSE}

show.the.cells_females_dom <- ggplot(whole.cell.data_manual[(whole.cell.data_manual$sex == "female") & (whole.cell.data_manual$subsp == "Dom"),], aes(x=fileName, y= SC.length, color=strain))+geom_point()+
  
   geom_point(data=manual_summary_stats[(manual_summary_stats$sex == "female") &(manual_summary_stats$subsp == "Dom"),],aes(y=mean_SC, x=fileName, col="red", size=2))+
  geom_point(data=manual_summary_stats[(manual_summary_stats$sex == "female") & (manual_summary_stats$subsp == "Dom"),],aes(y=median_SC, x=fileName, col="black", size=1.5))+
  
   geom_point(data=manual_summary_stats[(manual_summary_stats$sex == "female") & (manual_summary_stats$subsp == "Dom"),],aes(y=Qar.4, x=fileName, col="black", size=2.6), shape=4)+
  
  scale_color_manual(values=colors_of_strains)+
  facet_wrap(~strain, scales="free")+ylim(c(30,200))+ggtitle("Female whole cell manual measures")


show.the.cells_females_dom
#PWD seems to have the biggest variance across cells


almost_cells_DOM_f <- ggplot(almost_whole_cell_BivData[(almost_whole_cell_BivData$sex == "female") & (almost_whole_cell_BivData$subsp == "Dom"),], aes(x=fileName, y= chromosomeLength, color=strain))+
  geom_point()+ 
  geom_point(data=Biv_Data_mean_table[(Biv_Data_mean_table$sex == "female") & (Biv_Data_mean_table$subsp == "Dom"),],aes(y=mean_SC, x=fileName, col="red", size=2))+
  geom_point(data=Biv_Data_mean_table[(Biv_Data_mean_table$sex == "female") & (Biv_Data_mean_table$subsp == "Dom"),],aes(y=median_SC, x=fileName, col="black", size=1.5))+
   geom_point(data=Biv_Data_mean_table[(Biv_Data_mean_table$sex == "female") & (Biv_Data_mean_table$subsp == "Dom"),],aes(y=Qar.4, x=fileName, col="black", size=2.6), shape=4)+
  scale_color_manual(values=colors_of_strains)+
  facet_wrap(~strain, scales="free")+ylim(c(30,200))+ggtitle("Dom female, almost whole cell \nMean, median, 4th Quartille ploted")
almost_cells_DOM_f


#### Musc female

show.the.cells_females_musc <- ggplot(whole.cell.data_manual[(whole.cell.data_manual$sex == "female") & (whole.cell.data_manual$subsp == "Musc"),], aes(x=fileName, y= SC.length, color=strain))+geom_point()+
  
   geom_point(data=manual_summary_stats[(manual_summary_stats$sex == "female") &(manual_summary_stats$subsp == "Musc"),],aes(y=mean_SC, x=fileName, col="red", size=2))+
  geom_point(data=manual_summary_stats[(manual_summary_stats$sex == "female") & (manual_summary_stats$subsp == "Musc"),],aes(y=median_SC, x=fileName, col="black", size=1.5))+
  
   geom_point(data=manual_summary_stats[(manual_summary_stats$sex == "female") & (manual_summary_stats$subsp == "Musc"),],aes(y=Qar.4, x=fileName, col="black", size=2.6), shape=4)+
  
  facet_wrap(~strain, scales="free")+ylim(c(30,200))+ggtitle("Female Musc whole cell manual measures")



almost_cells_Musc_f <- ggplot(almost_whole_cell_BivData[(almost_whole_cell_BivData$sex == "female") & (almost_whole_cell_BivData$subsp == "Musc"),], aes(x=fileName, y= chromosomeLength, color=strain))+
  
  geom_point()+ 
  geom_point(data=Biv_Data_mean_table[(Biv_Data_mean_table$sex == "female") & (Biv_Data_mean_table$subsp == "Musc"),],aes(y=mean_SC, x=fileName, col="red", size=2))+
  geom_point(data=Biv_Data_mean_table[(Biv_Data_mean_table$sex == "female") & (Biv_Data_mean_table$subsp == "Musc"),],aes(y=median_SC, x=fileName, col="black", size=1.5))+
   geom_point(data=Biv_Data_mean_table[(Biv_Data_mean_table$sex == "female") & (Biv_Data_mean_table$subsp == "Musc"),],aes(y=Qar.4, x=fileName, col="black", size=2.6), shape=4)+
  
  facet_wrap(~strain, scales="free")+ylim(c(30,200))+ggtitle("Musc female, almost whole cell \nMean, median, 4th Quartille ploted")



### Dom Male
show.the.cells_male_dom <- ggplot(whole.cell.data_manual[(whole.cell.data_manual$sex == "male") & (whole.cell.data_manual$subsp == "Dom"),], aes(x=fileName, y= SC.length, color=strain))+geom_point()+
  
   geom_point(data=manual_summary_stats[(manual_summary_stats$sex == "male") &(manual_summary_stats$subsp == "Dom"),],aes(y=mean_SC, x=fileName, col="red", size=2))+
  geom_point(data=manual_summary_stats[(manual_summary_stats$sex == "male") & (manual_summary_stats$subsp == "Dom"),],aes(y=median_SC, x=fileName, col="black", size=1.5))+
  
   geom_point(data=manual_summary_stats[(manual_summary_stats$sex == "male") & (manual_summary_stats$subsp == "Dom"),],aes(y=Qar.4, x=fileName, col="black", size=2.6), shape=4)+
  
  facet_wrap(~strain, scales="free")+ylim(c(30,150))+ggtitle("Male Dom whole cell manual measures")



almost_cells_Dom_m <- ggplot(almost_whole_cell_BivData[(almost_whole_cell_BivData$sex == "male") & (almost_whole_cell_BivData$subsp == "Dom"),], aes(x=fileName, y= chromosomeLength, color=strain))+
  
  geom_point()+ 
  geom_point(data=Biv_Data_mean_table[(Biv_Data_mean_table$sex == "male") & (Biv_Data_mean_table$subsp == "Dom"),],aes(y=mean_SC, x=fileName, col="red", size=2))+
  geom_point(data=Biv_Data_mean_table[(Biv_Data_mean_table$sex == "male") & (Biv_Data_mean_table$subsp == "Dom"),],aes(y=median_SC, x=fileName, col="black", size=1.5))+
   geom_point(data=Biv_Data_mean_table[(Biv_Data_mean_table$sex == "male") & (Biv_Data_mean_table$subsp == "Dom"),],aes(y=Qar.4, x=fileName, col="black", size=2.6), shape=4)+
  
  facet_wrap(~strain, scales="free")+ylim(c(30,150))+ggtitle("Dom male, almost whole cell \nMean, median, 4th Quartille ploted")



#MUSC male
show.the.cells_male_musc <- ggplot(whole.cell.data_manual[(whole.cell.data_manual$sex == "male") & (whole.cell.data_manual$subsp == "Musc"),], aes(x=fileName, y= SC.length, color=strain))+geom_point()+
  
   geom_point(data=manual_summary_stats[(manual_summary_stats$sex == "male") &(manual_summary_stats$subsp == "Musc"),],aes(y=mean_SC, x=fileName, col="red", size=2))+
  geom_point(data=manual_summary_stats[(manual_summary_stats$sex == "male") & (manual_summary_stats$subsp == "Musc"),],aes(y=median_SC, x=fileName, col="black", size=1.5))+
  
   geom_point(data=manual_summary_stats[(manual_summary_stats$sex == "male") & (manual_summary_stats$subsp == "Musc"),],aes(y=Qar.4, x=fileName, col="black", size=2.6), shape=4)+
  
  facet_wrap(~strain, scales="free")+ylim(c(30,150))+ggtitle("Musc Dom whole cell manual measures")



almost_cells_Musc_m <- ggplot(almost_whole_cell_BivData[(almost_whole_cell_BivData$sex == "male") & (almost_whole_cell_BivData$subsp == "Musc"),], aes(x=fileName, y= chromosomeLength, color=strain))+
  
  geom_point()+ 
  geom_point(data=Biv_Data_mean_table[(Biv_Data_mean_table$sex == "male") & (Biv_Data_mean_table$subsp == "Musc"),],aes(y=mean_SC, x=fileName, col="red", size=2))+
  geom_point(data=Biv_Data_mean_table[(Biv_Data_mean_table$sex == "male") & (Biv_Data_mean_table$subsp == "Musc"),],aes(y=median_SC, x=fileName, col="black", size=1.5))+
   geom_point(data=Biv_Data_mean_table[(Biv_Data_mean_table$sex == "male") & (Biv_Data_mean_table$subsp == "Musc"),],aes(y=Qar.4, x=fileName, col="black", size=2.6), shape=4)+
  
  facet_wrap(~strain, scales="free")+ylim(c(30,150))+ggtitle("Musc male, almost whole cell \nMean, median, 4th Quartille ploted")

#show.the.cells_male <- ggplot(whole.cell.data_manual[whole.cell.data_manual$sex == "male",], aes(x=fileName, y= SC.length, color=strain))+geom_point()+
  
#   geom_point(data=manual_summary_stats[manual_summary_stats$sex == "male",],aes(y=mean_SC, x=fileName, col="red", size=2))+
  
#  geom_point(data=manual_summary_stats[manual_summary_stats$sex == "male",],aes(y=median_SC, x=fileName, col="black", size=1.5))+
  
#   geom_point(data=manual_summary_stats[manual_summary_stats$sex == "male",],aes(y=Qar.4, x=fileName, col="black", size=2.6), shape=4)+
  
 # facet_wrap(~strain, scales="free")+ylim(c(30,150))+ggtitle("Male whole cell manual measures")
#show.the.cells_male

#almost_whole_cell_BivData
#almost_whole_cell_wsb <- almost_whole_cell_BivData[almost_whole_cell_BivData$strain == "WSB",]
#almost_whole_cell_DOM <- almost_whole_cell_BivData[almost_whole_cell_BivData$subsp == "Dom",]
#almost_whole_cell_MUSC <- almost_whole_cell_BivData[almost_whole_cell_BivData$subsp == "Musc",]


#`r data.frame(table(droplevels(real.long.bivData$category)))`.  
```

<All the plots above show the distributions of manually whole cell measured SC lengths compared to the SC length distributions from the automated bivalent data. It shows the amount of within cell variance across strains. There is a bit of variance across the SC length distributions in the PWD females.
This data set might be noisy, given the amount of variance in the SC length distributions across cells (PWD females, WSB females).
The DF **real.long.bivData** contains `r length(real.long.bivData$Obj.ID)` bivalent measures. The full data set is `r length(Curated_BivData$Obj.ID)`. This is the breakdown of bivalent observations by category for this long dataset are 

**- Try to merge this DF with the whole.cell manual measures.**  
- Try estimating which 'loose' bivalent observations might be within the long class of bivalents.  
in code chunk above I ran the mouse averages for the longest bivalents. (680 bivalents, from 54 mice.  10202 bivalents from 86 mice.>



```{r show.long.SC, echo=FALSE, eval=FALSE}

female_long_biv_plot <- ggplot(real.long.bivData[real.long.bivData$sex == "female",], aes(x=fileName, y=chromosomeLength, color=strain))+geom_point()+facet_wrap(~category, scales="free")+ylim(c(30,200))+ggtitle("Long Bivalent Dataset, \nFemale")+scale_color_manual(values=colors_of_strains)



male_long_biv_plot <- ggplot(real.long.bivData[real.long.bivData$sex == "male",], aes(x=fileName, y=chromosomeLength, color=strain))+geom_point()+facet_wrap(~category, scales="free")+ylim(c(30,200))+ggtitle("Long Bivalent Dataset, \nmale")+scale_color_manual(values=colors_of_strains)


```

<ABOVE plots show the SC lengths for the 'long SC data set'. They are supposed to be the longest 4-5 SC from cells where I could get good measures. These longer bivalents are useful because their patterns shouldn't be affect by chromosome size effect (which effects, CO position). Hopefully this data set will have less noise from chromosome identity, but there was still data missing (they don't come from whole cell measures).>


<!-- unused chunkcs -->

```{r SC.distb_manual_Auto, echo=FALSE, eval=FALSE}

#calq the mouse means
manual_summary_stats <- ddply(whole.cell.data_manual, c("fileName"), summarise,

                        mean_SC = as.numeric(format(round(  mean(SC.length, na.rm = TRUE), 3 ), nsmall=3) ),
                        median_SC = median(SC.length, na.rm = TRUE),
                        Qar.4 = quantile(SC.length, na.rm = TRUE)[[4]]
#                        var_SC = format(round(   as.numeric(var(SC.length)),3), nsmall=3)
                   
)

manual_summary_stats <- add_mouse(manual_summary_stats)
manual_summary_stats <- add_strain(manual_summary_stats)
manual_summary_stats <- add_sex(manual_summary_stats)
manual_summary_stats <- add_subsp(manual_summary_stats)

###
#calq the mouse means
Biv_Data_mean_table <- ddply(almost_whole_cell_BivData, c("fileName"), summarise,

                        mean_SC = as.numeric(format(round(  mean(chromosomeLength), 3 ), nsmall=3) ),
                        median_SC = median(chromosomeLength),
                        Qar.4 = quantile(chromosomeLength)[[4]],
                        
                        var_SC = format(round(   as.numeric(var(chromosomeLength)),3), nsmall=3)
                        
)
Biv_Data_mean_table <- add_mouse(Biv_Data_mean_table)
Biv_Data_mean_table <- add_strain(Biv_Data_mean_table)
Biv_Data_mean_table <- add_sex(Biv_Data_mean_table)
Biv_Data_mean_table <- add_subsp(Biv_Data_mean_table)

#colors_of_strains
#scale_color_manual(values=colors_of_strains)

#un-order these factors
```




<!-- Whole Cell Manual Comparison -->

```{r XX.adj_manual.whole.cell, echo=FALSE, eval=FALSE}

#load and examine the the whole cell hand measures here

female.whole.cell <- whole.cell.data_manual[whole.cell.data_manual$sex == "female",]
data.frame(table(droplevels(female.whole.cell$category)))

# cell is 22nov16_13nov16_MSM_f2_sp1_19 is missing a biv


#cell 5aug17_14jul17_LEW_f1_sp1_11_rev has broken SC - could be used to compare broken SC

data.frame(table(droplevels(female.whole.cell$category), female.whole.cell$rank))
#missing some strains, most have 3 cells (min). MSM has 5 full cells


#remove the incomplete LEW female cell
female.whole.cell <- female.whole.cell[female.whole.cell$fileName != "5aug17_14jul17_LEW_f1_sp1_11_rev",]

female.whole.cell <- female.whole.cell[female.whole.cell$fileName != "22nov16_13nov16_MSM_f2_sp1_19",]

#calq mean SC lengths by rank

mean.SC_rank.strain <- ddply(female.whole.cell, c("strain", "rank"), summarise,
         
                       mean.sc =   mean(SC.length, na.rm = TRUE),
                         nbiv = length(Obj.ID),
                       sd.SC_length   = sd(SC.length, na.rm = TRUE),
                         se.SC_length   = sd.SC_length / sqrt(nbiv)
)
         #   mean.SC_rank1 = mean(SC.length[which(rank == 20)], na.rm = 
       
top.biv <- mean.SC_rank.strain[mean.SC_rank.strain$rank > 15,]


show.SC_by.rank_top <- ggplot(data = top.biv, aes(x=strain, y=mean.sc, color = as.factor(rank)  )) +
    geom_pointrange(aes(ymin = mean.sc-(se.SC_length*2), ymax = mean.sc+(se.SC_length*2)), 
                    position=position_dodge(width=0.7), 
                    linetype='solid') +    theme_bw() + facet_wrap(~strain, scales="free")+
          ylim(c(71,210) )+ggtitle("mean SC Length by ranks \n2SE as line")

show.SC_by.rank_top

bottom.biv <- mean.SC_rank.strain[mean.SC_rank.strain$rank < 6,]

show.SC_by.rank_bottom <- ggplot(data = bottom.biv, aes(x=strain, y=mean.sc, color = as.factor(rank)  )) +
    geom_pointrange(aes(ymin = mean.sc-(se.SC_length*2), ymax = mean.sc+(se.SC_length*2)), 
                    position=position_dodge(width=0.7), 
                    linetype='solid') +    theme_bw() + facet_wrap(~strain, scales="free")+
          ylim(c(35,99) )+ggtitle("Bottom Bivalents mean SC Length by ranks \n2SE as line")




#CALQ PERCENT TOTAL SC

CELL.SC.proportions <- ddply(female.whole.cell, c("fileName"), summarise,
         
                       total.sc.cell =   sum(SC.length, na.rm = TRUE),
                         nbiv = length(Obj.ID),
                       #rank.by.per
                   
                    rank1_prop = (SC.length[which(rank == 20)] / total.sc.cell),
                    rank2_prop = (SC.length[which(rank == 19)] / total.sc.cell),
                    rank3_prop = (SC.length[which(rank == 18)] / total.sc.cell),
                    rank4_prop = (SC.length[which(rank == 17)] / total.sc.cell),
                      rank5_prop = (SC.length[which(rank == 16)] / total.sc.cell)
                       #sd.SC_length   = sd(SC.length, na.rm = TRUE),
                      #   se.SC_length   = sd.SC_length / sqrt(nbiv)
)
    
#MELT ABOVE
MELT_Biv.SC_proporion <- melt(CELL.SC.proportions[,c( "fileName","total.sc.cell","nbiv",
              #leave out the Foci
               "rank1_prop","rank2_prop","rank3_prop","rank4_prop","rank5_prop")], 
                 
            id=c( "fileName","total.sc.cell","nbiv"))

MELT_Biv.SC_proporion <- MELT_Biv.SC_proporion[(!is.na(MELT_Biv.SC_proporion$value)),]

MELT_Biv.SC_proporion <- add_mouse(MELT_Biv.SC_proporion)
MELT_Biv.SC_proporion <- add_strain(MELT_Biv.SC_proporion)


show_long.sc.proportions <- ggplot(data = MELT_Biv.SC_proporion, aes(y = value, x = fileName, fill=as.factor(variable)) ) + geom_col()+ggtitle("Breakdown of total SC per cell")+ylim(c(0,1)) +theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+facet_wrap(~strain, scales = "free")

show_long.sc.proportions


#original plot code 
#ggplot(data = df1, aes(x, y, color = Group)) +
#    geom_pointrange(aes(ymin = lb, ymax = ub), 
#                    position=position_jitter(width=0.5), 
#                    linetype='dotted') +
#    theme_bw()
```

The plots above show the mean SC lengths and 2SE error bars for single bivalents which have been given within cell rank.

The first plot showing the mean SC lengths by the rank (most all of these cells have 3, MSM has 5 cells (observations)). 

The purpose of these plots is to display the variance of single bivalents when they are assigned a within cell rank. For the longest bivalents, XX is predicted to be the 3rd longest (according to physical length Mb).

(use the value for the 3rd bivalent to adjust the single bivalent traits for XX -- then compare to males values -- or re-run in the MM).

The other figure shows of each single bivalent contributes to the total SC area. Each column is a cell and each color is the percent of total SC area for the longest 5 bivalents in that cell. on average, each of the top longest bivalents make up ~10% of the cell's total SC area. So for cells all 20 bivalents, of it's total SC area, 5-7% is due to a XX, 

- Is the difference between cell averages for males and females less that 10%?

- also interesting, the pwd and msm don't have longer SC, compared to other strains.

<!-- Automated BivData Comparison -->

```{r XX.Automated.biv.stuff, echo=FALSE}

#make automated female data to compare to
female.Cur.Bivdata <- Curated_BivData[(Curated_BivData$sex == "female"),]

female.Bivdata_comp <- female.Cur.Bivdata[ ( (female.Cur.Bivdata$strain == "WSB") | (female.Cur.Bivdata$strain == "G") | (female.Cur.Bivdata$strain == "LEW") | (female.Cur.Bivdata$strain == "PWD")| (female.Cur.Bivdata$strain == "MSM") ),]

#table(female.Bivdata_comp$strain)

#female.Bivdata_comp


#1. calq the rate of chrm segmentation in female categories in automateted data

box.rates <- ddply(female.Bivdata_comp, c("fileName"), summarise,
                        
                   Nmice = length(unique(mouse)),
                   mx.box.num = max(boxNumber), # need to run this at the cell level
                    N.biv  = length(Obj.ID),
                   pass.biv = N.biv / mx.box.num
)
box.rates <- add_mouse(box.rates)
box.rates <- add_strain(box.rates)
box.rates <- add_category(box.rates)

cell.pass.rates <- ggplot(data = box.rates, aes(y=pass.biv, x=mouse, color=strain))+geom_jitter()+ggtitle("Auto biv pass rate per cell")+facet_wrap(~strain, scales="free")+ylim(c(0,1.1))+ theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())

cell.pass.rates


category.box.rates <- ddply(box.rates, c("category"), summarise,
                        
                   mean.pass.rate = mean(pass.biv)
)
#category.box.rates


#2. pull the 3rd rank biv's calculate the SC means
#calq means by rank (focus on the top 5)

```

For the Automated data set, I like to measure the rate of passing bivalent per cell. The mean pass rate will be multiplied to the estimated XX mean_SC. 

The table above shows the number of bivalents from the same strains as in the manual whole cell data. The plot shows the bivalent passing rate across all of the individual cells from this female data set. For each strain, I'll calculate the mean bivalent passing rate (maybe I should look at the mouse levels).  

(some of the mice have different ranges of per cell passing rate) - given this ranges, i think the xx adjustment factor should be called on the mouse level. (it could even be extended to cell level -- except i don't think the XX SC length estimates wont be good.)

strain.XX.adjustment.factor = per_cell_passing rate \* 1 of 20 random biv will be XX \*  


** It might be simplier to compare the male and female means, and test it they are greater than the whole cell proprotion of the XX in females cells.**  The XX in a whole female cell contributes ~ 7% of total SC, if the female means for a type of total SC measure are from XX. But I am not using 'whole cell' summaries to compare female and males.


**What is the effect of an extra XX-autosome on single bivalent means?**

use a permutation approach: Make a True data set to start with, same(similar) number of cells, mice and bivalents. Make fake data sets which sample 19 bivalents, for 'in silico' cells for males and females. Also Run cntrl-female data set, where 20 bivalents are sampled, but randomly. Run the same bivalent level summaries for each 'permuted data set'; male avSC, 19Female_avSC, and rand.20_Female_avSC.  The difference between the rand.20 and rand.19 female -permuted data sets should indicate the influence of having an extra 'XX-autosome' in the total data set. 


# Note on Heterochiasmy Definition

<!-- add outline for new XX adjust section 
adjust the colors for the plot below (standardize)
-->

I present heterochiasmy as a comparison of oocyte to spermatocyte MLH1 counts, but the sex chromosomes/bivalents complicate this comparison. In females the XX bivalent is indistinguishable from the autosomes. To the meiotic recombination machinery, it is an autosome and has a similar REC landscape. Whereas in spermatocytes the XY bivalent is visually distinct and any MLH1 where not included in the count). (I note if the and Y are paired, which they are at a high rate). The XY pair triggers a response to un-paired chromosomes and only has MLH1 foci within the PAR (the the tips of X and Y). To make a more equivalent comparison I will estimate which bivalent is the XX in oocytes, and subtract that average REC from the category average of each strain.

1. Compile full-cell data from females (all 20 bivalents measured)
2. Look at the SC length -ranked data, extract the 3rd longest estimate average REC for this bivalent, 
3. check how variable the REC is across the 1st,2nd,4th, and 5th are.

According to mouse genome website, the X is the 3rd largest chromosome by total amount of DNA (Mb).

(Put the XX adjustment section here)



There is now MOLF, which has female biased hetC
3 of my Musc strains have male biased patter; SKIVE, PWD and MSM.
1 of the musc strains has female biased heterochiasmy, KAZ. 


```{r mouse.specific.Qplot, echo=FALSE, message=FALSE, results='hide', warning=TRUE, include=FALSE, eval=FALSE}

#REMOVE This BLOCK?

#### Distributions by mouse
#There are code chunks that make a big dfs of CO count by quality score plot. The #code has been commented out, and moved to it's own doc.

#remove this block since it has it's own src/script

### to integrate 'pass'/fail
### this should be in setup data

#mice that don't have female data will break the chain-loop
#MOLF female, HMI, CZECH female, CAROLI female, Tom male

#instead of adding fake data, remove the categories from list

#filler_data_MOLF.f = c("BatchX", "1jan19_fake_MOLF_f1_sp1", "99999", "3","20","20","","","","","","1jan19_fake_MOLF_f1_sp1_rev.tif",
#                       "fake_MOLF_f1", "MOLF female", "MOLF","female","Musc", "20","")
#filler_data_HMI.f = c("BatchX", "1jan19_fake_HMI_f1_sp1", "99999", "3","20","20","","","","","","1jan19_fake_HMI_f1_sp1_rev.tif",
#                       "fake_HMI_f1", "HMI female", "HMI","female","Cast", "20","")
#filler_data_CZECH.f = c("BatchX", "1jan19_fake_CZECH_f1_sp1", "99999", "3","20","20","","","","","","1jan19_fake_CZECH_f1_sp1_rev.tif",
#                       "fake_CZECH_f1", "CZECH female", "CZECH","female","Musc", "20","")
#filler_data_CAROLI.f = c("BatchX", "1jan19_fake_CAROLI_f1_sp1", "99999", "3","20","20","","","","","","1jan19_fake_CAROLI_f1_sp1_rev.tif",
#                       "fake_CAROLI_f1", "CAROLI female", "CAROLI","female","Caroli", "20","")
#MLH1_data_table <- rbind(MLH1_data, filler_data_MOLF.f,filler_data_HMI.f,filler_data_CZECH.f,filler_data_CAROLI.f)#error from dataset thing

pass_mice <- c("28sep18_CZECH_m3","26oct18_MOLF_m1","22jun15_G_m2")


#this loop makes a quant_status col for a list of mice
for(i in 1:length(MLH1_data$mouse)){
  MLH1_data$quant_status[i] <- ifelse( is.element(MLH1_data$mouse[i], pass_mice), "pass", NA)
}


df_list <- split(MLH1_data, (MLH1_data$category))
#remove sections which are in category but have no data
#this is where you remove the non-data data!
df_list <- df_list[ -c(7,12,14,18,20,21,22,24,26,28,30) ]

#7 PERC, 12 MOLF female, 
#

#df list, is a list of df's which were split() from BIG df by category
df_list 
o=0
i=0

MLH1_data$quant_status <- as.factor(MLH1_data$quant_status)

#make a new empty df. create a ddply thing with mean calculations (this is like a list for matching up later)
for(i in 1:(length(df_list))){
 # only include data with quality values, then create a ddply for specific sub-df that calqs mean by each mouse
  df_list[[i]] <- df_list[[i]][!(is.na(df_list[[i]]$quality) | df_list[[i]]$quality==""), ]
  sub_mouse_means <- ddply(df_list[[i]], .(mouse), summarise,
                      mouse_nMLH1 = mean(adj_nMLH1.foci)
  )
# this loop is going though each row of dataframes within the df.list and assigning the mouse mean
  for(o in 1:length(df_list[[i]]$Original.Name)){
    df_list[[i]]$mouse_mean_MLH1[o] <- sub_mouse_means$mouse_nMLH1[ (df_list[[i]]$mouse[o] == sub_mouse_means$mouse) ]
        }
}


plot_list = list()
for (h in 1:length(df_list)) {
  pf <- ggplot(data = df_list[[h]], aes(colour = factor(Batch)), width = 0.25)+ 
    geom_jitter(aes(y= adj_nMLH1.foci, x= quality)) + 
    geom_rect(data = df_list[[h]], aes(fill=as.factor(quant_status)), alpha=0.5, xmin=-100, xmax=100, ymin=-100, ymax=100)+
    facet_wrap(~ mouse, drop = TRUE) + 
    geom_hline(aes(yintercept = mouse_mean_MLH1), color="red") +scale_fill_manual(values=c(NA, "red"), breaks=NULL) +    geom_hline(yintercept = c( mean(df_list[[h]]$adj_nMLH1.foci) ), color="black" )
  #theme(plot.background = element_rect(fill = "green")) #theme for indiviual mice, 
  #make an empty ggplot? for missing data
  
  plot_list[[h]] =   pf
}

#orig (I think this is the same as above)
#plot_list = list()
#for (h in 1:length(df_list)) {
#  pf <- ggplot(df_list[[h]],(aes(y= adj_nMLH1.foci, x= quality)))
#  pf <- pf +geom_rect(data = df_list[[h]], aes(fill=quant_status), xmin=-100, xmax=100, ymin=-100, ymax=100, alpha=0.005)

#    pf <- pf + geom_jitter(width = 0.25, aes(colour = factor(Batch)) )
#  pf <- pf + geom_hline(yintercept = c( mean(df_list[[h]]$adj_nMLH1.foci) ), color="black" )
#  pf <- pf + facet_wrap(~ mouse) + geom_hline(aes(yintercept = mouse_mean_MLH1), color="red") + 
    
 #   scale_fill_manual(values=alpha(c("red"), .005), breaks=NULL) +

    #theme for indiviual mice, 
#theme(plot.background = element_rect(fill = "green"))
#  plot_list[[h]] =   pf
#}
#can't change the alpha/transparency of the red rec

#aes must be same length ---   #geom_point(stat = "identity") +
# the last one is correct....?

#so plot list is some wride dataframe, but indeces 1:12 in plot list are ggplots
```

```{r show.plots, echo=FALSE, message=FALSE,results='hide', warning=FALSE, include=FALSE, eval=FALSE}
geom_hline(aes(yintercept = med, group = gr), colour = 'red')
#no plot is made that goes in the list -- so it can't be shown
for (gg in 1:length(plot_list) ){
  show(plot_list[gg])
 # invisible(lapply(obj, function(x) plot(x,main="some plot")))
}

#this stops at MSM
```

The mouse specific scatter plots aren't show here because there are too bulky. These plots are in a different document.

Making all of these scatter plots, allows us to look at the whole distributions of the data for each mouse. The distance of the red line from the black could be a indicator of slides or mice with slide specific technical noise.

```{r HetC.plot22, echo=FALSE,  fig.width=10, fig.height=5, fig.caption="Sex-Specific MLH1 average", eval=FALSE}


no_female <- c("HMI", "SKIVE", "CZECH", "MOLF", "CAROLI", "TOM","AST")

MixedModel_CO_data <- AP_mouse_table_w.Ages[AP_mouse_table_w.Ages$strain.x != "other",]
CO_data_HetC_w.old <- AP_mouse_table_w.Ages[AP_mouse_table_w.Ages$strain.x != "other",]
CO_data_HetC_w.old <- CO_data_HetC_w.old[ ! CO_data_HetC_w.old$strain.x %in% no_female, ]
CO_data_HetC_w.old <- CO_data_HetC_w.old[!(is.na(CO_data_HetC_w.old$Nmice) | CO_data_HetC_w.old$Nmice==""), ]

MixedModel_CO_data <- MixedModel_CO_data[MixedModel_CO_data$age.weeks < 15,]

MixedModel_CO_data <- MixedModel_CO_data[!(is.na(MixedModel_CO_data$mouse) | MixedModel_CO_data$mouse==""), ]


MixedModel_CO_data <- MixedModel_CO_data[ ! MixedModel_CO_data$strain.x %in% no_female, ]

kuku <- ggplot(MixedModel_CO_data, aes(y=mean_co, x=mouse, color=sex.y))+geom_point()+facet_wrap(~strain.x)

bubu <- ggplot(data=CO_data_HetC_w.old)+geom_point(aes(y=mean_co, x=mouse, color=sex.x))+facet_wrap(~strain.x)

#add titles
#use multiplot
#remove X axis

multiplot(kuku,bubu, cols=2)
```

```{r F1.by.F2, eval=FALSE}

#try remaking the plot Megan suggested
# for 2CO positions, Foci1, Position  on x and Foci 2 position on y

CurBivData_2CO <- Curated_BivData[Curated_BivData$hand.foci.count == 2,]

CurBivData_2CO <- CurBivData_2CO[!(is.na(CurBivData_2CO$Foci2) | CurBivData_2CO$Foci2==""), ]

#isolate 2COs
#facet by sex and subsp

F1.x.F2 <- ggplot(CurBivData_2CO, aes(x=Foci1,y=Foci2, color=strain) ) + geom_point()+ facet_wrap(~sex)+ggtitle("test plot")


```

<!-- Deleted notes -->

```{r variance.hist, echo=FALSE, eval=FALSE}
#figures of histograms, 

#make series for total dataset?
Var.Hist.total.mice <- ggplot(DF.HetC.MixedModel, aes(x = var, fill=sex)) + 
  geom_histogram(binwidth = 1, alpha=.6)  +
  ggtitle("Histograms of within mouse variance, all mice")
Var.Hist.total.mice <- Var.Hist.total.mice   + facet_wrap(~ strain, scales="free")
Var.Hist.total.mice

#DF.HetC.MixedModel.HQ
Var.Hist <- ggplot(DF.HetC.MixedModel.HQ, aes(x = var, fill=sex)) + 
  geom_histogram(binwidth = 1, alpha=.6)  + 
  ggtitle("Histograms of within mouse variance, 10 cell min")
Var.Hist <- Var.Hist   + facet_wrap(~ strain, scales="free")


Var.Hist.15cells <- ggplot(DF.HetC.MixedModel.HQ[DF.HetC.MixedModel.HQ$Ncells >14,], aes(x = var, fill=sex)) + 
  geom_histogram(binwidth = 1, alpha=.6)  +   ggtitle("Histograms of within mouse variance, 15 cell min")
Var.Hist.15cells <- Var.Hist.15cells   + facet_wrap(~ strain, scales="free")



Q12.var.hist.plot <- ggplot(Q12_mouse_table_w.Ages, aes(x = var, fill=sex)) + 
  geom_histogram(binwidth = 1, alpha=.6)  +   ggtitle("Histograms of within mouse variance, Q12")

Q12.var.hist.plot <- Q12.var.hist.plot   + facet_wrap(~ strain, scales="free")
Q12.var.hist.plot

#some LEW female still have v high within mouse variance (2 mice, )


```





# References